<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Charlotte Jones-Todd">
<meta name="dcterms.date" content="2025-01-01">

<title>Advanced Biological Data Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="index_files/libs/clipboard/clipboard.min.js"></script>
<script src="index_files/libs/quarto-html/quarto.js"></script>
<script src="index_files/libs/quarto-html/popper.min.js"></script>
<script src="index_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="index_files/libs/quarto-html/anchor.min.js"></script>
<link href="index_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="index_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="index_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="index_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="index_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<style>

      .quarto-title-block .quarto-title-banner {
        background: #c7dbe2;
      }
</style>
<script src="index_files/libs/quarto-diagram/mermaid.min.js"></script>
<script src="index_files/libs/quarto-diagram/mermaid-init.js"></script>
<link href="index_files/libs/quarto-diagram/mermaid.css" rel="stylesheet">
<script src="index_files/libs/fitvids-2.1.1/fitvids.min.js"></script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body>

<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Advanced Biological Data Analysis</h1>
            <p class="subtitle lead">BIOSCI 738</p>
                      </div>
  </div>
    
  <div class="quarto-title-meta-author">
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-heading">Affiliation</div>
    
      <div class="quarto-title-meta-contents">
      <p class="author">Charlotte Jones-Todd </p>
    </div>
    <div class="quarto-title-meta-contents">
          <p class="affiliation">
              University of Auckland
            </p>
        </div>
    </div>

  <div class="quarto-title-meta">

        
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul class="collapse">
  <li><a href="#notes" id="toc-notes" class="nav-link active" data-scroll-target="#notes">Notes</a></li>
  <li><a href="#march-10th" id="toc-march-10th" class="nav-link" data-scroll-target="#march-10th">March 10th</a>
  <ul class="collapse">
  <li><a href="#rough-timeline" id="toc-rough-timeline" class="nav-link" data-scroll-target="#rough-timeline">Rough timeline</a></li>
  <li><a href="#sec-gorilla" id="toc-sec-gorilla" class="nav-link" data-scroll-target="#sec-gorilla">Gorilla Gambit</a></li>
  <li><a href="#sec-coc" id="toc-sec-coc" class="nav-link" data-scroll-target="#sec-coc">Group Code of Conduct</a></li>
  <li><a href="#sec-intro" id="toc-sec-intro" class="nav-link" data-scroll-target="#sec-intro">Introduction to BIOSCI 738</a></li>
  <li><a href="#sec-ai" id="toc-sec-ai" class="nav-link" data-scroll-target="#sec-ai">GenAI in the BIOSCI 738 classroom</a></li>
  <li><a href="#sec-peershare01" id="toc-sec-peershare01" class="nav-link" data-scroll-target="#sec-peershare01">Q&amp;A and peer-share</a></li>
  </ul></li>
  <li><a href="#march-14th" id="toc-march-14th" class="nav-link" data-scroll-target="#march-14th">March 14th</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-1" id="toc-rough-timeline-1" class="nav-link" data-scroll-target="#rough-timeline-1">Rough timeline</a></li>
  <li><a href="#sec-coccompleted" id="toc-sec-coccompleted" class="nav-link" data-scroll-target="#sec-coccompleted">Activity &amp; AI expectations</a></li>
  <li><a href="#sec-viz" id="toc-sec-viz" class="nav-link" data-scroll-target="#sec-viz">Data visualization</a></li>
  <li><a href="#sec-prog_practice" id="toc-sec-prog_practice" class="nav-link" data-scroll-target="#sec-prog_practice">Good programming practice</a></li>
  <li><a href="#sec-honesty" id="toc-sec-honesty" class="nav-link" data-scroll-target="#sec-honesty"><em>Reproducible</em> analysis</a></li>
  <li><a href="#sec-peershare02" id="toc-sec-peershare02" class="nav-link" data-scroll-target="#sec-peershare02">Q&amp;A and peer-share</a></li>
  </ul></li>
  <li><a href="#march-17th---guest-lecturer-dr.-liza-bolton" id="toc-march-17th---guest-lecturer-dr.-liza-bolton" class="nav-link" data-scroll-target="#march-17th---guest-lecturer-dr.-liza-bolton">March 17th - Guest Lecturer Dr.&nbsp;Liza Bolton!</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-2" id="toc-rough-timeline-2" class="nav-link" data-scroll-target="#rough-timeline-2">Rough timeline</a></li>
  <li><a href="#sec-slidesliza" id="toc-sec-slidesliza" class="nav-link" data-scroll-target="#sec-slidesliza">Slides</a></li>
  <li><a href="#sec-caseliza" id="toc-sec-caseliza" class="nav-link" data-scroll-target="#sec-caseliza">Case studies</a></li>
  <li><a href="#finished-early" id="toc-finished-early" class="nav-link" data-scroll-target="#finished-early">Finished early?</a></li>
  </ul></li>
  <li><a href="#march-21st" id="toc-march-21st" class="nav-link" data-scroll-target="#march-21st">March 21st</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-3" id="toc-rough-timeline-3" class="nav-link" data-scroll-target="#rough-timeline-3">Rough timeline</a></li>
  <li><a href="#sec-significance" id="toc-sec-significance" class="nav-link" data-scroll-target="#sec-significance"><em>Significance</em></a></li>
  <li><a href="#sec-resampling" id="toc-sec-resampling" class="nav-link" data-scroll-target="#sec-resampling">Hypothesis testing via resampling</a></li>
  <li><a href="#sec-bootstrap" id="toc-sec-bootstrap" class="nav-link" data-scroll-target="#sec-bootstrap">Bootstrapping</a></li>
  <li><a href="#sec-peershare04" id="toc-sec-peershare04" class="nav-link" data-scroll-target="#sec-peershare04">Q&amp;A/peer-share/Task 03</a></li>
  </ul></li>
  <li><a href="#march-24th" id="toc-march-24th" class="nav-link" data-scroll-target="#march-24th">March 24th</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-4" id="toc-rough-timeline-4" class="nav-link" data-scroll-target="#rough-timeline-4">Rough timeline</a></li>
  <li><a href="#rough-timeline-5" id="toc-rough-timeline-5" class="nav-link" data-scroll-target="#rough-timeline-5">Rough timeline</a></li>
  <li><a href="#sec-draw" id="toc-sec-draw" class="nav-link" data-scroll-target="#sec-draw">From numbers to plots</a></li>
  <li><a href="#linear-model-diagnostics" id="toc-linear-model-diagnostics" class="nav-link" data-scroll-target="#linear-model-diagnostics">Linear model diagnostics</a></li>
  <li><a href="#sec-comm" id="toc-sec-comm" class="nav-link" data-scroll-target="#sec-comm">Communicating statistics</a></li>
  <li><a href="#sec-project" id="toc-sec-project" class="nav-link" data-scroll-target="#sec-project">Data discussion</a></li>
  <li><a href="#sec-ioa" id="toc-sec-ioa" class="nav-link" data-scroll-target="#sec-ioa">Interactive Oral Assessments (IOAs)</a></li>
  </ul></li>
  <li><a href="#march-28th" id="toc-march-28th" class="nav-link" data-scroll-target="#march-28th">March 28th</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-6" id="toc-rough-timeline-6" class="nav-link" data-scroll-target="#rough-timeline-6">Rough timeline</a></li>
  <li><a href="#sec-poly" id="toc-sec-poly" class="nav-link" data-scroll-target="#sec-poly">Polynomials</a></li>
  <li><a href="#sec-fry02" id="toc-sec-fry02" class="nav-link" data-scroll-target="#sec-fry02">Polynomial regression</a></li>
  <li><a href="#revisiting-sec-comm" id="toc-revisiting-sec-comm" class="nav-link" data-scroll-target="#revisiting-sec-comm">Revisiting Section&nbsp;6.5</a></li>
  <li><a href="#sec-peershare06" id="toc-sec-peershare06" class="nav-link" data-scroll-target="#sec-peershare06">Q&amp;A and Task 04</a></li>
  </ul></li>
  <li><a href="#march-31st" id="toc-march-31st" class="nav-link" data-scroll-target="#march-31st">March 31st</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-7" id="toc-rough-timeline-7" class="nav-link" data-scroll-target="#rough-timeline-7">Rough timeline</a></li>
  <li><a href="#sec-mods" id="toc-sec-mods" class="nav-link" data-scroll-target="#sec-mods">Modeller vs Designer</a></li>
  <li><a href="#sec-multi" id="toc-sec-multi" class="nav-link" data-scroll-target="#sec-multi">Multiple comparisons</a></li>
  <li><a href="#sec-peershare07" id="toc-sec-peershare07" class="nav-link" data-scroll-target="#sec-peershare07">Q&amp;A and peer-share</a></li>
  </ul></li>
  <li><a href="#april-4th" id="toc-april-4th" class="nav-link" data-scroll-target="#april-4th">April 4th</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-8" id="toc-rough-timeline-8" class="nav-link" data-scroll-target="#rough-timeline-8">Rough timeline</a></li>
  <li><a href="#sec-multi02" id="toc-sec-multi02" class="nav-link" data-scroll-target="#sec-multi02">Multiple Comparisons</a></li>
  <li><a href="#sec-df" id="toc-sec-df" class="nav-link" data-scroll-target="#sec-df">Degrees of Freedom</a></li>
  <li><a href="#sec-randomeff" id="toc-sec-randomeff" class="nav-link" data-scroll-target="#sec-randomeff">Introducing random effcets</a></li>
  <li><a href="#sec-peershare08" id="toc-sec-peershare08" class="nav-link" data-scroll-target="#sec-peershare08">Q&amp;A/peer-share/Task 05</a></li>
  </ul></li>
  <li><a href="#april-7th" id="toc-april-7th" class="nav-link" data-scroll-target="#april-7th">April 7th</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-9" id="toc-rough-timeline-9" class="nav-link" data-scroll-target="#rough-timeline-9">Rough timeline</a></li>
  <li><a href="#sec-casestudies" id="toc-sec-casestudies" class="nav-link" data-scroll-target="#sec-casestudies">Case studies</a></li>
  <li><a href="#sec-edterms" id="toc-sec-edterms" class="nav-link" data-scroll-target="#sec-edterms">The Terminology of Experimental Design</a></li>
  <li><a href="#using-r-to-design-an-experiment" id="toc-using-r-to-design-an-experiment" class="nav-link" data-scroll-target="#using-r-to-design-an-experiment">Using <code>R</code> to design an experiment</a></li>
  <li><a href="#sec-peershare09" id="toc-sec-peershare09" class="nav-link" data-scroll-target="#sec-peershare09">Q&amp;A and peer-share</a></li>
  </ul></li>
  <li><a href="#april-11th" id="toc-april-11th" class="nav-link" data-scroll-target="#april-11th">April 11th</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-10" id="toc-rough-timeline-10" class="nav-link" data-scroll-target="#rough-timeline-10">Rough timeline</a></li>
  <li><a href="#sec-distributions" id="toc-sec-distributions" class="nav-link" data-scroll-target="#sec-distributions">Distributions</a></li>
  <li><a href="#sec-glms" id="toc-sec-glms" class="nav-link" data-scroll-target="#sec-glms">Introduction to Generalised Linear Models (GLMs)</a></li>
  <li><a href="#sec-recap" id="toc-sec-recap" class="nav-link" data-scroll-target="#sec-recap">Recap</a></li>
  </ul></li>
  <li><a href="#april-28th" id="toc-april-28th" class="nav-link" data-scroll-target="#april-28th">April 28th</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-11" id="toc-rough-timeline-11" class="nav-link" data-scroll-target="#rough-timeline-11">Rough timeline</a></li>
  <li><a href="#sec-distribution" id="toc-sec-distribution" class="nav-link" data-scroll-target="#sec-distribution">Guess the distribution</a></li>
  <li><a href="#sec-writingglm" id="toc-sec-writingglm" class="nav-link" data-scroll-target="#sec-writingglm">Writing out (G)LM(m)s</a></li>
  <li><a href="#sec-glmfit" id="toc-sec-glmfit" class="nav-link" data-scroll-target="#sec-glmfit">Assessing model fit</a></li>
  <li><a href="#sec-peershare11" id="toc-sec-peershare11" class="nav-link" data-scroll-target="#sec-peershare11">Q&amp;A and peer-share</a></li>
  </ul></li>
  <li><a href="#may-2nd" id="toc-may-2nd" class="nav-link" data-scroll-target="#may-2nd">May 2nd</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-12" id="toc-rough-timeline-12" class="nav-link" data-scroll-target="#rough-timeline-12">Rough timeline</a></li>
  <li><a href="#sec-recap02" id="toc-sec-recap02" class="nav-link" data-scroll-target="#sec-recap02">Recap</a></li>
  <li><a href="#sec-mle" id="toc-sec-mle" class="nav-link" data-scroll-target="#sec-mle">Maximum Likelihood Estimation (MLE)</a></li>
  </ul></li>
  <li><a href="#may-5th" id="toc-may-5th" class="nav-link" data-scroll-target="#may-5th">May 5th</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-13" id="toc-rough-timeline-13" class="nav-link" data-scroll-target="#rough-timeline-13">Rough timeline</a></li>
  <li><a href="#sec-bayes" id="toc-sec-bayes" class="nav-link" data-scroll-target="#sec-bayes">Introduction to Bayesian Statistics</a></li>
  <li><a href="#sec-bayesestudies" id="toc-sec-bayesestudies" class="nav-link" data-scroll-target="#sec-bayesestudies">Casestudies from a Bayesian viewpoint</a></li>
  <li><a href="#sec-recap03" id="toc-sec-recap03" class="nav-link" data-scroll-target="#sec-recap03">Recap</a></li>
  </ul></li>
  <li><a href="#may-9th" id="toc-may-9th" class="nav-link" data-scroll-target="#may-9th">May 9th</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-14" id="toc-rough-timeline-14" class="nav-link" data-scroll-target="#rough-timeline-14">Rough timeline</a></li>
  <li><a href="#sec-model" id="toc-sec-model" class="nav-link" data-scroll-target="#sec-model">Modelling data, where to start</a></li>
  <li><a href="#sec-ica" id="toc-sec-ica" class="nav-link" data-scroll-target="#sec-ica">Interactive Coding Assessments (ICA)</a></li>
  </ul></li>
  <li><a href="#may-12th" id="toc-may-12th" class="nav-link" data-scroll-target="#may-12th">May 12th</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-15" id="toc-rough-timeline-15" class="nav-link" data-scroll-target="#rough-timeline-15">Rough timeline</a></li>
  <li><a href="#sec-multivariate" id="toc-sec-multivariate" class="nav-link" data-scroll-target="#sec-multivariate">Introduction to Multivariate Data Analysis</a></li>
  <li><a href="#sec-pca" id="toc-sec-pca" class="nav-link" data-scroll-target="#sec-pca">Principal Component Analysis (PCA)</a></li>
  <li><a href="#sec-eigenfaces" id="toc-sec-eigenfaces" class="nav-link" data-scroll-target="#sec-eigenfaces">Eigenfaces</a></li>
  </ul></li>
  <li><a href="#may-16th" id="toc-may-16th" class="nav-link" data-scroll-target="#may-16th">May 16th</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-16" id="toc-rough-timeline-16" class="nav-link" data-scroll-target="#rough-timeline-16">Rough timeline</a></li>
  <li><a href="#sec-misc" id="toc-sec-misc" class="nav-link" data-scroll-target="#sec-misc">Miscellaneous activities</a></li>
  <li><a href="#sec-mds" id="toc-sec-mds" class="nav-link" data-scroll-target="#sec-mds">Multidimensional Scaling</a></li>
  <li><a href="#sec-mdscasestudies" id="toc-sec-mdscasestudies" class="nav-link" data-scroll-target="#sec-mdscasestudies">Casestudies</a></li>
  </ul></li>
  <li><a href="#may-19th" id="toc-may-19th" class="nav-link" data-scroll-target="#may-19th">May 19th</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-17" id="toc-rough-timeline-17" class="nav-link" data-scroll-target="#rough-timeline-17">Rough timeline</a></li>
  <li><a href="#sec-clustering" id="toc-sec-clustering" class="nav-link" data-scroll-target="#sec-clustering"><em>Grouping</em> multivariate data</a></li>
  <li><a href="#sec-div" id="toc-sec-div" class="nav-link" data-scroll-target="#sec-div">Divisive vs.&nbsp;Hierarchical</a></li>
  <li><a href="#sec-clusterducks" id="toc-sec-clusterducks" class="nav-link" data-scroll-target="#sec-clusterducks">Clusterducks</a></li>
  <li><a href="#sec-peershare17" id="toc-sec-peershare17" class="nav-link" data-scroll-target="#sec-peershare17">Q&amp;A and peer-share</a></li>
  </ul></li>
  <li><a href="#may-23rd" id="toc-may-23rd" class="nav-link" data-scroll-target="#may-23rd">May 23rd</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-18" id="toc-rough-timeline-18" class="nav-link" data-scroll-target="#rough-timeline-18">Rough timeline</a></li>
  <li><a href="#sec-vizmult" id="toc-sec-vizmult" class="nav-link" data-scroll-target="#sec-vizmult">Multivariate data</a></li>
  <li><a href="#sec-mdscasestudies02" id="toc-sec-mdscasestudies02" class="nav-link" data-scroll-target="#sec-mdscasestudies02">Casestudies</a></li>
  <li><a href="#sec-modelbasedmult" id="toc-sec-modelbasedmult" class="nav-link" data-scroll-target="#sec-modelbasedmult">Model based multivariate analysis</a></li>
  </ul></li>
  <li><a href="#may-26th" id="toc-may-26th" class="nav-link" data-scroll-target="#may-26th">May 26th</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-19" id="toc-rough-timeline-19" class="nav-link" data-scroll-target="#rough-timeline-19">Rough timeline</a></li>
  <li><a href="#sec-discrim" id="toc-sec-discrim" class="nav-link" data-scroll-target="#sec-discrim">Linear Discriminant Analysis (LDA)</a></li>
  <li><a href="#sec-caselda" id="toc-sec-caselda" class="nav-link" data-scroll-target="#sec-caselda">Casestudies</a></li>
  <li><a href="#sec-peershare19" id="toc-sec-peershare19" class="nav-link" data-scroll-target="#sec-peershare19">Q&amp;A and peer-share</a></li>
  </ul></li>
  <li><a href="#may-30th" id="toc-may-30th" class="nav-link" data-scroll-target="#may-30th">May 30th</a>
  <ul class="collapse">
  <li><a href="#rough-timeline-20" id="toc-rough-timeline-20" class="nav-link" data-scroll-target="#rough-timeline-20">Rough timeline</a></li>
  </ul></li>
  <li><a href="#sec-solutions" id="toc-sec-solutions" class="nav-link" data-scroll-target="#sec-solutions">Activity solutions</a>
  <ul class="collapse">
  <li><a href="#suggested-solution-to-sec-gorilla" id="toc-suggested-solution-to-sec-gorilla" class="nav-link" data-scroll-target="#suggested-solution-to-sec-gorilla">Suggested solution to Section&nbsp;2.2</a></li>
  <li><a href="#suggested-solution-to-sec-honesty" id="toc-suggested-solution-to-sec-honesty" class="nav-link" data-scroll-target="#suggested-solution-to-sec-honesty">Suggested solution to Section&nbsp;3.5</a></li>
  <li><a href="#suggested-solution-to-sec-draw" id="toc-suggested-solution-to-sec-draw" class="nav-link" data-scroll-target="#suggested-solution-to-sec-draw">Suggested solution to Section&nbsp;6.3</a></li>
  <li><a href="#suggested-solution-to-sec-fry" id="toc-suggested-solution-to-sec-fry" class="nav-link" data-scroll-target="#suggested-solution-to-sec-fry">Suggested solution to Section&nbsp;6.4.2</a></li>
  <li><a href="#code-example-to-support-sec-mods-sec-ss" id="toc-code-example-to-support-sec-mods-sec-ss" class="nav-link" data-scroll-target="#code-example-to-support-sec-mods-sec-ss">Code example to support Section&nbsp;8.2 / Section&nbsp;8.2.2</a></li>
  <li><a href="#suggested-solution-to-sec-ss" id="toc-suggested-solution-to-sec-ss" class="nav-link" data-scroll-target="#suggested-solution-to-sec-ss">Suggested solution to Section&nbsp;8.2.2</a></li>
  <li><a href="#suggested-reasoning-for-the-sec-distributions-chooks" id="toc-suggested-reasoning-for-the-sec-distributions-chooks" class="nav-link" data-scroll-target="#suggested-reasoning-for-the-sec-distributions-chooks">Suggested reasoning for the Section&nbsp;11.2 chooks</a></li>
  <li><a href="#suggested-solutions-to-sec-distribution" id="toc-suggested-solutions-to-sec-distribution" class="nav-link" data-scroll-target="#suggested-solutions-to-sec-distribution">Suggested solutions to Section&nbsp;12.2</a></li>
  <li><a href="#scoring-for-quiz-in-sec-bayes" id="toc-scoring-for-quiz-in-sec-bayes" class="nav-link" data-scroll-target="#scoring-for-quiz-in-sec-bayes">Scoring for quiz in Section&nbsp;14.2</a></li>
  </ul></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
</nav>
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">





<p><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"></p>
<section id="notes" class="level1">
<h1>Notes</h1>
<p>All BIOSCI 738 lectures require your <strong>active involvement</strong>! See the <a href="https://biosci738.github.io/BIOSCI738/key-information.html#course-policies">course policies</a> for further infomation.</p>
<p>Throughout this runsheet you’ll find a number of different callout boxes:</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
this box will contain the expected learning objectives and/or outcomes for the material
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
this box will contain some useful information
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> this box will ask you discuss/write/sketch an answer within your randomly allocated group
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
<section id="first-things-first" class="level3">
<h3 class="anchored" data-anchor-id="first-things-first">First things first…</h3>
<p>Note the <strong>randomly allocated group</strong> mentioned above. Each lecture I’ll have run the following <code>R</code> function that randomly allocates you into groups. <strong>It’s your responsibility to get into them, just look for the cows!</strong></p>
<center>
<a href="https://biosci738.github.io/cowstats/"><img src="https://raw.githubusercontent.com/BIOSCI738/cowstats/main/img/coxhex.png" alt="BIOSCI738 hex" width="200" height="200"></a>
</center>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>group_allocation <span class="ot">&lt;-</span> <span class="cf">function</span>(names, seed, <span class="at">groups =</span> <span class="dv">9</span> ){</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set.seed</span>(seed)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> <span class="fu">sample</span>(names, <span class="at">replace =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">split</span>(tmp, <span class="fu">cut</span>(<span class="fu">seq_along</span>(tmp), groups, <span class="at">labels =</span> <span class="fu">paste</span>(<span class="st">"Group"</span>, <span class="dv">1</span><span class="sc">:</span>groups))) </span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Accessing activities
</div>
</div>
<div class="callout-body-container callout-body">
<p>Scan the QR code below, depending on the activity the site will update with what I want to share (e.g., this runsheet etc.)</p>
<center>
<p>
<a href="https://dub.sh/biosci738" target="_blank">dub.sh/biosci738</a>
</p>
<img src="runsheets/img/qr_biosci738.svg" alt="dub.sh.biosci738">
</center>
</div>
</div>
</section>
</section>
<section id="march-10th" class="level1">
<h1>March 10th</h1>
<section id="rough-timeline" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline">Rough timeline</h2>
<section id="am---1055-am" class="level3">
<h3 class="anchored" data-anchor-id="am---1055-am"><strong>09:05 am - 10:55 am</strong></h3>
<ul>
<li><p><strong>09:05–10.25 am</strong> Gorilla Gambit (<a href="#sec-gorilla" class="quarto-xref">Section&nbsp;2.2</a>)</p></li>
<li><p><strong>10.25–10.55 am</strong> Group Code of Conduct (<a href="#sec-coc" class="quarto-xref">Section&nbsp;2.3</a>)</p>
<ul>
<li>Classwide discussion (<a href="#sec-coc01" class="quarto-xref">Section&nbsp;2.3.1</a>)</li>
<li>Group activity (<a href="#sec-coc02" class="quarto-xref">Section&nbsp;2.3.2</a>)</li>
</ul></li>
<li><p><strong>09:55–10:05 am</strong> Break</p></li>
<li><p><strong>10.05–10.15 am</strong> Introduction to BIOSCI 738 (<a href="#sec-intro" class="quarto-xref">Section&nbsp;2.4</a>)</p></li>
<li><p><strong>10.15–10.40 am</strong> AI in the BOISCI 738 classroom (<a href="#sec-ai" class="quarto-xref">Section&nbsp;2.5</a>)</p>
<ul>
<li>Classwide discussion (<a href="#sec-ai01" class="quarto-xref">Section&nbsp;2.5.1</a>)</li>
<li>Group activity (<a href="#sec-ai02" class="quarto-xref">Section&nbsp;2.5.2</a>)</li>
</ul></li>
<li><p><strong>10.40–10.55 am</strong> Q&amp;A and peer-share (<a href="#sec-peershare01" class="quarto-xref">Section&nbsp;2.6</a>)</p></li>
</ul>
</section>
<section id="randomly-allocated-groups" class="level3">
<h3 class="anchored" data-anchor-id="randomly-allocated-groups">Randomly allocated groups</h3>
<p><strong>It’s your responsibility to get into these, just look for the cows!</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">group_allocation</span>(names, <span class="at">seed =</span> <span class="dv">100325</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Group 1`
[1] "Yutao Yang"   "Lana Najar"   "Emma Akeroyd" "Shoule Du"   

$`Group 2`
[1] "Robert Mihok" "Laura Munck"  "Alex Johnson" "Sunny Chen"  

$`Group 3`
[1] "Julia Thain"        "Xinyue Zhang"       "Zichang Li"        
[4] "Barnaby Clegg-Shaw"

$`Group 4`
[1] "Pikitangarangi Ratapu"        "Zhixiao Zhang"               
[3] "Anqi Cheng"                   "Antonia Schebek-Fuerstenberg"

$`Group 5`
[1] "Congkai Meng" "Jiaxin Liu"   "Yingyi Wang" 

$`Group 6`
[1] "Tongzheng Yao"    "Jianing Chen"     "Charlotte Sutton" "YOGAPRIYA S"     

$`Group 7`
[1] "Ziyi Wang"      "Yutong Wang"    "He Li"          "Bayu Pitandoyo"

$`Group 8`
[1] "Karie Labidon" "Happy Liang"   "Angel Loh"     "Yue Ma"       

$`Group 9`
[1] "Leyang Liu"  "Keyi Sun"    "Yi Han"      "Shuxiang He"</code></pre>
</div>
</div>
</section>
</section>
<section id="sec-gorilla" class="level2">
<h2 class="anchored" data-anchor-id="sec-gorilla">Gorilla Gambit</h2>
<p>Group activity roles include</p>
<ul>
<li><strong>A runner</strong>, responsible for running (<em>or at least getting the next instruction from Charlotte</em>)</li>
<li><strong>A coder</strong>, responsible for implementing the code for the group this does <strong>NOT</strong> mean the sole coder, rather the person responsible for running any suggested code etc.</li>
<li><strong>A timekeeper</strong>, responsible for managing time for the group, making sure you stay on track</li>
<li><strong>A facilitator</strong>, responsible for seeking out opinions from all group members and ensuring that everyone has the opportunity to contribute</li>
</ul>
<p>In your groups <strong>quickly</strong> determine who is filling what role by order of next birthdays: the person with the next birthday will be the <strong>runner</strong>, the next birthday after that will be the <strong>coder</strong>, the third birthday will be <strong>timekeeper</strong>, and if your group has four members, the fourth member will be the <strong>facilitator</strong>.</p>
<p>Once you have allocated group roles your <strong>runner</strong> should come to me to retrieve the first instruction.</p>
</section>
<section id="sec-coc" class="level2">
<h2 class="anchored" data-anchor-id="sec-coc">Group Code of Conduct</h2>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
A classwide agreed group working Code of Conduct
</div>
</div>
<div class="callout-body-container callout-body">
<p>Upon the completion of this activity I will summarise the main themes/suggestions from <a href="#sec-coc02" class="quarto-xref">Section&nbsp;2.3.2</a> (<em>that I deem appropriate</em>). This will become the class-agreed group working Code of Conduct that you are expected to adhere to during each activity.</p>
</div>
</div>
<section id="sec-coc01" class="level3">
<h3 class="anchored" data-anchor-id="sec-coc01">Why I think this is important</h3>
<p>As a student of University of Auckland student, you are responsible for understanding and abiding by the requirements of the <a href="https://www.auckland.ac.nz/en/students/forms-policies-and-guidelines/student-policies-and-guidelines/student-charter.html">Student Charter</a>.</p>
<p>In this activity we’re going to be talking about my and your expectation when working in a group during this class, see <a href="https://biosci738.github.io/BIOSCI738/key-information.html#participation">this section of the course guide for further details</a>.</p>
<p>A Code of Conduct is not just a strange thing the university make you sign. They are a large part of many professional and research-focused bodies beyond university. The following lists just a few examples of societies or institutes you will likely come across during a biostats career in NZ.</p>
<ul>
<li><a href="https://newzealandecology.org/about/council-members/society-rules">The NZ Ecological Society CoC</a></li>
<li><a href="https://www.stats.org.nz/code-of-conduct/">The NZ Statistical Association CoC</a></li>
<li><a href="https://www.doc.govt.nz/about-us/our-policies-and-plans/code-of-ethical-conduct/">DoCs Code of Ethical Conduct</a></li>
</ul>
</section>
<section id="sec-coc02" class="level3">
<h3 class="anchored" data-anchor-id="sec-coc02">Write your own</h3>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Propose your own code of conduct for group work in this class
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
<p>In your groups head to <a href="https://docs.google.com/document/d/1_mAmTSuFFiCV-LV8xOuWgduEjmPnmJpbsD2KXm2v68I/edit?usp=sharing"><strong>this shared Google Doc</strong></a>.</p>
<ol type="1">
<li><p>Follow the instructions given around assigning group role.</p></li>
<li><p>Your notetaker should then navigate to your groups’ section.</p></li>
<li><p>As a group discuss and write down your expectations of yourself and each other during group work activities. You may want to think about</p>
<ul>
<li>how to ensure all members feel welcome,</li>
<li>your expectations around being an active (<em>and equal</em>) contributor, and</li>
<li>your expectations around use of technologies during activities.</li>
</ul></li>
</ol>
</section>
</section>
<section id="sec-intro" class="level2">
<h2 class="anchored" data-anchor-id="sec-intro">Introduction to BIOSCI 738</h2>
<ul>
<li>📢 Call for class rep! 📢 First meeting <strong>25 March, 1-2pm</strong></li>
<li>The <a href="https://biosci738.github.io/BIOSCI738/">course guide</a>, how it works</li>
<li><a href="https://biosci738.github.io/BIOSCI738/key-information.html#assessment-summary">Assessments in BIOSCI 738</a></li>
</ul>
</section>
<section id="sec-ai" class="level2">
<h2 class="anchored" data-anchor-id="sec-ai">GenAI in the BIOSCI 738 classroom</h2>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Expectations of genAI use
</div>
</div>
<div class="callout-body-container callout-body">
<p>Upon the completion of this activity I will summarise the main themes/suggestions from <a href="#sec-ai02" class="quarto-xref">Section&nbsp;2.5.2</a> (<em>that I deem appropriate</em>). This will become the class-agreed group working Code of Conduct that you are expected to adhere to during each activity.</p>
</div>
</div>
<section id="sec-ai01" class="level3">
<h3 class="anchored" data-anchor-id="sec-ai01">Who’s heard of <em>bullshit</em>?</h3>
<blockquote class="blockquote">
<p><em>Recently, there has been considerable interest in large language models: machine learning systems which produce human-like text and dialogue. Applications of these systems have been plagued by persistent inaccuracies in their output; these are often called “AI hallucinations”. We argue that these falsehoods, and the overall activity of large language models, is better understood as bullshit in the sense explored by Frankfurt (On Bullshit, Princeton, 2005): the models are in an important way indifferent to the truth of their outputs</em></p>
</blockquote>
<p>This is an excerpt from the abstract of <span class="citation" data-cites="bullshit">Hicks, Humphries, and Slater (<a href="#ref-bullshit" role="doc-biblioref">2024</a>)</span>.</p>
<blockquote class="blockquote">
<p><em>If a student is confused about a concept, they can sit with ChatGPT and it will talk to them for hours about that particular concept.</em></p>
</blockquote>
<blockquote class="blockquote">
<p><em>It is a really great tool to create code but also a really great tool to prevent yourself from learning.</em></p>
</blockquote>
<blockquote class="blockquote">
<p><em>It’s teachers’ responsibility to motivate them and make such a problem that [students] are keen to solve and in a way that they actually would like to learn something and realize that they need these skills also in the future.</em></p>
</blockquote>
<blockquote class="blockquote">
<p><em>…there have always been so many ways of cheating, but I don’t think I’ve ever been aware of such an obvious, cheap, and easy way of cheating. Students can get [an AI tool] to answer any question I can ask them at the moment and therefore I have lost my ability to confidently assess any work that students hand in.</em></p>
</blockquote>
<blockquote class="blockquote">
<p><em>I think we need different kinds of professionals with different understandings of computing. Some need to be very deeply involved with how our programming languages work … others might only need some kind of overall understanding. They are not programmers by themselves, but they still should understand how software is produced.</em></p>
</blockquote>
<p>All the above are quotes garnered in <span class="citation" data-cites="instructor">Sheard et al. (<a href="#ref-instructor" role="doc-biblioref">2024</a>)</span>.</p>
<p>Remember back to last week when I asked you to complete a <em>pre-course</em> quiz or two? Below are a summary of your responses to the genAI related questions. These questions were, by design, worded in the same way those in <span class="citation" data-cites="robots">Prather et al. (<a href="#ref-robots" role="doc-biblioref">2023</a>)</span> (see Figure 3 of that paper for a summary of their findings).</p>
<section id="a-summary-of-your-responses" class="level4">
<h4 class="anchored" data-anchor-id="a-summary-of-your-responses">A summary of your responses</h4>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
</div>
<table class="caption-top table">
<caption>For programming assignments, I believe GenAI should be…</caption>
<thead>
<tr class="header">
<th style="text-align: left;">Answer</th>
<th style="text-align: right;">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Allowed/Disallowed</td>
<td style="text-align: right;">37</td>
</tr>
<tr class="even">
<td style="text-align: left;">Always allowed</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Always disallowed</td>
<td style="text-align: right;">3</td>
</tr>
</tbody>
</table>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="1440"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<table class="caption-top table">
<colgroup>
<col style="width: 3%">
<col style="width: 96%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Abbr</th>
<th style="text-align: left;">Question</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">A 6</td>
<td style="text-align: left;">It is unethical to auto-generate a solution for a given problem (or a large portion of it) and submit it without understanding it.</td>
</tr>
<tr class="even">
<td style="text-align: left;">A 7</td>
<td style="text-align: left;">It is unethical to auto-generate a solution for a given problem (or a large portion of it) and submit it after reading it and completely understanding it.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">A 8</td>
<td style="text-align: left;">It is unethical to provide code to GenAI tools and ask them to help fix a bug.</td>
</tr>
<tr class="even">
<td style="text-align: left;">A 9</td>
<td style="text-align: left;">It is unethical to ask GenAI tools to comment, tidy and improve the style of the code.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">A 10</td>
<td style="text-align: left;">It is unethical to auto-generate a solution even for small parts of the assignment.</td>
</tr>
<tr class="even">
<td style="text-align: left;">A 11</td>
<td style="text-align: left;">It is unethical to use GenAI tools to explain to you (step-by-step) how to solve a given problem.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">A 12</td>
<td style="text-align: left;">It is unethical to reuse large portions of code verbatim from any source (e.g., the course guide, another course, a peer, online help forums etc.).</td>
</tr>
<tr class="even">
<td style="text-align: left;">A 13</td>
<td style="text-align: left;">It is unethical to reuse code verbatim from any source (e.g., the course guide, another course, a peer, online help forums etc.).</td>
</tr>
</tbody>
</table>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="runsheets/ai_wordcloud.png" class="img-fluid figure-img"></p>
<figcaption><em>The theme of your responses</em></figcaption>
</figure>
</div>
<p><strong>Some specific answers I particularly liked or wanted to discuss</strong></p>
<blockquote class="blockquote">
<p>In my opinion, the controversy behind LLMs and other AIs is rather blown out of proportion, if people use it to do all their assignments for them it should be EASILY detectable for any marker/instructor…</p>
</blockquote>
<blockquote class="blockquote">
<p>I’m getting increasingly concerned by assignments at uni with the basic premise of “get chatgpt to generate this, then do [whatever is relevant for the assignment which is almost never critiquing the output]”…</p>
</blockquote>
<blockquote class="blockquote">
<p>I don’t enjoy the increasing prevalence and reliance on AI in academic settings and in day to day life due the large environmental impacts and blurry ethics…</p>
</blockquote>
<blockquote class="blockquote">
<p>GenAI’s kinda like a double-edged sword for jobs…AI’s the shortcut, but my brain’s still the GPS!</p>
</blockquote>
<blockquote class="blockquote">
<p>I have been using Grammarly (to my understanding, it’s also a generative AI),…</p>
</blockquote>
<blockquote class="blockquote">
<p>I’m worried that non-native English speakers using translation tools might be identified as AI…</p>
</blockquote>
<p><strong>Do you have anything else that you’d to mention about your experiences with GenAI tools?</strong></p>
</section>
</section>
<section id="sec-ai02" class="level3">
<h3 class="anchored" data-anchor-id="sec-ai02">Share and discuss</h3>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Discuss in your group how much you agree/disagree with the opinions and summaries presented above.
</div>
</div>
<div class="callout-body-container callout-body">
<p>If you are comfortable doing so you may wish to share some of your personal experiences. Feel free to expand on the answers you gave in the <em>pre-course</em> quiz.</p>
</div>
</div>
<p>Again in your groups head to <a href="https://docs.google.com/document/d/1_mAmTSuFFiCV-LV8xOuWgduEjmPnmJpbsD2KXm2v68I/edit?usp=sharing"><strong>the shared Google Doc</strong></a>.</p>
<ol type="1">
<li><p>As a group discuss and write down your expectations of yourself and each other with regards to use of GenAI for classroom activities, and assignment activities. You may want to think about</p>
<ul>
<li>how to ensure GenAI tools are use for the benefit of everyone’s learning, or</li>
<li>what your expectations are of your peers who use gen AI are.</li>
</ul></li>
</ol>
</section>
</section>
<section id="sec-peershare01" class="level2">
<h2 class="anchored" data-anchor-id="sec-peershare01">Q&amp;A and peer-share</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Share your approach to the diagnostic coding task
</div>
</div>
<div class="callout-body-container callout-body">
<p>Below is my solution to the diagnostic coding task. This is not the only solution! In your groups run through my approach and share your approach (especially if you got stuck in a few places)! Sharing and talking things through will make things clearer. Trust me! 😉</p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/dicots_proportions.csv"</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(url)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="fu">c</span>(<span class="st">"Calluna"</span>, <span class="st">"Treat"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>    <span class="do">## select only heather &amp; treatment cols</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(<span class="st">`</span><span class="at">Treat!</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>    <span class="do">## group by treatment so calcs are done by group</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise_all</span>(<span class="fu">list</span>(<span class="at">mean =</span> mean, <span class="at">sd =</span> sd)) <span class="sc">%&gt;%</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>    <span class="do">## cal mean and sd of each group</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">!</span><span class="st">`</span><span class="at">Treat!</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>    <span class="do">## flip the data frame to "long" format</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">separate</span>(name, <span class="fu">c</span>(<span class="cn">NA</span>, <span class="st">"year"</span>, <span class="st">"calc"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>    <span class="do">## extract and separate info from name column</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">year =</span> <span class="fu">as.numeric</span>(<span class="fu">str_remove</span>(year, <span class="st">"vulgaris"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>    <span class="do">## keep numeric year info only</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">"calc"</span>, <span class="at">values_from =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>    <span class="do">## data into wider format based on mean &amp; sd</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Treat!</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_replace</span>(<span class="st">`</span><span class="at">Treat!</span><span class="st">`</span>, <span class="st">"HB"</span>, <span class="st">"BH"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>    <span class="do">## change treatment label HB to BH</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Treat!</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">fct_relevel</span>(<span class="st">`</span><span class="at">Treat!</span><span class="st">`</span>,<span class="fu">c</span>(<span class="st">"C"</span>, <span class="st">"B"</span>, <span class="st">"H"</span>, <span class="st">"BH"</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>    <span class="do">## relevel treatment to help with legend ordering later on</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> year, <span class="at">y =</span> mean, <span class="at">group =</span> <span class="st">`</span><span class="at">Treat!</span><span class="st">`</span>)) <span class="sc">+</span> <span class="do">## set up plot</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">pch =</span> <span class="st">`</span><span class="at">Treat!</span><span class="st">`</span>)) <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">linetype =</span> <span class="st">`</span><span class="at">Treat!</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>    <span class="do">## add mean points &amp; lines</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">"Percentage cover"</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Year"</span>) <span class="sc">+</span> <span class="do">## axis labels</span></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin =</span> mean <span class="sc">-</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(<span class="dv">6</span>),</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ymax =</span> mean <span class="sc">+</span> sd<span class="sc">/</span><span class="fu">sqrt</span>(<span class="dv">6</span>)), <span class="at">width =</span> .<span class="dv">05</span>) <span class="sc">+</span></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>    <span class="do">## add error-bars note we need the standard error of the mean of the proportion</span></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">8</span>, <span class="dv">12</span>, <span class="dv">1</span>),</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>                       <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">2008</span>, <span class="dv">2012</span>, <span class="dv">1</span>),</span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>                       <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="fl">0.01</span>),</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>                       <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">08</span>, <span class="dv">12</span>)) <span class="sc">+</span> <span class="do">## match x-axis labels and limits</span></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">0.1</span>), <span class="at">labels =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">100</span>, <span class="dv">10</span>),</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>                       <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="fl">0.01</span>),</span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>                       <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.7</span>)) <span class="sc">+</span> <span class="do">## match y-axis labels and limits</span></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span> <span class="do">## closest in-built ggplot theme I could find</span></span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.title =</span> <span class="fu">element_blank</span>()) <span class="do">## remove legend title</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="march-14th" class="level1">
<h1>March 14th</h1>
<section id="rough-timeline-1" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-1">Rough timeline</h2>
<section id="pm---0355-pm" class="level3">
<h3 class="anchored" data-anchor-id="pm---0355-pm"><strong>02:05 pm - 03:55 pm</strong></h3>
<ul>
<li><p><strong>02:05–02.20 pm</strong> Activity &amp; AI expectations (<a href="#sec-coccompleted" class="quarto-xref">Section&nbsp;3.2</a>)</p></li>
<li><p><strong>02.20–02.45 pm</strong> Data visualization (<a href="#sec-viz" class="quarto-xref">Section&nbsp;3.3</a>)</p>
<ul>
<li>What <em>makes</em> a plot? (<a href="#sec-vibes" class="quarto-xref">Section&nbsp;3.3.1</a>)</li>
<li>Roast my plots (<a href="#sec-roast" class="quarto-xref">Section&nbsp;3.3.2</a>)</li>
</ul></li>
<li><p><strong>02:45–02:55 pm</strong> Break</p></li>
<li><p><strong>02.55–03.15 pm</strong> Good programming practice (<a href="#sec-prog_practice" class="quarto-xref">Section&nbsp;3.4</a>)</p>
<ul>
<li>Classwide discussion (<a href="#sec-prog_practice01" class="quarto-xref">Section&nbsp;3.4.1</a>)</li>
<li>Group activity (<a href="#sec-prog_practice02" class="quarto-xref">Section&nbsp;3.4.2</a>)</li>
</ul></li>
<li><p><strong>03.15–03.45 pm</strong> Reproducible / retracted analysis (<a href="#sec-honesty" class="quarto-xref">Section&nbsp;3.5</a>)</p></li>
<li><p><strong>03.45–03.55 pm</strong> Q&amp;A and peer-share (<a href="#sec-peershare02" class="quarto-xref">Section&nbsp;3.6</a>)</p></li>
</ul>
</section>
<section id="randomly-allocated-groups-1" class="level3">
<h3 class="anchored" data-anchor-id="randomly-allocated-groups-1">Randomly allocated groups</h3>
<p><strong>It’s your responsibility to get into these, just look for the cows!</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">group_allocation</span>(names, <span class="at">seed =</span> <span class="dv">1312</span>, <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Group 1`
[1] "Jiaxin Liu"            "Daneshajeya Jeyavalan" "Lana Najar"           
[4] "Julia Thain"          

$`Group 2`
[1] "Hnin Lwin"          "He Li"              "Ziyi Wang"         
[4] "Francesca Domanska"

$`Group 3`
[1] "Keyi Sun"     "Zichang Li"   "Yutong Wang"  "Alex Johnson"

$`Group 4`
[1] "Sunny Chen"     "Leyang Liu"     "Anqi Cheng"     "Bayu Pitandoyo"

$`Group 5`
[1] "Laura Munck"        "Charlotte Sutton"   "Barnaby Clegg-Shaw"
[4] "Shoule Du"         

$`Group 6`
[1] "Yi Han"       "Yue Ma"       "Robert Mihok"

$`Group 7`
[1] "Tongzheng Yao"         "Pikitangarangi Ratapu" "YOGAPRIYA S"          
[4] "Vivian Qian"          

$`Group 8`
[1] "Zhixiao Zhang" "Yutao Yang"    "Angel Loh"     "Congkai Meng" 

$`Group 9`
[1] "Shuxiang He"                  "Antonia Schebek-Fuerstenberg"
[3] "Xinyue Zhang"                 "Yingyi Wang"                 

$`Group 10`
[1] "Emma Akeroyd"  "Karie Labidon" "Happy Liang"   "Jianing Chen" </code></pre>
</div>
</div>
</section>
</section>
<section id="sec-coccompleted" class="level2">
<h2 class="anchored" data-anchor-id="sec-coccompleted">Activity &amp; AI expectations</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-users" aria-hidden="true"></i> Class rep!!
</div>
</div>
<div class="callout-body-container callout-body">
<p>BIOSCI 738 needs a class rep!</p>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
A classwide agreed group working Code of Conduct
</div>
</div>
<div class="callout-body-container callout-body">
<p>As decided by you in last lecture’s activity <a href="#sec-coc" class="quarto-xref">Section&nbsp;2.3</a>. I have <a href="https://biosci738.github.io/BIOSCI738/key-information.html#course-policies">updated the course guide accordingly</a>!</p>
</div>
</div>
<p><strong>Some points of note</strong></p>
<ol type="1">
<li>One thing that I think gets overlooked around the use of genAI (over and above preserving “academic integrity”) is that it’s often not the use of these tools that endangers academic integrity, more that the output generated (more often than not) does not answer the question posed.</li>
<li>Whilst using genAI tools to help your writing is fine for the most part, I would be particularly careful around overusing them (particularly for more technical content like stats inference). This is as often subtle differences in sentence structure changes the statistical meaning, which can make the inference incorrect.</li>
<li>Quite a few papers exist (up yo you to decide if these are reputable or not) stating that large language models have teaching and grading capabilities. This is true, however, typically these papers discuss how instructors use LLM capabilities to provide students with personalized learning opportunities. This requires expertly “training” specific models to ensure that specific learning objectives etc. are being met and given in the style specific to a particular course’s assessment. Personally, I would be very wary using genAI tools yourself to validate your learning, especially in regards to more complex material where common misconceptions infect the internet!</li>
<li>IMO genAI is just as overly verbose and often complicated in terms of generating code as it is renowned for in generating writing. For example, think back to the diagnostic coding task. I saw many submissions that included this syntax <code>"\\d{2}$"</code>. This is correct syntax (based on what it was used for) but I’d be keen for anyone who used it to explain what it’s doing!</li>
</ol>
<p>IMO these tools are here to stay and I do think that part of your future career will involve having to navigate working with peers/colleagues who may readily use them or choose not to. This is unavoidable and something I’d recommend starting to think about how you (they) deal with this if you’ve not already.</p>
</section>
<section id="sec-viz" class="level2">
<h2 class="anchored" data-anchor-id="sec-viz">Data visualization</h2>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Interpret, critique and communicate using visualization!
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
<section id="sec-vibes" class="level3">
<h3 class="anchored" data-anchor-id="sec-vibes">What <em>makes</em> a plot?</h3>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Scan the QR code below and based on <em>vibes</em> alone choose your favourite until I tell you to stop
</div>
</div>
<div class="callout-body-container callout-body">
<center>
<p>
<a href="https://dub.sh/biosci738" target="_blank">dub.sh/biosci738</a>
</p>
<img src="runsheets/img/qr_biosci738.svg" alt="dub.sh.biosci738">
</center>
</div>
</div>
</section>
<section id="sec-roast" class="level3">
<h3 class="anchored" data-anchor-id="sec-roast">Roast my plots</h3>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i>
</div>
</div>
<div class="callout-body-container callout-body">
<p>In your groups head to <a href="https://docs.google.com/document/d/1VVVii7_TQ4J1Yqu_aJ8yWe1i2DAyl6rtAH5AebA5XFQ/edit?usp=sharing"><strong>this shared Google Doc</strong></a> and as a group, <strong>roast</strong> (yes you can be as mean as you like, honestly) or admire a plot from <strong>one</strong> of the following publications<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a></p>
<ul>
<li><a href="https://esajournals.onlinelibrary.wiley.com/doi/abs/10.1002/eap.2475"><em>Discrete-space continuous-time models of marine mammal exposure to Navy sonar</em></a> (<span class="citation" data-cites="whales">Jones-Todd et al. (<a href="#ref-whales" role="doc-biblioref">2021</a>)</span>)</li>
<li><a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/ece3.11005"><code>stelfi</code>: <em>An <code>R</code> package for fitting Hawkes and log-Gaussian Cox point process models</em></a> (<span class="citation" data-cites="stelfi">Jones-Todd and Helsdingen (<a href="#ref-stelfi" role="doc-biblioref">2024</a>)</span>)</li>
<li><a href="https://eprints.gla.ac.uk/193326/7/193326.pdf">Identifying prognostic structural features in tissue sections of colon cancer patients using point pattern analysis</a> (<span class="citation" data-cites="cancer">Jones-Todd et al. (<a href="#ref-cancer" role="doc-biblioref">2018</a>)</span>)</li>
</ul>
</div>
</div>
</section>
</section>
<section id="sec-prog_practice" class="level2">
<h2 class="anchored" data-anchor-id="sec-prog_practice">Good programming practice</h2>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Abide by good programming practices to avoid future heartaches!
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
<section id="sec-prog_practice01" class="level3">
<h3 class="anchored" data-anchor-id="sec-prog_practice01">IMO</h3>
<p>Following <a href="https://biosci738.github.io/BIOSCI738/accuracy-and-honesty.html#good-coding-practice">this section of the course guide</a> let’s talk about what good programming practice looks like in this course. For a more in-depth and general discussion I recommend <span class="citation" data-cites="goodprogramming">Wilson et al. (<a href="#ref-goodprogramming" role="doc-biblioref">2017</a>)</span>.</p>
<section id="you-should-always-start-with-a-clean-work-space." class="level4">
<h4 class="anchored" data-anchor-id="you-should-always-start-with-a-clean-work-space.">You should always start with a <strong>clean work space</strong>.</h4>
<p>Honestly, I think the default RStudio behaviour of restoring <code>.RData</code> files etc. just makes everyone lazy…</p>
</section>
<section id="devise-a-workflow" class="level4">
<h4 class="anchored" data-anchor-id="devise-a-workflow">Devise a workflow</h4>
<p>Remember the pre-course tasks? I mentioned the following</p>
<blockquote class="blockquote">
<p>During this course, very likely in other courses you’ll be taking this semester and in your future careers you will have to deal with many different datasets, wrangle “dirty” data and deal with data from different sources (at the very least). The key thing is to ensure that ANY ANALYSIS YOU CARRY OUT is TRANSPARENT and FULLY REPRODICIBLE (either for your peers or future you). This is where setting good foundations and devising a well-thought-out workflow is imperative!</p>
</blockquote>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>A summary of your responses</caption>
<colgroup>
<col style="width: 96%">
<col style="width: 3%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Answer</th>
<th style="text-align: right;">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">The tasks have been bundled into a project and I have a whole GitHub repo giving a step-by-step show-and-tell of the answers</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">WTF is a workspace</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">I kept on forgetting to read in the new data and got very frustrated when my calculations were incorrect despite doing the correct thing</td>
<td style="text-align: right;">7</td>
</tr>
<tr class="even">
<td style="text-align: left;">I also have all of my othe work I have carried out in R there for safe keeping</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">My workspace is full of all the objects I created for the diagnostic tasks\</td>
<td style="text-align: right;">8</td>
</tr>
<tr class="even">
<td style="text-align: left;">I have multiple different RStudio sessions running</td>
<td style="text-align: right;">11</td>
</tr>
<tr class="odd">
<td style="text-align: left;">My workspace is full of all the objects I created for this diagnostic tasks but I remaned them sensibly</td>
<td style="text-align: right;">11</td>
</tr>
<tr class="even">
<td style="text-align: left;">My workspace only has objects relevant to the last question I attempted</td>
<td style="text-align: right;">12</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Each question had a fresh new session</td>
<td style="text-align: right;">17</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="its-not-all-about-length" class="level4">
<h4 class="anchored" data-anchor-id="its-not-all-about-length">It’s not all about length</h4>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Lines of code written is NOT a measure of skill!
</div>
</div>
<div class="callout-body-container callout-body">
<p>Some people thing that a writing a large number of lines of code demonstrates prowess. It does not. Surely we’ve all added nonsense to essays to “fill up” the word count!</p>
<p>On the over hand some people strive for carrying out operations in the fewest number of lines possible. This typically makes their work impossible to follow!</p>
</div>
</div>
<p>I recommend finding a spot you’re comfortable in between the following code snippets.<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a> Most importantly keep your style readable &amp; consistent!</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grepl</span>(<span class="st">"^[[:alnum:]._-]+@[[:alnum:].-]+[:alnum:]+$"</span>, <span class="fu">c</span>(<span class="st">"larry@gmail.com"</span>, <span class="st">"larry-sally@sally.com"</span>, <span class="st">"larry@sally.larry.com"</span>, <span class="st">"test@test.com."</span>, <span class="st">"charlottejones-todd"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  TRUE  TRUE  TRUE FALSE FALSE</code></pre>
</div>
</div>
<p>or</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>email_addresses <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"larry@gmail.com"</span>, <span class="st">"larry-sally@sally.com"</span>, <span class="st">"larry@sally.larry.com"</span>, <span class="st">"test@test.com."</span>, <span class="st">"charlottejones-todd"</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>contain_at <span class="ot">&lt;-</span> <span class="cf">function</span>(x){</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grep</span>(<span class="st">"@"</span>, x)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="fu">contain_at</span>(email_addresses)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>correct_email <span class="ot">&lt;-</span> email_addresses[idx]</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>correct_email</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "larry@gmail.com"       "larry-sally@sally.com" "larry@sally.larry.com"
[4] "test@test.com."       </code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>contain_notrailing <span class="ot">&lt;-</span> <span class="cf">function</span>(x){</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grep</span>(<span class="st">"^[:alnum:]+"</span>, x)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>idx01 <span class="ot">&lt;-</span> <span class="fu">contain_notrailing</span>(correct_email)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>final_correct_email <span class="ot">&lt;-</span> correct_email[idx01]</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>final_correct_email</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "larry@gmail.com"       "larry-sally@sally.com" "larry@sally.larry.com"</code></pre>
</div>
</div>
</section>
<section id="keep-the-inside-thoughts-inside" class="level4">
<h4 class="anchored" data-anchor-id="keep-the-inside-thoughts-inside">Keep the <em>inside</em> thoughts <em>inside</em></h4>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
An <code>R</code> script (or equivalent) is a roadmap to your work.
</div>
</div>
<div class="callout-body-container callout-body">
<p>You should present the cleanest most direct route you can!</p>
</div>
</div>
<p>I recommend the latter approach to the second task in <a href="#sec-gorilla" class="quarto-xref">Section&nbsp;2.2</a> (if you were going to pass on your solution that is). It’s not that each step shouldn’t be carried out. On the contrary, exploring your data via printing and plotting it is very important! But when you have a solution, pare down your script! no need to take everyone on your journey.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="do">## read in data</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/BMI.csv"</span>)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="do">## printing data</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(data)</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(data) <span class="do">##view opens up new window</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>bmi</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Create new data object</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>BMIdata <span class="ot">&lt;-</span> data<span class="sc">$</span>bmi</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>BMIdata</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="do">## plot data</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(BMIdata)</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="do">## calculate mean</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>mean <span class="ot">&lt;-</span> <span class="fu">mean</span>(BMIdata)</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(mean)</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a><span class="do">## round</span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fu">mean</span>(BMIdata))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>vs</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/BMI.csv"</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fu">mean</span>(data<span class="sc">$</span>bmi))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="make-it-easy-for-me-to-give-you-marks" class="level4">
<h4 class="anchored" data-anchor-id="make-it-easy-for-me-to-give-you-marks">Make it EASY for me to give you marks</h4>
<ul>
<li><strong>Think</strong> about your audience!</li>
<li>Ensure your code is <strong>reproducible</strong>!</li>
<li><strong>Do not include any irrelevant or overly verbose output</strong>; this makes it difficult to find and therefore award the relevant components!</li>
<li>Keep your code <strong>tidy</strong> and your plots <strong>neat</strong> and <strong>professional</strong>. For example, it’s very useful for the reader if you use informative, readable axis labels rather than allowing the default behaviour of printing the R object name.</li>
</ul>
</section>
</section>
<section id="sec-prog_practice02" class="level3">
<h3 class="anchored" data-anchor-id="sec-prog_practice02">Your own experience</h3>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i>
</div>
</div>
<div class="callout-body-container callout-body">
<p>Again, head to <a href="https://docs.google.com/document/d/1VVVii7_TQ4J1Yqu_aJ8yWe1i2DAyl6rtAH5AebA5XFQ/edit?usp=sharing"><strong>this shared Google Doc</strong></a> and based on personal experience (or otherwise) detail a scenario where you have <em>dug yourself into a hole</em> (i.e., faced avoidable issues) as you’ve not fully abided by best programming practices. Also outline how you plan to avoid this in future.</p>
</div>
</div>
</section>
</section>
<section id="sec-honesty" class="level2">
<h2 class="anchored" data-anchor-id="sec-honesty"><em>Reproducible</em> analysis</h2>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Exploring data
</div>
</div>
<div class="callout-body-container callout-body">
<p>Building on <a href="#sec-gorilla" class="quarto-xref">Section&nbsp;2.2</a> here you’ll be carrying out some basic EDA and interpreting hypothesis tests.</p>
</div>
</div>
<p>In this task you’re going to be using <a href="https://canvas.auckland.ac.nz/courses/120054/files/folder/data?preview=14841854">these data</a> collected as part of a (now retracted) study (<span class="citation" data-cites="retracted">Shu et al. (<a href="#ref-retracted" role="doc-biblioref">2012</a>)</span>)<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a>.</p>
<blockquote class="blockquote">
<p>The data comes from a study that claims to show that people are less likely to act dishonestly when they sign an honesty pledge at the top of a form rather than at the bottom of a form. Participants received a worksheet with 20 math puzzles and were offered $1 for each puzzle they (reported to have) solved correctly within 5 minutes. After the 5 minutes passed, participants were asked to count how many puzzles they solved correctly and then throw away their worksheets. The goal was to mislead participants into thinking that the experimenter could not observe their true performance, when in fact they could because each worksheet had a unique identifier. Thus, participants could cheat (and earn more money) without fear of being caught, while the researchers could observe how much each participant had cheated. Participants then completed a “tax” form reporting how much money they had earned, and also how much time and money they spent coming to the lab. The experimenters partially compensated participants for those costs.</p>
</blockquote>
<blockquote class="blockquote">
<p>The paper reported very large effects. Signing at the top vs.&nbsp;the bottom lowered the share of people over-reporting their math puzzle performance from 79% to 37% (p = .0013), and lowered the average amount of over-reporting from 3.94 puzzles to 0.77 puzzles (p &lt; .00001). Similarly, it nearly halved the average amount of claimed commuting expenses, from $9.62 to $5.27 (p = .0014).</p>
</blockquote>
<p><strong>Variables of interest</strong></p>
<ul>
<li><code>Cond</code>: <code>= 0</code>, No signature; <code>= 1</code>, Signed at the top; and <code>= 2</code>, Signed at the bottom.</li>
<li><code>CheatedOnMatrixTax</code>: <code>= 0</code>, participants didn’t <em>cheat</em>/over-report their maths puzzle performance; and <code>= 1</code>, participants did <em>cheat</em>/over-report their maths puzzle performance.</li>
<li><code>OverReport</code>: the amount of puzzles solved participants over-reported by.</li>
<li><code>SumDeduction</code>: the amount of claimed commuting expenses $US (corrected for the <em>true</em> amount).</li>
</ul>
<p>As a group:</p>
<ol type="1">
<li><p>Discuss what to the presented results indicate/claim?</p></li>
<li><p>Use the variables listed above and reproduce the results quoted above using the appropriate statistical analysis/test. How easy was it to replicate these results given the way the results were presented? How might you improve the way the results were presented?</p></li>
<li><p>There is an additional column of interest <code>flag</code>; this is a binary variable that indicates if the observations were considered dodgy (i.e., fraudulent) or not. Create a visualization/carry out some analysis that <em>could</em> be used as evidence for/against this belief.</p></li>
</ol>
</section>
<section id="sec-peershare02" class="level2">
<h2 class="anchored" data-anchor-id="sec-peershare02">Q&amp;A and peer-share</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> If you’ve not already got RStudio open, open it up. Show your workspace to your neighbour and justify its contents! Share any tips you have about ensuring an efficient workflow.
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
</section>
</section>
<section id="march-17th---guest-lecturer-dr.-liza-bolton" class="level1">
<h1>March 17th - Guest Lecturer Dr.&nbsp;Liza Bolton!</h1>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
By the end of today’s seminar, you will be able to:
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Recognise that while individuals (and societies) may have different opinions and approaches to ethics, we can use codes of conduct to support our professional ethical decision making.</li>
<li>Describe some of the values, principles, responsibilities, and standards from the Royal Society Te Apārangi Code of Professional Standards and Ethics in Science, Technology, and the Humanities.</li>
<li>Describe how ethical decisions can appear throughout different stages of study and research in biology.</li>
<li>Identify issues and provide advice for ethical decision-making in biology contexts.</li>
</ul>
</div>
</div>
<section id="rough-timeline-2" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-2">Rough timeline</h2>
<ul>
<li><strong>09:05–09:20 am</strong> Introduction (see <a href="#sec-slidesliza" class="quarto-xref">Section&nbsp;4.2</a>)</li>
<li><strong>09:20–09:55 am</strong> Your ethical experiences and codes of conduct (<a href="#sec-coc2liza" class="quarto-xref">Section&nbsp;4.2.1</a>)
<ul>
<li>Group discussion</li>
<li>Class discussion</li>
<li>Start on ethical data practices and introduction to the case studies (see slides) if time</li>
</ul></li>
<li><strong>09:55–10:05 am</strong> Break</li>
<li><strong>10.05–10.20 am</strong> Ethical data practices and introduction to the case studies (see slides)</li>
<li><strong>10.15–10.45 am</strong> Case studies (<a href="#sec-caseliza" class="quarto-xref">Section&nbsp;4.3</a>)
<ul>
<li>Groups work through the case studies</li>
<li>Class discussion</li>
</ul></li>
<li><strong>10.45–10.55 am</strong> Wrap-up (and Code of Conduct meme-off if anyone has enough time to make them)</li>
</ul>
<section id="randomly-allocated-groups-2" class="level3">
<h3 class="anchored" data-anchor-id="randomly-allocated-groups-2">Randomly allocated groups</h3>
<p><strong>It’s your responsibility to get into these, just look for the cows!</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">group_allocation</span>(names, <span class="at">seed =</span> <span class="dv">1992</span>, <span class="at">groups =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Group 1`
[1] "Alex Johnson"     "Charlotte Sutton" "Yi Han"           "Keyi Sun"        

$`Group 2`
[1] "Yutao Yang"            "Tongzheng Yao"         "Daneshajeya Jeyavalan"
[4] "He Li"                

$`Group 3`
[1] "Bayu Pitandoyo" "Angel Loh"      "Lana Najar"     "Zichang Li"    

$`Group 4`
[1] "Sunny Chen"    "Anqi Cheng"    "Happy Liang"   "Karie Labidon"

$`Group 5`
[1] "Antonia Schebek-Fuerstenberg" "YOGAPRIYA S"                 
[3] "Hnin Lwin"                    "Leyang Liu"                  

$`Group 6`
[1] "Julia Thain"  "Emma Akeroyd" "Yingyi Wang" 

$`Group 7`
[1] "Jianing Chen" "Yue Ma"       "Yutong Wang"  "Laura Munck" 

$`Group 8`
[1] "Vivian Qian"   "Zhixiao Zhang" "Jiaxin Liu"    "Shoule Du"    

$`Group 9`
[1] "Shuxiang He"           "Xinyue Zhang"          "Pikitangarangi Ratapu"
[4] "Congkai Meng"         

$`Group 10`
[1] "Francesca Domanska" "Robert Mihok"       "Barnaby Clegg-Shaw"
[4] "Ziyi Wang"         </code></pre>
</div>
</div>
</section>
</section>
<section id="sec-slidesliza" class="level2">
<h2 class="anchored" data-anchor-id="sec-slidesliza">Slides</h2>
<div class="cell">
<div class="cell-output-display">
<div class="shareagain" style="min-width:300px;margin:1em auto;" data-exeternal="1">
<iframe src="https://lizabolton.github.io/2025-biosci738-ethics/#/title-slide" width="1600" height="900" style="border:2px solid currentColor;" loading="lazy" allowfullscreen=""></iframe>
<script>fitvids('.shareagain', {players: 'iframe'});</script>
</div>
</div>
</div>
<section id="sec-coc2liza" class="level3">
<h3 class="anchored" data-anchor-id="sec-coc2liza">Group discussion: Codes of conduct</h3>
<ol type="1">
<li><p>What experiences have you had with ethical decisions as a student and/or early career researcher and/or employee?</p></li>
<li><p>Read over the <a href="https://www.royalsociety.org.nz/assets/Uploads/Code-Overview-A3-web.pdf">Royal Society Te Apārangi</a> Code of Professional Standards and Ethics in Science, Technology, and the Humanities.</p></li>
<li><p>Describe a possible workflow for biological research. Come up with possible ethical decisions for each stage.</p></li>
</ol>
</section>
<section id="sec-bonus" class="level3">
<h3 class="anchored" data-anchor-id="sec-bonus">Bonus activity</h3>
<p>If you have time (now or in the second half), do one of the following:</p>
<ul>
<li>Rewrite one of the values &amp; principles of responsibility &amp; standards using the most intense slang you can (Gen-Z, 80s valley girl, up to you, just keep it ‘family friendly’).</li>
</ul>
<p>OR</p>
<ul>
<li>Make a cheesy motivational poster / ‘hopecore’ edit for one of the principles from the <a href="https://www.royalsociety.org.nz/assets/Uploads/Code-Overview-A3-web.pdf">Royal Society Te Apārangi Code</a>.
<ul>
<li>There is a generator you can use <a href="https://bighugelabs.com/motivator.php">here</a> (and you can get free photos from <a href="https://unsplash.com/">Unsplash</a>).</li>
</ul></li>
</ul>
<p><img src="runsheets/img/nonmaleficence.jpg" class="img-fluid"></p>
</section>
</section>
<section id="sec-caseliza" class="level2">
<h2 class="anchored" data-anchor-id="sec-caseliza">Case studies</h2>
<p>For each of the following case studies, discuss the situation and identify potential ethical issues. For some, you may need to run the associated code in <code>R</code>.</p>
<section id="case-1" class="level3">
<h3 class="anchored" data-anchor-id="case-1">Case 1</h3>
<p><img src="runsheets/img/frog.jpg" class="img-fluid"></p>
<p>Image attribution: Hochstetter’s frog (Waitakere Ranges, Auckland). © Nick Harker (via https://www.reptiles.org.nz/)</p>
<p>A mine has been accused of managing their tailings poorly, causing dangerous chemicals to enter the nearby swamp. Local councillors, S. Hrek and D. Onkey, have complained that it is bad for the Hochstetter’s frog | Pangokereia living there. You have been hired by the mine to help them push back on these allegations.</p>
<p>You are contracted to measure the weight of the frogs in the area, with the idea that if the frogs are unhealthy, their weight will be lower than was found in a previous survey of the area. That previous study found that the average weight of the Hochstetter’s frogs in this location is 6.2 grams.</p>
<p>While you are catching and measuring the live frogs, you notice that there are not many juvenile frogs (‘froglets’) despite it being the season when many should be maturing from tadpoles into adult frogs. As Hochstetter’s frogs mature their colouring darkens and you notice that most of the frogs have quite dark colouring.</p>
<p>You can run the code below to find the mean weight of the frogs you studied and perform a t-test.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># The weights of 30 frogs</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>frog_weights <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">5.6</span>, <span class="fl">8.7</span>, <span class="fl">6.1</span>, <span class="fl">6.2</span>, <span class="fl">5.1</span>, <span class="fl">6.8</span>, <span class="fl">6.8</span>, <span class="fl">6.5</span>, <span class="fl">7.0</span>, <span class="fl">7.5</span>, <span class="fl">6.3</span>, <span class="fl">5.8</span>, <span class="fl">6.3</span>, <span class="fl">7.1</span>, <span class="fl">6.9</span>, <span class="fl">6.4</span>, <span class="fl">8.3</span>, <span class="fl">7.4</span>, <span class="fl">7.6</span>, <span class="fl">6.8</span>, <span class="fl">7.3</span>, <span class="fl">9.1</span>, <span class="fl">8.1</span>, <span class="fl">6.8</span>, <span class="fl">6.1</span>, <span class="fl">6.8</span>, <span class="fl">6.9</span>, <span class="fl">7.2</span>, <span class="fl">7.2</span>, <span class="fl">7.5</span>)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the average weight in this sample</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(frog_weights)</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co"># One sample t-test</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(frog_weights, <span class="at">mu =</span> <span class="fl">6.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Look up R documentation
</div>
</div>
<div class="callout-body-container callout-body">
<p>You can run <code>?t.test</code> in your R console to see the documentation for the basic t-test used above.</p>
</div>
</div>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In the <a href="https://docs.google.com/document/d/1QC44oQvFROL-UI1yZu9xvhP8OnG1TFbWBYFmGVJMZSM/edit?usp=sharing">Google Doc</a>…
</div>
</div>
<div class="callout-body-container callout-body">
<p>Write the null and alternative hypothesis for this test in context.</p>
</div>
</div>
<p>A mine executive is very relieved by your results and interprets them as indicating that the frogs are doing very well as they are even larger and healthier than they were before.</p>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In the <a href="https://docs.google.com/document/d/1QC44oQvFROL-UI1yZu9xvhP8OnG1TFbWBYFmGVJMZSM/edit?usp=sharing">Google Doc</a>…
</div>
</div>
<div class="callout-body-container callout-body">
<p>As a group, identify some possible issues with this interpretation based on limitations of your data collection. Write a short explanation for the mine executive about how these issues could affect the interpretation of your result.</p>
</div>
</div>
</section>
<section id="case-2" class="level3">
<h3 class="anchored" data-anchor-id="case-2">Case 2</h3>
<p><img src="runsheets/img/white-kiwi.png" class="img-fluid"></p>
<p>Image attribution: Getty images via https://www.theguardian.com/world/2020/dec/28/mankura-the-rare-white-kiwi-dies-after-surgery-in-new-zealand</p>
<p>You have been hired to work with a local iwi to collect data on the locations and habits of a rare white kiwi bird with a distinctive call. Data is collected across an array of microphones and is used to create a location heat map for where this kiwi spends time.</p>
<p>The iwi are kaitiaki (caretakers) of the area where this kiwi lives. In addition to wanting to learn more about the kiwi, they are worried about increasing attempts from tourists and others to get pictures of this rare bird when people post on social media about locations it has been seen. Lots of people looking for the kiwi and trying to touch it or take photos risks disturbing and distressing the animal, which could severely hurt its health and that of other kiwi in the area.</p>
<p>You finish up the work and your clients are very happy with it, especially complimentary about the data visualisation showing the common locations the kiwi visits by longitude and latitude.</p>
<p>You are preparing an update to your CV to apply for a job, and you notice that transperancy and open data practices are key values of the research institute you are applying to. You write a blog post for your website outlining your findings from the kiwi research, clearly explaining the data collection processes, analytical choices, and limitations. You create a public GitHub repository that provides all the code and data to create the location heat maps.</p>
<p>You’re about to press publish so you can link to it in your application, but a little voice in your head stops you…</p>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In the <a href="https://docs.google.com/document/d/1QC44oQvFROL-UI1yZu9xvhP8OnG1TFbWBYFmGVJMZSM/edit?usp=sharing">Google Doc</a>…
</div>
</div>
<div class="callout-body-container callout-body">
<p>Be the little voice in your head. Is it right to publish this? Make some arguments about potential risks and principles to consider.</p>
</div>
</div>
</section>
<section id="case-3" class="level3">
<h3 class="anchored" data-anchor-id="case-3">Case 3</h3>
<p><img src="runsheets/img/tussock.jpg" class="img-fluid"></p>
<p>Image attribution: Te Ara https://teara.govt.nz/en/grasslands</p>
<p>Concerns have been raised that an off-roading adventure course has caused vegetation loss and increased the risk of wind erosion due to exposed soil. The area where the course has been developed is a large, flat tussock grassland in the South Island. The grassland is fairly similar throughout, and has not been disturbed other than the dirt track for the off-road vehicles to travel on — this track area is not included in the coverage measures, but estimates are taken next to it.</p>
<p>High resolution drone photography of the area has been used to estimate vegetation coverage both close to the course and further away in the undisturbed grassland. To gather the data, 100 meter lines parallel to the course, spaced every 5 meters, are drawn on the aerial imagery. Five one meter square areas are randomly chosen along each line and the vegetation coverage estimated using an appropriate machine learning tool.</p>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In the <a href="https://docs.google.com/document/d/1QC44oQvFROL-UI1yZu9xvhP8OnG1TFbWBYFmGVJMZSM/edit?usp=sharing">Google Doc</a>…
</div>
</div>
<div class="callout-body-container callout-body">
<ol type="1">
<li><p>Run the following code to load the data and view the visualisation provided in a report about the impact of the road on the local vegetation.</p></li>
<li><p>Write down (bullet points are fine) about some of the specific choices made when presenting this data.</p></li>
<li><p>Create your own version of this visualisation. Aim to answer the question about whether the adventure course appears to have had an impact on the vegetation coverage in this grassland area.</p></li>
</ol>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># You need the library tidyverse installed to run this code</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>veg_cov <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">dist_from_track =</span> <span class="fu">c</span>(</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>    <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">5</span>,</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>    <span class="dv">10</span>, <span class="dv">10</span>, <span class="dv">10</span>, <span class="dv">10</span>, <span class="dv">10</span>, <span class="dv">10</span>, <span class="dv">10</span>, <span class="dv">10</span>, <span class="dv">10</span>, <span class="dv">10</span>, <span class="dv">15</span>, <span class="dv">15</span>, <span class="dv">15</span>, <span class="dv">15</span>, <span class="dv">15</span>, <span class="dv">15</span>, <span class="dv">15</span>, <span class="dv">15</span>, <span class="dv">15</span>, <span class="dv">15</span>,</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>    <span class="dv">20</span>, <span class="dv">20</span>, <span class="dv">20</span>, <span class="dv">20</span>, <span class="dv">20</span>, <span class="dv">20</span>, <span class="dv">20</span>, <span class="dv">20</span>, <span class="dv">20</span>, <span class="dv">20</span>, <span class="dv">25</span>, <span class="dv">25</span>, <span class="dv">25</span>, <span class="dv">25</span>, <span class="dv">25</span>, <span class="dv">25</span>, <span class="dv">25</span>, <span class="dv">25</span>, <span class="dv">25</span>, <span class="dv">25</span>,</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>    <span class="dv">30</span>, <span class="dv">30</span>, <span class="dv">30</span>, <span class="dv">30</span>, <span class="dv">30</span>, <span class="dv">30</span>, <span class="dv">30</span>, <span class="dv">30</span>, <span class="dv">30</span>, <span class="dv">30</span>, <span class="dv">35</span>, <span class="dv">35</span>, <span class="dv">35</span>, <span class="dv">35</span>, <span class="dv">35</span>, <span class="dv">35</span>, <span class="dv">35</span>, <span class="dv">35</span>, <span class="dv">35</span>, <span class="dv">35</span>,</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>    <span class="dv">40</span>, <span class="dv">40</span>, <span class="dv">40</span>, <span class="dv">40</span>, <span class="dv">40</span>, <span class="dv">40</span>, <span class="dv">40</span>, <span class="dv">40</span>, <span class="dv">40</span>, <span class="dv">40</span>, <span class="dv">45</span>, <span class="dv">45</span>, <span class="dv">45</span>, <span class="dv">45</span>, <span class="dv">45</span>, <span class="dv">45</span>, <span class="dv">45</span>, <span class="dv">45</span>, <span class="dv">45</span>, <span class="dv">45</span>,</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>    <span class="dv">50</span>, <span class="dv">50</span>, <span class="dv">50</span>, <span class="dv">50</span>, <span class="dv">50</span>, <span class="dv">50</span>, <span class="dv">50</span>, <span class="dv">50</span>, <span class="dv">50</span>, <span class="dv">50</span>, <span class="dv">55</span>, <span class="dv">55</span>, <span class="dv">55</span>, <span class="dv">55</span>, <span class="dv">55</span>, <span class="dv">55</span>, <span class="dv">55</span>, <span class="dv">55</span>, <span class="dv">55</span>, <span class="dv">55</span>,</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>    <span class="dv">60</span>, <span class="dv">60</span>, <span class="dv">60</span>, <span class="dv">60</span>, <span class="dv">60</span>, <span class="dv">60</span>, <span class="dv">60</span>, <span class="dv">60</span>, <span class="dv">60</span>, <span class="dv">60</span>, <span class="dv">65</span>, <span class="dv">65</span>, <span class="dv">65</span>, <span class="dv">65</span>, <span class="dv">65</span>, <span class="dv">65</span>, <span class="dv">65</span>, <span class="dv">65</span>, <span class="dv">65</span>, <span class="dv">65</span>,</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>    <span class="dv">70</span>, <span class="dv">70</span>, <span class="dv">70</span>, <span class="dv">70</span>, <span class="dv">70</span>, <span class="dv">70</span>, <span class="dv">70</span>, <span class="dv">70</span>, <span class="dv">70</span>, <span class="dv">70</span>, <span class="dv">75</span>, <span class="dv">75</span>, <span class="dv">75</span>, <span class="dv">75</span>, <span class="dv">75</span>, <span class="dv">75</span>, <span class="dv">75</span>, <span class="dv">75</span>, <span class="dv">75</span>, <span class="dv">75</span>,</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>    <span class="dv">80</span>, <span class="dv">80</span>, <span class="dv">80</span>, <span class="dv">80</span>, <span class="dv">80</span>, <span class="dv">80</span>, <span class="dv">80</span>, <span class="dv">80</span>, <span class="dv">80</span>, <span class="dv">80</span>, <span class="dv">85</span>, <span class="dv">85</span>, <span class="dv">85</span>, <span class="dv">85</span>, <span class="dv">85</span>, <span class="dv">85</span>, <span class="dv">85</span>, <span class="dv">85</span>, <span class="dv">85</span>, <span class="dv">85</span>,</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>    <span class="dv">90</span>, <span class="dv">90</span>, <span class="dv">90</span>, <span class="dv">90</span>, <span class="dv">90</span>, <span class="dv">90</span>, <span class="dv">90</span>, <span class="dv">90</span>, <span class="dv">90</span>, <span class="dv">90</span>, <span class="dv">95</span>, <span class="dv">95</span>, <span class="dv">95</span>, <span class="dv">95</span>, <span class="dv">95</span>, <span class="dv">95</span>, <span class="dv">95</span>, <span class="dv">95</span>, <span class="dv">95</span>, <span class="dv">95</span>,</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>    <span class="dv">100</span>, <span class="dv">100</span>, <span class="dv">100</span>, <span class="dv">100</span>, <span class="dv">100</span>, <span class="dv">100</span>, <span class="dv">100</span>, <span class="dv">100</span>, <span class="dv">100</span>, <span class="dv">100</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">coverage =</span> <span class="fu">c</span>(</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>    <span class="fl">27.9</span>, <span class="fl">22.5</span>, <span class="dv">27</span>, <span class="fl">21.8</span>, <span class="fl">29.6</span>, <span class="fl">29.2</span>, <span class="dv">21</span>, <span class="fl">21.7</span>, <span class="fl">29.9</span>, <span class="fl">28.5</span>, <span class="fl">26.7</span>, <span class="fl">29.4</span>, <span class="fl">20.6</span>, <span class="fl">26.2</span>, <span class="fl">21.7</span>, <span class="fl">20.4</span>, <span class="fl">25.3</span>, <span class="fl">22.8</span>, <span class="dv">25</span>, <span class="fl">26.3</span>,</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>    <span class="fl">20.1</span>, <span class="fl">26.1</span>, <span class="fl">27.8</span>, <span class="fl">29.2</span>, <span class="fl">22.9</span>, <span class="fl">27.9</span>, <span class="fl">25.7</span>, <span class="fl">27.8</span>, <span class="fl">27.1</span>, <span class="fl">26.7</span>, <span class="fl">29.3</span>, <span class="fl">25.1</span>, <span class="fl">27.5</span>, <span class="fl">28.4</span>, <span class="fl">28.7</span>, <span class="fl">21.9</span>, <span class="fl">22.2</span>, <span class="fl">26.5</span>, <span class="fl">23.4</span>, <span class="fl">25.1</span>,</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>    <span class="fl">26.5</span>, <span class="fl">29.7</span>, <span class="fl">25.1</span>, <span class="fl">20.6</span>, <span class="fl">21.5</span>, <span class="fl">26.4</span>, <span class="dv">21</span>, <span class="fl">27.7</span>, <span class="fl">24.1</span>, <span class="fl">28.7</span>, <span class="fl">27.7</span>, <span class="fl">26.5</span>, <span class="fl">26.3</span>, <span class="fl">28.3</span>, <span class="fl">21.3</span>, <span class="fl">28.8</span>, <span class="fl">22.4</span>, <span class="fl">26.5</span>, <span class="fl">22.3</span>, <span class="fl">20.7</span>,</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>    <span class="fl">21.3</span>, <span class="fl">56.9</span>, <span class="fl">33.7</span>, <span class="fl">57.5</span>, <span class="fl">40.4</span>, <span class="fl">25.3</span>, <span class="fl">52.8</span>, <span class="fl">56.9</span>, <span class="dv">58</span>, <span class="fl">42.3</span>, <span class="dv">41</span>, <span class="fl">41.4</span>, <span class="fl">58.7</span>, <span class="fl">41.1</span>, <span class="fl">41.3</span>, <span class="fl">44.8</span>, <span class="fl">56.9</span>, <span class="fl">29.1</span>, <span class="fl">57.1</span>, <span class="fl">57.2</span>,</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>    <span class="fl">40.9</span>, <span class="fl">59.6</span>, <span class="fl">22.6</span>, <span class="fl">54.5</span>, <span class="fl">39.2</span>, <span class="fl">41.4</span>, <span class="fl">36.6</span>, <span class="fl">29.1</span>, <span class="fl">59.3</span>, <span class="fl">46.2</span>, <span class="fl">43.2</span>, <span class="fl">25.6</span>, <span class="fl">47.9</span>, <span class="fl">49.9</span>, <span class="fl">58.1</span>, <span class="fl">25.2</span>, <span class="dv">46</span>, <span class="fl">48.3</span>, <span class="dv">24</span>, <span class="fl">22.8</span>,</span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>    <span class="fl">31.2</span>, <span class="fl">49.4</span>, <span class="fl">58.5</span>, <span class="fl">52.6</span>, <span class="fl">36.7</span>, <span class="fl">38.2</span>, <span class="fl">34.5</span>, <span class="fl">21.8</span>, <span class="fl">48.8</span>, <span class="fl">55.2</span>, <span class="fl">75.3</span>, <span class="fl">51.7</span>, <span class="fl">64.7</span>, <span class="fl">70.4</span>, <span class="fl">55.5</span>, <span class="fl">59.1</span>, <span class="fl">74.5</span>, <span class="fl">77.4</span>, <span class="fl">76.3</span>, <span class="fl">76.5</span>,</span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>    <span class="fl">59.7</span>, <span class="fl">59.4</span>, <span class="fl">74.7</span>, <span class="fl">50.2</span>, <span class="fl">50.5</span>, <span class="fl">51.1</span>, <span class="fl">51.4</span>, <span class="fl">79.3</span>, <span class="fl">78.8</span>, <span class="fl">71.2</span>, <span class="fl">73.6</span>, <span class="dv">71</span>, <span class="fl">59.4</span>, <span class="fl">66.4</span>, <span class="fl">79.2</span>, <span class="fl">65.9</span>, <span class="fl">60.5</span>, <span class="fl">78.8</span>, <span class="fl">51.7</span>, <span class="fl">57.2</span>,</span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>    <span class="fl">67.5</span>, <span class="fl">55.3</span>, <span class="fl">53.8</span>, <span class="fl">65.1</span>, <span class="fl">70.6</span>, <span class="fl">73.1</span>, <span class="fl">51.2</span>, <span class="fl">75.3</span>, <span class="fl">70.8</span>, <span class="fl">72.3</span>, <span class="dv">66</span>, <span class="fl">56.6</span>, <span class="fl">67.3</span>, <span class="fl">66.3</span>, <span class="fl">73.9</span>, <span class="fl">56.5</span>, <span class="fl">57.5</span>, <span class="fl">75.1</span>, <span class="fl">68.5</span>, <span class="dv">59</span>,</span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>    <span class="dv">76</span>, <span class="fl">60.6</span>, <span class="fl">69.2</span>, <span class="fl">75.8</span>, <span class="fl">61.8</span>, <span class="fl">60.8</span>, <span class="fl">62.7</span>, <span class="dv">66</span>, <span class="fl">65.8</span>, <span class="fl">60.1</span>, <span class="fl">55.8</span>, <span class="fl">74.1</span>, <span class="fl">60.8</span>, <span class="dv">79</span>, <span class="fl">53.8</span>, <span class="fl">73.7</span>, <span class="fl">56.2</span>, <span class="fl">57.3</span>, <span class="fl">77.1</span>, <span class="fl">53.6</span>,</span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>    <span class="fl">61.9</span>, <span class="fl">71.9</span>, <span class="fl">71.8</span>, <span class="fl">51.8</span>, <span class="fl">68.7</span>, <span class="fl">66.4</span>, <span class="fl">65.6</span>, <span class="fl">77.5</span>, <span class="fl">76.1</span>, <span class="fl">66.4</span>, <span class="fl">58.7</span>, <span class="fl">71.2</span>, <span class="fl">78.1</span>, <span class="fl">51.1</span>, <span class="fl">64.3</span>, <span class="fl">51.4</span>, <span class="fl">65.3</span>, <span class="fl">50.5</span>, <span class="fl">50.5</span>, <span class="fl">61.8</span>, <span class="fl">56.3</span>, <span class="fl">64.9</span>, <span class="dv">50</span>, <span class="fl">61.3</span>, <span class="fl">64.8</span>, <span class="fl">57.7</span>, <span class="fl">60.8</span>, <span class="fl">62.6</span>,<span class="fl">65.4</span>,<span class="dv">69</span>))</span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(veg_cov, <span class="fu">aes</span>(<span class="at">x =</span> dist_from_track, <span class="at">y =</span> coverage)) <span class="sc">+</span></span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="dv">25</span>) <span class="sc">+</span></span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">formula =</span> <span class="st">'y ~ x'</span>, <span class="at">col=</span><span class="st">"darkgreen"</span>) <span class="sc">+</span></span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Distance from road (m)"</span>, <span class="at">y =</span> <span class="st">"Coverage (%)"</span>,</span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"Vegetation coverage near to adventure course similar to surrounding area"</span>) <span class="sc">+</span></span>
<span id="cb18-37"><a href="#cb18-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressWarnings</span>(<span class="fu">print</span>(plot))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="finished-early" class="level2">
<h2 class="anchored" data-anchor-id="finished-early">Finished early?</h2>
<p>Work on the <a href="@sec-bonus">bonus activity</a>.</p>
</section>
</section>
<section id="march-21st" class="level1">
<h1>March 21st</h1>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
By the end of today’s seminar, you should be able to
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Carry out, interpret, and communicate the result of
<ul>
<li>a randomization hypothesis test,</li>
<li>a permutation test, and</li>
<li>a bootstrap.</li>
</ul></li>
<li>Identify which of, and when, the above resampling procedures are appropriate to use</li>
<li>Interpret <em>p-values</em> correctly!</li>
</ul>
</div>
</div>
<section id="rough-timeline-3" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-3">Rough timeline</h2>
<section id="pm---0355-pm-1" class="level3">
<h3 class="anchored" data-anchor-id="pm---0355-pm-1"><strong>02:05 pm - 03:55 pm</strong></h3>
<ul>
<li><p><strong>02:05–02.25 pm</strong> <em>Significance</em> (<a href="#sec-significance" class="quarto-xref">Section&nbsp;5.2</a>)</p></li>
<li><p><strong>02.25–02.55 pm</strong> Hypothesis testing via resampling (<a href="#sec-resampling" class="quarto-xref">Section&nbsp;5.3</a>)</p>
<ul>
<li>Permutation (<em>exact</em>) tests (<a href="#sec-perm" class="quarto-xref">Section&nbsp;5.3.1</a>)</li>
<li>Randomization tests (<a href="#sec-random" class="quarto-xref">Section&nbsp;5.3.2</a>)</li>
</ul></li>
<li><p><strong>02:55–03:05 pm</strong> Break</p></li>
<li><p><strong>03.05–03.45 pm</strong> Bootstrapping (<a href="#sec-bootstrap" class="quarto-xref">Section&nbsp;5.4</a>)</p>
<ul>
<li>Case study (<a href="#sec-boot01" class="quarto-xref">Section&nbsp;5.4.1</a>)</li>
</ul></li>
<li><p><strong>03.45–03.55 pm</strong> Q&amp;A and peer-share (<a href="#sec-peershare04" class="quarto-xref">Section&nbsp;5.5</a>)</p></li>
</ul>
</section>
<section id="randomly-allocated-groups-3" class="level3">
<h3 class="anchored" data-anchor-id="randomly-allocated-groups-3">Randomly allocated groups</h3>
<p><strong>It’s your responsibility to get into these, just look for the cows!</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">group_allocation</span>(names, <span class="at">seed =</span> <span class="dv">1131</span>, <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Group 1`
[1] "Pikitangarangi Ratapu" "YOGAPRIYA S"           "Yue Ma"               
[4] "Daneshajeya Jeyavalan"

$`Group 2`
[1] "Robert Mihok" "Jianing Chen" "Vivian Qian"  "Happy Liang" 

$`Group 3`
[1] "Zichang Li"                   "Antonia Schebek-Fuerstenberg"
[3] "Anqi Cheng"                   "Shoule Du"                   

$`Group 4`
[1] "Alex Johnson" "Jiaxin Liu"   "Yutao Yang"  

$`Group 5`
[1] "Tongzheng Yao" "Keyi Sun"      "Ziyi Wang"     "Yingyi Wang"  

$`Group 6`
[1] "Zhixiao Zhang"      "Charlotte Sutton"   "Barnaby Clegg-Shaw"
[4] "Lana Najar"        

$`Group 7`
[1] "He Li"        "Laura Munck"  "Congkai Meng"

$`Group 8`
[1] "Hnin Lwin"     "Xinyue Zhang"  "Leyang Liu"    "Karie Labidon"

$`Group 9`
[1] "Angel Loh"          "Sunny Chen"         "Emma Akeroyd"      
[4] "Francesca Domanska"

$`Group 10`
[1] "Shuxiang He" "Yutong Wang" "Julia Thain" "Yi Han"     </code></pre>
</div>
</div>
</section>
</section>
<section id="sec-significance" class="level2">
<h2 class="anchored" data-anchor-id="sec-significance"><em>Significance</em></h2>
<section id="how-good-are-you-at-assessing-significance-test-yourself." class="level4">
<h4 class="anchored" data-anchor-id="how-good-are-you-at-assessing-significance-test-yourself.">How good are you at assessing <em>significance</em>? <a href="https://statbiscuit.github.io/mini_games/shady/index.html">Test yourself</a>.</h4>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In this <a href="https://docs.google.com/document/d/1ph7vSEel0r7u_W1a7HaI8TewKRqY-DjrPidJGxrXONY/edit?usp=sharing">Google Doc</a>…
</div>
</div>
<div class="callout-body-container callout-body">
<p>Rerun the analysis discussed in <a href="#sec-honesty" class="quarto-xref">Section&nbsp;3.5</a> (see <a href="#sec-solutions" class="quarto-xref">Section&nbsp;22</a> for relevant <code>R</code> code). Compare the two p-values and discuss the inference around them from the hypothesis tests carried out with and without the flagged data values. Discuss why the observations were manipulated in the way they were?</p>
</div>
</div>
<p><strong>5% is NOT some magic threshold</strong>! Below is an excerpt taken from <span class="citation" data-cites="fisher">Fisher (<a href="#ref-fisher" role="doc-biblioref">1926</a>)</span>.<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a></p>
<blockquote class="blockquote">
<p>If one in twenty does not seem high enough odds, we may, if we prefer it, draw the line at one in fifty (the 2 per cent. point), or one in a hundred (the 1 per cent. point). Personally, the writer prefers to set a low standard of significance at the 5 per cent. point, and ignore entirely all results which fail to reach this level. A scientific fact should be regarded as experimentally established only if a properly designed experiment <em>rarely fails</em> to give this level of significance. The very high odds sometimes claimed for experimental results should usually be discounted, for inaccurate methods of estimating error have far more influence than has the particular standard of significance chosen. – Page 504, <span class="citation" data-cites="fisher">Fisher (<a href="#ref-fisher" role="doc-biblioref">1926</a>)</span>.</p>
</blockquote>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In this <a href="https://docs.google.com/document/d/1ph7vSEel0r7u_W1a7HaI8TewKRqY-DjrPidJGxrXONY/edit?usp=sharing">Google Doc</a>…
</div>
</div>
<div class="callout-body-container callout-body">
<p>Write out a inferential statement that includes the magical word <em>significance</em> (as well as quoting the appropriate quantities)! It can be from the analysis we’ve covered in this course, or from a paper, or from your other classes.</p>
</div>
</div>
<p>Remember back to <a href="#sec-caseliza" class="quarto-xref">Section&nbsp;4.3</a> where you faced some potential ethical issues in the given case studies? These ranged from data availability to drawing &amp; reporting conclusions. Your obligations in being an responsible (<em>ethical</em>) also applies to your use of language when reporting.</p>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In this <a href="https://docs.google.com/document/d/1ph7vSEel0r7u_W1a7HaI8TewKRqY-DjrPidJGxrXONY/edit?usp=sharing">Google Doc</a>…
</div>
</div>
<div class="callout-body-container callout-body">
<p>Re-write out your inferential statements from above that switching out the magical word <em>significance</em> to something you believe is more likely to be immediately understood/is less obscure, yet is still concise and meaningful.</p>
</div>
</div>
</section>
</section>
<section id="sec-resampling" class="level2">
<h2 class="anchored" data-anchor-id="sec-resampling">Hypothesis testing via resampling</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(tidyverse)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Group =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"BIOSCI220"</span>, <span class="st">"BIOSCI738"</span>), <span class="at">each =</span> <span class="dv">4</span>),</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">Value =</span> <span class="fu">c</span>(<span class="dv">7</span>,<span class="dv">6</span>,<span class="dv">4</span>,<span class="fl">1.5</span>,<span class="dv">5</span>,<span class="dv">6</span>,<span class="fl">9.5</span>,<span class="dv">3</span>))</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>means <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Group) <span class="sc">%&gt;%</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">means =</span> <span class="fu">round</span>(<span class="fu">mean</span>(Value), <span class="dv">3</span>))</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>vars <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Group) <span class="sc">%&gt;%</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">vars =</span> <span class="fu">round</span>(<span class="fu">var</span>(Value), <span class="dv">3</span>))</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> Group, <span class="at">y =</span> Value, <span class="at">col =</span> Group)) <span class="sc">+</span> <span class="fu">geom_violin</span>() <span class="sc">+</span> <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">col =</span> Group), <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Group"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Value"</span>) <span class="sc">+</span> <span class="fu">theme_light</span>() <span class="sc">+</span></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">x =</span> means<span class="sc">$</span>Group, <span class="at">y =</span> means<span class="sc">$</span>means, <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"mu == "</span>, means<span class="sc">$</span>means), <span class="at">parse =</span> <span class="cn">TRUE</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">"#80CFF0"</span>, <span class="st">"#DFC0D0"</span>), <span class="at">size =</span> <span class="dv">6</span>) <span class="sc">+</span></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">x =</span> vars<span class="sc">$</span>Group, <span class="at">y =</span> (means<span class="sc">$</span>means) <span class="sc">-</span> <span class="fl">0.75</span>, <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"sigma^2 == "</span>, vars<span class="sc">$</span>vars), <span class="at">parse =</span> <span class="cn">TRUE</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">"#80CFF0"</span>, <span class="st">"#DFC0D0"</span>), <span class="at">size =</span> <span class="dv">6</span>)  <span class="sc">+</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#80CFF0"</span>, <span class="st">"#DFC0D0"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>Q</strong> Do you think the means of each group are <em>significantly</em> different from each other? Why or why not?</p>
<p><strong>Q</strong> Do you think the variances of each group are <em>significantly</em> different from each other? Why or why not?</p>
<section id="sec-perm" class="level3">
<h3 class="anchored" data-anchor-id="sec-perm">Permutation (<em>exact</em>) tests</h3>
<p>How many times can you rearrange 8 values into two groups, each of size 4? Remember <span class="math inline">\(\binom{n}{k} = \frac{n!}{k!(n-k)!}\)</span> the binomial coefficient! In our case, <span class="math inline">\(\binom{8}{4}\)</span>. Luckily <code>R</code> has the function <code>choose(8,4)</code>!</p>
<p>Using a permutation test (i.e., one that considers all possible re-combinations of our data) we test the following hypothesis</p>
<p>NULL <span class="math inline">\(H_0: \mu_\text{BIOSCI738} = \mu_\text{BIOSCI220}\)</span> vs alternative <span class="math inline">\(H_1: \mu_\text{BIOSCI738} \neq \mu_\text{BIOSCI220}\)</span>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>mean_diff <span class="ot">&lt;-</span> means <span class="sc">%&gt;%</span> </span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">diff =</span> <span class="fu">diff</span>(means)) <span class="sc">%&gt;%</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.numeric</span>()</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>mean_diff</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1.25</code></pre>
</div>
</div>
<p>The observed statistic in our case is <span class="math inline">\(\mu_\text{BIOSCI738} - \mu_\text{BIOSCI220} =\)</span> 1.25. Below we carry out a permutation test.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>combinations <span class="ot">&lt;-</span> <span class="fu">combn</span>(<span class="dv">8</span>,<span class="dv">4</span>) <span class="do">## 70 in total</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>permtest_combinations <span class="ot">&lt;-</span> <span class="fu">apply</span>(combinations, <span class="dv">2</span>, <span class="cf">function</span>(x)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mean</span>(data<span class="sc">$</span>Value[x]) <span class="sc">-</span> <span class="fu">mean</span>(data<span class="sc">$</span>Value[<span class="sc">-</span>x]))</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>p_val <span class="ot">&lt;-</span> <span class="fu">length</span>(permtest_combinations[<span class="fu">abs</span>(permtest_combinations) <span class="sc">&gt;=</span> mean_diff]) <span class="sc">/</span> <span class="fu">choose</span>(<span class="dv">8</span>,<span class="dv">4</span>)</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>p_val</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.5714286</code></pre>
</div>
</div>
<p>TBH all we’ve really done is carried out a t-test without the associated assumptions, compare this to the output from <code>t.test()</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(Value <span class="sc">~</span> Group, <span class="at">data =</span> data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Welch Two Sample t-test

data:  Value by Group
t = -0.68573, df = 5.9245, p-value = 0.5188
alternative hypothesis: true difference in means between group BIOSCI220 and group BIOSCI738 is not equal to 0
95 percent confidence interval:
 -5.72425  3.22425
sample estimates:
mean in group BIOSCI220 mean in group BIOSCI738 
                  4.625                   5.875 </code></pre>
</div>
</div>
<p><strong>BUT</strong> what if it wasn’t just a vanilla stat we were interested in?</p>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Carry out…
</div>
</div>
<div class="callout-body-container callout-body">
<p>a permutation test that compares the variances of the groups.</p>
</div>
</div>
</section>
<section id="sec-random" class="level3">
<h3 class="anchored" data-anchor-id="sec-random">Randomization tests</h3>
<p>A randomization test is really just a incomplete permutation test (i.e., one where we can’t be bothered to compute all possible combinations). For a randomization test we reshuffle randomly and hope that we do this enough times to construct a sampling distribution of our test statistic under the NULL hypothesis.</p>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Head to the <a href="https://docs.google.com/document/d/1ph7vSEel0r7u_W1a7HaI8TewKRqY-DjrPidJGxrXONY/edit?usp=sharing">Google Doc</a>…
</div>
</div>
<div class="callout-body-container callout-body">
<ol type="1">
<li>``Read” <a href="https://raw.githubusercontent.com/BIOSCI738/cowstats/main/img/randomisation.png">this cheatsheet</a> and summarize the procedure explained.</li>
<li>Given the data below, carry out
<ol type="a">
<li>A permutation test that compares the <strong>medians</strong> of the groups</li>
<li>A randomization test for the same statistic (<a href="https://biosci738.github.io/BIOSCI738/permutation-and-randomisation-tests.html#a-randomisation-test-p%C4%81ua-shell-length">Module 2 of the courseguide give some pretty strong hints</a>)</li>
</ol>
<ul>
<li>think about how many times you want to resample</li>
<li>create a plot of your sampling distribution of the test statistic</li>
<li>what do you infer?</li>
</ul></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">weight =</span> <span class="fu">c</span>(<span class="fl">7.1</span>, <span class="fl">5.2</span>, <span class="fl">6.3</span>, <span class="fl">6.7</span>, <span class="fl">7.4</span>, </span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>                              <span class="fl">5.9</span>, <span class="fl">5.3</span>, <span class="fl">6.1</span>, <span class="fl">2.1</span>, <span class="fl">4.9</span>,</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>                              <span class="fl">4.8</span>, <span class="fl">5.4</span>, <span class="fl">3.6</span>, <span class="fl">4.1</span>, <span class="fl">4.4</span>, </span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>                              <span class="fl">4.8</span>, <span class="fl">3.4</span>, <span class="fl">5.1</span>, <span class="fl">5.3</span>, <span class="fl">5.2</span>),</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">group =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"Group_1"</span>,<span class="st">"Group_2"</span>), <span class="at">each =</span> <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</section>
</section>
<section id="sec-bootstrap" class="level2">
<h2 class="anchored" data-anchor-id="sec-bootstrap">Bootstrapping</h2>
<p>A bootstrap is a procedure for finding the (approximate) sampling distribution of a <strong>statistic/parameter of interest</strong> from a single data sample. We assume that,</p>
<ol type="1">
<li>the original sample represents the distribution of the population from which it was drawn, therefore</li>
<li>resamples, taken with replacement from the original sample are representative of what we would get from drawing many samples from the population (<em>the distribution of the statistics calculated from each resample is known as the bootstrap distribution of the statistic</em>), so</li>
<li>the bootstrap distribution of a statistic represents that statistic’s sampling distribution.</li>
</ol>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Head to the <a href="https://docs.google.com/document/d/1ph7vSEel0r7u_W1a7HaI8TewKRqY-DjrPidJGxrXONY/edit?usp=sharing">Google Doc</a>…
</div>
</div>
<div class="callout-body-container callout-body">
<p>“Read” <a href="https://raw.githubusercontent.com/BIOSCI738/cowstats/main/img/bootstrapping.png">this cheatsheet</a> and summarize the procedure explained. What are the key differences to the resampling procedures in <a href="#sec-resampling" class="quarto-xref">Section&nbsp;5.3</a></p>
</div>
</div>
<section id="sec-boot01" class="level3">
<h3 class="anchored" data-anchor-id="sec-boot01">Case study</h3>
<center>
<p>
Some of my 2022 pumpkin haul
</p>
<img src="runsheets/img/pumpkins.jpg" alt="Some of my 2022 pumpkin haul" height="400px">
</center>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Run through the <code>R</code> code below
</div>
</div>
<div class="callout-body-container callout-body">
<p>Discuss/answer each question posed in the comments. Figure out what each line of code does and replace the comments with your own.</p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="do">## data about the weight, height, and width of some of my homegrown 2022 pumpkins</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/pumpkins.csv"</span>)</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="do">#####################</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="do">## pairsplot</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>GGally<span class="sc">::</span><span class="fu">ggpairs</span>(data)</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a><span class="do">## change to kg</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">weight_kg =</span> weight_g<span class="sc">/</span><span class="dv">1000</span>)</span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a><span class="do">## Pairwise relationships between the three</span></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a><span class="do">## dimensions all appear approximately linear,</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a><span class="do">## with a high correlation</span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a><span class="do">######################</span></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a><span class="do">## linear model</span></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Fitted model.</span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lm</span>(weight_kg <span class="sc">~</span> height_mm <span class="sc">+</span> width_mm, <span class="at">data =</span> data)</span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>fit <span class="sc">|&gt;</span></span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>()</span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a><span class="do">## fitted values</span></span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a><span class="fu">fitted</span>(fit) <span class="do">## notice anything strange?</span></span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a><span class="do">## estimated error variance</span></span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(fit)<span class="sc">$</span>sigma<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a><span class="do">## standard errors for the estimated coefficients</span></span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(fit)<span class="sc">$</span>coefficients[, <span class="dv">2</span>]</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a><span class="do">## New pumpkins</span></span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a><span class="do">## pumpkin 1, height_mm 150, width_mm 240</span></span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a><span class="do">## pumpkin 2, height_mm 100, width_mm 160</span></span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a>new <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">height_mm =</span> <span class="fu">c</span>(<span class="dv">150</span>, <span class="dv">100</span>), <span class="at">width_mm =</span> <span class="fu">c</span>(<span class="dv">240</span>, <span class="dv">160</span>))</span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a><span class="do">## pumpkin 1 (~) 60% bigger than pumpkin 2</span></span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">y  =</span> height_mm , <span class="at">x =</span> width_mm)) <span class="sc">+</span></span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>) <span class="sc">+</span></span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">data =</span> new, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a><span class="do">## point predictions</span></span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(fit, <span class="at">newdata =</span> new)</span>
<span id="cb29-36"><a href="#cb29-36" aria-hidden="true" tabindex="-1"></a><span class="do">## confidence intervals</span></span>
<span id="cb29-37"><a href="#cb29-37" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(fit, <span class="at">newdata =</span> new, <span class="at">interval =</span> <span class="st">"confidence"</span>)[, <span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb29-38"><a href="#cb29-38" aria-hidden="true" tabindex="-1"></a><span class="do">## prediction intervals</span></span>
<span id="cb29-39"><a href="#cb29-39" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(fit, <span class="at">newdata =</span> new, <span class="at">interval =</span> <span class="st">"prediction"</span>)[, <span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb29-40"><a href="#cb29-40" aria-hidden="true" tabindex="-1"></a><span class="do">## pumpkin 1 is 60% larger than pumpkin 2 in terms of height and width. If</span></span>
<span id="cb29-41"><a href="#cb29-41" aria-hidden="true" tabindex="-1"></a><span class="do">## pumpkin growth is isometric, then pumpkin 1's expected weight will be 60% larger than</span></span>
<span id="cb29-42"><a href="#cb29-42" aria-hidden="true" tabindex="-1"></a><span class="do">## pumpkin 2s (ratio 1.6). If the ratio is less than 60% (1.6), then we have negative allometric growth</span></span>
<span id="cb29-43"><a href="#cb29-43" aria-hidden="true" tabindex="-1"></a><span class="do">## (pumpkins get less heavy , relative to the other dimensions), otherwise we have</span></span>
<span id="cb29-44"><a href="#cb29-44" aria-hidden="true" tabindex="-1"></a><span class="do">## positive allometric growth (&gt; 1/6) (pumpkins tend to get heavier as they grow, relative</span></span>
<span id="cb29-45"><a href="#cb29-45" aria-hidden="true" tabindex="-1"></a><span class="do">## to the other dimensions).</span></span>
<span id="cb29-46"><a href="#cb29-46" aria-hidden="true" tabindex="-1"></a><span class="do">## under model above what is the estimated ratio of pumpkin 1 weight (kg) to pumpkin 2 weight (kg)</span></span>
<span id="cb29-47"><a href="#cb29-47" aria-hidden="true" tabindex="-1"></a>mu <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit, <span class="at">newdata =</span> new)</span>
<span id="cb29-48"><a href="#cb29-48" aria-hidden="true" tabindex="-1"></a>mu[<span class="dv">1</span>]<span class="sc">/</span>mu[<span class="dv">2</span>]</span>
<span id="cb29-49"><a href="#cb29-49" aria-hidden="true" tabindex="-1"></a><span class="do">## Compute a confidence interval for this ratio using 1) a parametric and 2)a  non-parametric bootstrap.</span></span>
<span id="cb29-50"><a href="#cb29-50" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################################</span></span>
<span id="cb29-51"><a href="#cb29-51" aria-hidden="true" tabindex="-1"></a><span class="do">## Parametric (resample from assumed response distribution)</span></span>
<span id="cb29-52"><a href="#cb29-52" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################################</span></span>
<span id="cb29-53"><a href="#cb29-53" aria-hidden="true" tabindex="-1"></a><span class="do">## seed</span></span>
<span id="cb29-54"><a href="#cb29-54" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1567</span>)</span>
<span id="cb29-55"><a href="#cb29-55" aria-hidden="true" tabindex="-1"></a><span class="do">## number of bootstrap iterations</span></span>
<span id="cb29-56"><a href="#cb29-56" aria-hidden="true" tabindex="-1"></a>nreps <span class="ot">&lt;-</span> <span class="dv">1000</span></span>
<span id="cb29-57"><a href="#cb29-57" aria-hidden="true" tabindex="-1"></a><span class="do">## initialize empty array to hold results</span></span>
<span id="cb29-58"><a href="#cb29-58" aria-hidden="true" tabindex="-1"></a>bootstrap_ratios <span class="ot">&lt;-</span> <span class="fu">numeric</span>(nreps)</span>
<span id="cb29-59"><a href="#cb29-59" aria-hidden="true" tabindex="-1"></a><span class="do">##  bootstrap iterations.</span></span>
<span id="cb29-60"><a href="#cb29-60" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nreps){</span>
<span id="cb29-61"><a href="#cb29-61" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Simulating new response data assuming Normal response</span></span>
<span id="cb29-62"><a href="#cb29-62" aria-hidden="true" tabindex="-1"></a>    data<span class="sc">$</span>boot <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="fu">nrow</span>(data), <span class="fu">fitted</span>(fit), <span class="fu">summary</span>(fit)<span class="sc">$</span>sigma)</span>
<span id="cb29-63"><a href="#cb29-63" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Fitting the model to the bootstrapped response</span></span>
<span id="cb29-64"><a href="#cb29-64" aria-hidden="true" tabindex="-1"></a>    fit_boot <span class="ot">&lt;-</span> <span class="fu">lm</span>(boot <span class="sc">~</span>  height_mm <span class="sc">+</span> width_mm, <span class="at">data =</span> data)</span>
<span id="cb29-65"><a href="#cb29-65" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Calculating the estimated expectations for new pumpkins</span></span>
<span id="cb29-66"><a href="#cb29-66" aria-hidden="true" tabindex="-1"></a>    mu_new <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit_boot, <span class="at">newdata =</span> new)</span>
<span id="cb29-67"><a href="#cb29-67" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Saving the estimated ratio from the bootstrap model fit.</span></span>
<span id="cb29-68"><a href="#cb29-68" aria-hidden="true" tabindex="-1"></a>    bootstrap_ratios[i] <span class="ot">&lt;-</span> mu_new[<span class="dv">1</span>]<span class="sc">/</span>mu_new[<span class="dv">2</span>]</span>
<span id="cb29-69"><a href="#cb29-69" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-70"><a href="#cb29-70" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(bootstrap_ratios)</span>
<span id="cb29-71"><a href="#cb29-71" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> mu[<span class="dv">1</span>]<span class="sc">/</span>mu[<span class="dv">2</span>], <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb29-72"><a href="#cb29-72" aria-hidden="true" tabindex="-1"></a><span class="do">## CI</span></span>
<span id="cb29-73"><a href="#cb29-73" aria-hidden="true" tabindex="-1"></a>CI_parametric <span class="ot">&lt;-</span> <span class="fu">quantile</span>(bootstrap_ratios, <span class="fu">c</span>(<span class="fl">0.025</span>, <span class="fl">0.975</span>))</span>
<span id="cb29-74"><a href="#cb29-74" aria-hidden="true" tabindex="-1"></a>CI_parametric</span>
<span id="cb29-75"><a href="#cb29-75" aria-hidden="true" tabindex="-1"></a><span class="do">## The confidence interval from the parametric bootstrap does not contain 1.6. We</span></span>
<span id="cb29-76"><a href="#cb29-76" aria-hidden="true" tabindex="-1"></a><span class="do">## therefore have evidence in favour of positive allometric growth.</span></span>
<span id="cb29-77"><a href="#cb29-77" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################################</span></span>
<span id="cb29-78"><a href="#cb29-78" aria-hidden="true" tabindex="-1"></a><span class="do">## Non-parametric (resample from observed data)</span></span>
<span id="cb29-79"><a href="#cb29-79" aria-hidden="true" tabindex="-1"></a><span class="do">###########################################################</span></span>
<span id="cb29-80"><a href="#cb29-80" aria-hidden="true" tabindex="-1"></a><span class="do">## seed</span></span>
<span id="cb29-81"><a href="#cb29-81" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">7651</span>)</span>
<span id="cb29-82"><a href="#cb29-82" aria-hidden="true" tabindex="-1"></a><span class="do">## number of bootstrap iterations</span></span>
<span id="cb29-83"><a href="#cb29-83" aria-hidden="true" tabindex="-1"></a>nreps <span class="ot">&lt;-</span> <span class="dv">1000</span></span>
<span id="cb29-84"><a href="#cb29-84" aria-hidden="true" tabindex="-1"></a><span class="do">## initialize empty array to hold results</span></span>
<span id="cb29-85"><a href="#cb29-85" aria-hidden="true" tabindex="-1"></a>bootstrap_ratios <span class="ot">&lt;-</span> <span class="fu">numeric</span>(nreps)</span>
<span id="cb29-86"><a href="#cb29-86" aria-hidden="true" tabindex="-1"></a><span class="do">##  bootstrap iterations.</span></span>
<span id="cb29-87"><a href="#cb29-87" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nreps){</span>
<span id="cb29-88"><a href="#cb29-88" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Bootstrap resample</span></span>
<span id="cb29-89"><a href="#cb29-89" aria-hidden="true" tabindex="-1"></a>    boot.df <span class="ot">&lt;-</span> data[<span class="fu">sample</span>(<span class="fu">nrow</span>(data), <span class="at">replace =</span> <span class="cn">TRUE</span>), ]</span>
<span id="cb29-90"><a href="#cb29-90" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Fitting the model to the bootstrapped response</span></span>
<span id="cb29-91"><a href="#cb29-91" aria-hidden="true" tabindex="-1"></a>    fit_boot <span class="ot">&lt;-</span> <span class="fu">lm</span>(weight_kg <span class="sc">~</span>  height_mm <span class="sc">+</span> width_mm, <span class="at">data =</span> boot.df)</span>
<span id="cb29-92"><a href="#cb29-92" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Calculating the estimated expectations for new pumpkins</span></span>
<span id="cb29-93"><a href="#cb29-93" aria-hidden="true" tabindex="-1"></a>    mu_new <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit_boot, <span class="at">newdata =</span> new)</span>
<span id="cb29-94"><a href="#cb29-94" aria-hidden="true" tabindex="-1"></a>    <span class="do">## Saving the estimated ratio from the bootstrap model fit.</span></span>
<span id="cb29-95"><a href="#cb29-95" aria-hidden="true" tabindex="-1"></a>    bootstrap_ratios[i] <span class="ot">&lt;-</span> mu_new[<span class="dv">1</span>]<span class="sc">/</span>mu_new[<span class="dv">2</span>]</span>
<span id="cb29-96"><a href="#cb29-96" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-97"><a href="#cb29-97" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(bootstrap_ratios)</span>
<span id="cb29-98"><a href="#cb29-98" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> mu[<span class="dv">1</span>]<span class="sc">/</span>mu[<span class="dv">2</span>], <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb29-99"><a href="#cb29-99" aria-hidden="true" tabindex="-1"></a><span class="do">## CI</span></span>
<span id="cb29-100"><a href="#cb29-100" aria-hidden="true" tabindex="-1"></a>CI_non_parametric <span class="ot">&lt;-</span> <span class="fu">quantile</span>(bootstrap_ratios, <span class="fu">c</span>(<span class="fl">0.025</span>, <span class="fl">0.975</span>))</span>
<span id="cb29-101"><a href="#cb29-101" aria-hidden="true" tabindex="-1"></a>CI_non_parametric</span>
<span id="cb29-102"><a href="#cb29-102" aria-hidden="true" tabindex="-1"></a><span class="do">## again, evidence in favour of positive allometric growth (i.e., pumpkins get rounder the</span></span>
<span id="cb29-103"><a href="#cb29-103" aria-hidden="true" tabindex="-1"></a><span class="do">## bigger they get)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="sec-peershare04" class="level2">
<h2 class="anchored" data-anchor-id="sec-peershare04">Q&amp;A/peer-share/Task 03</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Peer-Share
</div>
</div>
<div class="callout-body-container callout-body">
<p>Go over an activity from the class again or spend some time on Task 03, due today!</p>
</div>
</div>
</section>
</section>
<section id="march-24th" class="level1">
<h1>March 24th</h1>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
By the end of today’s seminar, you should be able to
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Include polynomial terms in a linear model</li>
<li>Critically evaluate linear model fit</li>
<li>Communicate &amp; discuss data structure</li>
</ul>
</div>
</div>
<section id="rough-timeline-4" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-4">Rough timeline</h2>
<ul>
<li><strong>09:05–09:45 am</strong> Reverse engineer (<a href="#sec-draw" class="quarto-xref">Section&nbsp;6.3</a>)</li>
<li><strong>09:45–10:55 am</strong> Break</li>
<li><strong>09:55–10:30 am</strong> Linear model diagnostics
<ul>
<li>Recap residual plots (<a href="#sec-qq" class="quarto-xref">Section&nbsp;6.4.1</a>)</li>
<li>Fry time (<a href="#sec-fry" class="quarto-xref">Section&nbsp;6.4.2</a>)</li>
</ul></li>
<li><strong>10.30–10.55 am</strong> Communicating statistics (<a href="#sec-comm" class="quarto-xref">Section&nbsp;6.5</a>)
<ul>
<li>Project paper choice (<a href="#sec-project" class="quarto-xref">Section&nbsp;6.6</a>)</li>
<li>Consulting a statistician (<a href="#sec-ioa" class="quarto-xref">Section&nbsp;6.7</a>)</li>
</ul></li>
</ul>
<section id="randomly-allocated-groups-4" class="level3">
<h3 class="anchored" data-anchor-id="randomly-allocated-groups-4">Randomly allocated groups</h3>
<p><strong>It’s your responsibility to get into these, just look for the cows!</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">group_allocation</span>(names, <span class="at">seed =</span> <span class="dv">1458</span>, <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Group 1`
[1] "Jiaxin Liu"   "Yi Han"       "Alex Johnson" "Xinyue Zhang"

$`Group 2`
[1] "Karie Labidon" "He Li"         "Julia Thain"   "Yingyi Wang"  

$`Group 3`
[1] "Zichang Li"            "Anqi Cheng"            "Daneshajeya Jeyavalan"
[4] "Congkai Meng"         

$`Group 4`
[1] "Shuxiang He"                  "Zhixiao Zhang"               
[3] "Antonia Schebek-Fuerstenberg"

$`Group 5`
[1] "Pikitangarangi Ratapu" "Tongzheng Yao"         "Robert Mihok"         
[4] "Laura Munck"          

$`Group 6`
[1] "YOGAPRIYA S" "Ziyi Wang"   "Vivian Qian" "Leyang Liu" 

$`Group 7`
[1] "Keyi Sun"     "Yutong Wang"  "Jianing Chen"

$`Group 8`
[1] "Yue Ma"       "Shoule Du"    "Sunny Chen"   "Emma Akeroyd"

$`Group 9`
[1] "Barnaby Clegg-Shaw" "Angel Loh"          "Charlotte Sutton"  
[4] "Yutao Yang"        

$`Group 10`
[1] "Happy Liang"        "Francesca Domanska" "Hnin Lwin"         
[4] "Lana Najar"        </code></pre>
</div>
</div>
</section>
</section>
<section id="rough-timeline-5" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-5">Rough timeline</h2>
<ul>
<li><strong>09:05–09:45 am</strong> Reverse engineer (<a href="#sec-draw" class="quarto-xref">Section&nbsp;6.3</a>)</li>
<li><strong>09:45–10:55 am</strong> Break</li>
<li><strong>09:55–10:30 am</strong> Linear model diagnostics
<ul>
<li>Recap residual plots (<a href="#sec-qq" class="quarto-xref">Section&nbsp;6.4.1</a>)</li>
<li>Fry time (<a href="#sec-fry" class="quarto-xref">Section&nbsp;6.4.2</a>)</li>
</ul></li>
<li><strong>10.30–10.55 am</strong> Communicating statistics (<a href="#sec-comm" class="quarto-xref">Section&nbsp;6.5</a>)
<ul>
<li>Project paper choice (<a href="#sec-project" class="quarto-xref">Section&nbsp;6.6</a>)</li>
<li>Consulting a statistician (<a href="#sec-ioa" class="quarto-xref">Section&nbsp;6.7</a>)</li>
</ul></li>
</ul>
<section id="randomly-allocated-groups-5" class="level3">
<h3 class="anchored" data-anchor-id="randomly-allocated-groups-5">Randomly allocated groups</h3>
<p><strong>It’s your responsibility to get into these, just look for the cows!</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">group_allocation</span>(names, <span class="at">seed =</span> <span class="dv">1458</span>, <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Group 1`
[1] "Jiaxin Liu"   "Yi Han"       "Alex Johnson" "Xinyue Zhang"

$`Group 2`
[1] "Karie Labidon" "He Li"         "Julia Thain"   "Yingyi Wang"  

$`Group 3`
[1] "Zichang Li"            "Anqi Cheng"            "Daneshajeya Jeyavalan"
[4] "Congkai Meng"         

$`Group 4`
[1] "Shuxiang He"                  "Zhixiao Zhang"               
[3] "Antonia Schebek-Fuerstenberg"

$`Group 5`
[1] "Pikitangarangi Ratapu" "Tongzheng Yao"         "Robert Mihok"         
[4] "Laura Munck"          

$`Group 6`
[1] "YOGAPRIYA S" "Ziyi Wang"   "Vivian Qian" "Leyang Liu" 

$`Group 7`
[1] "Keyi Sun"     "Yutong Wang"  "Jianing Chen"

$`Group 8`
[1] "Yue Ma"       "Shoule Du"    "Sunny Chen"   "Emma Akeroyd"

$`Group 9`
[1] "Barnaby Clegg-Shaw" "Angel Loh"          "Charlotte Sutton"  
[4] "Yutao Yang"        

$`Group 10`
[1] "Happy Liang"        "Francesca Domanska" "Hnin Lwin"         
[4] "Lana Najar"        </code></pre>
</div>
</div>
</section>
</section>
<section id="sec-draw" class="level2">
<h2 class="anchored" data-anchor-id="sec-draw">From numbers to plots</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> As a group
</div>
</div>
<div class="callout-body-container callout-body">
<p><a href="https://canvas.auckland.ac.nz/courses/120054/files/14903576?module_item_id=2564916">Here is a handout</a> with nothing but data summaries and linear model summary output given. On the whiteboards sketch the model fit (including uncertainty), for each case.</p>
</div>
</div>
</section>
<section id="linear-model-diagnostics" class="level2">
<h2 class="anchored" data-anchor-id="linear-model-diagnostics">Linear model diagnostics</h2>
<section id="sec-qq" class="level3">
<h3 class="anchored" data-anchor-id="sec-qq">Residual plots (<em>a recap</em>)</h3>
<center>
<img src="https://github.com/statbiscuit/swots/blob/master/figs_n_gifs/lm.gif?raw=true" alt="GIF of linear model residuals">
</center>
<p>Otherwise known as <em>least squares</em> regression, by default <code>lm()</code> seeks to minimize the squared Euclidean distance between the observations and the fitted line.</p>
<center>
<img src="runsheets/img/blank_diagnostic.png" height="700px" alt="blank lm diagnostic plots">
</center>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Diagnostic cheatsheets
</div>
</div>
<div class="callout-body-container callout-body">
<center>
<img src="runsheets/img/outlawing_outliers.png" height="400px" alt="QQ residual plots"><img src="runsheets/img/qq_residuals.png" height="400px" alt="QQ residual plots">
</center>
</div>
</div>
</section>
<section id="sec-fry" class="level3">
<h3 class="anchored" data-anchor-id="sec-fry">Fry time</h3>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i>
</div>
</div>
<div class="callout-body-container callout-body">
<center>
<img src="https://ars.els-cdn.com/content/image/1-s2.0-S1658077X17302655-gr1.jpg" alt="Figure 1 from  article doi.org/10.1016/j.jssas.2018.01.002">
<p>
Figure 1 from <span class="citation" data-cites="fries">Abdul Hamid, Omar, and Sanny (<a href="#ref-fries" role="doc-biblioref">2019</a>)</span>
</p>
</center>
<p>Individually, or in a group, use <code>lm()</code> to model the data from <span class="citation" data-cites="fries">Abdul Hamid, Omar, and Sanny (<a href="#ref-fries" role="doc-biblioref">2019</a>)</span> (plotted above), these can be read into your <code>R</code> session as follows.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>fries <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/fries.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol type="1">
<li>Choose and then plot your best-fitting model. How did you decide on your “best” model?</li>
<li>The function <code>dredge</code> is from the <code>R</code> package <code>MuMIn</code>; it is very useful for model selection. <code>dredge()</code> generates a model selection table of models with combinations of fixed effect terms in the given global (full) model. By default <code>dredge()</code> uses AICc as its model comparison metric. Supplying your “best” model from 1) write out the top two selected models as chosen by dredge (including the estimated coefficients). From all the considered models which would you select? Justify your answer. Note, you should use residual plots in your justification. (p.s. you may need to set <code>na.action = na.fail</code> in your <code>lm()</code> call for <code>dregde()</code> to work)</li>
</ol>
</div>
</div>
</section>
</section>
<section id="sec-comm" class="level2">
<h2 class="anchored" data-anchor-id="sec-comm">Communicating statistics</h2>
<p>In our very first lecture (<a href="#sec-intro" class="quarto-xref">Section&nbsp;2.4</a>) we discussed the <a href="https://biosci738.github.io/BIOSCI738/key-information.html#assessment-summary">assessment structure of BIOSCI 738</a> and we saw that communication-focused activities/assessment made up 50% of the total assessments:</p>
<img src="https://biosci738.github.io/BIOSCI738/img/assessment_overview.png" alt="assessment structure of BIOSCI738" height="400px">

</section>
<section id="sec-project" class="level2">
<h2 class="anchored" data-anchor-id="sec-project">Data discussion</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Consultant and Client
</div>
</div>
<div class="callout-body-container callout-body">
<p>In turn, take on the following roles: <strong>Consultant</strong> and <strong>Client</strong>.</p>
<p>As the <strong>Client</strong> answer the following questions about your chosen paper for your <a href="https://canvas.auckland.ac.nz/courses/120054/assignments/436873">written report</a> (<em>or if you’ve not decided yet then use the heather data that you plotted in your diagnostic task (<a href="#sec-peershare01" class="quarto-xref">Section&nbsp;2.6</a>)</em>).</p>
<ul>
<li>What are the primary objectives of the/your project/analysis?</li>
<li>What specific questions or problems are you trying to address with your analysis?</li>
<li>What variables are included in your dataset?</li>
<li>Are there any known issues with the data/variables?</li>
</ul>
<p>As the <strong>Consultant</strong>, your role is to question the <strong>Client</strong> and ask for specific details/clarifications along the way. After the Q&amp;A session, you should be able to surmise your <strong>Client’s</strong> project in your own words.</p>
</div>
</div>
</section>
<section id="sec-ioa" class="level2">
<h2 class="anchored" data-anchor-id="sec-ioa">Interactive Oral Assessments (IOAs)</h2>
<p>Don’t worry too much if you’ve not completely delved into/chosen your <a href="https://canvas.auckland.ac.nz/courses/120054/assignments/436873">project data</a>. However, the activity from above (<a href="#sec-project" class="quarto-xref">Section&nbsp;6.6</a>) will basically be mirrored in your <a href="https://canvas.auckland.ac.nz/courses/120054/assignments/436870">Interactive 01 assessment</a>.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
What is an IOA
</div>
</div>
<div class="callout-body-container callout-body">
<blockquote class="blockquote">
<p>IOAs are novel, low-stakes authentic assessments, designed to promote academic integrity and prepare students for the workplace. Communication as a key skill in the UoA’s graduate profile and is integral for students’ skill development and employability. An IOA is authentic because it is based on a real-world scenario and promotes student engagement and facilitates higher-order thinking. It also preserves academic integrity through its unscripted conversation prompts based on student responses. — <span class="citation" data-cites="ioa">Ward et al. (<a href="#ref-ioa" role="doc-biblioref">2024</a>)</span></p>
</blockquote>
</div>
</div>
<section id="role-play-ioa-01" class="level4">
<h4 class="anchored" data-anchor-id="role-play-ioa-01"><a href="https://auckland.au.panopto.com/Panopto/Pages/Viewer.aspx?id=2d4d4354-9102-4d7d-9b26-b29e003a14db">Role play IOA 01</a></h4>
</section>
<section id="role-play-ioa-02" class="level4">
<h4 class="anchored" data-anchor-id="role-play-ioa-02"><a href="https://auckland.au.panopto.com/Panopto/Pages/Viewer.aspx?id=c90b7f74-cf30-4483-a26e-b29e003a14d7">Role play IOA 02</a></h4>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Head over to this <a href="https://docs.google.com/document/d/1QCAqp8EIxCZwh4PImcpVz9LzGGaO29Q3ET3EszV8kWg/edit?usp=sharing">Google Doc</a>…
</div>
</div>
<div class="callout-body-container callout-body">
<ol type="1">
<li>Read through and summarise <a href="https://canvas.auckland.ac.nz/courses/120054/assignments/436870">this assessment activity</a>,</li>
<li>(Re)watch the example IOAs from above and discuss as a group how you would grade Rishika according to the given rubric, and</li>
<li>Discuss/write out what amendments would you like to see to the proposed rubric (see assignment) so that, as you see it, it better reflects your input (I will update the rubric accordingly if I deem the suggestions a better reflection of effort!)</li>
</ol>
</div>
</div>
</section>
</section>
</section>
<section id="march-28th" class="level1">
<h1>March 28th</h1>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
By the end of today’s seminar, you should be able to
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Fit, draw inference from and critique polynomial regression models</li>
</ul>
</div>
</div>
<section id="rough-timeline-6" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-6">Rough timeline</h2>
<section id="pm---0355-pm-2" class="level3">
<h3 class="anchored" data-anchor-id="pm---0355-pm-2"><strong>02:05 pm - 03:55 pm</strong></h3>
<ul>
<li><strong>02:05–02.20 pm</strong> Polynomials (<a href="#sec-poly" class="quarto-xref">Section&nbsp;7.2</a>)</li>
<li><strong>02.20–02.45 pm</strong> Fry time revisited (<a href="#sec-fry02" class="quarto-xref">Section&nbsp;7.3</a>)</li>
<li><strong>02:45–02:55 pm</strong> Break</li>
<li><strong>02.55–03.30 pm</strong> Revisiting <a href="#sec-comm" class="quarto-xref">Section&nbsp;6.5</a></li>
<li><strong>03.30–03.55 pm</strong> Q&amp;A and Task 04 (<a href="#sec-peershare06" class="quarto-xref">Section&nbsp;7.5</a>)</li>
</ul>
</section>
<section id="randomly-allocated-groups-6" class="level3">
<h3 class="anchored" data-anchor-id="randomly-allocated-groups-6">Randomly allocated groups</h3>
<p><strong>It’s your responsibility to get into these, just look for the cows!</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">group_allocation</span>(names, <span class="at">seed =</span> <span class="dv">0833</span>, <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Group 1`
[1] "Leyang Liu"   "Congkai Meng" "Yutao Yang"   "Zichang Li"  

$`Group 2`
[1] "Robert Mihok"                 "Emma Akeroyd"                
[3] "Antonia Schebek-Fuerstenberg" "Xinyue Zhang"                

$`Group 3`
[1] "He Li"            "Shoule Du"        "Charlotte Sutton" "Yi Han"          

$`Group 4`
[1] "Anqi Cheng"  "Laura Munck" "Angel Loh"  

$`Group 5`
[1] "Yogapriya"          "Zhixiao Zhang"      "Keyi Sun"          
[4] "Barnaby Clegg-Shaw"

$`Group 6`
[1] "Jianing Chen"  "Tongzheng Yao" "Yutong Wang"   "Happy Liang"  

$`Group 7`
[1] "Karie Labidon" "Hnin Lwin"     "Ziyi Wang"    

$`Group 8`
[1] "Daneshajeya Jeyavalan" "Lana Najar"            "Jiaxin Liu"           
[4] "Vivian Qian"          

$`Group 9`
[1] "Yingyi Wang"           "Sunny Chen"            "Pikitangarangi Ratapu"
[4] "Alex Johnson"         

$`Group 10`
[1] "Julia Thain"        "Francesca Domanska" "Shuxiang He"       
[4] "Yue Ma"            </code></pre>
</div>
</div>
</section>
</section>
<section id="sec-poly" class="level2">
<h2 class="anchored" data-anchor-id="sec-poly">Polynomials</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Head to this <a href="https://docs.google.com/document/d/1QCAqp8EIxCZwh4PImcpVz9LzGGaO29Q3ET3EszV8kWg/edit?usp=sharing">Google Doc</a>…
</div>
</div>
<div class="callout-body-container callout-body">
<p>Discuss this cheatsheet below, what do you think it is attempting to show/compare?</p>
<p><img src="runsheets/img/poly_vs_I.png" class="img-fluid"></p>
</div>
</div>
</section>
<section id="sec-fry02" class="level2">
<h2 class="anchored" data-anchor-id="sec-fry02">Polynomial regression</h2>
<p>Individually, or as a group, revisit <a href="#sec-fry" class="quarto-xref">Section&nbsp;6.4.2</a> alongside the suggested solution given in <a href="#sec-solutions" class="quarto-xref">Section&nbsp;22</a>. Discuss and summarise each fitted model and modelling choice made. Chose and justify your choice of preferred/“best” model?</p>
</section>
<section id="revisiting-sec-comm" class="level2">
<h2 class="anchored" data-anchor-id="revisiting-sec-comm">Revisiting <a href="#sec-comm" class="quarto-xref">Section&nbsp;6.5</a></h2>
</section>
<section id="sec-peershare06" class="level2">
<h2 class="anchored" data-anchor-id="sec-peershare06">Q&amp;A and Task 04</h2>
<p><a href="https://canvas.auckland.ac.nz/courses/120054/assignments/436860"><strong>Task 04</strong></a></p>
<blockquote class="blockquote">
<p>Git is a version control system that manages the evolution of a set of files, called a repository (repo), in a structured way (<em>think of Word’s Track Changes</em>).</p>
</blockquote>
<blockquote class="blockquote">
<p>GitHub is a hosting service that provides a hgme for your git-based projects on the internet (<em>think of Dropbox</em>).</p>
</blockquote>
<p>The template repo you’ve been invited to in <a href="https://canvas.auckland.ac.nz/courses/120054/assignments/436860">Task 04</a> is set up as a <code>RStudio</code> Project. It contains the basics for you to get started on the analysis for your assessed project! For this <a href="https://canvas.auckland.ac.nz/courses/120054/assignments/436860">Task 04</a> all I’d like you to do is 1) connect to the repo assignment you’ve been invited to, 2) make a change locally in the repo, 3) then commit and push this change to the remote repo. <strong>We’ll do this in class if you’ve not yet already.</strong></p>
</section>
</section>
<section id="march-31st" class="level1">
<h1>March 31st</h1>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
By the end of today’s seminar, you should be able to
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Discuss the role of Sums of Squares (SS) in variance decomposition</li>
<li>“Manually” calculate within, between, and total SS</li>
<li>Discuss the issues associated with multiple comparison techniques and the standard adjustment techniques employed to combat the issues</li>
</ul>
</div>
</div>
<section id="rough-timeline-7" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-7">Rough timeline</h2>
<ul>
<li><strong>09:05–09:55 am</strong> Modeller vs Designer (<a href="#sec-mods" class="quarto-xref">Section&nbsp;8.2</a>)
<ul>
<li>Group activity (<a href="#sec-ss" class="quarto-xref">Section&nbsp;8.2.2</a>)</li>
</ul></li>
<li><strong>09:55–11:05 am</strong> Break</li>
<li><strong>11:00–11:45 am</strong> Multiple comparisons (<a href="#sec-multi" class="quarto-xref">Section&nbsp;8.3</a>)
<ul>
<li>Class discussion</li>
</ul></li>
<li><strong>11.45–11.55 am</strong> Q&amp;A and peer-share (<a href="#sec-peershare07" class="quarto-xref">Section&nbsp;8.4</a>)</li>
</ul>
<section id="randomly-allocated-groups-7" class="level3">
<h3 class="anchored" data-anchor-id="randomly-allocated-groups-7">Randomly allocated groups</h3>
<p><strong>It’s your responsibility to get into these, just look for the cows!</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">group_allocation</span>(names, <span class="at">seed =</span> <span class="dv">1631</span>, <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Group 1`
[1] "Congkai Meng"  "Yogapriya"     "He Li"         "Tongzheng Yao"

$`Group 2`
[1] "Karie Labidon"                "Antonia Schebek-Fuerstenberg"
[3] "Yutong Wang"                  "Zichang Li"                  

$`Group 3`
[1] "Xinyue Zhang" "Lana Najar"   "Jianing Chen" "Sunny Chen"  

$`Group 4`
[1] "Yue Ma"             "Hnin Lwin"          "Francesca Domanska"

$`Group 5`
[1] "Yi Han"        "Zhixiao Zhang" "Shoule Du"     "Yingyi Wang"  

$`Group 6`
[1] "Emma Akeroyd"          "Keyi Sun"              "Happy Liang"          
[4] "Daneshajeya Jeyavalan"

$`Group 7`
[1] "Pikitangarangi Ratapu" "Yutao Yang"            "Leyang Liu"           

$`Group 8`
[1] "Julia Thain"      "Charlotte Sutton" "Anqi Cheng"       "Laura Munck"     

$`Group 9`
[1] "Barnaby Clegg-Shaw" "Shuxiang He"        "Angel Loh"         
[4] "Alex Johnson"      

$`Group 10`
[1] "Jiaxin Liu"   "Ziyi Wang"    "Robert Mihok" "Vivian Qian" </code></pre>
</div>
</div>
</section>
</section>
<section id="sec-mods" class="level2">
<h2 class="anchored" data-anchor-id="sec-mods">Modeller vs Designer</h2>
<p align="center">
<iframe src="slides/05-01_modller-vs-designer.pdf" height="400" width="700" title="Modeller-vs-Designer slides">
</iframe>
</p>
<p align="center">
<a href="https://canvas.auckland.ac.nz/courses/120054/files/15042516?module_item_id=2582752"><strong>CANVAS slides</strong></a>
</p>
<section id="partitioning-variation" class="level3">
<h3 class="anchored" data-anchor-id="partitioning-variation">Partitioning Variation</h3>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="runsheets/img/well.jpg" class="img-fluid figure-img"></p>
<figcaption>📷 <a href="Adobe%20Stock">source</a></figcaption>
</figure>
</div>
<p><a href="https://canvas.auckland.ac.nz/courses/120054/files/15015121?module_item_id=2581434"><strong>Partitioning Variation</strong></a> some examples.</p>
</section>
<section id="sec-ss" class="level3">
<h3 class="anchored" data-anchor-id="sec-ss">Sums of Squares activity</h3>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Head to this <a href="https://docs.google.com/document/d/1Hby1zVqv39pZSaXwBifZ7W_Mz9fmTWYP8LQ6U8BUQ54/edit?usp=sharing">Google Doc</a>…
</div>
</div>
<div class="callout-body-container callout-body">
<p>Together, work through the code snippets from <a href="https://biosci738.github.io/BIOSCI738/modelling-experimental-data.html#sums-of-squares-ss">this section of the course guide Sums of squares (SS)</a>, you may have to refer to previous sections (e.g., the <a href="https://biosci738.github.io/BIOSCI738/parametric-hypothesis-testing.html#one-way-analysis-of-variance-anova">One-Way Analysis of Variance (ANOVA) section from Module 2</a>) if you are unfamiliar with some of the terminology, and discuss/answer the following</p>
<ul>
<li>What are sums of squares (SS)?</li>
<li>What role do SS play in statistics?</li>
<li>What is meant by the term sequential Sums of Squares? And what effect, if any, might this have on inference?</li>
<li>Explain the reasoning/answers above for an <a href="https://www.reddit.com/r/explainlikeimfive/?rdt=41742">r/ExplainLikeImFive</a> audience.</li>
</ul>
<p>You may find the pseudo code below useful.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="do">## template only, you will have to rename variables accordingly</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="do">## run segments/lines one at a time to see the steps</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span> </span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">grand_mean =</span> <span class="fu">mean</span>(response)) <span class="sc">%&gt;%</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">sstotal =</span> (response <span class="sc">-</span> grand_mean) <span class="sc">^</span><span class="dv">2</span> ) <span class="sc">%&gt;%</span></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(treatment) <span class="sc">%&gt;%</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">treatment_mean =</span> <span class="fu">mean</span>(response),</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>          <span class="at">sse =</span> (response <span class="sc">-</span> treatment_mean)<span class="sc">^</span><span class="dv">2</span>,</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>          <span class="at">tmp =</span> (treatment_mean <span class="sc">-</span> grand_mean)<span class="sc">^</span><span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>   <span class="fu">summarise</span>(<span class="at">ss_tot =</span> <span class="fu">sum</span>(sstotal), <span class="at">ss_e =</span> <span class="fu">sum</span>(sse), <span class="at">ss_treat =</span> <span class="fu">sum</span>(tmp)) <span class="sc">%&gt;%</span></span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>   <span class="fu">pull</span>(ss_treat) <span class="sc">%&gt;%</span> <span class="fu">sum</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Bonus activity
</div>
</div>
<div class="callout-body-container callout-body">
<p>If you have some spare time sketch a concept map or illustration about sequential Sums of Squares.</p>
</div>
</div>
</section>
</section>
<section id="sec-multi" class="level2">
<h2 class="anchored" data-anchor-id="sec-multi">Multiple comparisons</h2>
<p><strong>Slides</strong></p>
<p align="center">
<iframe src="slides/05-02_multiple-comparisons.pdf" height="400" width="700" title="Modeller-vs-Designer slides">
</iframe>
</p>
<p align="center">
<a href="https://canvas.auckland.ac.nz/courses/120054/files/15042517?module_item_id=2582753"><strong>CANVAS slides</strong></a>
</p>
</section>
<section id="sec-peershare07" class="level2">
<h2 class="anchored" data-anchor-id="sec-peershare07">Q&amp;A and peer-share</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Peer-Share
</div>
</div>
<div class="callout-body-container callout-body">
<p>Go over an activity from the class again. Feel free to ask Charlotte a question or two about it!</p>
</div>
</div>
</section>
</section>
<section id="april-4th" class="level1">
<h1>April 4th</h1>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Remember to sign up for your <a href="https://canvas.auckland.ac.nz/courses/120054/assignments/436870">Interactive 01</a> time slot!
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
By the end of today’s seminar, you should be able to
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Apply and compare standard adjustment techniques (e.g., Tukey, Bonferroni and Fisher corrections) employed in multiple testing</li>
<li>Discuss and compare definitions of fixed and random effects</li>
</ul>
</div>
</div>
<section id="rough-timeline-8" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-8">Rough timeline</h2>
<section id="pm---0355-pm-3" class="level3">
<h3 class="anchored" data-anchor-id="pm---0355-pm-3"><strong>02:05 pm - 03:55 pm</strong></h3>
<ul>
<li><strong>02:05–02.40 pm</strong> Multiple Comparisons (<a href="#sec-multi02" class="quarto-xref">Section&nbsp;9.2</a>)
<ul>
<li>Group activities</li>
</ul></li>
<li><strong>02.40–02.55 pm</strong> Degrees of Freedom (<a href="#sec-df" class="quarto-xref">Section&nbsp;9.3</a>)</li>
<li><strong>02:55–03:05 pm</strong> Break</li>
<li><strong>03.05–03.30 pm</strong> Introducing Random Effects (<a href="#sec-randomeff" class="quarto-xref">Section&nbsp;9.4</a>)
<ul>
<li>Class discussion</li>
<li>Group activity</li>
</ul></li>
<li><strong>03.45–03.55 pm</strong> Q&amp;A and Task 05 (<a href="#sec-peershare08" class="quarto-xref">Section&nbsp;9.5</a>)</li>
</ul>
</section>
<section id="randomly-allocated-groups-8" class="level3">
<h3 class="anchored" data-anchor-id="randomly-allocated-groups-8">Randomly allocated groups</h3>
<p><strong>It’s your responsibility to get into these, just look for the cows!</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">group_allocation</span>(names, <span class="at">seed =</span> <span class="dv">1001</span>, <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Group 1`
[1] "Laura Munck" "Yutao Yang"  "Happy Liang" "Shoule Du"  

$`Group 2`
[1] "Jiaxin Liu"         "Alex Johnson"       "Francesca Domanska"
[4] "Zichang Li"        

$`Group 3`
[1] "Anqi Cheng"       "Karie Labidon"    "Keyi Sun"         "Charlotte Sutton"

$`Group 4`
[1] "Robert Mihok" "Hnin Lwin"    "Angel Loh"   

$`Group 5`
[1] "Daneshajeya Jeyavalan"        "Antonia Schebek-Fuerstenberg"
[3] "Sunny Chen"                   "Emma Akeroyd"                

$`Group 6`
[1] "Yi Han"        "Lana Najar"    "Tongzheng Yao" "Julia Thain"  

$`Group 7`
[1] "Ziyi Wang"   "Yutong Wang" "He Li"      

$`Group 8`
[1] "Yingyi Wang" "Vivian Qian" "Yue Ma"      "Leyang Liu" 

$`Group 9`
[1] "Jianing Chen"          "Congkai Meng"          "Pikitangarangi Ratapu"
[4] "Shuxiang He"          

$`Group 10`
[1] "Barnaby Clegg-Shaw" "Zhixiao Zhang"      "Yogapriya"         
[4] "Xinyue Zhang"      </code></pre>
</div>
</div>
</section>
</section>
<section id="sec-multi02" class="level2">
<h2 class="anchored" data-anchor-id="sec-multi02">Multiple Comparisons</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Head to this <a href="https://docs.google.com/document/d/1j9b8n2NwN7C15oZjnveFbTnAaYdo1SvZNRqy-f5XZhs/edit?usp=sharing">Google Doc</a>…
</div>
</div>
<div class="callout-body-container callout-body">
<p>In your groups discuss and answer the following points/questions, which relate the the snippet of code that follows.</p>
<ol type="1">
<li>What is each line doing?</li>
<li>What is the relevance of 0.05 in the final line of the loop?</li>
<li>What happens if you change this value?</li>
<li>What happens if you change <code>n</code>, and why?</li>
<li>What is this an example of? Why is this the case? Can you explain this reasoning for an <a href="https://www.reddit.com/r/explainlikeimfive/?rdt=41742">r/ExplainLikeImFive</a> audience?</li>
</ol>
<p>Now, amend the code to ``test” the same thing using another statistical test of your choice.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="dv">10000</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>t1err <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n){</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set.seed</span>(<span class="dv">1432</span> <span class="sc">+</span> i)</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">100</span>, <span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>   <span class="cf">if</span> (((<span class="fu">t.test</span>(x, <span class="at">mu =</span> <span class="dv">0</span>))<span class="sc">$</span>p.value)<span class="sc">&lt;=</span><span class="fl">0.05</span>) (<span class="at">t1err =</span> t1err <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Type I error rate in percentage is"</span>, (t1err<span class="sc">/</span>n)<span class="sc">*</span><span class="dv">100</span>,<span class="st">"%"</span>, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Finally, think back to <a href="#sec-significance" class="quarto-xref">Section&nbsp;5.2</a>. What are your thoughts on <em>significance</em>, have they changed?</p>
</div>
</div>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Head to this <a href="https://docs.google.com/document/d/1j9b8n2NwN7C15oZjnveFbTnAaYdo1SvZNRqy-f5XZhs/edit?usp=sharing">Google Doc</a>…
</div>
</div>
<div class="callout-body-container callout-body">
<p>[Note: you may find <a href="https://biosci738.github.io/BIOSCI738/multiple-comparisons.html#adjustments-for-multiple-testing">this section of the course guide</a> useful for this activity]</p>
<ol type="1">
<li><p>Using any suitable data from the lectures or the course guide perform the pairwise comparisons of means using Fisher’s LSD with <span class="math inline">\(\alpha = 0.05\)</span> level of significance.</p></li>
<li><p>Summarise and present these results in a table with the following column names <strong>Comparison</strong>, <strong>Calculated difference</strong> (in means), <strong>SED</strong> (standard error of the difference), <strong>LSDs</strong>, <strong>Lower 95% CI</strong> &amp; <strong>Upper 95% CI</strong> (lower &amp; upper 95% confidence interval), <strong>t-statistic</strong>, <strong>P-value</strong>.</p></li>
<li><p>Calculate the scale factor (i.e., critical value of the assumed distribution) using 1) Bonferroni’s correction method and 2) Tukey’s HSD method for any pairwise comparison 95% CI from above. Discuss what effect the multipliers might have on inference.</p></li>
</ol>
</div>
</div>
</section>
<section id="sec-df" class="level2">
<h2 class="anchored" data-anchor-id="sec-df">Degrees of Freedom</h2>
<p><strong>Slides</strong></p>
<p align="center">
<iframe src="slides/05-03_degrees-of-freedom.pdf" height="400" width="700" title="Degrees of Freedom slides">
</iframe>
</p>
<p align="center">
<a href="https://canvas.auckland.ac.nz/courses/120054/files/15079931?module_item_id=2586885"><strong>CANVAS slides</strong></a>
</p>
</section>
<section id="sec-randomeff" class="level2">
<h2 class="anchored" data-anchor-id="sec-randomeff">Introducing random effcets</h2>
<p>Below is a list collated by <span class="citation" data-cites="gelman2005analysis">Gelman (<a href="#ref-gelman2005analysis" role="doc-biblioref">2005</a>)</span> (p.&nbsp;20) who discuss fixed and random effects in the context of ANOVA-type analysis (Note that some use terminology we’re yet to cover!).</p>
<blockquote class="blockquote">
<ol type="1">
<li>Fixed effects are constant across individuals, and random effects vary. For example, in a growth study, a model with random intercepts <span class="math inline">\(\alpha_i\)</span> and fixed slope <span class="math inline">\(\beta\)</span> corresponds to parallel lines for different individuals <span class="math inline">\(i\)</span>, or the model <span class="math inline">\(y_{it} = \alpha_i + \beta t\)</span>. Kreft and de Leeuw [(1998), page 12] thus distinguish between fixed and random coefficients.</li>
<li>Effects are fixed if they are interesting in themselves or random if there is interest in the underlying population. Searle, Casella, and McCulloch [(1992), Section 1.4] explore this distinction in depth.</li>
<li>When a sample exhausts the population, the corresponding variable is <em>fixed</em>; when the sample is a small (i.e., negligible) part of the population, the corresponding variable is <em>random</em> [Green and Tukey (1960)].</li>
<li>If an effect is assumed to be a realized value of a random variable, it is called a random effect” [LaMotte (1983)].</li>
<li>Fixed effects are estimated using least squares (or, more generally, maximum likelihood), and random effects are estimated with shrinkage [“linear unbiased prediction” in the terminology of Robinson (1991)]. This definition is standard in the multilevel modeling literature [see, e.g., Snijders and Bosker (1999), Section 4.2] and in econometrics. In the Bayesian framework, this definition implies that fixed effects <span class="math inline">\(\beta_j^{(m)}\)</span> are estimated conditional on <span class="math inline">\(\sigma_m = \infty\)</span>, and random effects <span class="math inline">\(\beta_j^{(m)}\)</span> are estimated conditional on <span class="math inline">\(\sigma_m\)</span> from the posterior distribution.</li>
</ol>
</blockquote>
<p>The author of <span class="citation" data-cites="gelman2005analysis">Gelman (<a href="#ref-gelman2005analysis" role="doc-biblioref">2005</a>)</span> moves on to the following conclusion (see p.&nbsp;21).</p>
<blockquote class="blockquote">
<p>We prefer to sidestep the overloaded terms “fixed” and “random” with a cleaner distinction by simply renaming the terms in definition 1 above. We define effects (or coefficients) in a multilevel model as <strong>constant</strong> if they are identical for all groups in a population and <strong>varying</strong> if they are allowed to differ from group to group. For example, the model <span class="math inline">\(y_{ij} = \alpha + \beta x_{ij}\)</span> (of units <span class="math inline">\(i\)</span> in groups <span class="math inline">\(j\)</span>) has a constant slope and varying intercepts, and <span class="math inline">\(y_{ij} = \alpha_{j} + \beta_j x_{ij}\)</span> has varying slopes and intercepts. In this terminology (which we would apply at any level of the hierarchy in a multilevel model), varying effects occur in batches, whether or not the effects are interesting in themselves (definition 2), and whether or not they are a sample from a larger set (definition 3). Definitions 4 and 5 do not arise for us since we estimate all batches of effects hierarchically, with the variance components <span class="math inline">\(\sigma_m\)</span> estimated from data.</p>
</blockquote>
<p><strong>Slides</strong></p>
<p align="center">
<iframe src="slides/05-04_fixed-vs-random.pdf" height="400" width="700" title="Fixed vs Random effects slides">
</iframe>
</p>
<p align="center">
<a href="https://canvas.auckland.ac.nz/courses/120054/files/15062504?module_item_id=2585389"><strong>CANVAS slides</strong></a>
</p>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Head to this <a href="https://docs.google.com/document/d/1j9b8n2NwN7C15oZjnveFbTnAaYdo1SvZNRqy-f5XZhs/edit?usp=sharing">Google Doc</a>…
</div>
</div>
<div class="callout-body-container callout-body">
<p>[Note You may find <a href="https://dynamicecology.wordpress.com/2015/11/04/is-it-a-fixed-or-random-effect/">this blog</a> offers some extra insights here<a href="#fn5" class="footnote-ref" id="fnref5" role="doc-noteref"><sup>5</sup></a> ]</p>
<ol type="1">
<li>Translate each definition in the list above to lay-person language (i.e., non-technical).</li>
<li>Make up a data example and allocate your fixed and random effects according to each definition.</li>
<li>Discuss the similarities and dissimilarities between the definitions.</li>
<li>Conclude with which definition <em>best</em> aligns with your current understanding of the distinction between fixed and random effects?</li>
</ol>
</div>
</div>
</section>
<section id="sec-peershare08" class="level2">
<h2 class="anchored" data-anchor-id="sec-peershare08">Q&amp;A/peer-share/Task 05</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Peer-Share
</div>
</div>
<div class="callout-body-container callout-body">
<p>Go over an activity from the class again or spend some time finishing off Task 05.</p>
</div>
</div>
</section>
</section>
<section id="april-7th" class="level1">
<h1>April 7th</h1>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
By the end of today’s seminar, you should be able to
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Fit and critique linear mixed effects models</li>
<li>Create an experimental design ‘recipe’ using <code>R</code></li>
</ul>
</div>
</div>
<section id="rough-timeline-9" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-9">Rough timeline</h2>
<ul>
<li><strong>09:05–09:55 am</strong> Case studies (<a href="#sec-casestudies" class="quarto-xref">Section&nbsp;10.2</a>)</li>
<li><strong>09:55–11:05 am</strong> Break</li>
<li><strong>11:00–11:45 am</strong> The Terminology of Experimental Design (<a href="#sec-edterms" class="quarto-xref">Section&nbsp;10.3</a>)
<ul>
<li>Class discussion</li>
<li>Group activity</li>
</ul></li>
<li><strong>11.45–11.55 am</strong> Q&amp;A and peer-share (<a href="#sec-peershare09" class="quarto-xref">Section&nbsp;10.5</a>)</li>
</ul>
<section id="randomly-allocated-groups-9" class="level3">
<h3 class="anchored" data-anchor-id="randomly-allocated-groups-9">Randomly allocated groups</h3>
<p><strong>It’s your responsibility to get into these, just look for the cows!</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">group_allocation</span>(names, <span class="at">seed =</span> <span class="dv">1121</span>, <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Group 1`
[1] "Emma Akeroyd" "He Li"        "Yingyi Wang"  "Robert Mihok"

$`Group 2`
[1] "Yutao Yang"    "Tongzheng Yao" "Karie Labidon" "Hnin Lwin"    

$`Group 3`
[1] "Antonia Schebek-Fuerstenberg" "Pikitangarangi Ratapu"       
[3] "Xinyue Zhang"                 "Charlotte Sutton"            

$`Group 4`
[1] "Ziyi Wang"    "Yue Ma"       "Alex Johnson"

$`Group 5`
[1] "Happy Liang" "Shoule Du"   "Leyang Liu"  "Jiaxin Liu" 

$`Group 6`
[1] "Congkai Meng"       "Barnaby Clegg-Shaw" "Shuxiang He"       
[4] "Vivian Qian"       

$`Group 7`
[1] "Yi Han"             "Francesca Domanska" "Yogapriya"         

$`Group 8`
[1] "Laura Munck" "Keyi Sun"    "Anqi Cheng"  "Yutong Wang"

$`Group 9`
[1] "Daneshajeya Jeyavalan" "Julia Thain"           "Jianing Chen"         
[4] "Lana Najar"           

$`Group 10`
[1] "Zichang Li"    "Angel Loh"     "Zhixiao Zhang" "Sunny Chen"   </code></pre>
</div>
</div>
</section>
</section>
<section id="sec-casestudies" class="level2">
<h2 class="anchored" data-anchor-id="sec-casestudies">Case studies</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In your groups…
</div>
</div>
<div class="callout-body-container callout-body">
<p>Work though each of the case studies given below answering/discussing the prompts as you go. You’ll notice a few extra packages and functions getting used; I suggest investigating! If you get stuck just grab me as I walk around!</p>
</div>
</div>
<section id="case-study-01" class="level3">
<h3 class="anchored" data-anchor-id="case-study-01">Case study 01</h3>
<p>Revisiting <a href="#sec-multi" class="quarto-xref">Section&nbsp;8.3</a> and <a href="#sec-multi02" class="quarto-xref">Section&nbsp;9.2</a> may be useful here.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(predictmeans) <span class="do">## install using install.packages("predictmeans")</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>rats <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/crd_rats_data.csv"</span>)</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>rats <span class="ot">&lt;-</span> rats <span class="sc">%&gt;%</span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Surgery =</span> <span class="fu">as.factor</span>(Surgery))</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a><span class="do">## CRD analysis</span></span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>rats_lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(logAUC <span class="sc">~</span> Surgery, <span class="at">data =</span> rats)</span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Prompt 1:  interpret the output from summary(rats_lm)$coef</span></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rats_lm)<span class="sc">$</span>coef</span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a><span class="do">## Prompt 2: interpret the output from anova(rats_lm)</span></span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(rats_lm)</span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a><span class="do">## Prompt 3, 4, and 5</span></span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a><span class="do">## what do the specified arguments to predictmeans() do? (HINT: use R help)</span></span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a><span class="do">## what are the elements returned in the object pm?</span></span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true" tabindex="-1"></a><span class="do">## specify plot = TRUE in the predictmeans() call, how would you interpret the resulting plot?</span></span>
<span id="cb45-16"><a href="#cb45-16" aria-hidden="true" tabindex="-1"></a>pm <span class="ot">&lt;-</span> predictmeans<span class="sc">::</span><span class="fu">predictmeans</span>(rats_lm , <span class="at">modelterm =</span> <span class="st">"Surgery"</span>, </span>
<span id="cb45-17"><a href="#cb45-17" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">pairwise =</span> <span class="cn">TRUE</span>, <span class="at">plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb45-18"><a href="#cb45-18" aria-hidden="true" tabindex="-1"></a>pm <span class="do">## a lot of output</span></span>
<span id="cb45-19"><a href="#cb45-19" aria-hidden="true" tabindex="-1"></a><span class="do">## Prompt 6: what are the three lines of code doing after this prompt?</span></span>
<span id="cb45-20"><a href="#cb45-20" aria-hidden="true" tabindex="-1"></a>url <span class="ot">&lt;-</span> <span class="st">"https://gist.github.com/cmjt/72f3941533a6bdad0701928cc2924b90"</span></span>
<span id="cb45-21"><a href="#cb45-21" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">source_gist</span>(url, <span class="at">quiet =</span> <span class="cn">TRUE</span>) <span class="do">## install using install.packages("devtools")</span></span>
<span id="cb45-22"><a href="#cb45-22" aria-hidden="true" tabindex="-1"></a><span class="fu">comparisons</span>(pm)</span>
<span id="cb45-23"><a href="#cb45-23" aria-hidden="true" tabindex="-1"></a><span class="do">## Prompt 7: explain each element of the calculations going into creating the object HSD</span></span>
<span id="cb45-24"><a href="#cb45-24" aria-hidden="true" tabindex="-1"></a>HSD <span class="ot">&lt;-</span> (<span class="fu">qtukey</span>(<span class="at">p =</span> <span class="dv">1</span> <span class="sc">-</span> <span class="fl">0.05</span>, <span class="at">nmeans =</span> <span class="dv">3</span>, <span class="at">df =</span> <span class="dv">12</span> <span class="sc">-</span> <span class="dv">3</span>)<span class="sc">/</span><span class="fu">sqrt</span>(<span class="dv">2</span>))<span class="sc">*</span><span class="fu">sqrt</span>(<span class="dv">2</span> <span class="sc">*</span> <span class="fu">anova</span>(rats_lm)[<span class="dv">2</span>,<span class="dv">3</span>] <span class="sc">/</span> <span class="dv">4</span>)</span>
<span id="cb45-25"><a href="#cb45-25" aria-hidden="true" tabindex="-1"></a><span class="fu">TukeyHSD</span>(<span class="fu">aov</span>(logAUC <span class="sc">~</span> Surgery, <span class="at">data =</span> rats))</span>
<span id="cb45-26"><a href="#cb45-26" aria-hidden="true" tabindex="-1"></a><span class="do">## Prompt 8: Use all the relevant pieces of the script (hint what is the alpha.adj object below?) to calculate the pairwise comparison between Surgery C and S, using Fisher’s, Bonferonni’s, and Tukey’s correction methods respectively. You should state the difference, and 95% CI in each case. What do you notice about these three intervals? How are they similar/dissimilar? Show your workings.</span></span>
<span id="cb45-27"><a href="#cb45-27" aria-hidden="true" tabindex="-1"></a>alpha.adj <span class="ot">&lt;-</span> <span class="fl">0.05</span><span class="sc">/</span><span class="fu">choose</span>(<span class="dv">3</span>,<span class="dv">2</span>)</span>
<span id="cb45-28"><a href="#cb45-28" aria-hidden="true" tabindex="-1"></a><span class="do">## Prompt 9: Produce a plot that compares the three methods above</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="case-study-02" class="level3">
<h3 class="anchored" data-anchor-id="case-study-02">Case study 02</h3>
<p>Revisiting <a href="#sec-mods" class="quarto-xref">Section&nbsp;8.2</a> may be useful here.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/factorial_expt.csv"</span>)</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="do">## below we change the treatments to factors, simply because later on if we</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="do">## use predictmeans it will have a tantrum if this isn't the case</span></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Disease =</span> <span class="fu">as.factor</span>(Disease)) <span class="sc">%&gt;%</span></span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Organ =</span> <span class="fu">as.factor</span>(Organ))</span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Prompt 1: what model has been fitted here and what does the output from the anova() call indicate?</span></span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">lm</span>(logAUC <span class="sc">~</span> Disease<span class="sc">*</span>Organ, <span class="at">data =</span> data)</span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(model)</span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Prompt 2: what model has been fitted here and what does the output from the anova() call indicate?</span></span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a>model_2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(logAUC <span class="sc">~</span> Organ<span class="sc">*</span>Disease, <span class="at">data =</span> data)</span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(model_2)</span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a><span class="do">#####################################</span></span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a><span class="do">##****making the data unbalanced***</span></span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a>data_sub <span class="ot">&lt;-</span> data[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">10</span>),]</span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a><span class="do">#####################################</span></span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a><span class="do">## Prompt 3: what model has been fitted here and what does the output from the anova() call indicate?</span></span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(logAUC <span class="sc">~</span> Disease<span class="sc">*</span>Organ, <span class="at">data =</span> data_sub)</span>
<span id="cb46-20"><a href="#cb46-20" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(mod)</span>
<span id="cb46-21"><a href="#cb46-21" aria-hidden="true" tabindex="-1"></a><span class="do">## Prompt 4: what model has been fitted here and what does the output from the anova() call indicate?</span></span>
<span id="cb46-22"><a href="#cb46-22" aria-hidden="true" tabindex="-1"></a>mod_2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(logAUC <span class="sc">~</span> Organ<span class="sc">*</span>Disease, <span class="at">data =</span> data_sub)</span>
<span id="cb46-23"><a href="#cb46-23" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(mod_2)</span>
<span id="cb46-24"><a href="#cb46-24" aria-hidden="true" tabindex="-1"></a><span class="do">## Prompt 5: run and interpret the output from the following code, what do you conclude?</span></span>
<span id="cb46-25"><a href="#cb46-25" aria-hidden="true" tabindex="-1"></a>pm <span class="ot">&lt;-</span> predictmeans<span class="sc">::</span><span class="fu">predictmeans</span>(<span class="at">model =</span> mod ,<span class="at">modelterm =</span> <span class="st">"Disease:Organ"</span>, </span>
<span id="cb46-26"><a href="#cb46-26" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">pairwise =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="case-study-03" class="level3">
<h3 class="anchored" data-anchor-id="case-study-03">Case study 03</h3>
<p>Revisiting <a href="#sec-randomeff" class="quarto-xref">Section&nbsp;9.4</a> and <a href="#sec-ss" class="quarto-xref">Section&nbsp;8.2.2</a> may be useful here as well as <a href="https://biosci738.github.io/BIOSCI738/linear-mixed-effect-models-lmms.html">this section of the course guide</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(tidyverse)</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(lme4)</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(predictmeans)</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>rcbd <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/rcbd.csv"</span>)</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a><span class="do">## turn appropriate variables into factors</span></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>rcbd <span class="ot">&lt;-</span> rcbd <span class="sc">%&gt;%</span></span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Run =</span> <span class="fu">as.factor</span>(Run)) <span class="sc">%&gt;%</span></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Surgery =</span> <span class="fu">as.factor</span>(Surgery))</span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a><span class="do">## run as a fixed effect</span></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a><span class="fu">lm</span>(logAUC8 <span class="sc">~</span> Run <span class="sc">+</span> Surgery, <span class="at">data =</span> rcbd) <span class="sc">|&gt;</span> <span class="fu">summary</span>()</span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a><span class="do">## run as a random effect, what's the difference?</span></span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a>lmm <span class="ot">&lt;-</span> lme4<span class="sc">::</span><span class="fu">lmer</span>(logAUC8 <span class="sc">~</span> Surgery <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>Run), <span class="at">data =</span> rcbd) </span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lmm)</span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a><span class="do">## diagnostics estimated variance partitioning and more...</span></span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a><span class="do">## these are really useful utility functions!</span></span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a>predictmeans<span class="sc">::</span><span class="fu">residplot</span>(lmm)</span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a>predictmeans<span class="sc">::</span><span class="fu">R2_glmm</span>(lmm)</span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a>predictmeans<span class="sc">::</span><span class="fu">se_ranef</span>(lmm)</span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a>predictmeans<span class="sc">::</span><span class="fu">varcomp</span>(lmm)</span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a>predictmeans<span class="sc">::</span><span class="fu">permmodels</span>(lmm) <span class="do">## remember permutation tests, what are we using them for here? Compare to a summary() output</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="case-study-04" class="level3">
<h3 class="anchored" data-anchor-id="case-study-04">Case study 04</h3>
<p>Now, we’re going to get a little more advanced and model some data from <span class="citation" data-cites="monkey">Bliss-Moreau and Baxter (<a href="#ref-monkey" role="doc-biblioref">2019</a>)</span> (data retrieved from <span class="citation" data-cites="monkeydata">Bliss-Moreau and Baxter (<a href="#ref-monkeydata" role="doc-biblioref">2020</a>)</span>). To do so we’re following along (somewhat) with the steps summarised in <span class="citation" data-cites="danblog">Dan (<a href="#ref-danblog" role="doc-biblioref">2022</a>)</span>. Revisiting <a href="#sec-randomeff" class="quarto-xref">Section&nbsp;9.4</a> may, again, be useful here.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="do">## wrangle as per Dan's blog</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(tidyverse)</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/monkey.csv"</span>)</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> age, <span class="at">y =</span> Activity)) <span class="sc">+</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() </span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Number of active intervals in the first two minutes (0-8)</span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>activity_2mins <span class="ot">&lt;-</span> data <span class="sc">|&gt;</span></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(obs<span class="sc">&lt;</span><span class="dv">9</span>) <span class="sc">|&gt;</span> <span class="fu">group_by</span>(subj_id, Day) <span class="sc">|&gt;</span></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">total=</span><span class="fu">sum</span>(Activity), </span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>            <span class="at">active_bins =</span> <span class="fu">sum</span>(Activity <span class="sc">&gt;</span> <span class="dv">0</span>), </span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>            <span class="at">age =</span> <span class="fu">min</span>(age)) <span class="sc">|&gt;</span></span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">monkey =</span> subj_id, <span class="at">day =</span> Day) <span class="sc">|&gt;</span></span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a><span class="fu">ungroup</span>()</span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">table</span>(activity_2mins<span class="sc">$</span>monkey))</span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>activity_2mins <span class="sc">%&gt;%</span></span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> age, <span class="at">y =</span> total, <span class="at">col =</span> <span class="fu">as.factor</span>(monkey))) <span class="sc">+</span></span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>day) <span class="sc">+</span></span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a><span class="do">## Linear model...</span></span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a>fit_lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(active_bins <span class="sc">~</span> age<span class="sc">*</span><span class="fu">factor</span>(day) <span class="sc">+</span> <span class="fu">factor</span>(monkey), <span class="at">data =</span> activity_2mins)</span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a>fit_lm <span class="sc">|&gt;</span></span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>()</span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a><span class="do">## ignore monkeys</span></span>
<span id="cb48-25"><a href="#cb48-25" aria-hidden="true" tabindex="-1"></a>fit_lm_pool <span class="ot">&lt;-</span> <span class="fu">lm</span>(active_bins <span class="sc">~</span> age<span class="sc">*</span><span class="fu">factor</span>(day), <span class="at">data =</span> activity_2mins)</span>
<span id="cb48-26"><a href="#cb48-26" aria-hidden="true" tabindex="-1"></a>fit_lm_pool <span class="sc">|&gt;</span></span>
<span id="cb48-27"><a href="#cb48-27" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>()</span>
<span id="cb48-28"><a href="#cb48-28" aria-hidden="true" tabindex="-1"></a>fit_lm_pool <span class="sc">|&gt;</span> gglm<span class="sc">::</span><span class="fu">gglm</span>() <span class="do">## and what do we think here?</span></span>
<span id="cb48-29"><a href="#cb48-29" aria-hidden="true" tabindex="-1"></a><span class="do">## plot</span></span>
<span id="cb48-30"><a href="#cb48-30" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(fit_lm_pool<span class="sc">$</span>fitted, fit_lm_pool<span class="sc">$</span>residuals)</span>
<span id="cb48-31"><a href="#cb48-31" aria-hidden="true" tabindex="-1"></a><span class="do">## why scale?</span></span>
<span id="cb48-32"><a href="#cb48-32" aria-hidden="true" tabindex="-1"></a>age_centre <span class="ot">&lt;-</span> <span class="fu">mean</span>(activity_2mins<span class="sc">$</span>age)</span>
<span id="cb48-33"><a href="#cb48-33" aria-hidden="true" tabindex="-1"></a>age_scale <span class="ot">&lt;-</span> <span class="fu">diff</span>(<span class="fu">range</span>(activity_2mins<span class="sc">$</span>age))<span class="sc">/</span><span class="dv">2</span></span>
<span id="cb48-34"><a href="#cb48-34" aria-hidden="true" tabindex="-1"></a>active_bins_centre <span class="ot">&lt;-</span> <span class="dv">4</span></span>
<span id="cb48-35"><a href="#cb48-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-36"><a href="#cb48-36" aria-hidden="true" tabindex="-1"></a>activity_2mins_scaled <span class="ot">&lt;-</span> activity_2mins <span class="sc">|&gt;</span></span>
<span id="cb48-37"><a href="#cb48-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">monkey =</span> <span class="fu">factor</span>(monkey),</span>
<span id="cb48-38"><a href="#cb48-38" aria-hidden="true" tabindex="-1"></a>         <span class="at">day =</span> <span class="fu">factor</span>(day),</span>
<span id="cb48-39"><a href="#cb48-39" aria-hidden="true" tabindex="-1"></a>         <span class="at">age_centred =</span> (age <span class="sc">-</span> age_centre)<span class="sc">/</span>age_scale,</span>
<span id="cb48-40"><a href="#cb48-40" aria-hidden="true" tabindex="-1"></a>         <span class="at">active_bins_scaled =</span> (active_bins <span class="sc">-</span> active_bins_centre)<span class="sc">/</span><span class="dv">4</span>)</span>
<span id="cb48-41"><a href="#cb48-41" aria-hidden="true" tabindex="-1"></a><span class="fu">glimpse</span>(activity_2mins_scaled)</span>
<span id="cb48-42"><a href="#cb48-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-43"><a href="#cb48-43" aria-hidden="true" tabindex="-1"></a><span class="do">## Monkey as a random effect, why? Is this sensible?</span></span>
<span id="cb48-44"><a href="#cb48-44" aria-hidden="true" tabindex="-1"></a><span class="fu">aov</span>(active_bins_scaled <span class="sc">~</span> age_centred<span class="sc">*</span>day <span class="sc">+</span> <span class="fu">Error</span>(monkey), <span class="at">data =</span> activity_2mins_scaled) <span class="sc">|&gt;</span></span>
<span id="cb48-45"><a href="#cb48-45" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>()</span>
<span id="cb48-46"><a href="#cb48-46" aria-hidden="true" tabindex="-1"></a><span class="do">## formula</span></span>
<span id="cb48-47"><a href="#cb48-47" aria-hidden="true" tabindex="-1"></a>formula <span class="ot">&lt;-</span> active_bins_scaled <span class="sc">~</span> age_centred<span class="sc">*</span>day <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> monkey)</span>
<span id="cb48-48"><a href="#cb48-48" aria-hidden="true" tabindex="-1"></a><span class="do">## lme4</span></span>
<span id="cb48-49"><a href="#cb48-49" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lme4)</span>
<span id="cb48-50"><a href="#cb48-50" aria-hidden="true" tabindex="-1"></a>fit_lme4 <span class="ot">&lt;-</span> lme4<span class="sc">::</span><span class="fu">lmer</span>(formula, activity_2mins_scaled)</span>
<span id="cb48-51"><a href="#cb48-51" aria-hidden="true" tabindex="-1"></a>fit_lme4 <span class="sc">|&gt;</span></span>
<span id="cb48-52"><a href="#cb48-52" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>()</span>
<span id="cb48-53"><a href="#cb48-53" aria-hidden="true" tabindex="-1"></a>predictmeans<span class="sc">::</span><span class="fu">predictmeans</span>(fit_lme4, <span class="st">"day"</span>)</span>
<span id="cb48-54"><a href="#cb48-54" aria-hidden="true" tabindex="-1"></a><span class="do">## lmerTest, is this the same model as above</span></span>
<span id="cb48-55"><a href="#cb48-55" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lmerTest)</span>
<span id="cb48-56"><a href="#cb48-56" aria-hidden="true" tabindex="-1"></a>fit_lmerTest <span class="ot">&lt;-</span> lmerTest<span class="sc">::</span><span class="fu">lmer</span>(formula, activity_2mins_scaled)</span>
<span id="cb48-57"><a href="#cb48-57" aria-hidden="true" tabindex="-1"></a>fit_lmerTest <span class="sc">|&gt;</span></span>
<span id="cb48-58"><a href="#cb48-58" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>()</span>
<span id="cb48-59"><a href="#cb48-59" aria-hidden="true" tabindex="-1"></a>predictmeans<span class="sc">::</span><span class="fu">predictmeans</span>(fit_lmerTest, <span class="st">"day"</span>)</span>
<span id="cb48-60"><a href="#cb48-60" aria-hidden="true" tabindex="-1"></a><span class="do">## glmmTMB, is this the same model as above</span></span>
<span id="cb48-61"><a href="#cb48-61" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmmTMB)</span>
<span id="cb48-62"><a href="#cb48-62" aria-hidden="true" tabindex="-1"></a>fit_glmmtmb <span class="ot">&lt;-</span> glmmTMB<span class="sc">::</span><span class="fu">glmmTMB</span>(formula, activity_2mins_scaled)</span>
<span id="cb48-63"><a href="#cb48-63" aria-hidden="true" tabindex="-1"></a>fit_glmmtmb <span class="sc">|&gt;</span> <span class="fu">summary</span>()</span>
<span id="cb48-64"><a href="#cb48-64" aria-hidden="true" tabindex="-1"></a><span class="do">## and what does this show?</span></span>
<span id="cb48-65"><a href="#cb48-65" aria-hidden="true" tabindex="-1"></a>emmeans<span class="sc">::</span><span class="fu">emmeans</span>(fit_glmmtmb, <span class="at">specs =</span> <span class="st">"day"</span>) <span class="sc">|&gt;</span></span>
<span id="cb48-66"><a href="#cb48-66" aria-hidden="true" tabindex="-1"></a> <span class="fu">plot</span>(<span class="at">pairwise =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="sec-edterms" class="level2">
<h2 class="anchored" data-anchor-id="sec-edterms">The Terminology of Experimental Design</h2>
<section id="the-grammar-of-experimental-designs" class="level3">
<h3 class="anchored" data-anchor-id="the-grammar-of-experimental-designs">The Grammar of Experimental Designs<a href="#fn6" class="footnote-ref" id="fnref6" role="doc-noteref"><sup>6</sup></a></h3>
<p>This section briefly summaries the terms and concepts discussed in <a href="https://biosci738.github.io/BIOSCI738/introduction-to-the-design-and-analysis-of-experiments.html">this section of the course guide</a> and <a href="https://emitanaka.org/edibble-book/">Emi Tanaka’s <code>edibble</code> ebook</a>. Refer tho these sources for more details definitions.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"><p></p>
<div>
<pre class="mermaid mermaid-js">flowchart TB
  A[Define Objectives] --&gt; B(Design Experiment)
  B --&gt; C[Procedures for Collecting Data]
  B --&gt; D[Model for Analysis of Data]
  C --&gt; D
  D --&gt; C
  C --&gt; E[Analysis of Data \n Interpretation of Results]
  D --&gt; E
  E --&gt; F(Conclusions)
</pre>
</div>
<p></p><figcaption> A typical sequence of events when carrying out an experiment<a href="#fn7" class="footnote-ref" id="fnref7" role="doc-noteref"><sup>7</sup></a>.</figcaption> </figure><p></p>
</div>
</div>
</div>
<p><strong>Experimental design FAQs</strong></p>
<p><strong>Why are specific objectives important?</strong> <em>Defining specific objectives direct you towards writing focused statements about the investigative questions you want your experiment to answer. The objective(s) are what it is you are aiming to accomplish in the experiment. The objective(s) should be written in terms of a specific verb that describes what you are doing (e.g., what you are hoping to measure/analyze/determine/test etc.).</em></p>
<p><strong>What is the response variable?</strong> <em>The response (or dependent) variable is the focus of your experiment; it is the variable that you will measure to accomplish your objective. It should be meaningful (i.e., represent the objective) and measurable.</em></p>
<p><strong>What is an experimental factor?</strong> <em>An experimental factor refers to a variable whose values that will be varied in the experiment with the goal of understanding how changes in the factor’s values explain difference in the responses in the measured variables.</em></p>
<p><strong>Why list the experimental factors?</strong> <em>Listing the experimental factors (or treatments or independent variables) you will study in your experiment helps to organise variables and work out how they may help to explain observed changes in your measurable response(s). It is important that the experimental factor can be controlled during and between experimental runs. Variables that are thought to affect the response, but cannot be controlled<a href="#fn8" class="footnote-ref" id="fnref8" role="doc-noteref"><sup>8</sup></a>, cannot be included as an experimental factor.</em></p>
<p><strong>What is the experimental material?</strong> <em>Experimental material (or units) is the material upon which something will be changed. To identify this you should answer the following questions. 1) Who or what are the subjects of your experiment? 2) How will you apply the experimental treatments to them? and 3) How will they be managed for taking measurements on them?</em></p>
<p><strong>Is a ‘shared environment’ an issue?</strong> <em>In some experiments, while the treatments are independently applied to subjects, subsets of subjects from the same treatment group share the same environment (e.g., are housed together). Since these subjects share environmental conditions, the measurements made on them cannot be considered to be independent of one another<a href="#fn9" class="footnote-ref" id="fnref9" role="doc-noteref"><sup>9</sup></a></em></p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Out of lecture activity!
</div>
</div>
<div class="callout-body-container callout-body">
<p>Read and critique this blog! <a href="https://www.statisticsdonewrong.com/pseudoreplication.html">Pseudoreplication: choose your data wisely</a></p>
</div>
</div>
</section>
</section>
<section id="using-r-to-design-an-experiment" class="level2">
<h2 class="anchored" data-anchor-id="using-r-to-design-an-experiment">Using <code>R</code> to design an experiment</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In your groups…
</div>
</div>
<div class="callout-body-container callout-body">
<p>Choose one or more of the following <code>R</code> packages and use the functionality offered to structure/design the following experiment (i.e., figure out how to allocated treatments to units etc.).</p>
<p>Some experimental design <code>R</code> package options (feel free to use any others you come across)</p>
<ul>
<li><code>edibble</code></li>
<li><code>agricolae</code></li>
<li><code>AlgDesign</code></li>
</ul>
<p>Description of experiment to create a design for (summary of the study from <span class="citation" data-cites="grass">Steinauer et al. (<a href="#ref-grass" role="doc-biblioref">2015</a>)</span>).</p>
<p><img src="runsheets/img/experiment.png" class="img-fluid"></p>
<p>[Note: you may find <a href="https://statbiscuit.shinyapps.io/doe_form/">this application useful</a> to structure your thoughts.]</p>
</div>
</div>
</section>
<section id="sec-peershare09" class="level2">
<h2 class="anchored" data-anchor-id="sec-peershare09">Q&amp;A and peer-share</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Peer-Share
</div>
</div>
<div class="callout-body-container callout-body">
<p>Go over an activity from the class again. Feel free to ask Charlotte a question or two about it!</p>
</div>
</div>
</section>
</section>
<section id="april-11th" class="level1">
<h1>April 11th</h1>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
By the end of today’s seminar, you should be able to
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Discuss, describe &amp; critique different probability distributions</li>
<li>Appropriately map a random variable to a (likely) distribution</li>
</ul>
</div>
</div>
<section id="rough-timeline-10" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-10">Rough timeline</h2>
<section id="pm---0355-pm-4" class="level3">
<h3 class="anchored" data-anchor-id="pm---0355-pm-4"><strong>02:05 pm - 03:55 pm</strong></h3>
<ul>
<li><strong>02:05–02.55 pm</strong> Distributions (<a href="#sec-distributions" class="quarto-xref">Section&nbsp;11.2</a>)</li>
<li><strong>02:55–03:05 pm</strong> Break</li>
<li><strong>03:05–03:20 pm</strong> Introduction to GLMs (<a href="#sec-glms" class="quarto-xref">Section&nbsp;11.3</a>)</li>
<li><strong>03.20–03.55 pm</strong> Recap (<a href="#sec-recap" class="quarto-xref">Section&nbsp;11.4</a>)</li>
</ul>
</section>
<section id="randomly-allocated-groups-10" class="level3">
<h3 class="anchored" data-anchor-id="randomly-allocated-groups-10">Randomly allocated groups</h3>
<p><strong>It’s your responsibility to get into these, just look for the cows!</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">group_allocation</span>(names, <span class="at">seed =</span> <span class="dv">1317</span>, <span class="dv">9</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$`Group 1`
[1] "Yutao Yang"   "Yue Ma"       "Hnin Lwin"    "Robert Mihok" "Vivian Qian" 

$`Group 2`
[1] "Yingyi Wang"           "Pikitangarangi Ratapu" "Jianing Chen"         
[4] "Angel Loh"            

$`Group 3`
[1] "Shoule Du"        "Charlotte Sutton" "Yogapriya"        "Sunny Chen"      

$`Group 4`
[1] "Leyang Liu"            "Anqi Cheng"            "Daneshajeya Jeyavalan"
[4] "Barnaby Clegg-Shaw"   

$`Group 5`
[1] "Shuxiang He"        "Francesca Domanska" "Ziyi Wang"         
[4] "Karie Labidon"     

$`Group 6`
[1] "Keyi Sun"      "Tongzheng Yao" "He Li"         "Zichang Li"   

$`Group 7`
[1] "Alex Johnson"                 "Antonia Schebek-Fuerstenberg"
[3] "Jiaxin Liu"                   "Xinyue Zhang"                

$`Group 8`
[1] "Laura Munck"   "Zhixiao Zhang" "Julia Thain"   "Emma Akeroyd" 

$`Group 9`
[1] "Happy Liang"  "Congkai Meng" "Yutong Wang"  "Lana Najar"   "Yi Han"      </code></pre>
</div>
</div>
</section>
</section>
<section id="sec-distributions" class="level2">
<h2 class="anchored" data-anchor-id="sec-distributions">Distributions</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Head to the <a href="https://docs.google.com/document/d/1D50lbL0gYLSoJ9kC260e9enbWaOSUWdqJ-J7dBDgAAA/edit?usp=sharing">Google Doc</a>…
</div>
</div>
<div class="callout-body-container callout-body">
<p><em>Read</em> <a href="https://raw.githubusercontent.com/BIOSCI738/cowstats/main/img/distributions.png">this cheatsheet</a> and summarize the following for each distribution:</p>
<ul>
<li>What is the assumed random variable (i.e., variable being measured/modelled)</li>
<li>What form does this variable take (e.g., positive-only numbers, integer values etc.)</li>
<li>What are the parameters of each distribution and how do they relate to the mean?</li>
<li>Where does the encircled blue cow fall in terms of this herd’s estimated population statistics?</li>
</ul>
<p>Your group reporter should make sure to be prepared to report!</p>
</div>
</div>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Head to the <a href="https://docs.google.com/document/d/1D50lbL0gYLSoJ9kC260e9enbWaOSUWdqJ-J7dBDgAAA/edit?usp=sharing">Google Doc</a>…
</div>
</div>
<div class="callout-body-container callout-body">
<p>Charlotte will soon allocate each group a distribution.</p>
<p>Using any means you wish read up about that particular distribution and note the following</p>
<ol type="1">
<li>Name of the distribution</li>
<li>Parameter(s) of the distribution</li>
<li>The mean and variance of the distribution in terms of the parameter(s)</li>
<li>Support/characteristics of the distribution</li>
<li>How parameter values change the distribution’s shape (hint use R to simulate the random variable using (a) parameter(s) of your choice and then plot this distribution)</li>
<li>When might this distribution be used</li>
<li>Answer the distribution-specific question(s)</li>
</ol>
<p><strong>Your group reporter should make sure to be prepared to report to the class!</strong></p>
<p>Note: You may find this <a href="https://cmjt.shinyapps.io/probable/">RShiny application useful</a> as well as the following cheat sheet</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="runsheets/img/pdqr.png" class="img-fluid figure-img"></p>
<figcaption>The <code>p..()</code>, <code>d..()</code>, <code>q..()</code>, <code>r..()</code> functions in <code>R</code></figcaption>
</figure>
</div>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Bonus activity
</div>
</div>
<div class="callout-body-container callout-body">
<p>I have 11 chickens. Here is a nice picture hanging out in an area between their coop and the feeder.</p>
<p><img src="runsheets/img/hens.JPG" class="img-fluid"></p>
<p>Because I love stats (and chickens) I’m interested in modelling the number of chickens visible in this particular area at different times of the day. What distribution should I use?</p>
</div>
</div>
</section>
<section id="sec-glms" class="level2">
<h2 class="anchored" data-anchor-id="sec-glms">Introduction to Generalised Linear Models (GLMs)</h2>
<p><strong>Slides</strong></p>
<p align="center">
<iframe src="slides/06-01-intro_to_glms.pdf" height="400" width="700" title="Introduction to Generalised Linear Models">
</iframe>
</p>
<p align="center">
<a href="https://canvas.auckland.ac.nz/courses/120054/files/15131336?module_item_id=2590952"><strong>CANVAS slides</strong></a>
</p>
</section>
<section id="sec-recap" class="level2">
<h2 class="anchored" data-anchor-id="sec-recap">Recap</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i>
</div>
</div>
<div class="callout-body-container callout-body">
<ol type="1">
<li>Choose a topic from the content covered so far that you have struggled the most with (e.g., random effects, Bonferonni’s correction, resampling etc.).</li>
<li>Revisit the associated materials (e.g., code/courseguide section/cheatsheets/runsheets etc.).</li>
<li>As a group design an activity/materials that explain/illustrate the topic/concept.</li>
</ol>
<p><strong>Be prepared to present your activity to me/the class</strong> (all group members should expect to contribute).</p>
</div>
</div>
</section>
</section>
<section id="april-28th" class="level1">
<h1>April 28th</h1>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
By the end of today’s seminar, you should be able to
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Write out GLMMs</li>
<li>Assess GLMM fit</li>
</ul>
</div>
</div>
<section id="rough-timeline-11" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-11">Rough timeline</h2>
<ul>
<li><strong>09:05–09:30 am</strong> Guess the distribution (<a href="#sec-distribution" class="quarto-xref">Section&nbsp;12.2</a>)</li>
<li><strong>09:30–09:55 am</strong> Writing out (G)LM(m)s (<a href="#sec-writingglm" class="quarto-xref">Section&nbsp;12.3</a>)</li>
<li><strong>09:55–11:05 am</strong> Break</li>
<li><strong>11:00–11:45 am</strong> Assessing model fit (<a href="#sec-glmfit" class="quarto-xref">Section&nbsp;12.4</a>)
<ul>
<li>Class discussion</li>
<li>Group activities</li>
</ul></li>
<li><strong>11.45–11.55 am</strong> Q&amp;A and peer-share (<a href="#sec-peershare11" class="quarto-xref">Section&nbsp;12.5</a>)</li>
</ul>
</section>
<section id="sec-distribution" class="level2">
<h2 class="anchored" data-anchor-id="sec-distribution">Guess the distribution</h2>
<p>Recall the chooks from <a href="#sec-distributions" class="quarto-xref">Section&nbsp;11.2</a>? Now, let’s play a game.</p>
<center>
<p>
SCAN ME or head to <a href="https://dub.sh/biosci738" target="_blank">dub.sh/biosci738</a>
</p>
<img src="runsheets/img/qr_biosci738.svg" alt="dub.sh.biosci738">
</center>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
When prompted <em>Quick Write</em> the distribution
</div>
</div>
<div class="callout-body-container callout-body">
<ol>
<li>
<a href="https://en.wikipedia.org/wiki/Fast_%26_Furious" target="_blank">Dominic Toretto</a> counts how many cars pass by on a highway every 10 minutes while scouting for races.
</li>
<li>
<a href="https://en.wikipedia.org/wiki/Squid_Game" target="_blank">Squid Game contestants</a> roll a die repeatedly, aiming to get a 6 to progress to the next level.
</li>
<li>
<a href="https://en.wikipedia.org/wiki/Willy_Wonka_%26_the_Chocolate_Factory" target="_blank">Charlie Bucket</a> opens chocolate bars randomly to find golden tickets hidden among them.
</li>
<li>
<a href="https://en.wikipedia.org/wiki/Stranger_Things" target="_blank">Dr.&nbsp;Owens</a> from Stranger Things tests a portal detection device 10 times on known rift locations.
</li>
<li>
<a href="https://en.wikipedia.org/wiki/The_Bachelor_(American_TV_series)" target="_blank">The Bachelor contestants</a> have their heights recorded to design custom suits.
</li>
<li>
<a href="https://en.wikipedia.org/wiki/Moana_(2016_film)" target="_blank">Moana</a> measures the time intervals between large waves hitting a reef to plan canoe launches.
</li>
<li>
<a href="https://en.wikipedia.org/wiki/The_Witcher_(TV_series)" target="_blank">Geralt of Rivia</a> wants to understand how long his battle potions last under different combat conditions.
</li>
<li>
<a href="https://en.wikipedia.org/wiki/Toy_Story" target="_blank">Buzz Lightyear</a> needs to estimate the total number of Buzz Lightyear toys manufactured, to do so he uses the serial numbers on a few toys he finds.
</li>
</ol>
<p><em>[NOTE: Scenarios were created with the assistance of ChatGPT (OpenAI, 2025).]</em></p>
</div>
</div>
</section>
<section id="sec-writingglm" class="level2">
<h2 class="anchored" data-anchor-id="sec-writingglm">Writing out (G)LM(m)s</h2>
<section id="linear-models-lms" class="level3">
<h3 class="anchored" data-anchor-id="linear-models-lms">Linear Models (LMs)</h3>
<p>Using the the well modelled <a href="https://biosci738.github.io/BIOSCI738/module-3.html">course guide rats</a> as an example!</p>
<p><strong><code>R</code> code</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>base_url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/"</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>rats <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="fu">paste</span>(base_url, <span class="st">"crd_rats_data.csv"</span>, <span class="at">sep =</span> <span class="st">""</span>))</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>rats<span class="sc">$</span>Surgery <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(rats<span class="sc">$</span>Surgery)</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>rats_lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(logAUC <span class="sc">~</span> Surgery, <span class="at">data =</span> rats)</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rats_lm)<span class="sc">$</span>coef</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>            Estimate Std. Error    t value     Pr(&gt;|t|)
(Intercept)   8.4600  0.4102531 20.6214144 6.930903e-09
SurgeryP      0.4900  0.5801856  0.8445574 4.202408e-01
SurgeryS      3.0875  0.5801856  5.3215734 4.799872e-04</code></pre>
</div>
</div>
<p><strong>Writing out the fitted model</strong></p>
<div class="cell">
<div class="cell-output-display">
<p><span class="math display">\[
\begin{aligned}
\operatorname{\widehat{logAUC}} &amp;= 8.46 + 0.49(\operatorname{Surgery}_{\operatorname{P}}) + 3.09(\operatorname{Surgery}_{\operatorname{S}})
\end{aligned}
\]</span></p>
</div>
</div>
</section>
<section id="linear-mixed-models-lmms" class="level3">
<h3 class="anchored" data-anchor-id="linear-mixed-models-lmms">Linear Mixed Models (LMMs)</h3>
<p>Again, using the the well modelled <a href="https://biosci738.github.io/BIOSCI738/module-3.html">course guide rats</a> as an example!</p>
<p><strong><code>R</code> code</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>rcbd <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="fu">paste</span>(base_url, <span class="st">"rcbd.csv"</span>, <span class="at">sep =</span> <span class="st">""</span>))</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>rcbd<span class="sc">$</span>Run <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(rcbd<span class="sc">$</span>Run); rcbd<span class="sc">$</span>Surgery <span class="ot">=</span> <span class="fu">as.factor</span>(rcbd<span class="sc">$</span>Surgery)</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>lmer4_mod <span class="ot">&lt;-</span> lme4<span class="sc">::</span><span class="fu">lmer</span>(logAUC8 <span class="sc">~</span> Surgery <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>Run), <span class="at">data =</span> rcbd)</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a><span class="do">## fixed effect coefficients</span></span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lmer4_mod)<span class="sc">$</span>coef</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>            Estimate Std. Error  t value
(Intercept)    7.580  0.8552404 8.863005
SurgeryP       1.975  0.8505913 2.321914
SurgeryS       3.850  0.8505913 4.526263</code></pre>
</div>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="do">## random effect standard deviation</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lmer4_mod)<span class="sc">$</span>varcor</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Groups   Name        Std.Dev.
 Run      (Intercept) 1.2160  
 Residual             1.2029  </code></pre>
</div>
</div>
<p><strong>Writing out the fitted model</strong></p>
<div class="cell">
<div class="cell-output-display">
<p><span class="math display">\[
\begin{aligned}
  \operatorname{\widehat{logAUC8}}_{i}  &amp;\sim N \left(7.58_{\alpha_{j[i]}} + 1.97_{\beta_{1}}(\operatorname{Surgery}_{\operatorname{P}}) + 3.85_{\beta_{2}}(\operatorname{Surgery}_{\operatorname{S}}), \sigma^2 \right) \\
    \alpha_{j}  &amp;\sim N \left(0, 1.22 \right)
    \text{, for Run j = 1,} \dots \text{,J}
\end{aligned}
\]</span></p>
</div>
</div>
</section>
<section id="generalised-linear-models-glms" class="level3">
<h3 class="anchored" data-anchor-id="generalised-linear-models-glms">Generalised Linear Models (GLMs)</h3>
<p>Now using the <a href="https://biosci738.github.io/BIOSCI738/module-4.html">course guide mice</a> as an example!</p>
<p><strong><code>R</code> code</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(tidyverse)</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>mice <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="fu">paste</span>(base_url, <span class="st">"autism.csv"</span>, <span class="at">sep =</span> <span class="st">""</span>))</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>mice <span class="ot">&lt;-</span> mice <span class="sc">%&gt;%</span></span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(., <span class="at">col =</span> Treatment, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"Diagnosis"</span>, <span class="st">"Sex"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><em>Additive model</em></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>glm_mod_add <span class="ot">&lt;-</span> <span class="fu">glm</span>(MB_buried <span class="sc">~</span> Sex <span class="sc">+</span> Diagnosis , <span class="at">data =</span> mice, <span class="at">family =</span> <span class="fu">poisson</span>(<span class="at">link =</span> <span class="st">"log"</span>))</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(glm_mod_add)<span class="sc">$</span>coef</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>              Estimate Std. Error   z value     Pr(&gt;|z|)
(Intercept)  2.2383670 0.03812209 58.715751 0.000000e+00
SexMale     -0.1478114 0.05101554 -2.897380 3.762939e-03
DiagnosisNT -0.4090481 0.05464912 -7.484989 7.155331e-14</code></pre>
</div>
</div>
<p><em>Interaction model</em></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>glm_mod_int <span class="ot">&lt;-</span> <span class="fu">glm</span>(MB_buried <span class="sc">~</span> Sex <span class="sc">*</span> Diagnosis , <span class="at">data =</span> mice, <span class="at">family =</span> <span class="fu">poisson</span>(<span class="at">link =</span> <span class="st">"log"</span>))</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(glm_mod_int)<span class="sc">$</span>coef</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                       Estimate Std. Error    z value     Pr(&gt;|z|)
(Intercept)          2.17943051 0.04237136 51.4364071 0.0000000000
SexMale             -0.01928335 0.06150938 -0.3135027 0.7538987994
DiagnosisNT         -0.22286307 0.07299932 -3.0529473 0.0022660570
SexMale:DiagnosisNT -0.40640850 0.11011204 -3.6908635 0.0002234941</code></pre>
</div>
</div>
<p><strong>Writing out the fitted model</strong></p>
<p><em>Additive model</em></p>
<div class="cell">
<div class="cell-output-display">
<p><span class="math display">\[
\begin{aligned}
\log ({ \widehat{E( \operatorname{MB_buried} )} })  &amp;= 2.24 - 0.15(\operatorname{Sex}_{\operatorname{Male}}) - 0.41(\operatorname{Diagnosis}_{\operatorname{NT}})
\end{aligned}
\]</span></p>
</div>
</div>
<p><em>Interaction model</em></p>
<div class="cell">
<div class="cell-output-display">
<p><span class="math display">\[
\begin{aligned}
\log ({ \widehat{E( \operatorname{MB_buried} )} })  &amp;= 2.18 - 0.02(\operatorname{Sex}_{\operatorname{Male}}) - 0.22(\operatorname{Diagnosis}_{\operatorname{NT}})\ - \\
&amp;\quad 0.41(\operatorname{Sex}_{\operatorname{Male}} \times \operatorname{Diagnosis}_{\operatorname{NT}})
\end{aligned}
\]</span></p>
</div>
</div>
</section>
<section id="generalised-linear-mixed-models-glmms" class="level3">
<h3 class="anchored" data-anchor-id="generalised-linear-mixed-models-glmms">Generalised Linear Mixed Models (GLMMs)</h3>
<p>Again, using the <a href="https://biosci738.github.io/BIOSCI738/module-4.html">course guide mice</a> as an example!</p>
<p><strong><code>R</code> code</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>glmer4_mod <span class="ot">&lt;-</span> lme4<span class="sc">::</span><span class="fu">glmer</span>(MB_buried <span class="sc">~</span> Sex <span class="sc">*</span> Diagnosis <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>Donor), <span class="at">data =</span> mice, <span class="at">family =</span> <span class="fu">poisson</span>(<span class="at">link =</span> <span class="st">"log"</span>))</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a><span class="do">## fixed effect coefficients</span></span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(glmer4_mod)<span class="sc">$</span>coef</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                      Estimate Std. Error    z value      Pr(&gt;|z|)
(Intercept)          2.1712351 0.07128425 30.4588336 9.150829e-204
SexMale             -0.0396138 0.06202255 -0.6386999  5.230182e-01
DiagnosisNT         -0.1774183 0.11815966 -1.5015134  1.332228e-01
SexMale:DiagnosisNT -0.3944359 0.11116199 -3.5482984  3.877286e-04</code></pre>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="do">## random effect standard deviation</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(glmer4_mod)<span class="sc">$</span>varcor</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Groups Name        Std.Dev.
 Donor  (Intercept) 0.12476 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Same model using glmmTMB</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>glmmTMB_mod <span class="ot">&lt;-</span> glmmTMB<span class="sc">::</span><span class="fu">glmmTMB</span>(MB_buried <span class="sc">~</span> Sex <span class="sc">*</span> Diagnosis <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>Donor), <span class="at">data =</span> mice, <span class="at">family =</span> <span class="fu">poisson</span>(<span class="at">link =</span> <span class="st">"log"</span>))</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(glmmTMB_mod)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Family: poisson  ( log )
Formula:          MB_buried ~ Sex * Diagnosis + (1 | Donor)
Data: mice

     AIC      BIC   logLik deviance df.resid 
  1333.0   1349.6   -661.5   1323.0      201 

Random effects:

Conditional model:
 Groups Name        Variance Std.Dev.
 Donor  (Intercept) 0.01556  0.1248  
Number of obs: 206, groups:  Donor, 8

Conditional model:
                    Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept)          2.17123    0.07130  30.451  &lt; 2e-16 ***
SexMale             -0.03962    0.06206  -0.638 0.523257    
DiagnosisNT         -0.17742    0.11820  -1.501 0.133351    
SexMale:DiagnosisNT -0.39443    0.11124  -3.546 0.000392 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<p><strong>Writing out the fitted model</strong></p>
<div class="cell">
<div class="cell-output-display">
<p><span class="math display">\[
\begin{aligned}
  \operatorname{\widehat{MB\_buried}}_{i}  &amp;\sim \operatorname{Poisson}(\lambda_i) \\
    \log(\lambda_i) &amp;=2.17_{\alpha_{j[i]}} - 0.04_{\beta_{1}}(\operatorname{Sex}_{\operatorname{Male}}) \\
    \alpha_{j}  &amp;\sim N \left(-0.18_{\gamma_{1}^{\alpha}}(\operatorname{Diagnosis}_{\operatorname{NT}}) - 0.39_{\gamma_{2}^{\alpha}}(\operatorname{Diagnosis}_{\operatorname{NT}} \times \operatorname{Sex}_{\operatorname{Male}}), 0.12 \right)
    \text{, for Donor j = 1,} \dots \text{,J}
\end{aligned}
\]</span></p>
</div>
</div>
</section>
</section>
<section id="sec-glmfit" class="level2">
<h2 class="anchored" data-anchor-id="sec-glmfit">Assessing model fit</h2>
<p><strong>Slides</strong></p>
<p align="center">
<iframe src="slides/07-01_glm-model-diagnostics.pdf" height="400" width="700" title="Generalised Linear Model Diagnostics">
</iframe>
</p>
<p align="center">
<a href="https://canvas.auckland.ac.nz/courses/120054/files/15191581?module_item_id=2596043"><strong>CANVAS slides</strong></a>
</p>
<section id="case-studies" class="level3">
<h3 class="anchored" data-anchor-id="case-studies">Case studies</h3>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
We’ll work through this example together! Note that this aligns with the content in <a href="https://biosci738.github.io/BIOSCI738/model-diagnostics.html">this section of the courseguide</a>; however, here we go beyond what is covered there!
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
<p><strong>Bat Abundance</strong> (subset of data from <span class="citation" data-cites="bats">Barlow et al. (<a href="#ref-bats" role="doc-biblioref">2015</a>)</span>)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>bats <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/bats.csv"</span>)</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot</span></span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(bats, <span class="fu">aes</span>(<span class="at">x =</span> TotalCount, <span class="at">fill =</span> SpeciesID)) <span class="sc">+</span> <span class="fu">geom_histogram</span>(<span class="at">alpha =</span> <span class="fl">0.4</span>, <span class="at">position =</span> <span class="st">"identity"</span>)</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a><span class="do">## summarise</span></span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>bats <span class="sc">%&gt;%</span></span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(SpeciesID) <span class="sc">%&gt;%</span></span>
<span id="cb68-8"><a href="#cb68-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">mean_roosts =</span> <span class="fu">mean</span>(TotalCount))</span>
<span id="cb68-9"><a href="#cb68-9" aria-hidden="true" tabindex="-1"></a><span class="do">## question, difference between number of roosts for each species?</span></span>
<span id="cb68-10"><a href="#cb68-10" aria-hidden="true" tabindex="-1"></a><span class="do">## So let's fit a Linear Model</span></span>
<span id="cb68-11"><a href="#cb68-11" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(TotalCount <span class="sc">~</span> SpeciesID, <span class="at">data =</span> bats)</span>
<span id="cb68-12"><a href="#cb68-12" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span> <span class="fu">summary</span>()</span>
<span id="cb68-13"><a href="#cb68-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Distribution of residuals</span></span>
<span id="cb68-14"><a href="#cb68-14" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">residuals</span>(mod), <span class="at">xlab =</span> <span class="st">"Model residuals"</span>)</span>
<span id="cb68-15"><a href="#cb68-15" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> <span class="dv">0</span>, <span class="at">lwd =</span> <span class="dv">4</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb68-16"><a href="#cb68-16" aria-hidden="true" tabindex="-1"></a><span class="do">## Other residual plots</span></span>
<span id="cb68-17"><a href="#cb68-17" aria-hidden="true" tabindex="-1"></a>gglm<span class="sc">::</span><span class="fu">gglm</span>(mod)</span>
<span id="cb68-18"><a href="#cb68-18" aria-hidden="true" tabindex="-1"></a><span class="do">## Poisson model</span></span>
<span id="cb68-19"><a href="#cb68-19" aria-hidden="true" tabindex="-1"></a>gmod <span class="ot">&lt;-</span> <span class="fu">glm</span>(TotalCount <span class="sc">~</span> SpeciesID, <span class="at">data =</span> bats, <span class="at">family =</span> <span class="st">"poisson"</span>)</span>
<span id="cb68-20"><a href="#cb68-20" aria-hidden="true" tabindex="-1"></a>gmod <span class="sc">|&gt;</span> <span class="fu">summary</span>()</span>
<span id="cb68-21"><a href="#cb68-21" aria-hidden="true" tabindex="-1"></a><span class="do">## quantile residual Q-Q plot</span></span>
<span id="cb68-22"><a href="#cb68-22" aria-hidden="true" tabindex="-1"></a>quantile_residuals <span class="ot">&lt;-</span> statmod<span class="sc">::</span><span class="fu">qresiduals</span>(gmod)</span>
<span id="cb68-23"><a href="#cb68-23" aria-hidden="true" tabindex="-1"></a><span class="fu">qqnorm</span>(<span class="sc">!</span><span class="fu">is.infinite</span>(quantile_residuals))</span>
<span id="cb68-24"><a href="#cb68-24" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb68-25"><a href="#cb68-25" aria-hidden="true" tabindex="-1"></a><span class="do">## Pearson residuals, recall that under a poisson model the variance increases with</span></span>
<span id="cb68-26"><a href="#cb68-26" aria-hidden="true" tabindex="-1"></a><span class="do">## the mean, so the raw resids should have a spread that increases with fitted vals</span></span>
<span id="cb68-27"><a href="#cb68-27" aria-hidden="true" tabindex="-1"></a><span class="do">## but a Pearson's residual are resids/sqrt(var) so if model is "correct"</span></span>
<span id="cb68-28"><a href="#cb68-28" aria-hidden="true" tabindex="-1"></a><span class="do">## then the Peasron residuals should have constant spread...</span></span>
<span id="cb68-29"><a href="#cb68-29" aria-hidden="true" tabindex="-1"></a>resids_pearson <span class="ot">&lt;-</span> <span class="fu">residuals</span>(gmod, <span class="at">type =</span> <span class="st">"pearson"</span>)</span>
<span id="cb68-30"><a href="#cb68-30" aria-hidden="true" tabindex="-1"></a>resids <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Fitted =</span> gmod<span class="sc">$</span>fitted.values,</span>
<span id="cb68-31"><a href="#cb68-31" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Pearson_residuals"</span> <span class="ot">=</span> resids_pearson)</span>
<span id="cb68-32"><a href="#cb68-32" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(resids, <span class="fu">aes</span>(<span class="at">x =</span> Fitted, <span class="at">y =</span> Pearson_residuals)) <span class="sc">+</span></span>
<span id="cb68-33"><a href="#cb68-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>) <span class="sc">+</span> <span class="fu">theme_linedraw</span>() <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Pearson residuals"</span>)</span>
<span id="cb68-34"><a href="#cb68-34" aria-hidden="true" tabindex="-1"></a><span class="do">## Deviance</span></span>
<span id="cb68-35"><a href="#cb68-35" aria-hidden="true" tabindex="-1"></a>D <span class="ot">&lt;-</span> gmod<span class="sc">$</span>deviance</span>
<span id="cb68-36"><a href="#cb68-36" aria-hidden="true" tabindex="-1"></a>D</span>
<span id="cb68-37"><a href="#cb68-37" aria-hidden="true" tabindex="-1"></a><span class="do">## extract the residual degrees of freedom (n-k)</span></span>
<span id="cb68-38"><a href="#cb68-38" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> gmod<span class="sc">$</span>df.residual</span>
<span id="cb68-39"><a href="#cb68-39" aria-hidden="true" tabindex="-1"></a>df</span>
<span id="cb68-40"><a href="#cb68-40" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span> <span class="sc">-</span> <span class="fu">pchisq</span>(D, df)</span>
<span id="cb68-41"><a href="#cb68-41" aria-hidden="true" tabindex="-1"></a><span class="do">## but are the chi-squared assumptions met?</span></span>
<span id="cb68-42"><a href="#cb68-42" aria-hidden="true" tabindex="-1"></a><span class="do">## dispersion plot (by simulation!!)</span></span>
<span id="cb68-43"><a href="#cb68-43" aria-hidden="true" tabindex="-1"></a><span class="do">## dispersion = residual deviance / degrees of freedom</span></span>
<span id="cb68-44"><a href="#cb68-44" aria-hidden="true" tabindex="-1"></a><span class="do">## expected value is 1, when larger than 1 Neg binomial more appropriate</span></span>
<span id="cb68-45"><a href="#cb68-45" aria-hidden="true" tabindex="-1"></a>observed_dispersion <span class="ot">&lt;-</span>  gmod<span class="sc">$</span>deviance<span class="sc">/</span>df</span>
<span id="cb68-46"><a href="#cb68-46" aria-hidden="true" tabindex="-1"></a>nreps <span class="ot">&lt;-</span> <span class="dv">1000</span></span>
<span id="cb68-47"><a href="#cb68-47" aria-hidden="true" tabindex="-1"></a>dispersion <span class="ot">&lt;-</span> <span class="fu">numeric</span>(nreps)</span>
<span id="cb68-48"><a href="#cb68-48" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nreps){</span>
<span id="cb68-49"><a href="#cb68-49" aria-hidden="true" tabindex="-1"></a>    <span class="do">## gererate nobvs poisson rvs using fitted mean</span></span>
<span id="cb68-50"><a href="#cb68-50" aria-hidden="true" tabindex="-1"></a>    tmp <span class="ot">&lt;-</span> <span class="fu">rpois</span>(<span class="fu">nrow</span>(bats), <span class="at">lambda =</span> <span class="fu">fitted</span>(gmod))</span>
<span id="cb68-51"><a href="#cb68-51" aria-hidden="true" tabindex="-1"></a>    tmp_mod <span class="ot">&lt;-</span> <span class="fu">glm</span>(tmp <span class="sc">~</span> bats<span class="sc">$</span>SpeciesID, <span class="at">family =</span> <span class="st">"poisson"</span>)</span>
<span id="cb68-52"><a href="#cb68-52" aria-hidden="true" tabindex="-1"></a>    df_tmp <span class="ot">&lt;-</span> tmp_mod<span class="sc">$</span>df.residual</span>
<span id="cb68-53"><a href="#cb68-53" aria-hidden="true" tabindex="-1"></a>    dispersion[i] <span class="ot">&lt;-</span> tmp_mod<span class="sc">$</span>deviance<span class="sc">/</span>df_tmp</span>
<span id="cb68-54"><a href="#cb68-54" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb68-55"><a href="#cb68-55" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">data.frame</span>(<span class="at">dispersion =</span> dispersion), <span class="fu">aes</span>(<span class="at">x =</span> dispersion)) <span class="sc">+</span></span>
<span id="cb68-56"><a href="#cb68-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_histogram</span>()</span>
<span id="cb68-57"><a href="#cb68-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-58"><a href="#cb68-58" aria-hidden="true" tabindex="-1"></a><span class="do">## dispersion plot (by simulation)</span></span>
<span id="cb68-59"><a href="#cb68-59" aria-hidden="true" tabindex="-1"></a><span class="do">## from the DHARMa package</span></span>
<span id="cb68-60"><a href="#cb68-60" aria-hidden="true" tabindex="-1"></a>simulation <span class="ot">&lt;-</span> DHARMa<span class="sc">::</span><span class="fu">simulateResiduals</span>(<span class="at">fittedModel =</span> gmod, <span class="at">n =</span> nreps, <span class="at">refit =</span> <span class="cn">TRUE</span>)</span>
<span id="cb68-61"><a href="#cb68-61" aria-hidden="true" tabindex="-1"></a>DHARMa<span class="sc">::</span><span class="fu">testDispersion</span>(simulation)</span>
<span id="cb68-62"><a href="#cb68-62" aria-hidden="true" tabindex="-1"></a><span class="do">####################</span></span>
<span id="cb68-63"><a href="#cb68-63" aria-hidden="true" tabindex="-1"></a><span class="do">## negative binomial</span></span>
<span id="cb68-64"><a href="#cb68-64" aria-hidden="true" tabindex="-1"></a>nbmod <span class="ot">&lt;-</span> MASS<span class="sc">::</span><span class="fu">glm.nb</span>(TotalCount <span class="sc">~</span> SpeciesID, <span class="at">data =</span> bats)</span>
<span id="cb68-65"><a href="#cb68-65" aria-hidden="true" tabindex="-1"></a>nbmod <span class="sc">|&gt;</span> <span class="fu">summary</span>()</span>
<span id="cb68-66"><a href="#cb68-66" aria-hidden="true" tabindex="-1"></a><span class="do">## slopes not very different from the Poisson model</span></span>
<span id="cb68-67"><a href="#cb68-67" aria-hidden="true" tabindex="-1"></a><span class="do">## quantile residual Q-Q plot</span></span>
<span id="cb68-68"><a href="#cb68-68" aria-hidden="true" tabindex="-1"></a>quantile_residuals <span class="ot">&lt;-</span> statmod<span class="sc">::</span><span class="fu">qresiduals</span>(nbmod)</span>
<span id="cb68-69"><a href="#cb68-69" aria-hidden="true" tabindex="-1"></a><span class="fu">qqnorm</span>(quantile_residuals)</span>
<span id="cb68-70"><a href="#cb68-70" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>) <span class="do">## PHEW</span></span>
<span id="cb68-71"><a href="#cb68-71" aria-hidden="true" tabindex="-1"></a><span class="do">## dispersion plot (by simulation)</span></span>
<span id="cb68-72"><a href="#cb68-72" aria-hidden="true" tabindex="-1"></a><span class="do">## from the DHARMa package</span></span>
<span id="cb68-73"><a href="#cb68-73" aria-hidden="true" tabindex="-1"></a>simulation <span class="ot">&lt;-</span> DHARMa<span class="sc">::</span><span class="fu">simulateResiduals</span>(<span class="at">fittedModel =</span> nbmod, <span class="at">n =</span> nreps, <span class="at">refit =</span> <span class="cn">TRUE</span>)</span>
<span id="cb68-74"><a href="#cb68-74" aria-hidden="true" tabindex="-1"></a>DHARMa<span class="sc">::</span><span class="fu">testDispersion</span>(simulation)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In your groups work through the examples below. After each model is fitted assess its fit using the code/functions provided. Discuss each model’s suitability. Discuss the similarity and/or differences to previous models.
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
<p><strong>Lobster Survival</strong> (data from <span class="citation" data-cites="lobster">Wilkinson et al. (<a href="#ref-lobster" role="doc-biblioref">2015</a>)</span>)</p>
<p>Before working through the code below recap <a href="https://biosci738.github.io/BIOSCI738/logistic-regression.html#an-example-lobsters">this section of the course guide</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/lobster.csv"</span>)</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Model 1 ##</span></span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a>glm_mod_bern <span class="ot">&lt;-</span> <span class="fu">glm</span>(survived <span class="sc">~</span> size, <span class="at">family =</span> <span class="st">"binomial"</span>, <span class="at">data =</span> data)</span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(glm_mod_bern)</span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a>qresids <span class="ot">&lt;-</span> statmod<span class="sc">::</span><span class="fu">qresiduals</span>(glm_mod_bern)</span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a><span class="fu">qqnorm</span>(qresids)</span>
<span id="cb69-10"><a href="#cb69-10" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb69-11"><a href="#cb69-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> size, <span class="at">y =</span> survived)) <span class="sc">+</span> </span>
<span id="cb69-12"><a href="#cb69-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> .<span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb69-13"><a href="#cb69-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_smooth</span>(<span class="at">method=</span><span class="st">"glm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>, <span class="at">method.args =</span> <span class="fu">list</span>(<span class="at">family=</span>binomial), <span class="at">col =</span> <span class="st">"#782c26"</span>) <span class="sc">+</span> </span>
<span id="cb69-14"><a href="#cb69-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Carapace length (mm)"</span>) <span class="sc">+</span></span>
<span id="cb69-15"><a href="#cb69-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Juvenile lobster survival"</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Fitted logistic regression model"</span>) <span class="sc">+</span></span>
<span id="cb69-16"><a href="#cb69-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb69-17"><a href="#cb69-17" aria-hidden="true" tabindex="-1"></a><span class="do">## Deviance</span></span>
<span id="cb69-18"><a href="#cb69-18" aria-hidden="true" tabindex="-1"></a>D <span class="ot">&lt;-</span> glm_mod_bern<span class="sc">$</span>deviance</span>
<span id="cb69-19"><a href="#cb69-19" aria-hidden="true" tabindex="-1"></a>D</span>
<span id="cb69-20"><a href="#cb69-20" aria-hidden="true" tabindex="-1"></a><span class="do">## extract the residual degrees of freedom (n-k)</span></span>
<span id="cb69-21"><a href="#cb69-21" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> glm_mod_bern<span class="sc">$</span>df.residual</span>
<span id="cb69-22"><a href="#cb69-22" aria-hidden="true" tabindex="-1"></a>df</span>
<span id="cb69-23"><a href="#cb69-23" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span> <span class="sc">-</span> <span class="fu">pchisq</span>(D, df)</span>
<span id="cb69-24"><a href="#cb69-24" aria-hidden="true" tabindex="-1"></a><span class="do">### BUT are my Chi squared assumptions met?</span></span>
<span id="cb69-25"><a href="#cb69-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-26"><a href="#cb69-26" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb69-27"><a href="#cb69-27" aria-hidden="true" tabindex="-1"></a><span class="do">## Model 2 ##</span></span>
<span id="cb69-28"><a href="#cb69-28" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb69-29"><a href="#cb69-29" aria-hidden="true" tabindex="-1"></a>grouped <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb69-30"><a href="#cb69-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(size) <span class="sc">%&gt;%</span></span>
<span id="cb69-31"><a href="#cb69-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">y =</span> <span class="fu">sum</span>(survived), <span class="at">n =</span> <span class="fu">length</span>(survived), <span class="at">p =</span> <span class="fu">mean</span>(survived))</span>
<span id="cb69-32"><a href="#cb69-32" aria-hidden="true" tabindex="-1"></a>grouped</span>
<span id="cb69-33"><a href="#cb69-33" aria-hidden="true" tabindex="-1"></a>grouped <span class="sc">%&gt;%</span></span>
<span id="cb69-34"><a href="#cb69-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> size, <span class="at">y =</span> p)) <span class="sc">+</span></span>
<span id="cb69-35"><a href="#cb69-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Carapace length (mm)"</span>) <span class="sc">+</span></span>
<span id="cb69-36"><a href="#cb69-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Proportion survived"</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Survival rates of juvenile lobsters"</span>) <span class="sc">+</span></span>
<span id="cb69-37"><a href="#cb69-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()</span>
<span id="cb69-38"><a href="#cb69-38" aria-hidden="true" tabindex="-1"></a>glm_mod_binom <span class="ot">&lt;-</span> <span class="fu">glm</span>(<span class="fu">cbind</span>(y, n <span class="sc">-</span> y) <span class="sc">~</span> size, <span class="at">family =</span> <span class="st">"binomial"</span>, <span class="at">data =</span> grouped)</span>
<span id="cb69-39"><a href="#cb69-39" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(glm_mod_binom)</span>
<span id="cb69-40"><a href="#cb69-40" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(grouped, <span class="fu">aes</span>(<span class="at">x =</span> size, <span class="at">y =</span> p)) <span class="sc">+</span> </span>
<span id="cb69-41"><a href="#cb69-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> .<span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb69-42"><a href="#cb69-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_smooth</span>(<span class="at">method =</span> <span class="st">"glm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>,</span>
<span id="cb69-43"><a href="#cb69-43" aria-hidden="true" tabindex="-1"></a>              <span class="at">method.args =</span> <span class="fu">list</span>(<span class="at">family=</span>binomial), </span>
<span id="cb69-44"><a href="#cb69-44" aria-hidden="true" tabindex="-1"></a>              <span class="at">col =</span> <span class="st">"#782c26"</span>) <span class="sc">+</span> </span>
<span id="cb69-45"><a href="#cb69-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Carapace length (mm)"</span>) <span class="sc">+</span></span>
<span id="cb69-46"><a href="#cb69-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Proportion survived"</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Fitted logistic regression model"</span>) <span class="sc">+</span></span>
<span id="cb69-47"><a href="#cb69-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb69-48"><a href="#cb69-48" aria-hidden="true" tabindex="-1"></a><span class="do">## Deviance</span></span>
<span id="cb69-49"><a href="#cb69-49" aria-hidden="true" tabindex="-1"></a>D <span class="ot">&lt;-</span> glm_mod_binom<span class="sc">$</span>deviance</span>
<span id="cb69-50"><a href="#cb69-50" aria-hidden="true" tabindex="-1"></a>D</span>
<span id="cb69-51"><a href="#cb69-51" aria-hidden="true" tabindex="-1"></a><span class="do">## extract the residual degrees of freedom (n-k)</span></span>
<span id="cb69-52"><a href="#cb69-52" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> glm_mod_binom<span class="sc">$</span>df.residual</span>
<span id="cb69-53"><a href="#cb69-53" aria-hidden="true" tabindex="-1"></a>df</span>
<span id="cb69-54"><a href="#cb69-54" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span> <span class="sc">-</span> <span class="fu">pchisq</span>(D, df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Bird Abundance</strong> (data from <span class="citation" data-cites="birds">García-Navas et al. (<a href="#ref-birds" role="doc-biblioref">2022</a>)</span>)</p>
<blockquote class="blockquote">
<p>We investigated taxonomic and functional beta diversity of bird communities inhabiting Mediterranean olive groves subject to either intensive or low-intensity management of the ground cover and located in landscapes with different degrees of complexity.</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(tidyverse)</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(glmmTMB)</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/bird_abundance.csv"</span></span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a>birds <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(url) <span class="sc">%&gt;%</span></span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(., <span class="fu">c</span>(<span class="sc">-</span>OliveFarm, <span class="sc">-</span>Management, <span class="sc">-</span>Complexity), <span class="at">names_to =</span> <span class="st">"Species"</span>, <span class="at">values_to =</span> <span class="st">"Count"</span>)</span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a><span class="do">## subset for simplicity</span></span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a>birds <span class="ot">&lt;-</span> <span class="fu">subset</span>(birds, birds<span class="sc">$</span>Species <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Anthus_pratensis"</span>, <span class="st">"Corvus_corax"</span>, <span class="st">"Passer_montanus"</span>))</span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb70-9"><a href="#cb70-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Model 1 ##</span></span>
<span id="cb70-10"><a href="#cb70-10" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb70-11"><a href="#cb70-11" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(Count <span class="sc">~</span> Species, <span class="at">data =</span> birds)</span>
<span id="cb70-12"><a href="#cb70-12" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span> <span class="fu">summary</span>()</span>
<span id="cb70-13"><a href="#cb70-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Residual plots</span></span>
<span id="cb70-14"><a href="#cb70-14" aria-hidden="true" tabindex="-1"></a>gglm<span class="sc">::</span><span class="fu">gglm</span>(mod)</span>
<span id="cb70-15"><a href="#cb70-15" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb70-16"><a href="#cb70-16" aria-hidden="true" tabindex="-1"></a><span class="do">## Model 2 ##</span></span>
<span id="cb70-17"><a href="#cb70-17" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb70-18"><a href="#cb70-18" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">glm</span>(Count <span class="sc">~</span> Species, <span class="at">data =</span> birds, <span class="at">family =</span> <span class="st">"poisson"</span>)</span>
<span id="cb70-19"><a href="#cb70-19" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span> <span class="fu">summary</span>()</span>
<span id="cb70-20"><a href="#cb70-20" aria-hidden="true" tabindex="-1"></a><span class="do">## Pearson's residuls</span></span>
<span id="cb70-21"><a href="#cb70-21" aria-hidden="true" tabindex="-1"></a>resids_pearson <span class="ot">&lt;-</span> <span class="fu">residuals</span>(mod, <span class="at">type =</span> <span class="st">"pearson"</span>)</span>
<span id="cb70-22"><a href="#cb70-22" aria-hidden="true" tabindex="-1"></a>resids <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Fitted =</span> mod<span class="sc">$</span>fitted.values,</span>
<span id="cb70-23"><a href="#cb70-23" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"Pearson_residuals"</span> <span class="ot">=</span> resids_pearson)</span>
<span id="cb70-24"><a href="#cb70-24" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(resids, <span class="fu">aes</span>(<span class="at">x =</span> Fitted, <span class="at">y =</span> Pearson_residuals)) <span class="sc">+</span></span>
<span id="cb70-25"><a href="#cb70-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>) <span class="sc">+</span> <span class="fu">theme_linedraw</span>() <span class="sc">+</span> </span>
<span id="cb70-26"><a href="#cb70-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Pearson residuals"</span>)</span>
<span id="cb70-27"><a href="#cb70-27" aria-hidden="true" tabindex="-1"></a><span class="do">## quantile residual Q-Q plot</span></span>
<span id="cb70-28"><a href="#cb70-28" aria-hidden="true" tabindex="-1"></a>quantile_residuals <span class="ot">&lt;-</span> statmod<span class="sc">::</span><span class="fu">qresiduals</span>(mod)</span>
<span id="cb70-29"><a href="#cb70-29" aria-hidden="true" tabindex="-1"></a><span class="fu">qqnorm</span>(<span class="sc">!</span><span class="fu">is.infinite</span>(quantile_residuals))</span>
<span id="cb70-30"><a href="#cb70-30" aria-hidden="true" tabindex="-1"></a><span class="do">## Deviance</span></span>
<span id="cb70-31"><a href="#cb70-31" aria-hidden="true" tabindex="-1"></a>D <span class="ot">&lt;-</span> mod<span class="sc">$</span>deviance</span>
<span id="cb70-32"><a href="#cb70-32" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> mod<span class="sc">$</span>df.residual</span>
<span id="cb70-33"><a href="#cb70-33" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span> <span class="sc">-</span> <span class="fu">pchisq</span>(D, df)</span>
<span id="cb70-34"><a href="#cb70-34" aria-hidden="true" tabindex="-1"></a><span class="do">## dispersion plot (by simulation)</span></span>
<span id="cb70-35"><a href="#cb70-35" aria-hidden="true" tabindex="-1"></a><span class="do">## from the DHARMa package</span></span>
<span id="cb70-36"><a href="#cb70-36" aria-hidden="true" tabindex="-1"></a>simulation <span class="ot">&lt;-</span> DHARMa<span class="sc">::</span><span class="fu">simulateResiduals</span>(<span class="at">fittedModel =</span> mod, <span class="at">n =</span> <span class="dv">1000</span>, <span class="at">refit =</span> <span class="cn">TRUE</span>)</span>
<span id="cb70-37"><a href="#cb70-37" aria-hidden="true" tabindex="-1"></a>DHARMa<span class="sc">::</span><span class="fu">testDispersion</span>(simulation)</span>
<span id="cb70-38"><a href="#cb70-38" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb70-39"><a href="#cb70-39" aria-hidden="true" tabindex="-1"></a><span class="do">## Model 3 ##</span></span>
<span id="cb70-40"><a href="#cb70-40" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb70-41"><a href="#cb70-41" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">glmmTMB</span>(Count <span class="sc">~</span> Species <span class="sc">+</span> Management <span class="sc">+</span> Complexity <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>OliveFarm), <span class="at">data =</span> birds, </span>
<span id="cb70-42"><a href="#cb70-42" aria-hidden="true" tabindex="-1"></a>               <span class="at">family =</span> <span class="st">"poisson"</span>)</span>
<span id="cb70-43"><a href="#cb70-43" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span> <span class="fu">summary</span>()</span>
<span id="cb70-44"><a href="#cb70-44" aria-hidden="true" tabindex="-1"></a><span class="do">## % variation explained</span></span>
<span id="cb70-45"><a href="#cb70-45" aria-hidden="true" tabindex="-1"></a>predictmeans<span class="sc">::</span><span class="fu">R2_glmm</span>(mod)</span>
<span id="cb70-46"><a href="#cb70-46" aria-hidden="true" tabindex="-1"></a><span class="do">## deviance ...</span></span>
<span id="cb70-47"><a href="#cb70-47" aria-hidden="true" tabindex="-1"></a>D <span class="ot">&lt;-</span> <span class="fu">summary</span>(mod)[[<span class="st">"AICtab"</span>]][<span class="dv">4</span>] <span class="sc">|&gt;</span> <span class="fu">as.numeric</span>()</span>
<span id="cb70-48"><a href="#cb70-48" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">summary</span>(mod)[[<span class="st">"AICtab"</span>]][<span class="dv">5</span>] <span class="sc">|&gt;</span> <span class="fu">as.numeric</span>()</span>
<span id="cb70-49"><a href="#cb70-49" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span> <span class="sc">-</span> <span class="fu">pchisq</span>(D, df)</span>
<span id="cb70-50"><a href="#cb70-50" aria-hidden="true" tabindex="-1"></a><span class="do">## dispersion plot (by simulation)</span></span>
<span id="cb70-51"><a href="#cb70-51" aria-hidden="true" tabindex="-1"></a><span class="do">## from the DHARMa package</span></span>
<span id="cb70-52"><a href="#cb70-52" aria-hidden="true" tabindex="-1"></a>simulation <span class="ot">&lt;-</span> DHARMa<span class="sc">::</span><span class="fu">simulateResiduals</span>(<span class="at">fittedModel =</span> mod, <span class="at">n =</span> <span class="dv">1000</span>)</span>
<span id="cb70-53"><a href="#cb70-53" aria-hidden="true" tabindex="-1"></a>DHARMa<span class="sc">::</span><span class="fu">testDispersion</span>(simulation)</span>
<span id="cb70-54"><a href="#cb70-54" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb70-55"><a href="#cb70-55" aria-hidden="true" tabindex="-1"></a><span class="do">## Model 4 ##</span></span>
<span id="cb70-56"><a href="#cb70-56" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb70-57"><a href="#cb70-57" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">glmmTMB</span>(Count <span class="sc">~</span> Species <span class="sc">+</span>  Management <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>OliveFarm<span class="sc">/</span>Complexity) , <span class="at">data =</span> birds,</span>
<span id="cb70-58"><a href="#cb70-58" aria-hidden="true" tabindex="-1"></a>               <span class="at">family =</span> <span class="st">"poisson"</span>)</span>
<span id="cb70-59"><a href="#cb70-59" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span> <span class="fu">summary</span>()</span>
<span id="cb70-60"><a href="#cb70-60" aria-hidden="true" tabindex="-1"></a><span class="do">## % variation explained</span></span>
<span id="cb70-61"><a href="#cb70-61" aria-hidden="true" tabindex="-1"></a>predictmeans<span class="sc">::</span><span class="fu">R2_glmm</span>(mod)</span>
<span id="cb70-62"><a href="#cb70-62" aria-hidden="true" tabindex="-1"></a><span class="do">## dispersion plot (by simulation)</span></span>
<span id="cb70-63"><a href="#cb70-63" aria-hidden="true" tabindex="-1"></a><span class="do">## from the DHARMa package</span></span>
<span id="cb70-64"><a href="#cb70-64" aria-hidden="true" tabindex="-1"></a>simulation <span class="ot">&lt;-</span> DHARMa<span class="sc">::</span><span class="fu">simulateResiduals</span>(<span class="at">fittedModel =</span> mod, <span class="at">n =</span> <span class="dv">1000</span>)</span>
<span id="cb70-65"><a href="#cb70-65" aria-hidden="true" tabindex="-1"></a>DHARMa<span class="sc">::</span><span class="fu">testDispersion</span>(simulation)</span>
<span id="cb70-66"><a href="#cb70-66" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb70-67"><a href="#cb70-67" aria-hidden="true" tabindex="-1"></a><span class="do">## Model 5 ##</span></span>
<span id="cb70-68"><a href="#cb70-68" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb70-69"><a href="#cb70-69" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">glmmTMB</span>(Count <span class="sc">~</span> Species <span class="sc">+</span>  Management <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>OliveFarm) , <span class="at">data =</span> birds, </span>
<span id="cb70-70"><a href="#cb70-70" aria-hidden="true" tabindex="-1"></a>               <span class="at">family =</span> <span class="st">"nbinom2"</span>)</span>
<span id="cb70-71"><a href="#cb70-71" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span> <span class="fu">summary</span>()</span>
<span id="cb70-72"><a href="#cb70-72" aria-hidden="true" tabindex="-1"></a><span class="do">## % variation explained</span></span>
<span id="cb70-73"><a href="#cb70-73" aria-hidden="true" tabindex="-1"></a>predictmeans<span class="sc">::</span><span class="fu">R2_glmm</span>(mod)</span>
<span id="cb70-74"><a href="#cb70-74" aria-hidden="true" tabindex="-1"></a><span class="do">## dispersion plot (by simulation)</span></span>
<span id="cb70-75"><a href="#cb70-75" aria-hidden="true" tabindex="-1"></a><span class="do">## from the DHARMa package</span></span>
<span id="cb70-76"><a href="#cb70-76" aria-hidden="true" tabindex="-1"></a>simulation <span class="ot">&lt;-</span> DHARMa<span class="sc">::</span><span class="fu">simulateResiduals</span>(<span class="at">fittedModel =</span> mod, <span class="at">n =</span> <span class="dv">1000</span>)</span>
<span id="cb70-77"><a href="#cb70-77" aria-hidden="true" tabindex="-1"></a>DHARMa<span class="sc">::</span><span class="fu">testDispersion</span>(simulation)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="sec-peershare11" class="level2">
<h2 class="anchored" data-anchor-id="sec-peershare11">Q&amp;A and peer-share</h2>
<section id="vote-r-or-me" class="level4">
<h4 class="anchored" data-anchor-id="vote-r-or-me">VOTE: <strong>R</strong> or <strong>ME</strong></h4>
<center>
<p>
<a href="https://dub.sh/biosci738" target="_blank">dub.sh/biosci738</a>
</p>
<img src="runsheets/img/qr_biosci738.svg" alt="dub.sh.biosci738">
</center>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Peer-Share
</div>
</div>
<div class="callout-body-container callout-body">
<p>Go over an activity from the class again or discuss task 07…Feel free to ask Charlotte a question or two!</p>
</div>
</div>
</section>
</section>
</section>
<section id="may-2nd" class="level1">
<h1>May 2nd</h1>
<section id="rough-timeline-12" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-12">Rough timeline</h2>
<section id="pm---0355-pm-5" class="level3">
<h3 class="anchored" data-anchor-id="pm---0355-pm-5"><strong>02:05 pm - 03:55 pm</strong></h3>
<ul>
<li><strong>02:05–02.50 pm</strong> Recap (<a href="#sec-recap02" class="quarto-xref">Section&nbsp;13.2</a>)</li>
<li><strong>02:50–03:00 pm</strong> Break</li>
<li><strong>03:00–03:55 pm</strong> Introduction to MLE (<a href="#sec-mle" class="quarto-xref">Section&nbsp;13.3</a>)
<ul>
<li>Class discussion</li>
<li>Group activity</li>
</ul></li>
</ul>
</section>
</section>
<section id="sec-recap02" class="level2">
<h2 class="anchored" data-anchor-id="sec-recap02">Recap</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Revisit…
</div>
</div>
<div class="callout-body-container callout-body">
<p>…the following case studies. In each case, write out your chosen model and create a plot/calculate a measure that demonstrates the “fit” of that model.</p>
<ol type="1">
<li><strong>Fry Time</strong> <a href="#sec-fry" class="quarto-xref">Section&nbsp;6.4.2</a> (data from <span class="citation" data-cites="fries">Abdul Hamid, Omar, and Sanny (<a href="#ref-fries" role="doc-biblioref">2019</a>)</span>)</li>
<li><strong>Monkeys</strong> case study 04 from <a href="#sec-casestudies" class="quarto-xref">Section&nbsp;10.2</a> (data from <span class="citation" data-cites="monkey">Bliss-Moreau and Baxter (<a href="#ref-monkey" role="doc-biblioref">2019</a>)</span>)</li>
<li><strong>Bird Abundance</strong> third case study from <a href="#sec-glmfit" class="quarto-xref">Section&nbsp;12.4</a> (data from <span class="citation" data-cites="birds">García-Navas et al. (<a href="#ref-birds" role="doc-biblioref">2022</a>)</span>)</li>
</ol>
<p><strong>Be prepared to present your activity to me/the class</strong> (all group members should expect to contribute).</p>
</div>
</div>
</section>
<section id="sec-mle" class="level2">
<h2 class="anchored" data-anchor-id="sec-mle">Maximum Likelihood Estimation (MLE)</h2>
<p align="center">
<iframe src="slides/mle_summary.html" height="600" width="800" title="Maximum Likelihood Estimation">
</iframe>
</p>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In your groups…
</div>
</div>
<div class="callout-body-container callout-body">
<p>“Read” and summarise the following cheatsheets.</p>
<ol type="1">
<li><a href="https://github.com/BIOSCI738/cowstats/blob/main/img/mle_p.png?raw=true">MLE (p)</a></li>
<li><a href="https://github.com/BIOSCI738/cowstats/blob/main/img/mle_mu_sigma.png?raw=true">MLE (<span class="math inline">\(\mu\)</span> &amp; <span class="math inline">\(\sigma\)</span>)</a></li>
</ol>
</div>
</div>
<p><strong>Slides</strong></p>
<p align="center">
<iframe src="slides/07-02_maximum-likelihood.pdf" height="400" width="700" title="Maximum Likelihood Estimation">
</iframe>
</p>
<p align="center">
<a href="https://canvas.auckland.ac.nz/courses/120054/files/15211515?module_item_id=2598102"><strong>CANVAS slides</strong></a>
</p>
<section id="case-studies-1" class="level3">
<h3 class="anchored" data-anchor-id="case-studies-1">Case studies</h3>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
We’ll work through this example together! Note that this aligns with the content in <a href="https://biosci738.github.io/BIOSCI738/maximum-likelihood-estimation.html">this section of the courseguide</a>.
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(tidyverse)</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(patchwork)</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Binomial #</span></span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a>likelihood <span class="ot">&lt;-</span> <span class="cf">function</span>(theta) <span class="fu">dbinom</span>(<span class="at">x =</span> <span class="dv">79</span>, <span class="at">size =</span> <span class="dv">159</span>, <span class="at">prob =</span> theta)</span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a><span class="fu">optimise</span>(likelihood, <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">maximum =</span> <span class="cn">TRUE</span>)</span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-10"><a href="#cb71-10" aria-hidden="true" tabindex="-1"></a>log_likelihood <span class="ot">&lt;-</span> <span class="cf">function</span>(theta) <span class="fu">dbinom</span>(<span class="at">x =</span> <span class="dv">79</span>, <span class="at">size =</span> <span class="dv">159</span>, <span class="at">prob =</span> theta, <span class="at">log =</span> <span class="cn">TRUE</span>)</span>
<span id="cb71-11"><a href="#cb71-11" aria-hidden="true" tabindex="-1"></a><span class="fu">optimise</span>(log_likelihood, <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">maximum =</span> <span class="cn">TRUE</span>)</span>
<span id="cb71-12"><a href="#cb71-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-13"><a href="#cb71-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-14"><a href="#cb71-14" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb71-15"><a href="#cb71-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Poisson ##</span></span>
<span id="cb71-16"><a href="#cb71-16" aria-hidden="true" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb71-17"><a href="#cb71-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-18"><a href="#cb71-18" aria-hidden="true" tabindex="-1"></a>likelihood <span class="ot">&lt;-</span> <span class="cf">function</span>(lambda) <span class="fu">dpois</span>(<span class="at">x =</span> <span class="dv">54</span>, lambda)</span>
<span id="cb71-19"><a href="#cb71-19" aria-hidden="true" tabindex="-1"></a><span class="fu">optimise</span>(likelihood, <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>), <span class="at">maximum =</span> <span class="cn">TRUE</span>)</span>
<span id="cb71-20"><a href="#cb71-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-21"><a href="#cb71-21" aria-hidden="true" tabindex="-1"></a>log_likelihood <span class="ot">&lt;-</span> <span class="cf">function</span>(lambda) <span class="fu">dpois</span>(<span class="at">x =</span> <span class="dv">54</span>, lambda, <span class="at">log =</span> <span class="cn">TRUE</span>)</span>
<span id="cb71-22"><a href="#cb71-22" aria-hidden="true" tabindex="-1"></a><span class="fu">optimise</span>(log_likelihood, <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>), <span class="at">maximum =</span> <span class="cn">TRUE</span>)</span>
<span id="cb71-23"><a href="#cb71-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-24"><a href="#cb71-24" aria-hidden="true" tabindex="-1"></a><span class="do">################################</span></span>
<span id="cb71-25"><a href="#cb71-25" aria-hidden="true" tabindex="-1"></a><span class="do">## plots for Poisson example </span><span class="al">###</span></span>
<span id="cb71-26"><a href="#cb71-26" aria-hidden="true" tabindex="-1"></a><span class="do">################################</span></span>
<span id="cb71-27"><a href="#cb71-27" aria-hidden="true" tabindex="-1"></a>lam <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">100</span>, <span class="at">length.out =</span> <span class="dv">500</span>)</span>
<span id="cb71-28"><a href="#cb71-28" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">lam =</span> lam, <span class="at">likelihood =</span>  <span class="fu">dpois</span>(<span class="at">x =</span> <span class="dv">54</span>, lam)) <span class="sc">%&gt;%</span></span>
<span id="cb71-29"><a href="#cb71-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb71-30"><a href="#cb71-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> lam, <span class="at">y =</span> likelihood)) <span class="sc">+</span></span>
<span id="cb71-31"><a href="#cb71-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_linedraw</span>() <span class="sc">+</span> <span class="fu">xlab</span>(<span class="fu">expression</span>(lambda)) <span class="sc">+</span></span>
<span id="cb71-32"><a href="#cb71-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">"L("</span>, <span class="sc">~</span>lambda, <span class="st">";s)"</span>, <span class="at">sep =</span> <span class="st">" "</span>))) <span class="sc">+</span></span>
<span id="cb71-33"><a href="#cb71-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">54</span>,  <span class="at">col =</span> <span class="st">"#782c26"</span>, <span class="at">linewidth =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb71-34"><a href="#cb71-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Likelihood for X ~ Poisson(54)"</span>) <span class="sc">+</span></span>
<span id="cb71-35"><a href="#cb71-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">75</span>, <span class="at">y =</span> <span class="fl">0.04</span>, <span class="at">label =</span> <span class="st">"Maximum likelihood </span><span class="sc">\n</span><span class="st">estimate"</span>, <span class="at">col =</span> <span class="st">"#782c26"</span>) <span class="sc">+</span></span>
<span id="cb71-36"><a href="#cb71-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">dpois</span>(<span class="at">x =</span> <span class="dv">54</span>, <span class="dv">54</span>),</span>
<span id="cb71-37"><a href="#cb71-37" aria-hidden="true" tabindex="-1"></a>             <span class="at">lty =</span> <span class="dv">2</span>, <span class="at">col =</span> <span class="st">"#26733a"</span>, <span class="at">linewidth =</span> <span class="fl">1.5</span> ) <span class="sc">+</span></span>
<span id="cb71-38"><a href="#cb71-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">5</span>, <span class="at">y =</span> <span class="fu">dpois</span>(<span class="at">x =</span> <span class="dv">54</span>, <span class="dv">54</span>) <span class="sc">-</span> <span class="fl">0.005</span> , </span>
<span id="cb71-39"><a href="#cb71-39" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="fu">frac</span>(<span class="st">"d L"</span>,<span class="st">"dx"</span>), <span class="st">" = 0"</span>)), <span class="at">col =</span> <span class="st">"#26733a"</span>)</span>
<span id="cb71-40"><a href="#cb71-40" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">lam =</span> lam, <span class="at">loglikelihood =</span>  <span class="fu">dpois</span>(<span class="at">x =</span> <span class="dv">54</span>, lam, <span class="at">log =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb71-41"><a href="#cb71-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb71-42"><a href="#cb71-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> lam, <span class="at">y =</span> loglikelihood)) <span class="sc">+</span></span>
<span id="cb71-43"><a href="#cb71-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_linedraw</span>() <span class="sc">+</span> <span class="fu">xlab</span>(<span class="fu">expression</span>(lambda)) <span class="sc">+</span></span>
<span id="cb71-44"><a href="#cb71-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">"log(L("</span>, <span class="sc">~</span>lambda, <span class="st">";s))"</span>, <span class="at">sep =</span> <span class="st">" "</span>))) <span class="sc">+</span></span>
<span id="cb71-45"><a href="#cb71-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">54</span>,  <span class="at">col =</span> <span class="st">"#782c26"</span>, <span class="at">linewidth =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb71-46"><a href="#cb71-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"log-likelihood for X ~ Poisson(54)"</span>) <span class="sc">+</span></span>
<span id="cb71-47"><a href="#cb71-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">70</span>, <span class="at">y =</span> <span class="sc">-</span><span class="dv">100</span>, <span class="at">label =</span> <span class="st">"Maximum likelihood </span><span class="sc">\n</span><span class="st">estimate"</span>, <span class="at">col =</span> <span class="st">"#782c26"</span>) <span class="sc">+</span></span>
<span id="cb71-48"><a href="#cb71-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">dpois</span>(<span class="at">x =</span> <span class="dv">54</span>, <span class="dv">54</span>, <span class="at">log =</span> <span class="cn">TRUE</span>),</span>
<span id="cb71-49"><a href="#cb71-49" aria-hidden="true" tabindex="-1"></a>             <span class="at">lty =</span> <span class="dv">2</span>, <span class="at">col =</span> <span class="st">"#26733a"</span>, <span class="at">linewidth =</span> <span class="fl">1.5</span> ) <span class="sc">+</span></span>
<span id="cb71-50"><a href="#cb71-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">5</span>, <span class="at">y =</span> <span class="sc">-</span><span class="dv">20</span>, </span>
<span id="cb71-51"><a href="#cb71-51" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">expression</span>(<span class="fu">paste</span>(<span class="fu">frac</span>(<span class="st">"d log(L)"</span>,<span class="st">"dx"</span>), <span class="st">" = 0"</span>)), <span class="at">col =</span> <span class="st">"#26733a"</span>)</span>
<span id="cb71-52"><a href="#cb71-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-53"><a href="#cb71-53" aria-hidden="true" tabindex="-1"></a>patchwork<span class="sc">::</span><span class="fu">wrap_plots</span>(p1, p2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In your groups work through the examples below (both from <a href="#sec-glmfit" class="quarto-xref">Section&nbsp;12.4</a>) and discuss the prompts and plots etc.
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
<p><strong>Bat Abundance</strong> (subset of data from <span class="citation" data-cites="bats">Barlow et al. (<a href="#ref-bats" role="doc-biblioref">2015</a>)</span>)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/bats.csv"</span>)</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">mean =</span> <span class="fu">mean</span>(TotalCount))</span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Assume Poisson distribution</span></span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a><span class="do">## mle for lambda (across all species) is the mean</span></span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a><span class="do">#############################################</span></span>
<span id="cb72-8"><a href="#cb72-8" aria-hidden="true" tabindex="-1"></a><span class="do">## So what is the MLE for lambda when x = 100?</span></span>
<span id="cb72-9"><a href="#cb72-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Does this make sense?</span></span>
<span id="cb72-10"><a href="#cb72-10" aria-hidden="true" tabindex="-1"></a><span class="do">## Recall that the (log)likelihood is a function of the parameter(s)</span></span>
<span id="cb72-11"><a href="#cb72-11" aria-hidden="true" tabindex="-1"></a><span class="do">## for some fixed data; it gives the probability of a fixed (set of)</span></span>
<span id="cb72-12"><a href="#cb72-12" aria-hidden="true" tabindex="-1"></a><span class="do">## observation(s) for every possible value of the parameter(s)</span></span>
<span id="cb72-13"><a href="#cb72-13" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="at">lambda =</span>  <span class="fu">seq</span>(<span class="dv">10</span>, <span class="dv">200</span>,<span class="at">by =</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb72-14"><a href="#cb72-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">ll =</span> <span class="fu">dpois</span>(<span class="at">x =</span> <span class="dv">100</span>, lambda , <span class="at">log =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb72-15"><a href="#cb72-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>( <span class="at">x =</span> lambda, <span class="at">y =</span> ll)) <span class="sc">+</span></span>
<span id="cb72-16"><a href="#cb72-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb72-17"><a href="#cb72-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_linedraw</span>() <span class="sc">+</span> <span class="fu">xlab</span>(<span class="fu">expression</span>(lambda)) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"log-likelihood value"</span>) <span class="sc">+</span></span>
<span id="cb72-18"><a href="#cb72-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">"Log-likelihood for X ~ Poisson("</span>, lambda,<span class="st">"), where x = 100"</span>)))</span>
<span id="cb72-19"><a href="#cb72-19" aria-hidden="true" tabindex="-1"></a><span class="do">## Now for our data</span></span>
<span id="cb72-20"><a href="#cb72-20" aria-hidden="true" tabindex="-1"></a><span class="do">## log likelihood for sum of poisson random variables</span></span>
<span id="cb72-21"><a href="#cb72-21" aria-hidden="true" tabindex="-1"></a><span class="do">## where observations are the observed number of roosts</span></span>
<span id="cb72-22"><a href="#cb72-22" aria-hidden="true" tabindex="-1"></a><span class="do">## for some unknown parameter lambda</span></span>
<span id="cb72-23"><a href="#cb72-23" aria-hidden="true" tabindex="-1"></a><span class="co">#' @param lambda unknown parameter lambda </span></span>
<span id="cb72-24"><a href="#cb72-24" aria-hidden="true" tabindex="-1"></a><span class="co">#' @param obvs vector of observations</span></span>
<span id="cb72-25"><a href="#cb72-25" aria-hidden="true" tabindex="-1"></a>log_lik_poisson <span class="ot">&lt;-</span> <span class="cf">function</span>(lambda, obvs){</span>
<span id="cb72-26"><a href="#cb72-26" aria-hidden="true" tabindex="-1"></a>  llh <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">dpois</span>(obvs, lambda, <span class="at">log =</span> <span class="cn">TRUE</span>))</span>
<span id="cb72-27"><a href="#cb72-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(llh)</span>
<span id="cb72-28"><a href="#cb72-28" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb72-29"><a href="#cb72-29" aria-hidden="true" tabindex="-1"></a><span class="do">## optimise the function above to find the MLE of  unknown param lambda</span></span>
<span id="cb72-30"><a href="#cb72-30" aria-hidden="true" tabindex="-1"></a>lam_hat <span class="ot">&lt;-</span> <span class="fu">optimise</span>(log_lik_poisson, <span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">200</span>) , <span class="at">obvs =</span> data<span class="sc">$</span>TotalCount, <span class="at">maximum =</span> <span class="cn">TRUE</span>)</span>
<span id="cb72-31"><a href="#cb72-31" aria-hidden="true" tabindex="-1"></a>lam_hat</span>
<span id="cb72-32"><a href="#cb72-32" aria-hidden="true" tabindex="-1"></a><span class="do">## what is out maximum likelihood estimate here?</span></span>
<span id="cb72-33"><a href="#cb72-33" aria-hidden="true" tabindex="-1"></a><span class="do">## plot the log-likelihood for the sum of our observed data</span></span>
<span id="cb72-34"><a href="#cb72-34" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="at">lambda =</span>  <span class="fu">seq</span>(<span class="dv">10</span>, <span class="dv">200</span>, <span class="at">by =</span> <span class="fl">0.5</span>) ) <span class="sc">%&gt;%</span></span>
<span id="cb72-35"><a href="#cb72-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">ll =</span> <span class="fu">sapply</span>(lambda, log_lik_poisson, <span class="at">obvs =</span> data<span class="sc">$</span>TotalCount)) <span class="sc">%&gt;%</span></span>
<span id="cb72-36"><a href="#cb72-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>( <span class="at">x =</span> lambda, <span class="at">y =</span> ll)) <span class="sc">+</span></span>
<span id="cb72-37"><a href="#cb72-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb72-38"><a href="#cb72-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_linedraw</span>() <span class="sc">+</span> <span class="fu">xlab</span>(<span class="fu">expression</span>(lambda)) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"log-likelihood value"</span>) <span class="sc">+</span></span>
<span id="cb72-39"><a href="#cb72-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> lam_hat<span class="sc">$</span>maximum, <span class="at">size =</span> <span class="dv">2</span>, <span class="at">col =</span> <span class="st">"purple"</span>) <span class="sc">+</span></span>
<span id="cb72-40"><a href="#cb72-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="at">geom =</span> <span class="st">"text"</span>, lam_hat<span class="sc">$</span>maximum, <span class="sc">-</span><span class="dv">40000</span>, <span class="at">col =</span> <span class="st">"purple"</span>, <span class="at">label =</span> <span class="st">"MLE"</span>) <span class="sc">+</span></span>
<span id="cb72-41"><a href="#cb72-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="st">"Log-likelihood for X ~ Poisson("</span>, lambda,<span class="st">")"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Lobster Survival</strong> (data from <span class="citation" data-cites="lobster">Wilkinson et al. (<a href="#ref-lobster" role="doc-biblioref">2015</a>)</span>)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/lobster.csv"</span>)</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">survival =</span> <span class="fu">ifelse</span>(survived <span class="sc">==</span> <span class="dv">0</span>, <span class="st">"consumed"</span>, <span class="st">"alive"</span>))</span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(data<span class="sc">$</span>survival)</span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>n_survived <span class="ot">&lt;-</span> <span class="dv">79</span></span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a>n_total <span class="ot">&lt;-</span> <span class="dv">159</span></span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a><span class="do">## define the likelihood function</span></span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a><span class="do">## note you will need to fill in the correct values</span></span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a>likelihood <span class="ot">&lt;-</span> <span class="cf">function</span>(theta) <span class="fu">dbinom</span>(<span class="at">x =</span> n_survived, <span class="at">size =</span> n_total, <span class="at">prob =</span> theta)</span>
<span id="cb73-10"><a href="#cb73-10" aria-hidden="true" tabindex="-1"></a><span class="fu">optimise</span>(likelihood, <span class="at">interval =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">maximum =</span> <span class="cn">TRUE</span>) <span class="do">## what have we estimated here?</span></span>
<span id="cb73-11"><a href="#cb73-11" aria-hidden="true" tabindex="-1"></a><span class="do">## define the likelihood function</span></span>
<span id="cb73-12"><a href="#cb73-12" aria-hidden="true" tabindex="-1"></a><span class="do">## note you will need to fill in the correct values</span></span>
<span id="cb73-13"><a href="#cb73-13" aria-hidden="true" tabindex="-1"></a>log_likelihood <span class="ot">&lt;-</span> <span class="cf">function</span>(theta) <span class="fu">dbinom</span>(<span class="at">x =</span> n_survived, <span class="at">size =</span> n_total, <span class="at">prob =</span> theta, <span class="at">log =</span> <span class="cn">TRUE</span>)</span>
<span id="cb73-14"><a href="#cb73-14" aria-hidden="true" tabindex="-1"></a><span class="fu">optimise</span>(log_likelihood, <span class="at">interval =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">maximum =</span> <span class="cn">TRUE</span>)  <span class="do">## what have we estimated here?</span></span>
<span id="cb73-15"><a href="#cb73-15" aria-hidden="true" tabindex="-1"></a><span class="do">## plot likelihood vs unknown param</span></span>
<span id="cb73-16"><a href="#cb73-16" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">length.out =</span> <span class="dv">100</span>)</span>
<span id="cb73-17"><a href="#cb73-17" aria-hidden="true" tabindex="-1"></a>lik <span class="ot">&lt;-</span> <span class="fu">dbinom</span>(<span class="at">x =</span> n_survived, <span class="at">size =</span> n_total, <span class="at">prob =</span> p)</span>
<span id="cb73-18"><a href="#cb73-18" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(p, lik, <span class="at">type =</span> <span class="st">"l"</span>)</span>
<span id="cb73-19"><a href="#cb73-19" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(p, lik)</span>
<span id="cb73-20"><a href="#cb73-20" aria-hidden="true" tabindex="-1"></a><span class="do">## Optimise for parameter p</span></span>
<span id="cb73-21"><a href="#cb73-21" aria-hidden="true" tabindex="-1"></a>llh <span class="ot">&lt;-</span> <span class="cf">function</span>(theta, data){</span>
<span id="cb73-22"><a href="#cb73-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sum</span>(<span class="fu">dbinom</span>(<span class="at">x =</span> data, <span class="at">size =</span> <span class="dv">1</span>, <span class="at">prob =</span> theta, <span class="at">log =</span> <span class="cn">TRUE</span>))</span>
<span id="cb73-23"><a href="#cb73-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb73-24"><a href="#cb73-24" aria-hidden="true" tabindex="-1"></a><span class="fu">optimise</span>(llh, <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">data =</span> data<span class="sc">$</span>survived,  <span class="at">maximum =</span> <span class="cn">TRUE</span>)</span>
<span id="cb73-25"><a href="#cb73-25" aria-hidden="true" tabindex="-1"></a><span class="do">## Now for prob or survival as a function of claw size.</span></span>
<span id="cb73-26"><a href="#cb73-26" aria-hidden="true" tabindex="-1"></a><span class="do">## What is the function I've specified? why the negative here??</span></span>
<span id="cb73-27"><a href="#cb73-27" aria-hidden="true" tabindex="-1"></a>llh <span class="ot">&lt;-</span> <span class="cf">function</span>(par, data){</span>
<span id="cb73-28"><a href="#cb73-28" aria-hidden="true" tabindex="-1"></a>    <span class="sc">-</span><span class="fu">sum</span>(<span class="fu">dbinom</span>(<span class="at">x =</span> data<span class="sc">$</span>survived,</span>
<span id="cb73-29"><a href="#cb73-29" aria-hidden="true" tabindex="-1"></a>               <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb73-30"><a href="#cb73-30" aria-hidden="true" tabindex="-1"></a>               <span class="at">prob =</span> <span class="fu">exp</span>(par[<span class="dv">1</span>] <span class="sc">+</span> par[<span class="dv">2</span>]<span class="sc">*</span>data<span class="sc">$</span>size)<span class="sc">/</span>(<span class="dv">1</span> <span class="sc">+</span> <span class="fu">exp</span>(par[<span class="dv">1</span>] <span class="sc">+</span> par[<span class="dv">2</span>]<span class="sc">*</span>data<span class="sc">$</span>size)),</span>
<span id="cb73-31"><a href="#cb73-31" aria-hidden="true" tabindex="-1"></a>               <span class="at">log =</span> <span class="cn">TRUE</span>))</span>
<span id="cb73-32"><a href="#cb73-32" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb73-33"><a href="#cb73-33" aria-hidden="true" tabindex="-1"></a><span class="do">## find the maximum likelihood estimate for the unknown parameters</span></span>
<span id="cb73-34"><a href="#cb73-34" aria-hidden="true" tabindex="-1"></a><span class="fu">optim</span>(<span class="at">fn =</span> llh, <span class="at">par =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="fl">0.5</span>), <span class="at">data =</span> data)</span>
<span id="cb73-35"><a href="#cb73-35" aria-hidden="true" tabindex="-1"></a><span class="do">## Compare to the ungrouped model in the course guide</span></span>
<span id="cb73-36"><a href="#cb73-36" aria-hidden="true" tabindex="-1"></a>glm_mod_ug <span class="ot">&lt;-</span> <span class="fu">glm</span>(survived <span class="sc">~</span> size, <span class="at">family =</span> <span class="st">"binomial"</span>, <span class="at">data =</span> data)</span>
<span id="cb73-37"><a href="#cb73-37" aria-hidden="true" tabindex="-1"></a><span class="fu">coef</span>(glm_mod_ug)</span>
<span id="cb73-38"><a href="#cb73-38" aria-hidden="true" tabindex="-1"></a><span class="do">## Briefly comment on why, or why not, your parameter estimates are the same.</span></span>
<span id="cb73-39"><a href="#cb73-39" aria-hidden="true" tabindex="-1"></a><span class="do">## Has the same model been fitted in each case?</span></span>
<span id="cb73-40"><a href="#cb73-40" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot your fitted models and compare!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
<section id="may-5th" class="level1">
<h1>May 5th</h1>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
By the end of today’s seminar, you should be able to
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Distinguish frequentist and Bayesian approaches to data analysis</li>
<li>Choose appropriate prior distributions for a Bayesian model</li>
</ul>
</div>
</div>
<section id="rough-timeline-13" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-13">Rough timeline</h2>
<ul>
<li><strong>09:05–09:35 am</strong> Introduction to Bayesian Statistics (<a href="#sec-bayes" class="quarto-xref">Section&nbsp;14.2</a>)
<ul>
<li>Class discussion</li>
<li>Class activities</li>
</ul></li>
<li><strong>09:55–11:05 am</strong> Break</li>
<li><strong>11:05–11:30 am</strong> Casestudies from a Bayesian viewpoint (<a href="#sec-bayesestudies" class="quarto-xref">Section&nbsp;14.3</a>)</li>
<li><strong>11.30–11.55 am</strong> Recap (<a href="#sec-recap03" class="quarto-xref">Section&nbsp;14.4</a>)</li>
</ul>
<p>NOTE: here we only briefly discuss Bayesian approaches using examples, for more details see <a href="https://biosci738.github.io/BIOSCI738/introduction-to-bayesian-statistics.html">this section of the courseguide</a> and for a much more rounded discussion I strongly recommend at least Chapter 1 of <span class="citation" data-cites="bayesrules">Johnson, Ott, and Dogucu (<a href="#ref-bayesrules" role="doc-biblioref">2021</a>)</span>. This runsheet only skims the basics!</p>
</section>
<section id="sec-bayes" class="level2">
<h2 class="anchored" data-anchor-id="sec-bayes">Introduction to Bayesian Statistics</h2>
<center>
<p>
SCAN ME or head to <a href="https://dub.sh/biosci738" target="_blank">dub.sh/biosci738</a>
</p>
<img src="runsheets/img/qr_biosci738.svg" alt="dub.sh.biosci738">
</center>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
When prompted <em>Quick Write</em>…
</div>
</div>
<div class="callout-body-container callout-body">
<p>…<strong>a</strong>, <strong>b</strong>, or <strong>c</strong> in answer to the following<a href="#fn10" class="footnote-ref" id="fnref10" role="doc-noteref"><sup>10</sup></a> (<em>record your answers</em>)</p>
<ol type="1">
<li>When flipping a fair coin, we say that “the probability of flipping Heads is 0.5.” How do you interpret this probability?
<ul>
<li><ol type="a">
<li>If I flip this coin over and over, roughly 50% will be Heads.<br>
</li>
</ol></li>
<li><ol start="2" type="a">
<li>Heads and Tails are equally plausible.<br>
</li>
</ol></li>
<li><ol start="3" type="a">
<li>Both a and b make sense.</li>
</ol></li>
</ul></li>
<li>An election is coming up and a pollster claims that candidate A has a 0.9 probability of winning. How do you interpret this probability?
<ul>
<li><ol type="a">
<li>If we observe the election over and over, candidate A will win roughly 90% of the time.<br>
</li>
</ol></li>
<li><ol start="2" type="a">
<li>Candidate A is much more likely to win than to lose.<br>
</li>
</ol></li>
<li><ol start="3" type="a">
<li>The pollster’s calculation is wrong. Candidate A will either win or lose, thus their probability of winning can only be 0 or 1.</li>
</ol></li>
</ul></li>
<li>Consider two claims.
<ol type="1">
<li>Zuofu claims that he can predict the outcome of a coin flip. To test his claim, you flip a fair coin 10 times and he correctly predicts all 10.<br>
</li>
<li>Kavya claims that she can distinguish natural and artificial sweeteners. To test her claim, you give her 10 sweetener samples and she correctly identifies each.<br>
In light of these experiments, what do you conclude?<br>
</li>
</ol>
<ul>
<li><ol type="a">
<li>You’re more confident in Kavya’s claim than Zuofu’s claim.<br>
</li>
</ol></li>
<li><ol start="2" type="a">
<li>The evidence supporting Zuofu’s claim is just as strong as the evidence supporting Kavya’s claim.</li>
</ol></li>
</ul></li>
<li>Suppose that during a recent doctor’s visit, you tested positive for a very rare disease. If you only get to ask the doctor one question, which would it be?
<ul>
<li><ol type="a">
<li>What’s the chance that I actually have the disease?<br>
</li>
</ol></li>
<li><ol start="2" type="a">
<li>If in fact I don’t have the disease, what’s the chance that I would’ve gotten this positive test result?</li>
</ol></li>
</ul></li>
</ol>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
The simplified distinction
</div>
</div>
<div class="callout-body-container callout-body">
<p><strong>Frequentist</strong>: The parameter <strong>is not</strong> a random variable (i.e., is fixed and has one <em>true</em> value). <strong>Bayesian</strong>: The parameter <strong>is</strong> a random variable (i.e., comes from some probability distribution).</p>
</div>
</div>
<section id="a-bayesian-beta-binomial-model" class="level3">
<h3 class="anchored" data-anchor-id="a-bayesian-beta-binomial-model">A Bayesian Beta-Binomial model</h3>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
We’ll work through this example together! Note that this aligns with the content in <a href="https://biosci738.github.io/BIOSCI738/introduction-to-bayesian-statistics.html">this section of the courseguide</a>; however, here we go beyond what is covered there!
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
<p>Let’s take a look again at the <strong>Lobster Survival</strong> data from <span class="citation" data-cites="lobster">Wilkinson et al. (<a href="#ref-lobster" role="doc-biblioref">2015</a>)</span>. The NULL model equivalent to the ungrouped model we considered in <a href="#sec-glmfit" class="quarto-xref">Section&nbsp;12.4</a> was</p>
<div class="cell">
<div class="cell-output-display">
<p><span class="math display">\[
\log\left[ \frac { P( \operatorname{survived} = \operatorname{1} ) }{ 1 - P( \operatorname{survived} = \operatorname{1} ) } \right] = \alpha
\]</span></p>
</div>
</div>
<p>In other words, let <span class="math inline">\(y\)</span> be either 1 or 0 depending on survival then <span class="math inline">\(y \sim \text{Binomial}(n = 1, p)\)</span> (i.e., <span class="math inline">\(y \sim \text{Bernoulli}(p)\)</span>) where <span class="math inline">\(p\)</span> is the probability of survival.</p>
<p>Here, <span class="math inline">\(y \sim \text{Binomial}(n = 1, p)\)</span> is the <strong>likelihood</strong> (see <a href="#sec-mle" class="quarto-xref">Section&nbsp;13.3</a>) and <span class="math inline">\(p\)</span> is an <strong>unknown parameter</strong> we wish to estimate. Under a Bayesian framework we assume that <span class="math inline">\(p\)</span> is a random variable, with it’s own distribution!</p>
<p><strong>So, what do we know about <span class="math inline">\(p\)</span>?</strong> It is a probability and therefore bounded between 0 and 1. At this stage we have no further information (i.e., nothing about what value this probability might be).</p>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In your groups suggest a candidate distribution for <span class="math inline">\(p\)</span>.
</div>
</div>
<div class="callout-body-container callout-body">

</div>
</div>
<p>This distribution is your <strong>prior</strong> distribution and reflect your beliefs about <span class="math inline">\(p\)</span>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(bayesrules)</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a><span class="do">## The data (i.e., likelihood)</span></span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_binomial_likelihood</span>(<span class="at">y =</span> <span class="dv">79</span>, <span class="at">n =</span> <span class="dv">159</span>)</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a><span class="do">## What about a Beta prior on p</span></span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_beta</span>(<span class="dv">10</span>,<span class="dv">3</span>) <span class="do">##??</span></span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_beta</span>(<span class="fl">0.5</span>,<span class="fl">0.5</span>) <span class="do">## favours extremes (actually called a Jeffreys prior)</span></span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_beta</span>(<span class="dv">1</span>,<span class="dv">1</span>) <span class="do">##?? (all equally likely)</span></span>
<span id="cb74-8"><a href="#cb74-8" aria-hidden="true" tabindex="-1"></a><span class="do">## The posterior with an uninformative prior</span></span>
<span id="cb74-9"><a href="#cb74-9" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_beta_binomial</span>(<span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">beta =</span> <span class="dv">1</span>, <span class="at">y =</span> <span class="dv">79</span>, <span class="at">n =</span> <span class="dv">159</span>)</span>
<span id="cb74-10"><a href="#cb74-10" aria-hidden="true" tabindex="-1"></a><span class="do">## What about an "extreme favouring" prior</span></span>
<span id="cb74-11"><a href="#cb74-11" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_beta_binomial</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">beta =</span> <span class="fl">0.5</span>, <span class="at">y =</span> <span class="dv">79</span>, <span class="at">n =</span> <span class="dv">159</span>)</span>
<span id="cb74-12"><a href="#cb74-12" aria-hidden="true" tabindex="-1"></a><span class="do">## The data usurps the prior!! As it should</span></span>
<span id="cb74-13"><a href="#cb74-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Inference</span></span>
<span id="cb74-14"><a href="#cb74-14" aria-hidden="true" tabindex="-1"></a><span class="fu">summarize_beta_binomial</span>(<span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">beta =</span> <span class="dv">1</span>, <span class="at">y =</span> <span class="dv">79</span>, <span class="at">n =</span> <span class="dv">159</span>)</span>
<span id="cb74-15"><a href="#cb74-15" aria-hidden="true" tabindex="-1"></a><span class="do">## The estimate we're interested in is the posterior mean</span></span>
<span id="cb74-16"><a href="#cb74-16" aria-hidden="true" tabindex="-1"></a><span class="do">## How does this compare to our MLE approach?</span></span>
<span id="cb74-17"><a href="#cb74-17" aria-hidden="true" tabindex="-1"></a>log_likelihood <span class="ot">&lt;-</span> <span class="cf">function</span>(theta) <span class="fu">dbinom</span>(<span class="at">x =</span> <span class="dv">79</span>, <span class="at">size =</span> <span class="dv">159</span>, <span class="at">prob =</span> theta, <span class="at">log =</span> <span class="cn">TRUE</span>)</span>
<span id="cb74-18"><a href="#cb74-18" aria-hidden="true" tabindex="-1"></a><span class="fu">optimise</span>(log_likelihood, <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">maximum =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Read more about prior sensitivity (i.e., your choice of prior affecting results) in <a href="https://biosci738.github.io/BIOSCI738/introduction-to-bayesian-statistics.html#prior-sensitivity">this section of the courseguide</a>.</p>
<p>What about frequentist approach vs Bayesian?</p>
<blockquote class="blockquote">
<p>Typically, Bayesian and frequentist estimates will always agree if there is sufficient data, so long as the likelihood is not explicitly ruled out by the prior.— Olivier Gimenez (<span class="citation" data-cites="olivier">Gimenez (<a href="#ref-olivier" role="doc-biblioref">2023</a>)</span>)</p>
</blockquote>
</section>
</section>
<section id="sec-bayesestudies" class="level2">
<h2 class="anchored" data-anchor-id="sec-bayesestudies">Casestudies from a Bayesian viewpoint</h2>
<section id="choosing-a-prior-distribution" class="level3">
<h3 class="anchored" data-anchor-id="choosing-a-prior-distribution">Choosing a prior distribution</h3>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In your groups discuss/answer the sceanrios below.
</div>
</div>
<div class="callout-body-container callout-body">
<ol type="1">
<li><strong>Test Scores Under a New Teaching Method</strong> A school is testing a new math teaching method. A sample of students is taught using the new method, and their standardized test scores are recorded. Assume test scores are normally distributed with known variance <span class="math inline">\(\sigma^2 = 100\)</span> (i.e., standard deviation = 10). We’re interested in the average score <span class="math inline">\(\mu\)</span> under the new method as follows <span class="math inline">\(y_i \sim \mathcal{N}(\mu, \sigma^2)\)</span>. What might be a suitable distribution to reflect the prior belief about the average score <span class="math inline">\(\mu\)</span>.</li>
<li><strong>Kiwi Call Monitoring</strong> Wildlife biologists are studying nocturnal kiwi activity in a forest reserve. On several nights, researchers record the number of kiwi calls heard per hour from a fixed listening post. Let <span class="math inline">\(y_i\)</span> be the number of kiwi calls heard during hour <span class="math inline">\(i\)</span>, modeled as <span class="math inline">\(y_i \sim \text{Poisson}(\lambda)\)</span>. What might be a suitable distribution to reflect the prior belief about the average call rate <span class="math inline">\(\lambda\)</span>.</li>
</ol>
<p>Now, a statistician has come along and said that things will be a lot easier (ask Charlotte why if you’re interested) and said that an appropriate prior for the average score (scenario 1) is <span class="math inline">\(\mu \sim \mathcal{N}(\mu_0, \tau^2)\)</span> and for the average call rate (scenario 2) is <span class="math inline">\(\lambda \sim \text{Gamma}(\alpha, \beta)\)</span>. In addition, the principal of the school in scenario 1 believes students taught with the new method will average around 75 points, but the level of certainty varies depending on context. And Ecologists familiar with the region in scenario 2 suggest that the rate of kiwi calls typically ranges from 2 to 4 calls per hour.</p>
<p>Assuming these prior distributions and the extra information</p>
<p>a.Based on the prior knowledge, how might you choose values for the parameters of each prior distribution? b. What prior reflects strong certainty about random variable? c.&nbsp;What prior reflects strong uncertainty about random variable?</p>
<p><em>[NOTE: Scenarios were created with the assistance of ChatGPT (OpenAI, 2025).]</em></p>
</div>
</div>
</section>
<section id="casestudy" class="level3">
<h3 class="anchored" data-anchor-id="casestudy">Casestudy</h3>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In your groups work through the section below (previously seen example from <a href="#sec-glmfit" class="quarto-xref">Section&nbsp;12.4</a>) and discuss the prompts and plots etc.
</div>
</div>
<div class="callout-body-container callout-body">
<p><em>NOTE that this is quite advanced and touches on the boundary of what is beyond the scope of this course, but you’re all up to it!</em></p>
</div>
</div>
<section id="a-gamma-poisson-model-i.e.-negative-binomial" class="level4">
<h4 class="anchored" data-anchor-id="a-gamma-poisson-model-i.e.-negative-binomial">A Gamma-Poisson Model (i.e., Negative Binomial)</h4>
<p><strong>Bat Abundance</strong> (subset of data from <span class="citation" data-cites="bats">Barlow et al. (<a href="#ref-bats" role="doc-biblioref">2015</a>)</span>)</p>
<p>In <a href="#sec-glmfit" class="quarto-xref">Section&nbsp;12.4</a> we fitted the following model in a frequentist framework</p>
<p><span class="math display">\[
\log ({ E( \operatorname{TotalCount} ) })  = \alpha + \beta_{1}(\operatorname{SpeciesID}_{\operatorname{Rhinolophus\ hipposideros}})
\]</span></p>
<p>where <span class="math inline">\(Y = y\)</span> was the nest count and <span class="math inline">\(y \sim \text{NegBin}(\mu, \phi)\)</span>, with <span class="math inline">\(\mu\)</span> the expected count (<span class="math inline">\(E( \operatorname{TotalCount} )\)</span>) and <span class="math inline">\(\phi\)</span> the dispersion parameter (i.e., how much variance exceeds the mean).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>bats <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/bats.csv"</span>)</span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a><span class="do">## frequentist approach</span></span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>nbmod <span class="ot">&lt;-</span> MASS<span class="sc">::</span><span class="fu">glm.nb</span>(TotalCount <span class="sc">~</span> SpeciesID, <span class="at">data =</span> bats)</span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a>nbmod <span class="sc">|&gt;</span> <span class="fu">summary</span>() <span class="sc">|&gt;</span> <span class="fu">coef</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Just a recap, why did we choose a Negative Binomial model? Because the was a lot more variation than we’d expect under a Poisson model. We saw this in <a href="#sec-glmfit" class="quarto-xref">Section&nbsp;12.4</a>, but let’s use simulation to <em>show</em> this again.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="do">## observed mean</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(bats<span class="sc">$</span>TotalCount) <span class="do">## ~174</span></span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a><span class="do">## observed variance</span></span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a><span class="fu">var</span>(bats<span class="sc">$</span>TotalCount) <span class="do">## a LOT higher</span></span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a><span class="do">## simulate from a Poisson with lambda = observed mean</span></span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a>sim <span class="ot">&lt;-</span> <span class="fu">replicate</span>(<span class="dv">100</span>, <span class="fu">rpois</span>(<span class="dv">100</span>, <span class="fu">mean</span>(bats<span class="sc">$</span>TotalCount))) <span class="do">## quicker than a 'for loop'</span></span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a><span class="do">## wrangle and plot</span></span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a>sim <span class="sc">|&gt;</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb76-9"><a href="#cb76-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(., <span class="fu">everything</span>(), <span class="at">names_to =</span> <span class="st">"column"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb76-10"><a href="#cb76-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> value, <span class="at">group =</span> column)) <span class="sc">+</span></span>
<span id="cb76-11"><a href="#cb76-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">linewidth =</span> <span class="fl">0.2</span>, <span class="at">alpha =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb76-12"><a href="#cb76-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">data =</span> bats, <span class="fu">aes</span>(<span class="at">x =</span> TotalCount), <span class="at">color =</span> <span class="st">"purple"</span>, </span>
<span id="cb76-13"><a href="#cb76-13" aria-hidden="true" tabindex="-1"></a>               <span class="at">linewidth =</span> <span class="dv">1</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>) </span>
<span id="cb76-14"><a href="#cb76-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb76-15"><a href="#cb76-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Clearly the observed (purple) doesn't match the assumed Poisson (all the black lines)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>OK, back to Bayesian modelling. We have <span class="math inline">\(y \sim \text{NegBin}(\mu, \phi)\)</span>. We have two unknown parameters <span class="math inline">\(\mu\)</span> and <span class="math inline">\(\phi\)</span>. Let’s first consider a NULL (i.e., intercept only) model <span class="math inline">\(\log (\mu)  = \alpha\)</span>.</p>
<p>We use the following priors</p>
<ol type="1">
<li><span class="math inline">\(\alpha \sim \text{N}(0,5)\)</span> and</li>
<li><span class="math inline">\(\phi \sim \text{Exp}(1)\)</span>.</li>
</ol>
<p>Do these seem appropriate?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_normal</span>(<span class="dv">0</span>, <span class="dv">5</span>)</span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_gamma</span>(<span class="dv">1</span>, <span class="dv">1</span>) <span class="do">## recall that exponential is a special case of Gamma with shape = 1 </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rstanarm)</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>fit_null <span class="ot">&lt;-</span> <span class="fu">stan_glm</span>(</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>  TotalCount <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">neg_binomial_2</span>(), <span class="do">## Likelihood</span></span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> bats,</span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">prior =</span> <span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>),             <span class="do">## Prior on log(mu) (i.e., alpha)</span></span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">prior_aux =</span> <span class="fu">exponential</span>(<span class="dv">1</span>),       <span class="do">## Prior on phi (overdispersion)</span></span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">iter =</span> <span class="dv">2000</span>, <span class="at">seed =</span> <span class="dv">123</span></span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb78-10"><a href="#cb78-10" aria-hidden="true" tabindex="-1"></a><span class="do">## coefficients</span></span>
<span id="cb78-11"><a href="#cb78-11" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(fit_null, <span class="at">digits =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, let’s consider a model with the factor explanatory variable <code>SpeciesID</code>. Here <span class="math inline">\(\log(\mu)  = \alpha + \beta_{1}(\operatorname{SpeciesID}_{\operatorname{Rhinolophus\ hipposideros}})\)</span>. So now we have three unknown parameters <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta_1\)</span> and <span class="math inline">\(\phi\)</span>.</p>
<p>We use the following priors</p>
<ol type="1">
<li><span class="math inline">\(\alpha \sim \text{N}(0, 2.5)\)</span>,</li>
<li><span class="math inline">\(\beta_1 \sim  \text{N}(0, 5)\)</span>, and</li>
<li><span class="math inline">\(\phi \sim \text{Exp}(1)\)</span>.</li>
</ol>
<p>Fitting the model:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>fit_species <span class="ot">&lt;-</span> <span class="fu">stan_glm</span>(</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>  TotalCount <span class="sc">~</span> SpeciesID,</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">neg_binomial_2</span>(),</span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> bats,</span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">prior =</span> <span class="fu">normal</span>(<span class="dv">0</span>, <span class="fl">2.5</span>),           <span class="co"># Prior on coefficients (log scale)</span></span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">prior_intercept =</span> <span class="fu">normal</span>(<span class="dv">0</span>, <span class="dv">5</span>),   <span class="co"># Prior on intercept</span></span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">prior_aux =</span> <span class="fu">exponential</span>(<span class="dv">1</span>),       <span class="co"># Prior on phi (overdispersion)</span></span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">chains =</span> <span class="dv">4</span>, <span class="at">iter =</span> <span class="dv">2000</span>, <span class="at">seed =</span> <span class="dv">123</span></span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a><span class="do">## coefficients</span></span>
<span id="cb79-11"><a href="#cb79-11" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(fit_species, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb79-12"><a href="#cb79-12" aria-hidden="true" tabindex="-1"></a><span class="do">## compare to</span></span>
<span id="cb79-13"><a href="#cb79-13" aria-hidden="true" tabindex="-1"></a>nbmod <span class="sc">|&gt;</span> <span class="fu">summary</span>() <span class="sc">|&gt;</span> <span class="fu">coef</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Assessing model fit</strong> the full details of this are beyond the scope of this course, but let’s use a common sense check of <em>does our data look similar to the model</em>. Special Bayesian terminology for this might be a <em>posterior predictive check</em>, we can do this in one step using <code>pp_check()</code>:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pp_check</span>(fit_species)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Bonus activity (Advanced)
</div>
</div>
<div class="callout-body-container callout-body">
<p>Revisit one of the following case studies and fit an appropriate model in a Bayesian framework</p>
<ol type="1">
<li><strong>Fry Time</strong> <a href="#sec-fry" class="quarto-xref">Section&nbsp;6.4.2</a> (data from <span class="citation" data-cites="fries">Abdul Hamid, Omar, and Sanny (<a href="#ref-fries" role="doc-biblioref">2019</a>)</span>)</li>
<li><strong>Monkeys</strong> case study 04 from <a href="#sec-casestudies" class="quarto-xref">Section&nbsp;10.2</a> (data from <span class="citation" data-cites="monkey">Bliss-Moreau and Baxter (<a href="#ref-monkey" role="doc-biblioref">2019</a>)</span>)</li>
<li><strong>Bird Abundance</strong> third case study from <a href="#sec-glmfit" class="quarto-xref">Section&nbsp;12.4</a> (data from <span class="citation" data-cites="birds">García-Navas et al. (<a href="#ref-birds" role="doc-biblioref">2022</a>)</span>)</li>
</ol>
</div>
</div>
</section>
</section>
</section>
<section id="sec-recap03" class="level2">
<h2 class="anchored" data-anchor-id="sec-recap03">Recap</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i>
</div>
</div>
<div class="callout-body-container callout-body">
<ol type="1">
<li>Choose a topic from the content covered so far that you wish to revisit.</li>
<li>Revisit the associated materials (e.g., code/courseguide section/cheatsheets/runsheets etc.).</li>
<li>As a group design an activity/materials that explain/illustrate the topic/concept.</li>
</ol>
<p><strong>Be prepared to present your activity to me/the class</strong> (all group members should expect to contribute).</p>
</div>
</div>
</section>
</section>
<section id="may-9th" class="level1">
<h1>May 9th</h1>
<section id="rough-timeline-14" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-14">Rough timeline</h2>
<section id="pm---0355-pm-6" class="level3">
<h3 class="anchored" data-anchor-id="pm---0355-pm-6"><strong>02:05 pm - 03:55 pm</strong></h3>
<ul>
<li><strong>02:05–02.50 pm</strong> Modelling Data (<a href="#sec-model" class="quarto-xref">Section&nbsp;15.2</a>)</li>
<li><strong>02:50–03:00 pm</strong> Break</li>
<li><strong>03:00–03:55 pm</strong> Coding on the fly (<a href="#sec-ica" class="quarto-xref">Section&nbsp;15.3</a>)
<ul>
<li>Class discussion</li>
<li>Group activity</li>
</ul></li>
</ul>
</section>
</section>
<section id="sec-model" class="level2">
<h2 class="anchored" data-anchor-id="sec-model">Modelling data, where to start</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In groups of two or three…
</div>
</div>
<div class="callout-body-container callout-body">
<p>…choose one of the three case studies outlined below then</p>
<ol type="1">
<li>discuss what questions you might be interested in answering with these data,</li>
<li>discuss how you might analyse these data so that you could answer those questions, and</li>
<li>and carry out any analysis you deem appropriate and infer your conclusions.</li>
</ol>
<p><strong>You will be expected to give a brief presentation about your approach and analysis to the class</strong>.</p>
</div>
</div>
<section id="case-study-1" class="level3">
<h3 class="anchored" data-anchor-id="case-study-1">Case study 1</h3>
<p>For this case study you will be considering data from a paper investigating the recovery of biological evidence from common drink containers (<span class="citation" data-cites="james">Abaz et al. (<a href="#ref-james" role="doc-biblioref">2002</a>)</span>).</p>
<p>The object <code>data</code> below contains 96 observations and 17 different variables. The <code>amylase</code> column gives the relative amount of <span class="math inline">\(\alpha\)</span> amylase activity extracted from the swabs, the <code>dna</code> column gives the DNA yield (as a proportion), and the <code>failed.profile</code> column are logical values indicating if obtaining a usable DNA profile was a failure or a success. Other variables are as discussed in the paper.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/amylase.csv"</span></span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(url)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>In your groups,</p>
<ol type="1">
<li>discuss what questions you might be interested in answering with these data,</li>
<li>discuss how you might analyse these data so that you could answer those questions, and</li>
<li>and carry out any analysis you deem appropriate.</li>
</ol>
<p>The profile intensity variable mentioned in the paper is a linear combination of the peak height measurements for each of the STR loci. The peak heights of <code>D3S1358</code>, <code>HUMvWF31/A</code>, <code>D16S539</code>, <code>D2S1338</code>, <code>Amelogenin</code>, <code>D8S1179</code>, <code>D21S11</code>, <code>D18S51</code>, <code>D19S433</code>, <code>HUMTH01</code> and <code>HUMFIBRA</code> in RFU quantify, to some extent, the amount of DNA present. Can you recreate this variable?</p>
<p><strong>NOTE: You will be expected to give a brief presentation about your approach and analysis.</strong></p>
</section>
<section id="case-study-2" class="level3">
<h3 class="anchored" data-anchor-id="case-study-2">Case study 2</h3>
<p>In this case study you are going to be be dealing with a subset of data from a project called ACCURATE: ACoustic CUe RATEs for Passive Acoustics Density Estimation (<span class="citation" data-cites="tiago">Marques et al. (<a href="#ref-tiago" role="doc-biblioref">2013</a>)</span>). (If you are interested the full data were modelled by my PhD student in <span class="citation" data-cites="alec">Helsdingen, Marques, and Jones-Todd (<a href="#ref-alec" role="doc-biblioref">2025</a>)</span>.)</p>
<p>The ACCURATE project aims to estimate marine mammal using passive acoustics (i.e., using passive tools to listen to the sounds the animals make rather than having to rely on sighting them). One such method relies on the estimation of an appropriate cue rate (number of cues per unit time). Attaching a Digital Acoustic Recording Tag (DTAG) to marine mammals is a non-invasive method by which we can monitor the behaviour of these animals continuously throughout the dive cycle.</p>
<p>The object <code>data</code> is an aggregated version of the full ACCURATE data:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/whale_clicks.csv"</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(url)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>These data contain 843 observations and three variables:</p>
<ul>
<li><code>minute</code> the minute index over the time period the DTAG was active</li>
<li><code>mean_depth</code> the average depth in meters (below the surface) for each minute</li>
<li><code>n_click</code> the number of clicks recorded in each minute</li>
</ul>
<p>In your groups,</p>
<ol type="1">
<li>discuss what questions you might be interested in answering with these data,</li>
<li>discuss how you might analyse these data so that you could answer those questions, and</li>
<li>and carry out any analysis you deem appropriate.</li>
</ol>
<p><strong>NOTE: You will be expected to give a brief presentation about your approach and analysis.</strong></p>
</section>
<section id="case-study-3" class="level3">
<h3 class="anchored" data-anchor-id="case-study-3">Case study 3</h3>
<p>In this case study you are going to be dealing with data that refer to 346 Pokémon (fictional creatures that feature is certain games, each with associated measures of their ability). The data includes individual Pokémon stats including their number, name, first and second type, the stat total and basic stats: HP (<code>hp</code>), Attack (<code>attack</code>), Defense (<code>defense</code>), Special Attack (<code>sp_attack</code>), Special Defense (<code>sp_defense</code>), and Speed (<code>speed</code>), generation, and legendary status.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/pokemon.csv"</span></span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(url)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>In your groups,</p>
<ol type="1">
<li>discuss what questions you might be interested in answering with these data,</li>
<li>discuss how you might analyse these data so that you could answer those questions, and</li>
<li>and carry out any analysis you deem appropriate.</li>
</ol>
<p><strong>NOTE: You will be expected to give a brief presentation about your approach and analysis.</strong></p>
</section>
</section>
<section id="sec-ica" class="level2">
<h2 class="anchored" data-anchor-id="sec-ica">Interactive Coding Assessments (ICA)</h2>
<p>Recall <a href="#sec-ioa" class="quarto-xref">Section&nbsp;6.7</a> and your first <a href="https://canvas.auckland.ac.nz/courses/120054/assignments/436870">interactive assessment</a>! Now it’s time for <a href="https://canvas.auckland.ac.nz/courses/120054/assignments/436871">your second, which will have a focus on coding practices</a>! How this is achieved is up you you!</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
What is an IOA
</div>
</div>
<div class="callout-body-container callout-body">
<blockquote class="blockquote">
<p>IOAs are novel, low-stakes authentic assessments, designed to promote academic integrity and prepare students for the workplace. Communication as a key skill in the UoA’s graduate profile and is integral for students’ skill development and employability. An IOA is authentic because it is based on a real-world scenario and promotes student engagement and facilitates higher-order thinking. It also preserves academic integrity through its unscripted conversation prompts based on student responses. — <span class="citation" data-cites="ioa">Ward et al. (<a href="#ref-ioa" role="doc-biblioref">2024</a>)</span></p>
</blockquote>
</div>
</div>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In groups of two or three…
</div>
</div>
<div class="callout-body-container callout-body">
<p>…draw on your experience from your first <a href="https://canvas.auckland.ac.nz/courses/120054/assignments/436870">interactive assessment</a> and</p>
<ol type="1">
<li><strong>Design an ICA</strong> that demonstrates the participants effective data management, wrangling and visualisation skills as well as clearly communicating their thought process whilst doing so. Things that are set (by me!)</li>
</ol>
<ul>
<li>the ICA should be designed to take no more than 10 minutes,</li>
<li>the ICA should be interactive and require participants to demonstrate <code>R</code> programming skills,</li>
<li>the ICA should be designed such that everyone <em>starts</em> on an equal footing (e.g., participants will all use a pre-provided laptop with the same software installed), and</li>
<li>the ICA should facilitate flexibility (e.g., no set questions) and reflect, as best as it can, an authentic scenario.</li>
</ul>
<ol start="2" type="1">
<li>In line with the <a href="https://www.auckland.ac.nz/en/about-us/about-the-university/policy-hub/education-student-experience/assessment/grade-descriptors-policy.html">University of Auckland’s Grade Descriptors Policy</a> <strong>design a rubric for the ICA</strong> outlining what you might expect for each Grade/GPA score.</li>
</ol>
</div>
</div>
<section id="some-example-ica-scenarios-you-may-want-to-consideramend" class="level3">
<h3 class="anchored" data-anchor-id="some-example-ica-scenarios-you-may-want-to-consideramend">Some example ICA scenarios you may want to consider/amend</h3>
<ol type="a">
<li>You are in a technical interview with a senior data analyst at a research lab. Your task is to wrangle and visualize a provided dataset. The focus will be on your approach to solving any problems, how you handle any challenges, and your coding process. The analyst is there to observe your methods and after support if needed, not to judge the final result.</li>
</ol>
<p><em>You have 10 minutes to complete the task. This task will focus on</em></p>
<ul>
<li>How you approach <strong>cleaning the data</strong> and dealing with any issues*<br>
</li>
<li>How you use <strong>external tools</strong> (e.g., Stack Overflow, AI assistance) to resolve challenges<br>
</li>
<li>How <strong>clean and readable</strong> your code is<br>
</li>
<li>How well you <strong>explain your process</strong>, reasoning, and decisions</li>
</ul>
<ol start="2" type="a">
<li>You have been asked to carry out a code review (script pre-provided) on a script used by a colleague for data cleaning, analysis, and visualization. Your task is to interpret the code and communicate your understanding what it does alongside identifying any bugs/inefficiencies and/or unclear coding practices.</li>
</ol>
<p><em>You have 10 minutes to complete the task. This task will focus on</em></p>
<ul>
<li>How you <strong>interpret</strong> what unfamiliar code is doing<br>
</li>
<li>Your ability to provide <strong>constructive feedback</strong> (technical and stylistic)<br>
</li>
<li>How well you <strong>communicate</strong> your reasoning, and decisions</li>
<li>Your ability to recognize opportunities to improve <strong>efficiency</strong> and/or <strong>reproducibility</strong> etc.</li>
</ul>
</section>
</section>
</section>
<section id="may-12th" class="level1">
<h1>May 12th</h1>
<section id="rough-timeline-15" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-15">Rough timeline</h2>
<ul>
<li><strong>09:05–09:25 am</strong> Multivariate Data Analysis (<a href="#sec-multivariate" class="quarto-xref">Section&nbsp;16.2</a>)
<ul>
<li>Class discussion</li>
<li>Class activities</li>
</ul></li>
<li><strong>09:25–09:55 am</strong> PCA (<a href="#sec-pca" class="quarto-xref">Section&nbsp;16.3</a>)</li>
<li><strong>09:55–11:05 am</strong> Break</li>
<li><strong>11:05–11:50 am</strong> Eigenfaces (<a href="#sec-eigenfaces" class="quarto-xref">Section&nbsp;16.4</a>)</li>
</ul>
</section>
<section id="sec-multivariate" class="level2">
<h2 class="anchored" data-anchor-id="sec-multivariate">Introduction to Multivariate Data Analysis</h2>
<blockquote class="blockquote">
<p>An assignment was given to BIOSCI220 students in 2022 that asked them to create an <code>R</code> script that produced a plot. This was marked using a rubric-based marking scheme that had 5 sections I1, I2, I3, I4, and I5. Submissions were graded by three peers (prefix P) and three tutors (prefix T), the average across the three grades was taken to be the scores (T and P) summarised in the following table and plot. <strong>What inference can you draw?</strong></p>
</blockquote>
<p><img src="runsheets/img/table.png" class="img-fluid"></p>
</section>
<section id="sec-pca" class="level2">
<h2 class="anchored" data-anchor-id="sec-pca">Principal Component Analysis (PCA)</h2>
<p>Compare the following two GIFs; <strong>what is the main difference?</strong></p>
<div>

</div>
<div class="quarto-layout-panel" data-layout-ncol="2">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="https://raw.githubusercontent.com/statbiscuit/swots/refs/heads/master/figs_n_gifs/lm.gif" class="img-fluid"></p>
</div>
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="https://raw.githubusercontent.com/statbiscuit/swots/refs/heads/master/figs_n_gifs/perp.gif" class="img-fluid"></p>
</div>
</div>
</div>
<p><strong>Principal Component Analysis (PCA)</strong> is a dimension reduction technique! In summary, we want to</p>
<ol type="1">
<li>find important features in a dataset, and</li>
<li>find the best low-dimensional representation of the variation in the multivariate dataset.</li>
</ol>
<p>The new variables we create with PCA are called Principal components (PCs)</p>
<p><em>Note that one problem with PCA is that the components are not scale invariant. So, we standardise (scale) by centering and convert variables into standard deviation units. See <a href="https://biosci738.github.io/BIOSCI738/principal-component-analysis-pca.html">this section of the course guide for more details</a>.</em></p>
<p>Two things we’re interested in:</p>
<ul>
<li>the <strong>direction</strong> of these PCs (otherwise called eigenvectors), which reflect the weight that each variable has on a particular PC.</li>
<li>the <strong>magnitude</strong> of theses PCs (otherwise called eigenvalues), which represent the variance explained by each PC.</li>
</ul>
<p>We’ll work through the code below together, it is that from <a href="https://biosci738.github.io/BIOSCI738/principal-component-analysis-pca.html">this section of the course guide for more details</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(tidyverse)</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(palmerpenguins)</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(GGally)</span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>penguins <span class="sc">%&gt;%</span></span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(species, <span class="fu">where</span>(is.numeric)) <span class="sc">%&gt;%</span> </span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggpairs</span>(<span class="fu">aes</span>(<span class="at">color =</span> species),</span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"flipper_length_mm"</span>, <span class="st">"body_mass_g"</span>, </span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"bill_length_mm"</span>, <span class="st">"bill_depth_mm"</span>)) </span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> penguins <span class="sc">%&gt;%</span> </span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">where</span>(is.numeric), <span class="sc">-</span>year) <span class="sc">%&gt;%</span> <span class="do">## year makes no sense here so we remove it and keep the other numeric variables</span></span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> <span class="do">## can't have NA values in any numeric</span></span>
<span id="cb84-12"><a href="#cb84-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale</span>() <span class="sc">%&gt;%</span> <span class="do">## scale the variables</span></span>
<span id="cb84-13"><a href="#cb84-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prcomp</span>()</span>
<span id="cb84-14"><a href="#cb84-14" aria-hidden="true" tabindex="-1"></a><span class="do">## print out a summary</span></span>
<span id="cb84-15"><a href="#cb84-15" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(pca)</span>
<span id="cb84-16"><a href="#cb84-16" aria-hidden="true" tabindex="-1"></a><span class="do">## loadings </span></span>
<span id="cb84-17"><a href="#cb84-17" aria-hidden="true" tabindex="-1"></a>pca<span class="sc">$</span>rotation</span>
<span id="cb84-18"><a href="#cb84-18" aria-hidden="true" tabindex="-1"></a><span class="do">## PC1 vs PC2</span></span>
<span id="cb84-19"><a href="#cb84-19" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(factoextra) <span class="do">## install this package first</span></span>
<span id="cb84-20"><a href="#cb84-20" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_pca_biplot</span>(pca, <span class="at">geom =</span> <span class="st">"point"</span>) <span class="sc">+</span></span>
<span id="cb84-21"><a href="#cb84-21" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_point</span> (<span class="at">alpha =</span> <span class="fl">0.2</span>)</span>
<span id="cb84-22"><a href="#cb84-22" aria-hidden="true" tabindex="-1"></a><span class="do">## PC2 vs PC3</span></span>
<span id="cb84-23"><a href="#cb84-23" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_pca_biplot</span>(pca, <span class="at">axes =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>),<span class="at">geom =</span> <span class="st">"point"</span>) <span class="sc">+</span></span>
<span id="cb84-24"><a href="#cb84-24" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_point</span> (<span class="at">alpha =</span> <span class="fl">0.2</span>)</span>
<span id="cb84-25"><a href="#cb84-25" aria-hidden="true" tabindex="-1"></a><span class="do">## how many are informative enough</span></span>
<span id="cb84-26"><a href="#cb84-26" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_screeplot</span>(pca)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, think back to <a href="#sec-multivariate" class="quarto-xref">Section&nbsp;16.2</a> and the BIOSCI220 data. What conclusions might you draw from the following plot of the loadings?</p>
<p><img src="runsheets/img/pca_loadings.png" class="img-fluid"></p>
</section>
<section id="sec-eigenfaces" class="level2">
<h2 class="anchored" data-anchor-id="sec-eigenfaces">Eigenfaces</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In groups of two or three..
</div>
</div>
<div class="callout-body-container callout-body">
<p>… work through the code below and discuss what each step is achieving, what do you conclude?</p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>data_url <span class="ot">&lt;-</span> <span class="st">"https://github.com/cmjt/statbiscuits/raw/master/swots/data/faces.RData"</span></span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">url</span>(data_url))</span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pixmap)</span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(faces)</span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">4</span>), <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">oma =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(faces)){</span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot</span>(faces[[i]])</span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">legend</span>(<span class="st">"bottom"</span>, <span class="at">bty =</span> <span class="st">"n"</span>, <span class="fu">names</span>(faces[i]))</span>
<span id="cb85-9"><a href="#cb85-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb85-10"><a href="#cb85-10" aria-hidden="true" tabindex="-1"></a><span class="do">## collapse the grey matrix</span></span>
<span id="cb85-11"><a href="#cb85-11" aria-hidden="true" tabindex="-1"></a>face_data <span class="ot">&lt;-</span> <span class="fu">lapply</span>(faces, <span class="cf">function</span>(x) <span class="fu">c</span>(x<span class="sc">@</span>grey))</span>
<span id="cb85-12"><a href="#cb85-12" aria-hidden="true" tabindex="-1"></a><span class="do">## each column is a face</span></span>
<span id="cb85-13"><a href="#cb85-13" aria-hidden="true" tabindex="-1"></a>face_data <span class="ot">&lt;-</span> <span class="fu">do.call</span>(<span class="st">'cbind'</span>,face_data)</span>
<span id="cb85-14"><a href="#cb85-14" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(face_data) <span class="ot">&lt;-</span> <span class="fu">names</span>(faces)</span>
<span id="cb85-15"><a href="#cb85-15" aria-hidden="true" tabindex="-1"></a><span class="do">## have a look at the data matrix</span></span>
<span id="cb85-16"><a href="#cb85-16" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(face_data)</span>
<span id="cb85-17"><a href="#cb85-17" aria-hidden="true" tabindex="-1"></a><span class="do">## center around mean</span></span>
<span id="cb85-18"><a href="#cb85-18" aria-hidden="true" tabindex="-1"></a>mean <span class="ot">&lt;-</span> <span class="fu">apply</span>(face_data, <span class="dv">1</span>, mean)</span>
<span id="cb85-19"><a href="#cb85-19" aria-hidden="true" tabindex="-1"></a>mean_face <span class="ot">&lt;-</span> <span class="fu">pixmapGrey</span>(mean, <span class="dv">64</span>, <span class="dv">64</span>, <span class="at">bbox =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">64</span>, <span class="dv">64</span>))</span>
<span id="cb85-20"><a href="#cb85-20" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mean_face)</span>
<span id="cb85-21"><a href="#cb85-21" aria-hidden="true" tabindex="-1"></a>centered <span class="ot">&lt;-</span> <span class="fu">apply</span>(face_data, <span class="dv">2</span>, <span class="cf">function</span>(x) x <span class="sc">-</span> mean)</span>
<span id="cb85-22"><a href="#cb85-22" aria-hidden="true" tabindex="-1"></a><span class="do">## centered images</span></span>
<span id="cb85-23"><a href="#cb85-23" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">4</span>), <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">oma =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb85-24"><a href="#cb85-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(faces)){</span>
<span id="cb85-25"><a href="#cb85-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot</span>(<span class="fu">pixmapGrey</span>(centered[,i], <span class="dv">64</span>, <span class="dv">64</span>, <span class="at">bbox =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">64</span>, <span class="dv">64</span>)))</span>
<span id="cb85-26"><a href="#cb85-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">legend</span>(<span class="st">"bottom"</span>, <span class="at">bty =</span> <span class="st">"n"</span>, <span class="fu">paste</span>(<span class="fu">names</span>(faces[i]), <span class="st">"centered"</span>, <span class="at">sep =</span> <span class="st">" "</span>))</span>
<span id="cb85-27"><a href="#cb85-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb85-28"><a href="#cb85-28" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(centered)</span>
<span id="cb85-29"><a href="#cb85-29" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(pca)</span>
<span id="cb85-30"><a href="#cb85-30" aria-hidden="true" tabindex="-1"></a><span class="do">## for a nice biplot</span></span>
<span id="cb85-31"><a href="#cb85-31" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggfortify)</span>
<span id="cb85-32"><a href="#cb85-32" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb85-33"><a href="#cb85-33" aria-hidden="true" tabindex="-1"></a><span class="do">## using autoplot</span></span>
<span id="cb85-34"><a href="#cb85-34" aria-hidden="true" tabindex="-1"></a><span class="do">## PC1 vs PC2 by default</span></span>
<span id="cb85-35"><a href="#cb85-35" aria-hidden="true" tabindex="-1"></a><span class="fu">autoplot</span>(pca,<span class="at">loadings =</span> <span class="cn">TRUE</span>,<span class="at">loadings.label =</span> <span class="cn">TRUE</span>,<span class="at">alpha =</span> <span class="fl">0.1</span>)</span>
<span id="cb85-36"><a href="#cb85-36" aria-hidden="true" tabindex="-1"></a><span class="do">## play around with the arguments to see what they control</span></span>
<span id="cb85-37"><a href="#cb85-37" aria-hidden="true" tabindex="-1"></a><span class="fu">autoplot</span>(pca,<span class="at">x =</span> <span class="dv">3</span>, <span class="at">y =</span> <span class="dv">4</span>,<span class="at">loadings =</span> <span class="cn">TRUE</span>,<span class="at">loadings.label =</span> <span class="cn">TRUE</span>,<span class="at">alpha =</span> <span class="fl">0.1</span>)</span>
<span id="cb85-38"><a href="#cb85-38" aria-hidden="true" tabindex="-1"></a><span class="do">## eigen faces</span></span>
<span id="cb85-39"><a href="#cb85-39" aria-hidden="true" tabindex="-1"></a>pcs <span class="ot">&lt;-</span> pca<span class="sc">$</span>x</span>
<span id="cb85-40"><a href="#cb85-40" aria-hidden="true" tabindex="-1"></a>eigenfaces <span class="ot">&lt;-</span> <span class="fu">apply</span>(pcs,<span class="dv">2</span>, <span class="cf">function</span>(x)  <span class="fu">pixmapGrey</span>(x, <span class="dv">64</span>, <span class="dv">64</span>, <span class="at">bbox =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">64</span>, <span class="dv">64</span>)))</span>
<span id="cb85-41"><a href="#cb85-41" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">4</span>), <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">oma =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb85-42"><a href="#cb85-42" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(eigenfaces)){</span>
<span id="cb85-43"><a href="#cb85-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot</span>(eigenfaces[[i]])</span>
<span id="cb85-44"><a href="#cb85-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">legend</span>(<span class="st">"bottom"</span>, <span class="at">bty =</span> <span class="st">"n"</span>, <span class="fu">paste</span>(<span class="st">"PC"</span>, i, <span class="at">sep =</span> <span class="st">" "</span>))</span>
<span id="cb85-45"><a href="#cb85-45" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb85-46"><a href="#cb85-46" aria-hidden="true" tabindex="-1"></a><span class="do">## Each face is a weighted (loadings) combintation of the eigen faces (PCs) above, </span></span>
<span id="cb85-47"><a href="#cb85-47" aria-hidden="true" tabindex="-1"></a><span class="do">## but how many PCs (eigenfaces) do we keep?</span></span>
<span id="cb85-48"><a href="#cb85-48" aria-hidden="true" tabindex="-1"></a><span class="do">## screeplot</span></span>
<span id="cb85-49"><a href="#cb85-49" aria-hidden="true" tabindex="-1"></a><span class="fu">screeplot</span>(pca,<span class="at">type =</span> <span class="st">"lines"</span>, <span class="at">pch =</span> <span class="dv">20</span>)</span>
<span id="cb85-50"><a href="#cb85-50" aria-hidden="true" tabindex="-1"></a><span class="do">## 2 PCs ?</span></span>
<span id="cb85-51"><a href="#cb85-51" aria-hidden="true" tabindex="-1"></a>n_pc <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb85-52"><a href="#cb85-52" aria-hidden="true" tabindex="-1"></a>recon <span class="ot">&lt;-</span> pixm <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb85-53"><a href="#cb85-53" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(faces)){</span>
<span id="cb85-54"><a href="#cb85-54" aria-hidden="true" tabindex="-1"></a>    recon[[i]] <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(pca<span class="sc">$</span>x[,<span class="dv">1</span><span class="sc">:</span>n_pc]<span class="sc">%*%</span>pca<span class="sc">$</span>rotation[i,<span class="dv">1</span><span class="sc">:</span>n_pc])</span>
<span id="cb85-55"><a href="#cb85-55" aria-hidden="true" tabindex="-1"></a>    pixm[[i]] <span class="ot">&lt;-</span> <span class="fu">pixmapGrey</span>(recon[[i]], <span class="dv">64</span>, <span class="dv">64</span>, <span class="at">bbox =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">64</span>, <span class="dv">64</span>))</span>
<span id="cb85-56"><a href="#cb85-56" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb85-57"><a href="#cb85-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-58"><a href="#cb85-58" aria-hidden="true" tabindex="-1"></a><span class="do">## 2PC to reconstruct centered images</span></span>
<span id="cb85-59"><a href="#cb85-59" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">4</span>), <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="at">oma =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb85-60"><a href="#cb85-60" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pixm)){</span>
<span id="cb85-61"><a href="#cb85-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot</span>(pixm[[i]])</span>
<span id="cb85-62"><a href="#cb85-62" aria-hidden="true" tabindex="-1"></a>    <span class="fu">legend</span>(<span class="st">"bottom"</span>, <span class="at">bty =</span> <span class="st">"n"</span>, <span class="fu">paste</span>(<span class="fu">names</span>(faces[i]), <span class="st">"2 PC"</span>, <span class="at">sep =</span> <span class="st">" "</span>))</span>
<span id="cb85-63"><a href="#cb85-63" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb85-64"><a href="#cb85-64" aria-hidden="true" tabindex="-1"></a><span class="do">## Now add on the "mean face"</span></span>
<span id="cb85-65"><a href="#cb85-65" aria-hidden="true" tabindex="-1"></a><span class="do">## Is 2 enough? Try 5... Or 10 PCs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="may-16th" class="level1">
<h1>May 16th</h1>
<section id="rough-timeline-16" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-16">Rough timeline</h2>
<section id="pm---0355-pm-7" class="level3">
<h3 class="anchored" data-anchor-id="pm---0355-pm-7"><strong>02:05 pm - 03:55 pm</strong></h3>
<ul>
<li><strong>02:05–02.30 pm</strong> Miscellaneous activities (<a href="#sec-misc" class="quarto-xref">Section&nbsp;17.2</a>)</li>
<li><strong>02:30–02.50 pm</strong> Multidimensional Scaling (<a href="#sec-mds" class="quarto-xref">Section&nbsp;17.3</a>)</li>
<li><strong>02:50–03:00 pm</strong> Break</li>
<li><strong>03:00–03:55 pm</strong> Casestudies (<a href="#sec-mdscasestudies" class="quarto-xref">Section&nbsp;17.4</a>)</li>
</ul>
</section>
</section>
<section id="sec-misc" class="level2">
<h2 class="anchored" data-anchor-id="sec-misc">Miscellaneous activities</h2>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
ICA
</div>
</div>
<div class="callout-body-container callout-body">
<p>Recall <a href="#sec-ica" class="quarto-xref">Section&nbsp;15.3</a>? Listen to hear the upshot. Any questions?</p>
</div>
</div>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Soon…
</div>
</div>
<div class="callout-body-container callout-body">
<p>Charlotte will soon share some PCA analysis carried out using the grades of this class to date! Obviously, I can’t give you the data; however, based on the outputs shown let’s discuss what inference PCA offers you!</p>
</div>
</div>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Fill out <a href="https://docs.google.com/forms/d/e/1FAIpQLSc2Lxlr4IwrooEoQ6SgWSdFuv_ULVL-XL0GBt6q3TO98FWyOQ/viewform?usp=sf_link">THIS QUIZ</a>
</div>
</div>
<div class="callout-body-container callout-body">
<p><strong>Note I will be using this data live in class, you do not have to answer honestly if you are uncomfortable doing so, but please be consistent! Perhaps answer as if you were your favourite gaming character. I will be sharing the responses with the class.</strong></p>
<p><em>Questions stolen from <a href="https://community.typeform.com/build-your-typeform-7/using-opinion-scales-for-personality-quizzes-1929">this page</a></em></p>
</div>
</div>
<p>Using <a href="https://docs.google.com/spreadsheets/d/1Oiyz5WgMKCKpNw7zkHN8Vw0-zMx-5tU9yUOwtWUhUMs/edit?usp=sharing">these data (i.e., that collected above)</a> we’ll now work through the code below together.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"person_quiz.csv"</span>) <span class="do">## or equivalent</span></span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(tidyverse)</span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a><span class="do">## set up names and correlations</span></span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(data)[<span class="dv">3</span><span class="sc">:</span><span class="dv">12</span>] <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_match</span>(<span class="fu">names</span>(data)[<span class="dv">3</span><span class="sc">:</span><span class="dv">12</span>], <span class="st">"as.</span><span class="sc">\\</span><span class="st">s*(.*?)</span><span class="sc">\\</span><span class="st">s*..Do"</span>)[,<span class="dv">2</span>]</span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a><span class="do">## variables</span></span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">where</span>(is.numeric)) <span class="sc">|&gt;</span> <span class="fu">cor</span>() <span class="sc">|&gt;</span> corrplot<span class="sc">::</span><span class="fu">corrplot</span>(<span class="at">addrect =</span> <span class="dv">2</span>, <span class="at">order =</span> <span class="st">'hclust'</span>)</span>
<span id="cb86-7"><a href="#cb86-7" aria-hidden="true" tabindex="-1"></a><span class="do">## people</span></span>
<span id="cb86-8"><a href="#cb86-8" aria-hidden="true" tabindex="-1"></a>people <span class="ot">&lt;-</span> <span class="fu">t</span>(data[,<span class="dv">3</span><span class="sc">:</span><span class="dv">12</span>])</span>
<span id="cb86-9"><a href="#cb86-9" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(people) <span class="ot">&lt;-</span> data[,<span class="dv">2</span>]</span>
<span id="cb86-10"><a href="#cb86-10" aria-hidden="true" tabindex="-1"></a>people <span class="sc">|&gt;</span> <span class="fu">scale</span>() <span class="sc">|&gt;</span> <span class="fu">cor</span>() <span class="sc">|&gt;</span> corrplot<span class="sc">::</span><span class="fu">corrplot</span>()</span>
<span id="cb86-11"><a href="#cb86-11" aria-hidden="true" tabindex="-1"></a><span class="do">## PCA</span></span>
<span id="cb86-12"><a href="#cb86-12" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb86-13"><a href="#cb86-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="fu">where</span>(is.numeric)) <span class="sc">|&gt;</span> <span class="fu">scale</span>() <span class="sc">|&gt;</span> <span class="fu">prcomp</span>()</span>
<span id="cb86-14"><a href="#cb86-14" aria-hidden="true" tabindex="-1"></a>factoextra<span class="sc">::</span><span class="fu">fviz_pca_biplot</span>(pca, <span class="at">geom =</span> <span class="st">"point"</span>)</span>
<span id="cb86-15"><a href="#cb86-15" aria-hidden="true" tabindex="-1"></a>factoextra<span class="sc">::</span><span class="fu">fviz_screeplot</span>(pca)</span>
<span id="cb86-16"><a href="#cb86-16" aria-hidden="true" tabindex="-1"></a>pca<span class="sc">$</span>rotation <span class="sc">%&gt;%</span></span>
<span id="cb86-17"><a href="#cb86-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb86-18"><a href="#cb86-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">variables =</span> <span class="fu">rownames</span>(.)) <span class="sc">%&gt;%</span></span>
<span id="cb86-19"><a href="#cb86-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>(PC,loading,PC1<span class="sc">:</span>PC3) <span class="sc">%&gt;%</span></span>
<span id="cb86-20"><a href="#cb86-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="fu">abs</span>(loading), variables, <span class="at">fill =</span> loading <span class="sc">&gt;</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb86-21"><a href="#cb86-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb86-22"><a href="#cb86-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>PC, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb86-23"><a href="#cb86-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Absolute value of loading"</span>,<span class="at">y =</span> <span class="cn">NULL</span>, <span class="at">fill =</span> <span class="st">"Positive?"</span>)</span>
<span id="cb86-24"><a href="#cb86-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-25"><a href="#cb86-25" aria-hidden="true" tabindex="-1"></a><span class="do">## Euclidean distance matrix using scaled data</span></span>
<span id="cb86-26"><a href="#cb86-26" aria-hidden="true" tabindex="-1"></a>dists <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb86-27"><a href="#cb86-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="fu">where</span>(is.numeric)) <span class="sc">|&gt;</span> <span class="fu">scale</span>() <span class="sc">|&gt;</span> <span class="fu">dist</span>(<span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">diag =</span> <span class="cn">TRUE</span>)</span>
<span id="cb86-28"><a href="#cb86-28" aria-hidden="true" tabindex="-1"></a>mds <span class="ot">&lt;-</span> <span class="fu">cmdscale</span>(dists, <span class="at">eig =</span> <span class="cn">TRUE</span>)</span>
<span id="cb86-29"><a href="#cb86-29" aria-hidden="true" tabindex="-1"></a>mds</span>
<span id="cb86-30"><a href="#cb86-30" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mds<span class="sc">$</span>points, <span class="at">cex =</span> <span class="dv">0</span>)</span>
<span id="cb86-31"><a href="#cb86-31" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(mds<span class="sc">$</span>points, <span class="at">label =</span> data[,<span class="dv">2</span>])</span>
<span id="cb86-32"><a href="#cb86-32" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggfortify)</span>
<span id="cb86-33"><a href="#cb86-33" aria-hidden="true" tabindex="-1"></a><span class="fu">autoplot</span>(mds, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">shape =</span> <span class="cn">FALSE</span>,</span>
<span id="cb86-34"><a href="#cb86-34" aria-hidden="true" tabindex="-1"></a>         <span class="at">label.size =</span> <span class="dv">3</span>)</span>
<span id="cb86-35"><a href="#cb86-35" aria-hidden="true" tabindex="-1"></a><span class="do">## Manhattan distance matrix using scaled data (typically used treating ordinal as interval)</span></span>
<span id="cb86-36"><a href="#cb86-36" aria-hidden="true" tabindex="-1"></a>dists <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb86-37"><a href="#cb86-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="fu">where</span>(is.numeric)) <span class="sc">|&gt;</span> <span class="fu">scale</span>() <span class="sc">|&gt;</span> <span class="fu">dist</span>(<span class="at">method =</span> <span class="st">"manhattan"</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">diag =</span> <span class="cn">TRUE</span>)</span>
<span id="cb86-38"><a href="#cb86-38" aria-hidden="true" tabindex="-1"></a>mds <span class="ot">&lt;-</span> <span class="fu">cmdscale</span>(dists, <span class="at">eig =</span> <span class="cn">TRUE</span>)</span>
<span id="cb86-39"><a href="#cb86-39" aria-hidden="true" tabindex="-1"></a>mds</span>
<span id="cb86-40"><a href="#cb86-40" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mds<span class="sc">$</span>points, <span class="at">cex =</span> <span class="dv">0</span>)</span>
<span id="cb86-41"><a href="#cb86-41" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(mds<span class="sc">$</span>points, <span class="at">label =</span> data[,<span class="dv">2</span>])</span>
<span id="cb86-42"><a href="#cb86-42" aria-hidden="true" tabindex="-1"></a><span class="do">## Minkowski distance matrix using scaled data (typically used)</span></span>
<span id="cb86-43"><a href="#cb86-43" aria-hidden="true" tabindex="-1"></a>dists <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb86-44"><a href="#cb86-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="fu">where</span>(is.numeric)) <span class="sc">|&gt;</span> <span class="fu">scale</span>() <span class="sc">|&gt;</span> <span class="fu">dist</span>(<span class="at">method =</span> <span class="st">"minkowski"</span>, <span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">diag =</span> <span class="cn">TRUE</span>)</span>
<span id="cb86-45"><a href="#cb86-45" aria-hidden="true" tabindex="-1"></a>mds <span class="ot">&lt;-</span> <span class="fu">cmdscale</span>(dists, <span class="at">eig =</span> <span class="cn">TRUE</span>)</span>
<span id="cb86-46"><a href="#cb86-46" aria-hidden="true" tabindex="-1"></a>mds</span>
<span id="cb86-47"><a href="#cb86-47" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="at">x =</span> mds<span class="sc">$</span>points[,<span class="dv">1</span>], <span class="at">y =</span> mds<span class="sc">$</span>points[,<span class="dv">2</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb86-48"><a href="#cb86-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">label =</span> data[,<span class="dv">2</span>])) <span class="sc">+</span></span>
<span id="cb86-49"><a href="#cb86-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">0.9</span>) <span class="sc">+</span> <span class="fu">coord_fixed</span>()  <span class="sc">+</span></span>
<span id="cb86-50"><a href="#cb86-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"BIOSCI738 Personality Map..."</span>) <span class="sc">+</span> <span class="fu">theme_void</span>() <span class="sc">+</span></span>
<span id="cb86-51"><a href="#cb86-51" aria-hidden="true" tabindex="-1"></a>   <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">y =</span> <span class="sc">-</span><span class="dv">4</span>, <span class="at">xend =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">yend =</span> <span class="fl">4.5</span>),</span>
<span id="cb86-52"><a href="#cb86-52" aria-hidden="true" tabindex="-1"></a>                <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">"cm"</span>)), <span class="at">col =</span> <span class="st">"grey"</span>) <span class="sc">+</span></span>
<span id="cb86-53"><a href="#cb86-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="sc">-</span><span class="dv">4</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> <span class="fl">2.5</span>, <span class="at">yend =</span> <span class="dv">0</span>),</span>
<span id="cb86-54"><a href="#cb86-54" aria-hidden="true" tabindex="-1"></a>                 <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">"cm"</span>)), <span class="at">col =</span> <span class="st">"grey"</span>) <span class="sc">+</span></span>
<span id="cb86-55"><a href="#cb86-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">size =</span> <span class="fl">3.5</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.8</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="sec-mds" class="level2">
<h2 class="anchored" data-anchor-id="sec-mds">Multidimensional Scaling</h2>
<p><strong>Slides</strong></p>
<p align="center">
<iframe src="slides/08-01_mds.pdf" height="400" width="700" title="Maximum Likelihood Estimation">
</iframe>
</p>
<p align="center">
<a href="https://canvas.auckland.ac.nz/courses/120054/files/15372455?module_item_id=2614789"><strong>CANVAS slides</strong></a>
</p>
<p><strong>Examples</strong> from <a href="https://biosci738.github.io/BIOSCI738/multidimensional-scaling-mds.html">this section of the courseguide</a></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(factoextra) </span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggfortify) </span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrplot)</span>
<span id="cb87-8"><a href="#cb87-8" aria-hidden="true" tabindex="-1"></a>base_url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/"</span></span>
<span id="cb87-9"><a href="#cb87-9" aria-hidden="true" tabindex="-1"></a>north_island <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">paste</span>(base_url, <span class="st">"north_islands_distances.csv"</span>, <span class="at">sep =</span> <span class="st">""</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb87-10"><a href="#cb87-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"...1"</span>)</span>
<span id="cb87-11"><a href="#cb87-11" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">"HairEyeColor"</span>, <span class="at">package =</span> <span class="st">"datasets"</span>)</span>
<span id="cb87-12"><a href="#cb87-12" aria-hidden="true" tabindex="-1"></a>ekman <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">paste</span>(base_url, <span class="st">"ekman.csv"</span>, <span class="at">sep =</span> <span class="st">""</span>))</span>
<span id="cb87-13"><a href="#cb87-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-14"><a href="#cb87-14" aria-hidden="true" tabindex="-1"></a><span class="do">## North Island cities</span></span>
<span id="cb87-15"><a href="#cb87-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Principal Coordinate Analysis (PCO): </span></span>
<span id="cb87-16"><a href="#cb87-16" aria-hidden="true" tabindex="-1"></a><span class="do">## also known as classical multidimensional scaling (classical MDS). </span></span>
<span id="cb87-17"><a href="#cb87-17" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(north_island)</span>
<span id="cb87-18"><a href="#cb87-18" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(north_island, <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,</span>
<span id="cb87-19"><a href="#cb87-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">treeheight_row =</span> <span class="dv">2</span>, <span class="at">treeheight_col =</span> <span class="dv">2</span>,</span>
<span id="cb87-20"><a href="#cb87-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">12</span>, <span class="at">fontsize_col =</span> <span class="dv">12</span>,</span>
<span id="cb87-21"><a href="#cb87-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">cellwidth =</span> <span class="dv">26</span>, <span class="at">cellheight =</span> <span class="dv">26</span>)</span>
<span id="cb87-22"><a href="#cb87-22" aria-hidden="true" tabindex="-1"></a>mds <span class="ot">&lt;-</span> <span class="fu">cmdscale</span>(north_island, <span class="at">eig =</span> <span class="cn">TRUE</span>)</span>
<span id="cb87-23"><a href="#cb87-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-24"><a href="#cb87-24" aria-hidden="true" tabindex="-1"></a><span class="do">## GOF based on eigenvalues (scree plot)</span></span>
<span id="cb87-25"><a href="#cb87-25" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mds<span class="sc">$</span>eig, <span class="at">type=</span><span class="st">"b"</span>,  <span class="at">ylab=</span><span class="st">"Eigenvalue"</span>, <span class="at">xlab=</span><span class="st">"Index"</span>, <span class="at">lwd=</span><span class="dv">3</span>)</span>
<span id="cb87-26"><a href="#cb87-26" aria-hidden="true" tabindex="-1"></a><span class="do">## GOF (proportion explained by first two PC)</span></span>
<span id="cb87-27"><a href="#cb87-27" aria-hidden="true" tabindex="-1"></a>mds<span class="sc">$</span>GOF</span>
<span id="cb87-28"><a href="#cb87-28" aria-hidden="true" tabindex="-1"></a><span class="do">## plot</span></span>
<span id="cb87-29"><a href="#cb87-29" aria-hidden="true" tabindex="-1"></a><span class="fu">autoplot</span>(mds, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">shape =</span> <span class="cn">FALSE</span>) <span class="do">## does this look right?</span></span>
<span id="cb87-30"><a href="#cb87-30" aria-hidden="true" tabindex="-1"></a><span class="do">## switch direction as distances are only relative</span></span>
<span id="cb87-31"><a href="#cb87-31" aria-hidden="true" tabindex="-1"></a><span class="do">## Note however that axes (principal coordinates) are rarely interpretable in</span></span>
<span id="cb87-32"><a href="#cb87-32" aria-hidden="true" tabindex="-1"></a><span class="do">## PCO</span></span>
<span id="cb87-33"><a href="#cb87-33" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="at">x =</span> <span class="sc">-</span>mds<span class="sc">$</span>points[,<span class="dv">2</span>], <span class="at">y =</span> mds<span class="sc">$</span>points[,<span class="dv">1</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb87-34"><a href="#cb87-34" aria-hidden="true" tabindex="-1"></a>   <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">label =</span> <span class="fu">rownames</span>(mds<span class="sc">$</span>points))) <span class="sc">+</span></span>
<span id="cb87-35"><a href="#cb87-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>() <span class="sc">+</span><span class="fu">coord_fixed</span>()</span>
<span id="cb87-36"><a href="#cb87-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-37"><a href="#cb87-37" aria-hidden="true" tabindex="-1"></a><span class="do">## HairEyeColor</span></span>
<span id="cb87-38"><a href="#cb87-38" aria-hidden="true" tabindex="-1"></a><span class="do">## Correspondence Analysis (CA) is a special case of metric </span></span>
<span id="cb87-39"><a href="#cb87-39" aria-hidden="true" tabindex="-1"></a><span class="do">## MDS where the distance measure is the chi-square distance </span></span>
<span id="cb87-40"><a href="#cb87-40" aria-hidden="true" tabindex="-1"></a><span class="do">## (again conceptually similar to PCA).</span></span>
<span id="cb87-41"><a href="#cb87-41" aria-hidden="true" tabindex="-1"></a>HC.df <span class="ot">&lt;-</span> <span class="fu">as.data.frame.matrix</span>(HairEyeColor[ , , <span class="dv">2</span>])</span>
<span id="cb87-42"><a href="#cb87-42" aria-hidden="true" tabindex="-1"></a>HC.df</span>
<span id="cb87-43"><a href="#cb87-43" aria-hidden="true" tabindex="-1"></a><span class="fu">chisq.test</span>(HC.df)</span>
<span id="cb87-44"><a href="#cb87-44" aria-hidden="true" tabindex="-1"></a>coaHC <span class="ot">&lt;-</span> <span class="fu">dudi.coa</span>(HC.df, <span class="at">scannf =</span> <span class="cn">FALSE</span>, <span class="at">nf =</span> <span class="dv">2</span>)</span>
<span id="cb87-45"><a href="#cb87-45" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(coaHC<span class="sc">$</span>eig, <span class="at">type=</span><span class="st">"b"</span>,  <span class="at">ylab=</span><span class="st">"Eigenvalue"</span>, <span class="at">xlab=</span><span class="st">"Index"</span>, <span class="at">lwd=</span><span class="dv">3</span>)</span>
<span id="cb87-46"><a href="#cb87-46" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_ca_row</span>(coaHC)</span>
<span id="cb87-47"><a href="#cb87-47" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_ca_col</span>(coaHC)</span>
<span id="cb87-48"><a href="#cb87-48" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_ca_biplot</span>(coaHC, <span class="at">repel =</span> <span class="cn">TRUE</span>, <span class="at">col.col =</span> <span class="st">"brown"</span>, <span class="at">col.row =</span> <span class="st">"purple"</span>,</span>
<span id="cb87-49"><a href="#cb87-49" aria-hidden="true" tabindex="-1"></a>               <span class="at">labelsize =</span> <span class="dv">5</span>, <span class="at">pointsize =</span> <span class="dv">5</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb87-50"><a href="#cb87-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">25</span>),</span>
<span id="cb87-51"><a href="#cb87-51" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x.bottom =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),</span>
<span id="cb87-52"><a href="#cb87-52" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y.left =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),</span>
<span id="cb87-53"><a href="#cb87-53" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)) <span class="sc">+</span></span>
<span id="cb87-54"><a href="#cb87-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.5</span>,<span class="fl">0.5</span>))</span>
<span id="cb87-55"><a href="#cb87-55" aria-hidden="true" tabindex="-1"></a><span class="do">## Colour perception</span></span>
<span id="cb87-56"><a href="#cb87-56" aria-hidden="true" tabindex="-1"></a><span class="do">### Metric</span></span>
<span id="cb87-57"><a href="#cb87-57" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(ekman)</span>
<span id="cb87-58"><a href="#cb87-58" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ekman) <span class="do">## nanometer wavelengths</span></span>
<span id="cb87-59"><a href="#cb87-59" aria-hidden="true" tabindex="-1"></a>mds <span class="ot">&lt;-</span> <span class="fu">cmdscale</span>(ekman, <span class="at">eig =</span> <span class="cn">TRUE</span>)</span>
<span id="cb87-60"><a href="#cb87-60" aria-hidden="true" tabindex="-1"></a><span class="do">## GOF based on eigenvalues (scree plot)</span></span>
<span id="cb87-61"><a href="#cb87-61" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mds<span class="sc">$</span>eig, <span class="at">type=</span><span class="st">"b"</span>,  <span class="at">ylab=</span><span class="st">"Eigenvalue"</span>, <span class="at">xlab=</span><span class="st">"Index"</span>, <span class="at">lwd=</span><span class="dv">3</span>)</span>
<span id="cb87-62"><a href="#cb87-62" aria-hidden="true" tabindex="-1"></a><span class="do">## GOF (proportion explained by first two PC)</span></span>
<span id="cb87-63"><a href="#cb87-63" aria-hidden="true" tabindex="-1"></a>mds<span class="sc">$</span>GOF</span>
<span id="cb87-64"><a href="#cb87-64" aria-hidden="true" tabindex="-1"></a><span class="do">## Non-metric</span></span>
<span id="cb87-65"><a href="#cb87-65" aria-hidden="true" tabindex="-1"></a><span class="do">## 2 d</span></span>
<span id="cb87-66"><a href="#cb87-66" aria-hidden="true" tabindex="-1"></a>mmds2d <span class="ot">&lt;-</span> <span class="fu">metaMDS</span>(ekman, <span class="at">k =</span> <span class="dv">2</span>, <span class="at">autotransform =</span> <span class="cn">FALSE</span>, <span class="at">trace =</span> <span class="cn">FALSE</span>)</span>
<span id="cb87-67"><a href="#cb87-67" aria-hidden="true" tabindex="-1"></a><span class="fu">stressplot</span>(mmds2d, <span class="at">pch =</span> <span class="dv">20</span>, <span class="at">cex =</span> <span class="dv">2</span>)</span>
<span id="cb87-68"><a href="#cb87-68" aria-hidden="true" tabindex="-1"></a>mmds2d<span class="sc">$</span>stress</span>
<span id="cb87-69"><a href="#cb87-69" aria-hidden="true" tabindex="-1"></a><span class="do">## 3 d</span></span>
<span id="cb87-70"><a href="#cb87-70" aria-hidden="true" tabindex="-1"></a>mmds3d <span class="ot">&lt;-</span> <span class="fu">metaMDS</span>(ekman, <span class="at">k =</span> <span class="dv">3</span>, <span class="at">autotransform =</span> <span class="cn">FALSE</span>, <span class="at">trace =</span> <span class="cn">FALSE</span>)</span>
<span id="cb87-71"><a href="#cb87-71" aria-hidden="true" tabindex="-1"></a><span class="fu">stressplot</span>(mmds3d, <span class="at">pch =</span> <span class="dv">20</span>, <span class="at">cex =</span> <span class="dv">2</span>)</span>
<span id="cb87-72"><a href="#cb87-72" aria-hidden="true" tabindex="-1"></a>mmds3d<span class="sc">$</span>stress</span>
<span id="cb87-73"><a href="#cb87-73" aria-hidden="true" tabindex="-1"></a>mmds2d<span class="sc">$</span>points <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb87-74"><a href="#cb87-74" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> MDS1, <span class="at">y =</span> MDS2, <span class="at">label =</span> <span class="fu">colnames</span>(ekman))) <span class="sc">+</span></span>
<span id="cb87-75"><a href="#cb87-75" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_text</span>()</span>
<span id="cb87-76"><a href="#cb87-76" aria-hidden="true" tabindex="-1"></a><span class="do">## rough wavelength conversion to hex code  based on</span></span>
<span id="cb87-77"><a href="#cb87-77" aria-hidden="true" tabindex="-1"></a><span class="do">## https://www.johndcook.com/wavelength_to_RGB.html</span></span>
<span id="cb87-78"><a href="#cb87-78" aria-hidden="true" tabindex="-1"></a><span class="do">## ekman wavelengths based from data(ekman,package = "smacof") attributes</span></span>
<span id="cb87-79"><a href="#cb87-79" aria-hidden="true" tabindex="-1"></a>wavelengths <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">434</span>, <span class="dv">445</span>, <span class="dv">465</span>, <span class="dv">472</span>, <span class="dv">490</span>, <span class="dv">504</span>, <span class="dv">537</span>, <span class="dv">555</span>, <span class="dv">584</span>, <span class="dv">600</span>, <span class="dv">610</span>, <span class="dv">628</span>, <span class="dv">651</span>, <span class="dv">674</span>)</span>
<span id="cb87-80"><a href="#cb87-80" aria-hidden="true" tabindex="-1"></a>hex <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#2800ff"</span>, <span class="st">"#0028ff"</span>, <span class="st">"#0092ff"</span>, <span class="st">"#00b2ff"</span>, <span class="st">"#00ffff"</span>, <span class="st">"#00ff61"</span>, <span class="st">"#77ff00"</span>, <span class="st">"#b3ff00"</span>, <span class="st">"#fff200"</span>, <span class="st">"#ffbe00"</span>, <span class="st">"#ff9b00"</span>, <span class="st">"#ff5700"</span>, <span class="st">"#ff0000"</span>, <span class="st">"#ff0000"</span>)</span>
<span id="cb87-81"><a href="#cb87-81" aria-hidden="true" tabindex="-1"></a><span class="do">## 2D points</span></span>
<span id="cb87-82"><a href="#cb87-82" aria-hidden="true" tabindex="-1"></a>mmds2d<span class="sc">$</span>points <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb87-83"><a href="#cb87-83" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> MDS1, <span class="at">y =</span> MDS2, <span class="at">label =</span> hex, <span class="at">col =</span> hex)) <span class="sc">+</span></span>
<span id="cb87-84"><a href="#cb87-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> hex) <span class="sc">+</span></span>
<span id="cb87-85"><a href="#cb87-85" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_text</span>()</span>
<span id="cb87-86"><a href="#cb87-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-87"><a href="#cb87-87" aria-hidden="true" tabindex="-1"></a><span class="do">## BONUS </span><span class="al">TASK</span><span class="do"> create a 3D plot (e.g., using plotly) showing the results from mmds3d</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="sec-mdscasestudies" class="level2">
<h2 class="anchored" data-anchor-id="sec-mdscasestudies">Casestudies</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In groups of two or three…
</div>
</div>
<div class="callout-body-container callout-body">
<p>… work through each case study below discuss what each step is achieving, what do you conclude?</p>
</div>
</div>
<p><strong>Bird Abundance</strong> (recall the example from <a href="#sec-glmfit" class="quarto-xref">Section&nbsp;12.4</a>; data from <span class="citation" data-cites="birds">García-Navas et al. (<a href="#ref-birds" role="doc-biblioref">2022</a>)</span>)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(tidyverse)</span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggfortify)</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(vegan)</span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>birds <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/bird_abundance.csv"</span>)</span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>birds <span class="sc">%&gt;%</span></span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">"OliveFarm"</span>, <span class="st">"Management"</span>, <span class="st">"Complexity"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cor</span>() <span class="sc">|&gt;</span></span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a>corrplot<span class="sc">::</span><span class="fu">corrplot</span>(<span class="at">method =</span> <span class="st">"ellipse"</span>, <span class="at">type =</span> <span class="st">"lower"</span>)</span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a><span class="do">### PCA</span></span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span>birds <span class="sc">%&gt;%</span></span>
<span id="cb88-12"><a href="#cb88-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">"OliveFarm"</span>, <span class="st">"Management"</span>, <span class="st">"Complexity"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb88-13"><a href="#cb88-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale</span>() <span class="sc">%&gt;%</span></span>
<span id="cb88-14"><a href="#cb88-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">prcomp</span>()</span>
<span id="cb88-15"><a href="#cb88-15" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(pca)</span>
<span id="cb88-16"><a href="#cb88-16" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_screeplot</span>(pca)</span>
<span id="cb88-17"><a href="#cb88-17" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_pca_biplot</span>(pca, <span class="at">geom =</span> <span class="st">"point"</span>) <span class="sc">+</span></span>
<span id="cb88-18"><a href="#cb88-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span> (<span class="at">alpha =</span> <span class="fl">0.2</span>)</span>
<span id="cb88-19"><a href="#cb88-19" aria-hidden="true" tabindex="-1"></a>pca<span class="sc">$</span>rotation</span>
<span id="cb88-20"><a href="#cb88-20" aria-hidden="true" tabindex="-1"></a>pca<span class="sc">$</span>rotation <span class="sc">%&gt;%</span></span>
<span id="cb88-21"><a href="#cb88-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb88-22"><a href="#cb88-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">variables =</span> <span class="fu">rownames</span>(.)) <span class="sc">%&gt;%</span></span>
<span id="cb88-23"><a href="#cb88-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>(PC,loading,PC1<span class="sc">:</span>PC4) <span class="sc">%&gt;%</span></span>
<span id="cb88-24"><a href="#cb88-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="fu">abs</span>(loading), variables, <span class="at">fill =</span> loading <span class="sc">&gt;</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb88-25"><a href="#cb88-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb88-26"><a href="#cb88-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>PC, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb88-27"><a href="#cb88-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Absolute value of loading"</span>,<span class="at">y =</span> <span class="cn">NULL</span>, <span class="at">fill =</span> <span class="st">"Positive?"</span>)</span>
<span id="cb88-28"><a href="#cb88-28" aria-hidden="true" tabindex="-1"></a>mmds <span class="ot">&lt;-</span> birds <span class="sc">%&gt;%</span></span>
<span id="cb88-29"><a href="#cb88-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">"OliveFarm"</span>, <span class="st">"Management"</span>, <span class="st">"Complexity"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb88-30"><a href="#cb88-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">metaMDS</span>( <span class="at">distance =</span> <span class="st">"bray"</span>, <span class="at">trace =</span> <span class="cn">FALSE</span>, <span class="at">autotransform =</span> <span class="cn">FALSE</span>)</span>
<span id="cb88-31"><a href="#cb88-31" aria-hidden="true" tabindex="-1"></a><span class="do">## 'badness-of-fit indicator'</span></span>
<span id="cb88-32"><a href="#cb88-32" aria-hidden="true" tabindex="-1"></a><span class="do">## stress provides a measure of the degree to which the distance between </span></span>
<span id="cb88-33"><a href="#cb88-33" aria-hidden="true" tabindex="-1"></a><span class="do">## samples in reduced (2) dimensional space matches the actual multivariate distance between </span></span>
<span id="cb88-34"><a href="#cb88-34" aria-hidden="true" tabindex="-1"></a><span class="do">## the samples. rule of thumb that stress &lt; 0.2 is ok</span></span>
<span id="cb88-35"><a href="#cb88-35" aria-hidden="true" tabindex="-1"></a>mmds<span class="sc">$</span>stress</span>
<span id="cb88-36"><a href="#cb88-36" aria-hidden="true" tabindex="-1"></a><span class="do">## 'Shepard' plot (i.e., scatter plot of the  of the interpoint distances</span></span>
<span id="cb88-37"><a href="#cb88-37" aria-hidden="true" tabindex="-1"></a><span class="do">## vs. the original dissimilarities)</span></span>
<span id="cb88-38"><a href="#cb88-38" aria-hidden="true" tabindex="-1"></a><span class="do">## In brief, a narrow scatter around the fitted line indicates a good fit of the distances to the</span></span>
<span id="cb88-39"><a href="#cb88-39" aria-hidden="true" tabindex="-1"></a><span class="do">## dissimilarities</span></span>
<span id="cb88-40"><a href="#cb88-40" aria-hidden="true" tabindex="-1"></a><span class="fu">stressplot</span>(mmds, <span class="at">pch =</span> <span class="dv">20</span>, <span class="at">cex =</span> <span class="dv">2</span>)</span>
<span id="cb88-41"><a href="#cb88-41" aria-hidden="true" tabindex="-1"></a><span class="do">## do you think there's a  difference in species composition by farm/management</span></span>
<span id="cb88-42"><a href="#cb88-42" aria-hidden="true" tabindex="-1"></a>mmds<span class="sc">$</span>points <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb88-43"><a href="#cb88-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> MDS1, <span class="at">y =</span> MDS2, <span class="at">label =</span> birds<span class="sc">$</span>OliveFarm)) <span class="sc">+</span></span>
<span id="cb88-44"><a href="#cb88-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>()</span>
<span id="cb88-45"><a href="#cb88-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-46"><a href="#cb88-46" aria-hidden="true" tabindex="-1"></a>mmds<span class="sc">$</span>points <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb88-47"><a href="#cb88-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> MDS1, <span class="at">y =</span> MDS2, <span class="at">label =</span> birds<span class="sc">$</span>Management, <span class="at">col =</span> birds<span class="sc">$</span>Management)) <span class="sc">+</span></span>
<span id="cb88-48"><a href="#cb88-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>DNA</strong> data from <span class="citation" data-cites="dna">Park et al. (<a href="#ref-dna" role="doc-biblioref">2016</a>)</span></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(tidyverse)</span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggfortify)</span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a>dna <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/dna_forensic.csv"</span>)</span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>dna <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Group) <span class="sc">%&gt;%</span></span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">mean =</span> <span class="fu">mean</span>(Age), <span class="at">sem =</span> <span class="fu">sd</span>(Age)<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">n</span>()), <span class="at">mad =</span> <span class="fu">sum</span>(<span class="fu">abs</span>((Age <span class="sc">-</span> <span class="fu">mean</span>(Age))))<span class="sc">/</span><span class="fu">n</span>())</span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a>dna <span class="sc">%&gt;%</span></span>
<span id="cb89-8"><a href="#cb89-8" aria-hidden="true" tabindex="-1"></a>   <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">"Group"</span>, <span class="st">"Sex"</span>, <span class="st">"NO"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb89-9"><a href="#cb89-9" aria-hidden="true" tabindex="-1"></a>   <span class="fu">cor</span>() <span class="sc">|&gt;</span></span>
<span id="cb89-10"><a href="#cb89-10" aria-hidden="true" tabindex="-1"></a>   corrplot<span class="sc">::</span><span class="fu">corrplot</span>(<span class="at">method =</span> <span class="st">"ellipse"</span>, <span class="at">type =</span> <span class="st">"lower"</span>)</span>
<span id="cb89-11"><a href="#cb89-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-12"><a href="#cb89-12" aria-hidden="true" tabindex="-1"></a><span class="do">## Replicating model as in the paper</span></span>
<span id="cb89-13"><a href="#cb89-13" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> dna</span>
<span id="cb89-14"><a href="#cb89-14" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(data)</span>
<span id="cb89-15"><a href="#cb89-15" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(data)[<span class="dv">5</span>] <span class="ot">&lt;-</span> <span class="st">"x1"</span></span>
<span id="cb89-16"><a href="#cb89-16" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(data)[<span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="st">"x2"</span></span>
<span id="cb89-17"><a href="#cb89-17" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(data)[<span class="dv">15</span>] <span class="ot">&lt;-</span> <span class="st">"x3"</span></span>
<span id="cb89-18"><a href="#cb89-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-19"><a href="#cb89-19" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">lm</span>(Age <span class="sc">~</span> x1 <span class="sc">+</span> x2 <span class="sc">+</span> x3, <span class="at">data =</span> data)</span>
<span id="cb89-20"><a href="#cb89-20" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model)</span>
<span id="cb89-21"><a href="#cb89-21" aria-hidden="true" tabindex="-1"></a>gglm<span class="sc">::</span><span class="fu">gglm</span>(model)</span>
<span id="cb89-22"><a href="#cb89-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-23"><a href="#cb89-23" aria-hidden="true" tabindex="-1"></a><span class="do">### PCA</span></span>
<span id="cb89-24"><a href="#cb89-24" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> dna <span class="sc">%&gt;%</span></span>
<span id="cb89-25"><a href="#cb89-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">"Group"</span>, <span class="st">"Sex"</span>, <span class="st">"NO"</span>, <span class="st">"Age"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb89-26"><a href="#cb89-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">prcomp</span>()</span>
<span id="cb89-27"><a href="#cb89-27" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(pca)</span>
<span id="cb89-28"><a href="#cb89-28" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_screeplot</span>(pca)</span>
<span id="cb89-29"><a href="#cb89-29" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_pca_biplot</span>(pca, <span class="at">geom =</span> <span class="st">"point"</span>) <span class="sc">+</span></span>
<span id="cb89-30"><a href="#cb89-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span> (<span class="at">alpha =</span> <span class="fl">0.2</span>)</span>
<span id="cb89-31"><a href="#cb89-31" aria-hidden="true" tabindex="-1"></a>pca<span class="sc">$</span>rotation</span>
<span id="cb89-32"><a href="#cb89-32" aria-hidden="true" tabindex="-1"></a>pca<span class="sc">$</span>rotation <span class="sc">%&gt;%</span></span>
<span id="cb89-33"><a href="#cb89-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb89-34"><a href="#cb89-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">variables =</span> <span class="fu">rownames</span>(.)) <span class="sc">%&gt;%</span></span>
<span id="cb89-35"><a href="#cb89-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>(PC,loading,PC1<span class="sc">:</span>PC4) <span class="sc">%&gt;%</span></span>
<span id="cb89-36"><a href="#cb89-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="fu">abs</span>(loading), variables, <span class="at">fill =</span> loading <span class="sc">&gt;</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb89-37"><a href="#cb89-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb89-38"><a href="#cb89-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>PC, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb89-39"><a href="#cb89-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Absolute value of loading"</span>,<span class="at">y =</span> <span class="cn">NULL</span>, <span class="at">fill =</span> <span class="st">"Positive?"</span>) </span>
<span id="cb89-40"><a href="#cb89-40" aria-hidden="true" tabindex="-1"></a><span class="do">## MDS</span></span>
<span id="cb89-41"><a href="#cb89-41" aria-hidden="true" tabindex="-1"></a>dna <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/dna_forensic.csv"</span>)</span>
<span id="cb89-42"><a href="#cb89-42" aria-hidden="true" tabindex="-1"></a>mds <span class="ot">&lt;-</span> dna <span class="sc">%&gt;%</span></span>
<span id="cb89-43"><a href="#cb89-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">"Group"</span>, <span class="st">"Sex"</span>, <span class="st">"NO"</span>, <span class="st">"Age"</span>)) <span class="sc">|&gt;</span> <span class="fu">scale</span>() <span class="sc">|&gt;</span></span>
<span id="cb89-44"><a href="#cb89-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dist</span>(<span class="at">upper =</span> <span class="cn">TRUE</span>, <span class="at">diag =</span> <span class="cn">TRUE</span>, <span class="at">method =</span> <span class="st">"manhattan"</span>) <span class="sc">|&gt;</span> <span class="fu">cmdscale</span>( <span class="at">eig =</span> <span class="cn">TRUE</span>)</span>
<span id="cb89-45"><a href="#cb89-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-46"><a href="#cb89-46" aria-hidden="true" tabindex="-1"></a><span class="do">## GOF gives us the proportion of variance explained by the first 2 dimensions</span></span>
<span id="cb89-47"><a href="#cb89-47" aria-hidden="true" tabindex="-1"></a><span class="do">## think of PCA</span></span>
<span id="cb89-48"><a href="#cb89-48" aria-hidden="true" tabindex="-1"></a>mds<span class="sc">$</span>GOF</span>
<span id="cb89-49"><a href="#cb89-49" aria-hidden="true" tabindex="-1"></a><span class="fu">autoplot</span>(mds, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">shape =</span> <span class="cn">FALSE</span>,</span>
<span id="cb89-50"><a href="#cb89-50" aria-hidden="true" tabindex="-1"></a>         <span class="at">label.size =</span> <span class="dv">3</span>)</span>
<span id="cb89-51"><a href="#cb89-51" aria-hidden="true" tabindex="-1"></a><span class="do">## Age</span></span>
<span id="cb89-52"><a href="#cb89-52" aria-hidden="true" tabindex="-1"></a>mds<span class="sc">$</span>points <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb89-53"><a href="#cb89-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> V1, <span class="at">y =</span> V2, <span class="at">label =</span> dna<span class="sc">$</span>Age, <span class="at">col =</span> dna<span class="sc">$</span>Age)) <span class="sc">+</span></span>
<span id="cb89-54"><a href="#cb89-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>()</span>
<span id="cb89-55"><a href="#cb89-55" aria-hidden="true" tabindex="-1"></a><span class="do">## Sex</span></span>
<span id="cb89-56"><a href="#cb89-56" aria-hidden="true" tabindex="-1"></a>mds<span class="sc">$</span>points <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb89-57"><a href="#cb89-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> V1, <span class="at">y =</span> V2, <span class="at">label =</span> dna<span class="sc">$</span>Sex, <span class="at">col =</span> dna<span class="sc">$</span>Sex)) <span class="sc">+</span></span>
<span id="cb89-58"><a href="#cb89-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="may-19th" class="level1">
<h1>May 19th</h1>
<section id="rough-timeline-17" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-17">Rough timeline</h2>
<ul>
<li><strong>09:05–09:25 am</strong> <em>Grouping</em> multivariate data (<a href="#sec-clustering" class="quarto-xref">Section&nbsp;18.2</a>)</li>
<li><strong>09:25–09:55 am</strong> Divisive vs.&nbsp;Hierarchical (<a href="#sec-div" class="quarto-xref">Section&nbsp;18.3</a>)</li>
<li><strong>09:55–11:05 am</strong> Break</li>
<li><strong>11:05–11:45 am</strong> Clusterducks (<a href="#sec-clusterducks" class="quarto-xref">Section&nbsp;18.4</a>)</li>
<li><strong>11.45–11.55 pm</strong> Q&amp;A and Task 09 (<a href="#sec-peershare17" class="quarto-xref">Section&nbsp;18.5</a>)</li>
</ul>
</section>
<section id="sec-clustering" class="level2">
<h2 class="anchored" data-anchor-id="sec-clustering"><em>Grouping</em> multivariate data</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Here are some celebrity stats.
</div>
</div>
<div class="callout-body-container callout-body">
<p><img src="runsheets/img/celebs.png" class="img-fluid"></p>
<p>In your groups sort the celebrities into clusters; outline your reasoning.</p>
</div>
</div>
<p>We’ll now work through the code below together.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GGally)</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/celebs.csv"</span>)</span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">column_to_rownames</span>(data, <span class="st">"name"</span>)</span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span> <span class="fu">ggpairs</span>()</span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a><span class="do">## k-means clustering</span></span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">4321</span>)</span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a><span class="do">## two clusters</span></span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a>k2 <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(data, <span class="at">centers =</span> <span class="dv">2</span>, <span class="at">nstart =</span> <span class="dv">25</span>)</span>
<span id="cb90-11"><a href="#cb90-11" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_cluster</span>(k2, <span class="at">data =</span> data)</span>
<span id="cb90-12"><a href="#cb90-12" aria-hidden="true" tabindex="-1"></a><span class="do">## three clusters</span></span>
<span id="cb90-13"><a href="#cb90-13" aria-hidden="true" tabindex="-1"></a>k3 <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(data, <span class="at">centers =</span> <span class="dv">3</span>, <span class="at">nstart =</span> <span class="dv">25</span>)</span>
<span id="cb90-14"><a href="#cb90-14" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_cluster</span>(k3, <span class="at">data =</span> data)</span>
<span id="cb90-15"><a href="#cb90-15" aria-hidden="true" tabindex="-1"></a><span class="do">## choose clusters?</span></span>
<span id="cb90-16"><a href="#cb90-16" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_nbclust</span>(data, kmeans, <span class="at">method =</span> <span class="st">"wss"</span>) <span class="sc">+</span></span>
<span id="cb90-17"><a href="#cb90-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">"Elbow method"</span>)</span>
<span id="cb90-18"><a href="#cb90-18" aria-hidden="true" tabindex="-1"></a><span class="do">## dimensions?</span></span>
<span id="cb90-19"><a href="#cb90-19" aria-hidden="true" tabindex="-1"></a><span class="do">## What about PCA</span></span>
<span id="cb90-20"><a href="#cb90-20" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span> <span class="fu">scale</span>() <span class="sc">%&gt;%</span> <span class="fu">prcomp</span>()</span>
<span id="cb90-21"><a href="#cb90-21" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(pca)</span>
<span id="cb90-22"><a href="#cb90-22" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_pca_biplot</span>(pca, <span class="at">geom =</span> <span class="fu">c</span>(<span class="st">"point"</span>, <span class="st">"text"</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="sec-div" class="level2">
<h2 class="anchored" data-anchor-id="sec-div">Divisive vs.&nbsp;Hierarchical</h2>
<p><strong>Slides</strong></p>
<p align="center">
<iframe src="slides/10-01_clustering.pdf" height="400" width="700" title="Maximum Likelihood Estimation">
</iframe>
</p>
<p align="center">
<a href="https://canvas.auckland.ac.nz/courses/120054/files/15377008?module_item_id=2615037"><strong>CANVAS slides</strong></a>
</p>
<p><strong>Ant example</strong> from <a href="https://biosci738.github.io/BIOSCI738/clustering.html#ants">this section of the courseguide</a></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ape)</span>
<span id="cb91-5"><a href="#cb91-5" aria-hidden="true" tabindex="-1"></a>base_url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/"</span></span>
<span id="cb91-6"><a href="#cb91-6" aria-hidden="true" tabindex="-1"></a>ants <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="fu">paste</span>(base_url, <span class="st">"pitfalls.csv"</span>, <span class="at">sep =</span> <span class="st">""</span>))</span>
<span id="cb91-7"><a href="#cb91-7" aria-hidden="true" tabindex="-1"></a>pitfall.dist <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">vegdist</span>(ants[,<span class="dv">5</span><span class="sc">:</span><span class="dv">8</span>], <span class="at">method =</span> <span class="st">"bray"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>)</span>
<span id="cb91-8"><a href="#cb91-8" aria-hidden="true" tabindex="-1"></a>factoextra<span class="sc">::</span><span class="fu">fviz_dist</span>(pitfall.dist)</span>
<span id="cb91-9"><a href="#cb91-9" aria-hidden="true" tabindex="-1"></a><span class="do">## single (nearest neighbour) linkage</span></span>
<span id="cb91-10"><a href="#cb91-10" aria-hidden="true" tabindex="-1"></a>single <span class="ot">&lt;-</span> ants[,<span class="dv">5</span><span class="sc">:</span><span class="dv">8</span>] <span class="sc">%&gt;%</span></span>
<span id="cb91-11"><a href="#cb91-11" aria-hidden="true" tabindex="-1"></a>  vegan<span class="sc">::</span><span class="fu">vegdist</span>(., <span class="at">method =</span> <span class="st">"bray"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb91-12"><a href="#cb91-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(<span class="at">method =</span> <span class="st">"single"</span>)</span>
<span id="cb91-13"><a href="#cb91-13" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(single, <span class="at">labels =</span> ants<span class="sc">$</span>Site)</span>
<span id="cb91-14"><a href="#cb91-14" aria-hidden="true" tabindex="-1"></a><span class="do">## complete (maximium neighbour) linkage</span></span>
<span id="cb91-15"><a href="#cb91-15" aria-hidden="true" tabindex="-1"></a>complete <span class="ot">&lt;-</span> ants[,<span class="dv">5</span><span class="sc">:</span><span class="dv">8</span>] <span class="sc">%&gt;%</span></span>
<span id="cb91-16"><a href="#cb91-16" aria-hidden="true" tabindex="-1"></a>  vegan<span class="sc">::</span><span class="fu">vegdist</span>(., <span class="at">method =</span> <span class="st">"bray"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb91-17"><a href="#cb91-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(<span class="at">method =</span> <span class="st">"complete"</span>)</span>
<span id="cb91-18"><a href="#cb91-18" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(complete, <span class="at">labels =</span> ants<span class="sc">$</span>Site)</span>
<span id="cb91-19"><a href="#cb91-19" aria-hidden="true" tabindex="-1"></a><span class="do">## average linkage</span></span>
<span id="cb91-20"><a href="#cb91-20" aria-hidden="true" tabindex="-1"></a>average <span class="ot">&lt;-</span> ants[,<span class="dv">5</span><span class="sc">:</span><span class="dv">8</span>] <span class="sc">%&gt;%</span></span>
<span id="cb91-21"><a href="#cb91-21" aria-hidden="true" tabindex="-1"></a>  vegan<span class="sc">::</span><span class="fu">vegdist</span>(., <span class="at">method =</span> <span class="st">"bray"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb91-22"><a href="#cb91-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(<span class="at">method =</span> <span class="st">"average"</span>)</span>
<span id="cb91-23"><a href="#cb91-23" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(average, <span class="at">labels =</span> ants<span class="sc">$</span>Site)</span>
<span id="cb91-24"><a href="#cb91-24" aria-hidden="true" tabindex="-1"></a><span class="do">## ward's method</span></span>
<span id="cb91-25"><a href="#cb91-25" aria-hidden="true" tabindex="-1"></a>ward <span class="ot">&lt;-</span> ants[,<span class="dv">5</span><span class="sc">:</span><span class="dv">8</span>] <span class="sc">%&gt;%</span></span>
<span id="cb91-26"><a href="#cb91-26" aria-hidden="true" tabindex="-1"></a>  vegan<span class="sc">::</span><span class="fu">vegdist</span>(., <span class="at">method =</span> <span class="st">"bray"</span>, <span class="at">binary =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb91-27"><a href="#cb91-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(<span class="at">method =</span> <span class="st">"ward.D"</span>)</span>
<span id="cb91-28"><a href="#cb91-28" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(ward, <span class="at">labels =</span> ants<span class="sc">$</span>Site)</span>
<span id="cb91-29"><a href="#cb91-29" aria-hidden="true" tabindex="-1"></a><span class="do">## split clusters on based on your choice of position</span></span>
<span id="cb91-30"><a href="#cb91-30" aria-hidden="true" tabindex="-1"></a>ants<span class="sc">$</span>clust4 <span class="ot">&lt;-</span> <span class="fu">cutree</span>(ward, <span class="at">k =</span> <span class="dv">4</span>)</span>
<span id="cb91-31"><a href="#cb91-31" aria-hidden="true" tabindex="-1"></a>pitfall.phylo <span class="ot">&lt;-</span> <span class="fu">as.phylo</span>(ward)</span>
<span id="cb91-32"><a href="#cb91-32" aria-hidden="true" tabindex="-1"></a>pitfall.phylo<span class="sc">$</span>tip.label <span class="ot">&lt;-</span> ants<span class="sc">$</span>Site</span>
<span id="cb91-33"><a href="#cb91-33" aria-hidden="true" tabindex="-1"></a><span class="do">## Set colours </span></span>
<span id="cb91-34"><a href="#cb91-34" aria-hidden="true" tabindex="-1"></a>colours  <span class="ot">&lt;-</span>  <span class="fu">c</span>(<span class="st">"red"</span>,<span class="st">"blue"</span>,<span class="st">"green"</span>,<span class="st">"black"</span>)</span>
<span id="cb91-35"><a href="#cb91-35" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pitfall.phylo, <span class="at">cex =</span> <span class="fl">0.6</span>, <span class="at">tip.color =</span> colours[ants<span class="sc">$</span>clust4], <span class="at">label.offset =</span> <span class="fl">0.05</span>)</span>
<span id="cb91-36"><a href="#cb91-36" aria-hidden="true" tabindex="-1"></a><span class="do">## Now there are other variables in the dataset, discuss what, if any,</span></span>
<span id="cb91-37"><a href="#cb91-37" aria-hidden="true" tabindex="-1"></a><span class="do">## might lead to the suggested "clusters" of sites.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="sec-clusterducks" class="level2">
<h2 class="anchored" data-anchor-id="sec-clusterducks">Clusterducks</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In groups of two or three..
</div>
</div>
<div class="callout-body-container callout-body">
<p>… work through the code below and discuss what each step is achieving, what do you conclude?</p>
<p><em>NOTE</em> <a href="https://statbiscuit.github.io/swots/clusterducks.html">this worksheet may help you a little</a></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a>data_url <span class="ot">&lt;-</span> <span class="st">"https://github.com/cmjt/statbiscuits/raw/master/swots/data/duck_rgbs.RData"</span></span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">url</span>(data_url))</span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a>cluster_ducks <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">attire  =</span> stringr<span class="sc">::</span><span class="fu">str_match</span>(<span class="fu">names</span>(duck_rgbs),<span class="st">"(.*?)-"</span>)[,<span class="dv">2</span>],</span>
<span id="cb92-6"><a href="#cb92-6" aria-hidden="true" tabindex="-1"></a>                            <span class="at">av_red =</span> <span class="fu">sapply</span>(duck_rgbs, <span class="cf">function</span>(x) <span class="fu">mean</span>(<span class="fu">c</span>(x[,,<span class="dv">1</span>]))),</span>
<span id="cb92-7"><a href="#cb92-7" aria-hidden="true" tabindex="-1"></a>                            <span class="at">av_green =</span> <span class="fu">sapply</span>(duck_rgbs, <span class="cf">function</span>(x) <span class="fu">mean</span>(<span class="fu">c</span>(x[,,<span class="dv">2</span>]))),</span>
<span id="cb92-8"><a href="#cb92-8" aria-hidden="true" tabindex="-1"></a>                            <span class="at">av_blue =</span> <span class="fu">sapply</span>(duck_rgbs, <span class="cf">function</span>(x) <span class="fu">mean</span>(<span class="fu">c</span>(x[,,<span class="dv">3</span>]))))</span>
<span id="cb92-9"><a href="#cb92-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cluster_ducks)</span>
<span id="cb92-10"><a href="#cb92-10" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_ly</span>(<span class="at">x =</span> cluster_ducks<span class="sc">$</span>av_red, <span class="at">y =</span> cluster_ducks<span class="sc">$</span>av_green, </span>
<span id="cb92-11"><a href="#cb92-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">z =</span> cluster_ducks<span class="sc">$</span>av_blue,</span>
<span id="cb92-12"><a href="#cb92-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">type =</span> <span class="st">"scatter3d"</span>, <span class="at">mode =</span> <span class="st">"markers"</span>, </span>
<span id="cb92-13"><a href="#cb92-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> cluster_ducks<span class="sc">$</span>attire)</span>
<span id="cb92-14"><a href="#cb92-14" aria-hidden="true" tabindex="-1"></a>prop.max <span class="ot">&lt;-</span> <span class="cf">function</span>(x){</span>
<span id="cb92-15"><a href="#cb92-15" aria-hidden="true" tabindex="-1"></a>    <span class="do">## matrix of index of max RGB values of x</span></span>
<span id="cb92-16"><a href="#cb92-16" aria-hidden="true" tabindex="-1"></a>    mat_max <span class="ot">&lt;-</span> <span class="fu">apply</span>(x,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>),which.max)</span>
<span id="cb92-17"><a href="#cb92-17" aria-hidden="true" tabindex="-1"></a>    <span class="do">## table of collapsed values</span></span>
<span id="cb92-18"><a href="#cb92-18" aria-hidden="true" tabindex="-1"></a>    tab <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="fu">c</span>(mat_max))</span>
<span id="cb92-19"><a href="#cb92-19" aria-hidden="true" tabindex="-1"></a>    <span class="do">## proportion of red</span></span>
<span id="cb92-20"><a href="#cb92-20" aria-hidden="true" tabindex="-1"></a>    prop_red <span class="ot">&lt;-</span> tab[<span class="dv">1</span>]<span class="sc">/</span><span class="fu">sum</span>(tab)</span>
<span id="cb92-21"><a href="#cb92-21" aria-hidden="true" tabindex="-1"></a>    prop_green <span class="ot">&lt;-</span> tab[<span class="dv">2</span>]<span class="sc">/</span><span class="fu">sum</span>(tab)</span>
<span id="cb92-22"><a href="#cb92-22" aria-hidden="true" tabindex="-1"></a>    prop_blue <span class="ot">&lt;-</span> tab[<span class="dv">3</span>]<span class="sc">/</span><span class="fu">sum</span>(tab)</span>
<span id="cb92-23"><a href="#cb92-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">c</span>(prop_red,prop_green,prop_blue))</span>
<span id="cb92-24"><a href="#cb92-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb92-25"><a href="#cb92-25" aria-hidden="true" tabindex="-1"></a><span class="do">## proportion of r, g, b in each image</span></span>
<span id="cb92-26"><a href="#cb92-26" aria-hidden="true" tabindex="-1"></a>prop <span class="ot">&lt;-</span> <span class="fu">do.call</span>(<span class="st">'rbind'</span>,<span class="fu">lapply</span>(duck_rgbs,prop.max))</span>
<span id="cb92-27"><a href="#cb92-27" aria-hidden="true" tabindex="-1"></a>cluster_ducks<span class="sc">$</span>prop_red <span class="ot">&lt;-</span> prop[,<span class="dv">1</span>]</span>
<span id="cb92-28"><a href="#cb92-28" aria-hidden="true" tabindex="-1"></a>cluster_ducks<span class="sc">$</span>prop_green <span class="ot">&lt;-</span> prop[,<span class="dv">2</span>]</span>
<span id="cb92-29"><a href="#cb92-29" aria-hidden="true" tabindex="-1"></a>cluster_ducks<span class="sc">$</span>prop_blue <span class="ot">&lt;-</span> prop[,<span class="dv">3</span>]</span>
<span id="cb92-30"><a href="#cb92-30" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_ly</span>(<span class="at">x =</span> cluster_ducks<span class="sc">$</span>prop_red, <span class="at">y =</span> cluster_ducks<span class="sc">$</span>prop_green, </span>
<span id="cb92-31"><a href="#cb92-31" aria-hidden="true" tabindex="-1"></a>        <span class="at">z =</span> cluster_ducks<span class="sc">$</span>prop_blue,</span>
<span id="cb92-32"><a href="#cb92-32" aria-hidden="true" tabindex="-1"></a>        <span class="at">type =</span> <span class="st">"scatter3d"</span>, <span class="at">mode =</span> <span class="st">"markers"</span>, </span>
<span id="cb92-33"><a href="#cb92-33" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> cluster_ducks<span class="sc">$</span>attire)</span>
<span id="cb92-34"><a href="#cb92-34" aria-hidden="true" tabindex="-1"></a><span class="do">## K-means clustering</span></span>
<span id="cb92-35"><a href="#cb92-35" aria-hidden="true" tabindex="-1"></a><span class="do">## re format data. We deal only with the numerics info</span></span>
<span id="cb92-36"><a href="#cb92-36" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> cluster_ducks[,<span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>]</span>
<span id="cb92-37"><a href="#cb92-37" aria-hidden="true" tabindex="-1"></a><span class="do">## specify rownames as image names</span></span>
<span id="cb92-38"><a href="#cb92-38" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(df) <span class="ot">&lt;-</span> <span class="fu">names</span>(duck_rgbs)</span>
<span id="cb92-39"><a href="#cb92-39" aria-hidden="true" tabindex="-1"></a>distance <span class="ot">&lt;-</span> <span class="fu">get_dist</span>(df)</span>
<span id="cb92-40"><a href="#cb92-40" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_dist</span>(distance, <span class="at">gradient =</span> <span class="fu">list</span>(<span class="at">low =</span> <span class="st">"#00AFBB"</span>, <span class="at">mid =</span> <span class="st">"white"</span>, <span class="at">high =</span> <span class="st">"#FC4E07"</span>))</span>
<span id="cb92-41"><a href="#cb92-41" aria-hidden="true" tabindex="-1"></a><span class="do">## from two clusters to 6 (can we separate out the images?)</span></span>
<span id="cb92-42"><a href="#cb92-42" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">4321</span>)</span>
<span id="cb92-43"><a href="#cb92-43" aria-hidden="true" tabindex="-1"></a>k2 <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(df, <span class="at">centers =</span> <span class="dv">2</span>, <span class="at">nstart =</span> <span class="dv">25</span>)</span>
<span id="cb92-44"><a href="#cb92-44" aria-hidden="true" tabindex="-1"></a>k3 <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(df, <span class="at">centers =</span> <span class="dv">3</span>, <span class="at">nstart =</span> <span class="dv">25</span>)</span>
<span id="cb92-45"><a href="#cb92-45" aria-hidden="true" tabindex="-1"></a>k4 <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(df, <span class="at">centers =</span> <span class="dv">4</span>, <span class="at">nstart =</span> <span class="dv">25</span>)</span>
<span id="cb92-46"><a href="#cb92-46" aria-hidden="true" tabindex="-1"></a>k5 <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(df, <span class="at">centers =</span> <span class="dv">5</span>, <span class="at">nstart =</span> <span class="dv">25</span>)</span>
<span id="cb92-47"><a href="#cb92-47" aria-hidden="true" tabindex="-1"></a>k6 <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(df, <span class="at">centers =</span> <span class="dv">6</span>, <span class="at">nstart =</span> <span class="dv">25</span>)</span>
<span id="cb92-48"><a href="#cb92-48" aria-hidden="true" tabindex="-1"></a><span class="do">## total tithin-cluster SS</span></span>
<span id="cb92-49"><a href="#cb92-49" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(<span class="fu">c</span>(k2<span class="sc">$</span>tot.withinss,k3<span class="sc">$</span>tot.withinss,k4<span class="sc">$</span>tot.withinss,</span>
<span id="cb92-50"><a href="#cb92-50" aria-hidden="true" tabindex="-1"></a>          k5<span class="sc">$</span>tot.withinss,k6<span class="sc">$</span>tot.withinss),</span>
<span id="cb92-51"><a href="#cb92-51" aria-hidden="true" tabindex="-1"></a>        <span class="at">names =</span> <span class="fu">paste</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>,<span class="st">" clusters"</span>))</span>
<span id="cb92-52"><a href="#cb92-52" aria-hidden="true" tabindex="-1"></a><span class="do">## eyeballing it</span></span>
<span id="cb92-53"><a href="#cb92-53" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">fviz_cluster</span>(k2, <span class="at">data =</span> df)</span>
<span id="cb92-54"><a href="#cb92-54" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">fviz_cluster</span>(k3, <span class="at">data =</span> df)</span>
<span id="cb92-55"><a href="#cb92-55" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">fviz_cluster</span>(k4, <span class="at">data =</span> df)</span>
<span id="cb92-56"><a href="#cb92-56" aria-hidden="true" tabindex="-1"></a>p5 <span class="ot">&lt;-</span> <span class="fu">fviz_cluster</span>(k5, <span class="at">data =</span> df)</span>
<span id="cb92-57"><a href="#cb92-57" aria-hidden="true" tabindex="-1"></a>p6 <span class="ot">&lt;-</span> <span class="fu">fviz_cluster</span>(k6, <span class="at">data =</span> df)</span>
<span id="cb92-58"><a href="#cb92-58" aria-hidden="true" tabindex="-1"></a><span class="do">## for arranging plots</span></span>
<span id="cb92-59"><a href="#cb92-59" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork) </span>
<span id="cb92-60"><a href="#cb92-60" aria-hidden="true" tabindex="-1"></a>p2<span class="sc">/</span> p3<span class="sc">/</span> p4<span class="sc">/</span> p5<span class="sc">/</span> p6</span>
<span id="cb92-61"><a href="#cb92-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Elbow method</span></span>
<span id="cb92-62"><a href="#cb92-62" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_nbclust</span>(df, kmeans, <span class="at">method =</span> <span class="st">"wss"</span>) <span class="sc">+</span></span>
<span id="cb92-63"><a href="#cb92-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">"Elbow method"</span>)</span>
<span id="cb92-64"><a href="#cb92-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-65"><a href="#cb92-65" aria-hidden="true" tabindex="-1"></a><span class="co"># Silhouette method</span></span>
<span id="cb92-66"><a href="#cb92-66" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_nbclust</span>(df, kmeans, <span class="at">method =</span> <span class="st">"silhouette"</span>)<span class="sc">+</span></span>
<span id="cb92-67"><a href="#cb92-67" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">"Silhouette method"</span>)</span>
<span id="cb92-68"><a href="#cb92-68" aria-hidden="true" tabindex="-1"></a><span class="co"># Gap statistic</span></span>
<span id="cb92-69"><a href="#cb92-69" aria-hidden="true" tabindex="-1"></a><span class="co"># recommended value: nboot= 500 for your analysis.</span></span>
<span id="cb92-70"><a href="#cb92-70" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb92-71"><a href="#cb92-71" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_nbclust</span>(df, kmeans, <span class="at">nstart =</span> <span class="dv">25</span>,  <span class="at">method =</span> <span class="st">"gap_stat"</span>, <span class="at">nboot =</span> <span class="dv">50</span>)<span class="sc">+</span></span>
<span id="cb92-72"><a href="#cb92-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">"Gap statistic method"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="sec-peershare17" class="level2">
<h2 class="anchored" data-anchor-id="sec-peershare17">Q&amp;A and peer-share</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Peer-Share
</div>
</div>
<div class="callout-body-container callout-body">
<p>Go over an activity from the class again. Feel free to ask Charlotte a question or two about it!</p>
</div>
</div>
</section>
</section>
<section id="may-23rd" class="level1">
<h1>May 23rd</h1>
<section id="rough-timeline-18" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-18">Rough timeline</h2>
<section id="pm---0355-pm-8" class="level3">
<h3 class="anchored" data-anchor-id="pm---0355-pm-8"><strong>02:05 pm - 03:55 pm</strong></h3>
<ul>
<li><strong>02:05–02.40 pm</strong> Multivariate data (<a href="#sec-vizmult" class="quarto-xref">Section&nbsp;19.2</a>)</li>
<li>What we’re interested in</li>
<li>Visualizing it</li>
<li><strong>02:40–03:30 pm</strong> Casestudies [<em>including a break</em>] (<a href="#sec-mdscasestudies02" class="quarto-xref">Section&nbsp;19.3</a>)</li>
<li><strong>03:30–03:55 pm</strong> Model based multivariate analysis (<a href="#sec-modelbasedmult" class="quarto-xref">Section&nbsp;19.4</a>)</li>
</ul>
</section>
</section>
<section id="sec-vizmult" class="level2">
<h2 class="anchored" data-anchor-id="sec-vizmult">Multivariate data</h2>
<blockquote class="blockquote">
<p>Multivariate analyses refers to a suite of statistical techniques designed to analyse data with more than one variable: either multiple dependent variables, or both multiple dependent and independent variables.</p>
</blockquote>
<p>A typical sequence of inference for multivariate data would be <strong>Raw Data</strong> <span class="math inline">\(^?\rightarrow\)</span> <strong>Scale/Standardize Raw Data</strong> <span class="math inline">\(^?\rightarrow\)</span> <strong>Create a Similarity/Dissimilarity Matrix</strong>. From the created matrix you might consider <strong>Cluster Analysis</strong> and/or <strong>Ordination</strong> (a fancy word for summarizing/visualizing/reducing multiple variables in/to a lower-dimensional space).</p>
<div>

</div>
<div class="quarto-layout-panel" data-layout-ncol="2">
<div class="quarto-layout-row">
<div class="cell quarto-layout-cell" style="flex-basis: 50.0%;justify-content: flex-start;">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"><p></p>
<div>
<pre class="mermaid mermaid-js">flowchart TD
    A[Cluster Analysis] --&gt; B1(Hierarchical)
    A --&gt; B2(Non hierarchical)
    B1 --&gt; C1[Dendrograms]
    C1 -- ? --&gt; D1[Single]
    C1 -- ? --&gt; D2[Complete]
    C1 -- ? --&gt; D3[UPGMA]
    C1 -- ? --&gt; D4[Ward]
    B2 -- ? --&gt; E1[Kmeans]
    style A fill:#f9f,stroke:#333,stroke-width:4px
</pre>
</div>
<p></p></figure><p></p>
</div>
</div>
</div>
<div class="cell quarto-layout-cell" style="flex-basis: 50.0%;justify-content: flex-start;">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"><p></p>
<div>
<pre class="mermaid mermaid-js">flowchart TD
    A[Ordination] --&gt; B1(Unconstrained)
    A --&gt; B2(Constrained)
    B1 --&gt; C1[Reduced dimensions]
    C1 -- ? --&gt; D1[PCA]
    C1 -- ? --&gt; D2[CA]
    C1 -- ? --&gt; D3[mMDS]
    C1 -- ? --&gt; D4[PCO]
    B2 --&gt; E1[Not covered here]
    style E1 fill:#808080,stroke:#333,stroke-width:0.5px
    style A fill:#f9f,stroke:#333,stroke-width:4px
</pre>
</div>
<p></p></figure><p></p>
</div>
</div>
</div>
</div>
</div>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> In groups of two or three…
</div>
</div>
<div class="callout-body-container callout-body">
<p>The sequences and flow charts above have the occasional <span class="math inline">\(?\)</span> visible in the interim steps. Discuss the decisions &amp; justifications required/made to traverse along each arrow (i.e., pass the <span class="math inline">\(?\)</span>). Elaborate on the what each stage achieves giving specific examples from the courseguide. D</p>
</div>
</div>
<section id="visualizing-multivariate-data" class="level3">
<h3 class="anchored" data-anchor-id="visualizing-multivariate-data">Visualizing multivariate data</h3>
<p>We’ll work through the code below together; the data refer to different Pokémon (seen in case study 3 of <a href="#sec-model" class="quarto-xref">Section&nbsp;15.2</a>). How useful do you find each plot?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(tidyverse)</span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(GGally)</span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a>url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/pokemon.csv"</span></span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(url)</span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a><span class="do">## pairs plots for continuous variables</span></span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb93-7"><a href="#cb93-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(is.numeric)) <span class="sc">|&gt;</span> <span class="fu">pairs</span>()</span>
<span id="cb93-8"><a href="#cb93-8" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb93-9"><a href="#cb93-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(is.numeric)) <span class="sc">|&gt;</span> <span class="fu">ggpairs</span>()</span>
<span id="cb93-10"><a href="#cb93-10" aria-hidden="true" tabindex="-1"></a><span class="do">## but what if you had a categorical variable</span></span>
<span id="cb93-11"><a href="#cb93-11" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb93-12"><a href="#cb93-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(legendary, <span class="fu">where</span>(is.numeric)) <span class="sc">|&gt;</span> <span class="fu">ggpairs</span>(<span class="fu">aes</span>(<span class="at">color =</span> legendary))</span>
<span id="cb93-13"><a href="#cb93-13" aria-hidden="true" tabindex="-1"></a><span class="do">## two (or more) categorical variables</span></span>
<span id="cb93-14"><a href="#cb93-14" aria-hidden="true" tabindex="-1"></a><span class="fu">with</span>(data, <span class="fu">table</span>(legendary,</span>
<span id="cb93-15"><a href="#cb93-15" aria-hidden="true" tabindex="-1"></a>    type2)) <span class="sc">|&gt;</span> <span class="fu">barplot</span>(<span class="at">legend =</span> <span class="cn">TRUE</span>)</span>
<span id="cb93-16"><a href="#cb93-16" aria-hidden="true" tabindex="-1"></a><span class="do">## conditional</span></span>
<span id="cb93-17"><a href="#cb93-17" aria-hidden="true" tabindex="-1"></a><span class="fu">with</span>(data, <span class="fu">prop.table</span>(<span class="fu">table</span>(legendary,</span>
<span id="cb93-18"><a href="#cb93-18" aria-hidden="true" tabindex="-1"></a>    type2), <span class="at">margin =</span> <span class="dv">2</span>)) <span class="sc">|&gt;</span> <span class="fu">barplot</span>(<span class="at">legend =</span> <span class="cn">TRUE</span>)</span>
<span id="cb93-19"><a href="#cb93-19" aria-hidden="true" tabindex="-1"></a><span class="do">## alluvial plots</span></span>
<span id="cb93-20"><a href="#cb93-20" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(alluvial)</span>
<span id="cb93-21"><a href="#cb93-21" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>Freq <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb93-22"><a href="#cb93-22" aria-hidden="true" tabindex="-1"></a>mult <span class="ot">&lt;-</span> <span class="fu">aggregate</span>(Freq <span class="sc">~</span> type2 <span class="sc">+</span> legendary,</span>
<span id="cb93-23"><a href="#cb93-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> data, <span class="at">FUN =</span> sum)</span>
<span id="cb93-24"><a href="#cb93-24" aria-hidden="true" tabindex="-1"></a><span class="fu">alluvial</span>(mult, <span class="at">freq =</span> mult<span class="sc">$</span>Freq, <span class="at">border =</span> <span class="cn">NA</span>, <span class="at">col =</span> <span class="fu">ifelse</span>(mult<span class="sc">$</span>legendary, <span class="st">"red"</span>, <span class="st">"blue"</span>))</span>
<span id="cb93-25"><a href="#cb93-25" aria-hidden="true" tabindex="-1"></a><span class="do">## What are some potential interpretation pitfalls</span></span>
<span id="cb93-26"><a href="#cb93-26" aria-hidden="true" tabindex="-1"></a><span class="do">## heatmaps</span></span>
<span id="cb93-27"><a href="#cb93-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb93-28"><a href="#cb93-28" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb93-29"><a href="#cb93-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(is.numeric), <span class="sc">-</span>Freq, <span class="sc">-</span> generation, <span class="sc">-</span>number) <span class="sc">|&gt;</span> <span class="fu">cor</span>() <span class="sc">|&gt;</span></span>
<span id="cb93-30"><a href="#cb93-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pheatmap</span>( <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>, <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>,</span>
<span id="cb93-31"><a href="#cb93-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">treeheight_row =</span> <span class="dv">0</span>, <span class="at">treeheight_col =</span> <span class="dv">0</span>)</span>
<span id="cb93-32"><a href="#cb93-32" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">where</span>(is.numeric), <span class="sc">-</span>Freq, <span class="sc">-</span>generation, <span class="sc">-</span>number) <span class="sc">|&gt;</span></span>
<span id="cb93-33"><a href="#cb93-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pheatmap</span>( , <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,</span>
<span id="cb93-34"><a href="#cb93-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>, <span class="at">labels_row =</span> data<span class="sc">$</span>name, <span class="at">cex =</span> <span class="fl">0.7</span>)</span>
<span id="cb93-35"><a href="#cb93-35" aria-hidden="true" tabindex="-1"></a><span class="do">## look at ?pheatmap</span></span>
<span id="cb93-36"><a href="#cb93-36" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrplot)</span>
<span id="cb93-37"><a href="#cb93-37" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">where</span>(is.numeric), <span class="sc">-</span>Freq, <span class="sc">-</span> generation, <span class="sc">-</span>number) <span class="sc">|&gt;</span> <span class="fu">cor</span>() <span class="sc">|&gt;</span> </span>
<span id="cb93-38"><a href="#cb93-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">corrplot</span>(<span class="at">addrect =</span> <span class="dv">3</span>, <span class="at">order =</span> <span class="st">'hclust'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="sec-mdscasestudies02" class="level2">
<h2 class="anchored" data-anchor-id="sec-mdscasestudies02">Casestudies</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Revisit…
</div>
</div>
<div class="callout-body-container callout-body">
<p>…one of the following case studies. Create an informative visualization that appropriately demonstrates the structure of these multivariate data. You may find some useful examples <a href="https://epurdom.github.io/Stat131A/book/visualizing-multivariate-data.html#relationships-between-continous-variables">here</a>.</p>
<ol type="1">
<li><strong>Eigenfaces</strong> from <a href="#sec-eigenfaces" class="quarto-xref">Section&nbsp;16.4</a></li>
<li><strong>Clusterducks</strong> from <a href="#sec-clusterducks" class="quarto-xref">Section&nbsp;18.4</a></li>
<li><strong>Bird Abundance</strong> first case study from revisited in <a href="#sec-mdscasestudies" class="quarto-xref">Section&nbsp;17.4</a> (data from <span class="citation" data-cites="birds">García-Navas et al. (<a href="#ref-birds" role="doc-biblioref">2022</a>)</span>)</li>
<li><strong>DNA</strong> second case study from <a href="#sec-mdscasestudies" class="quarto-xref">Section&nbsp;17.4</a> (data from <span class="citation" data-cites="dna">Park et al. (<a href="#ref-dna" role="doc-biblioref">2016</a>)</span>)</li>
</ol>
<p><strong>Be prepared to present your activity to me as I’ll be prepared to critique!</strong></p>
</div>
</div>
</section>
<section id="sec-modelbasedmult" class="level2">
<h2 class="anchored" data-anchor-id="sec-modelbasedmult">Model based multivariate analysis</h2>
<p>Note this is a bit beyond the scope of this course, but I include it here for fun! We’ll go through each example together. Here, we don’t reduce data to correlation or similarity matrix. Can calculate AIC, plot residuals, etc and can directly relate multiple responses to covariates in single analysis….</p>
<blockquote class="blockquote">
<p>The <code>mvabund</code> package provides tools for model-based analysis of multivariate abundance data in ecology (<span class="citation" data-cites="mvabund">Wang et al. (<a href="#ref-mvabund" role="doc-biblioref">2012</a>)</span>)</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(mvabund)</span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(Tasmania)</span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>tasmvabund <span class="ot">&lt;-</span> <span class="fu">mvabund</span>(Tasmania<span class="sc">$</span>copepods)</span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a><span class="do">## fit a model</span></span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>tas.nb <span class="ot">&lt;-</span> <span class="fu">manyglm</span>(Tasmania<span class="sc">$</span>copepods <span class="sc">~</span> Tasmania<span class="sc">$</span>block<span class="sc">*</span>Tasmania<span class="sc">$</span>treatment, <span class="at">family =</span> <span class="st">"negative.binomial"</span>)</span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(tas.nb)</span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a><span class="do">## predict</span></span>
<span id="cb94-8"><a href="#cb94-8" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(tas.nb, <span class="at">type =</span> <span class="st">"response"</span>)</span>
<span id="cb94-9"><a href="#cb94-9" aria-hidden="true" tabindex="-1"></a><span class="do">## model assumptions</span></span>
<span id="cb94-10"><a href="#cb94-10" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tas.nb)</span>
<span id="cb94-11"><a href="#cb94-11" aria-hidden="true" tabindex="-1"></a><span class="do">## multivariate hypothesis</span></span>
<span id="cb94-12"><a href="#cb94-12" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(tas.nb, <span class="at">p.uni =</span> <span class="st">"adjusted"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<blockquote class="blockquote">
<p>The <code>clustglm</code> package implements multivariate methods using mixtures combining correspondence analysis (CA), scaling (like multidimensional scaling, or MDS), and pattern detection techniques to analyze data with both categorical and continuous variables (<span class="citation" data-cites="clustglm">Pledger and Arnold (<a href="#ref-clustglm" role="doc-biblioref">2014</a>)</span>)</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="do">## devtools::install_github("vuw-clustering/clustglm")</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clustglm)</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(cottontails)</span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a><span class="do">## glm</span></span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a>glm.null <span class="ot">&lt;-</span> <span class="fu">glm</span>(<span class="at">formula =</span> <span class="fu">cbind</span>(nsucc,nfail) <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> cottontails,</span>
<span id="cb95-7"><a href="#cb95-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">family =</span> <span class="st">"binomial"</span>)</span>
<span id="cb95-8"><a href="#cb95-8" aria-hidden="true" tabindex="-1"></a>glm.null<span class="sc">$</span>deviance</span>
<span id="cb95-9"><a href="#cb95-9" aria-hidden="true" tabindex="-1"></a><span class="fu">logLik</span>(glm.null) </span>
<span id="cb95-10"><a href="#cb95-10" aria-hidden="true" tabindex="-1"></a><span class="do">## clustglm</span></span>
<span id="cb95-11"><a href="#cb95-11" aria-hidden="true" tabindex="-1"></a>clustglm.null <span class="ot">&lt;-</span> <span class="fu">clustglm</span>(<span class="at">formula =</span> <span class="fu">cbind</span>(nsucc,nfail) <span class="sc">~</span> <span class="fu">rep</span>(<span class="dv">1</span>,<span class="dv">74</span>),</span>
<span id="cb95-12"><a href="#cb95-12" aria-hidden="true" tabindex="-1"></a>                          <span class="at">data =</span> cottontails,</span>
<span id="cb95-13"><a href="#cb95-13" aria-hidden="true" tabindex="-1"></a>                          <span class="at">family =</span> <span class="st">"binomial"</span>)</span>
<span id="cb95-14"><a href="#cb95-14" aria-hidden="true" tabindex="-1"></a>clustglm.null<span class="sc">$</span>deviance   </span>
<span id="cb95-15"><a href="#cb95-15" aria-hidden="true" tabindex="-1"></a>clustglm.null<span class="sc">$</span>LL   <span class="do">## calculated slightly differently </span></span>
<span id="cb95-16"><a href="#cb95-16" aria-hidden="true" tabindex="-1"></a><span class="do">## two clusters...</span></span>
<span id="cb95-17"><a href="#cb95-17" aria-hidden="true" tabindex="-1"></a>cg<span class="fl">.2</span>clust <span class="ot">&lt;-</span> <span class="fu">clustglm</span>(<span class="at">formula =</span> <span class="fu">cbind</span>(nsucc,nfail) <span class="sc">~</span> IDclust,</span>
<span id="cb95-18"><a href="#cb95-18" aria-hidden="true" tabindex="-1"></a>                       <span class="at">family =</span> <span class="st">"binomial"</span>,</span>
<span id="cb95-19"><a href="#cb95-19" aria-hidden="true" tabindex="-1"></a>                   <span class="at">data =</span> cottontails,</span>
<span id="cb95-20"><a href="#cb95-20" aria-hidden="true" tabindex="-1"></a>                   <span class="at">fact4clust =</span> <span class="st">"ID"</span>, <span class="at">nclust =</span> <span class="dv">2</span>,</span>
<span id="cb95-21"><a href="#cb95-21" aria-hidden="true" tabindex="-1"></a>                   <span class="at">clustfactnames =</span> <span class="st">"IDclust"</span>,</span>
<span id="cb95-22"><a href="#cb95-22" aria-hidden="true" tabindex="-1"></a>                   <span class="at">start.control =</span> <span class="fu">list</span>(<span class="at">randstarts =</span> <span class="dv">100</span>),</span>
<span id="cb95-23"><a href="#cb95-23" aria-hidden="true" tabindex="-1"></a>                   <span class="at">verbose =</span> <span class="dv">1</span>)</span>
<span id="cb95-24"><a href="#cb95-24" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(cg<span class="fl">.2</span>clust)</span>
<span id="cb95-25"><a href="#cb95-25" aria-hidden="true" tabindex="-1"></a><span class="do">## three clusters?</span></span>
<span id="cb95-26"><a href="#cb95-26" aria-hidden="true" tabindex="-1"></a>cg<span class="fl">.3</span>clust <span class="ot">&lt;-</span> <span class="fu">clustglm</span>(<span class="at">formula =</span> <span class="fu">cbind</span>(nsucc,nfail) <span class="sc">~</span> IDclust,</span>
<span id="cb95-27"><a href="#cb95-27" aria-hidden="true" tabindex="-1"></a>                       <span class="at">family =</span> <span class="st">"binomial"</span>,</span>
<span id="cb95-28"><a href="#cb95-28" aria-hidden="true" tabindex="-1"></a>                   <span class="at">data =</span> cottontails,</span>
<span id="cb95-29"><a href="#cb95-29" aria-hidden="true" tabindex="-1"></a>                   <span class="at">fact4clust =</span> <span class="st">"ID"</span>, <span class="at">nclust =</span> <span class="dv">3</span>,</span>
<span id="cb95-30"><a href="#cb95-30" aria-hidden="true" tabindex="-1"></a>                   <span class="at">clustfactnames =</span> <span class="st">"IDclust"</span>,</span>
<span id="cb95-31"><a href="#cb95-31" aria-hidden="true" tabindex="-1"></a>                   <span class="at">start.control =</span> <span class="fu">list</span>(<span class="at">randstarts =</span> <span class="dv">100</span>),</span>
<span id="cb95-32"><a href="#cb95-32" aria-hidden="true" tabindex="-1"></a>                   <span class="at">verbose =</span> <span class="dv">1</span>)</span>
<span id="cb95-33"><a href="#cb95-33" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(cg<span class="fl">.3</span>clust)</span>
<span id="cb95-34"><a href="#cb95-34" aria-hidden="true" tabindex="-1"></a><span class="do">## choose between them</span></span>
<span id="cb95-35"><a href="#cb95-35" aria-hidden="true" tabindex="-1"></a>model.list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">cg.null =</span> clustglm.null,</span>
<span id="cb95-36"><a href="#cb95-36" aria-hidden="true" tabindex="-1"></a>                   <span class="at">cg.2clust =</span> cg<span class="fl">.2</span>clust,</span>
<span id="cb95-37"><a href="#cb95-37" aria-hidden="true" tabindex="-1"></a>                   <span class="at">cg.3clust =</span> cg<span class="fl">.3</span>clust)</span>
<span id="cb95-38"><a href="#cb95-38" aria-hidden="true" tabindex="-1"></a><span class="fu">comparison</span>(model.list)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="may-26th" class="level1">
<h1>May 26th</h1>
<section id="rough-timeline-19" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-19">Rough timeline</h2>
<ul>
<li><strong>09:05–09:55 am</strong> <em>Discriminating</em> multivariate data (<a href="#sec-discrim" class="quarto-xref">Section&nbsp;20.2</a>)</li>
<li><strong>09:55–11:05 am</strong> Break</li>
<li><strong>11:05–11:45 am</strong> Casestudies (<a href="#sec-caselda" class="quarto-xref">Section&nbsp;20.3</a>)</li>
<li><strong>11.45–11.55 pm</strong> Q&amp;A (<a href="#sec-peershare19" class="quarto-xref">Section&nbsp;20.4</a>)</li>
</ul>
</section>
<section id="sec-discrim" class="level2">
<h2 class="anchored" data-anchor-id="sec-discrim">Linear Discriminant Analysis (LDA)</h2>
<p>Compare the following two GIFs; <strong>what is the main difference?</strong></p>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Compare the following two GIFs…
</div>
</div>
<div class="callout-body-container callout-body">
<p>…<strong>what is the main difference?</strong></p>
<div>

</div>
<div class="quarto-layout-panel" data-layout-ncol="2">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="https://raw.githubusercontent.com/statbiscuit/swots/refs/heads/master/figs_n_gifs/lm.gif" class="img-fluid"></p>
</div>
<div class="quarto-layout-cell" style="flex-basis: 50.0%;justify-content: center;">
<p><img src="runsheets/img/fac_lm.gif" class="img-fluid"></p>
</div>
</div>
</div>
</div>
</div>
<p><strong>Linear discriminant analysis (LDA)</strong> is all about <span class="math inline">\(y = mx + c …\)</span></p>
<p align="center">
<img src="runsheets/img/lda.gif" class="img-fluid">
</p>
<p><strong>Slides</strong></p>
<iframe src="slides/11-01_lda.pdf" height="400" width="700" title="LDA">
</iframe>
<p></p>
<p align="center">
<a href="https://canvas.auckland.ac.nz/courses/120054/files/15372455?module_item_id=2614789"><strong>CANVAS slides</strong></a>
</p>
</section>
<section id="sec-caselda" class="level2">
<h2 class="anchored" data-anchor-id="sec-caselda">Casestudies</h2>
<p><strong>THC levels in cannabis</strong> (data from <span class="citation" data-cites="cannabis">Schwabe et al. (<a href="#ref-cannabis" role="doc-biblioref">2023</a>)</span>)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>thc <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/thc.csv"</span>)</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>thc <span class="ot">&lt;-</span> thc <span class="sc">%&gt;%</span></span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">reported =</span> <span class="fu">mean</span>(<span class="fu">c</span>(low, high, mid), <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(thc, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">"Lab tested"</span>, <span class="at">y =</span> total)) <span class="sc">+</span></span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">"Reported by dispensary"</span>, <span class="at">y =</span> reported)) <span class="sc">+</span></span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"THC"</span>) <span class="sc">+</span></span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> thc <span class="sc">%&gt;%</span></span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(strain <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Dankey Kong"</span>, <span class="st">"OG Kush"</span>, <span class="st">"Blue Dream"</span>))</span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-16"><a href="#cb96-16" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">y =</span> total, <span class="at">x =</span> strain, <span class="at">col =</span> strain)) <span class="sc">+</span></span>
<span id="cb96-17"><a href="#cb96-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb96-18"><a href="#cb96-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>() <span class="sc">+</span></span>
<span id="cb96-19"><a href="#cb96-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span>
<span id="cb96-20"><a href="#cb96-20" aria-hidden="true" tabindex="-1"></a><span class="do">### Kind of like regression!</span></span>
<span id="cb96-21"><a href="#cb96-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb96-22"><a href="#cb96-22" aria-hidden="true" tabindex="-1"></a>lda  <span class="ot">&lt;-</span>  <span class="fu">lda</span>(strain <span class="sc">~</span> total, <span class="at">data =</span> data)</span>
<span id="cb96-23"><a href="#cb96-23" aria-hidden="true" tabindex="-1"></a>lda</span>
<span id="cb96-24"><a href="#cb96-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-25"><a href="#cb96-25" aria-hidden="true" tabindex="-1"></a>means <span class="ot">&lt;-</span>  <span class="fu">data.frame</span>(<span class="at">total =</span> lda<span class="sc">$</span>means[,<span class="dv">1</span>],</span>
<span id="cb96-26"><a href="#cb96-26" aria-hidden="true" tabindex="-1"></a>                        <span class="at">strain =</span> <span class="fu">row.names</span>(lda<span class="sc">$</span>means))</span>
<span id="cb96-27"><a href="#cb96-27" aria-hidden="true" tabindex="-1"></a>                        </span>
<span id="cb96-28"><a href="#cb96-28" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">y =</span> total, <span class="at">x =</span> strain, <span class="at">col =</span> strain)) <span class="sc">+</span></span>
<span id="cb96-29"><a href="#cb96-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb96-30"><a href="#cb96-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>() <span class="sc">+</span></span>
<span id="cb96-31"><a href="#cb96-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb96-32"><a href="#cb96-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">data =</span> means, <span class="at">pch =</span> <span class="dv">18</span>, <span class="at">size =</span> <span class="dv">5</span>)</span>
<span id="cb96-33"><a href="#cb96-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-34"><a href="#cb96-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-35"><a href="#cb96-35" aria-hidden="true" tabindex="-1"></a><span class="do">### The LDA  score, which along with the prior is used to compute the</span></span>
<span id="cb96-36"><a href="#cb96-36" aria-hidden="true" tabindex="-1"></a><span class="do">## posterior probability of group membership (there are a number of different formulas for this).</span></span>
<span id="cb96-37"><a href="#cb96-37" aria-hidden="true" tabindex="-1"></a>lda<span class="sc">$</span>scaling</span>
<span id="cb96-38"><a href="#cb96-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-39"><a href="#cb96-39" aria-hidden="true" tabindex="-1"></a>pred <span class="ot">&lt;-</span> <span class="fu">predict</span>(lda)</span>
<span id="cb96-40"><a href="#cb96-40" aria-hidden="true" tabindex="-1"></a>pred</span>
<span id="cb96-41"><a href="#cb96-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-42"><a href="#cb96-42" aria-hidden="true" tabindex="-1"></a><span class="do">## plot prediction</span></span>
<span id="cb96-43"><a href="#cb96-43" aria-hidden="true" tabindex="-1"></a>make1Dgrid <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb96-44"><a href="#cb96-44" aria-hidden="true" tabindex="-1"></a>    rg <span class="ot">&lt;-</span> grDevices<span class="sc">::</span><span class="fu">extendrange</span>(x)</span>
<span id="cb96-45"><a href="#cb96-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">seq</span>(<span class="at">from =</span> rg[<span class="dv">1</span>], <span class="at">to =</span> rg[<span class="dv">2</span>], <span class="at">length.out =</span> <span class="dv">100</span>)</span>
<span id="cb96-46"><a href="#cb96-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb96-47"><a href="#cb96-47" aria-hidden="true" tabindex="-1"></a>                                        </span>
<span id="cb96-48"><a href="#cb96-48" aria-hidden="true" tabindex="-1"></a>grid.df <span class="ot">&lt;-</span> <span class="fu">with</span>(data, <span class="fu">expand.grid</span>(<span class="at">total =</span> <span class="fu">make1Dgrid</span>(total)))</span>
<span id="cb96-49"><a href="#cb96-49" aria-hidden="true" tabindex="-1"></a><span class="co"># Do the predictions</span></span>
<span id="cb96-50"><a href="#cb96-50" aria-hidden="true" tabindex="-1"></a>grid.df<span class="sc">$</span>ghat <span class="ot">&lt;-</span> <span class="fu">predict</span>(lda, <span class="at">newdata =</span> grid.df)<span class="sc">$</span>class</span>
<span id="cb96-51"><a href="#cb96-51" aria-hidden="true" tabindex="-1"></a><span class="do">##plot prediction surface</span></span>
<span id="cb96-52"><a href="#cb96-52" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">y =</span> total, <span class="at">x =</span> strain, <span class="at">col =</span> strain)) <span class="sc">+</span></span>
<span id="cb96-53"><a href="#cb96-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb96-54"><a href="#cb96-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>() <span class="sc">+</span></span>
<span id="cb96-55"><a href="#cb96-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb96-56"><a href="#cb96-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">data =</span> means, <span class="at">pch =</span> <span class="dv">18</span>, <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb96-57"><a href="#cb96-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_raster</span>(<span class="fu">aes</span>(<span class="at">fill =</span> ghat, <span class="at">x =</span> ghat, <span class="at">y =</span> total), <span class="at">data =</span> grid.df, <span class="at">alpha =</span> <span class="fl">0.25</span>, <span class="at">interpolate =</span> <span class="cn">TRUE</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Glass Identification</strong> (dataset from UCI, accessed <a href="https://www.kaggle.com/datasets/uciml/glass">here</a>)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>glass <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/glass.csv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Type =</span> <span class="fu">as.factor</span>(Type))</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(reshape2<span class="sc">::</span><span class="fu">melt</span>(glass, <span class="at">id.vars =</span> <span class="fu">c</span>(<span class="st">"Type"</span>)),</span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> value, <span class="at">colour =</span> Type)) <span class="sc">+</span></span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a>       <span class="fu">geom_density</span>() <span class="sc">+</span> <span class="fu">facet_wrap</span>( <span class="sc">~</span>variable, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a><span class="do">##Possible classification rules? RI &amp; Al??</span></span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(glass, <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> RI, <span class="at">y =</span> Al)) <span class="sc">+</span></span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb97-10"><a href="#cb97-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Type), <span class="at">size =</span> <span class="dv">3</span>)</span>
<span id="cb97-11"><a href="#cb97-11" aria-hidden="true" tabindex="-1"></a><span class="do">## LDA</span></span>
<span id="cb97-12"><a href="#cb97-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb97-13"><a href="#cb97-13" aria-hidden="true" tabindex="-1"></a>lda  <span class="ot">&lt;-</span>  <span class="fu">lda</span>(Type <span class="sc">~</span> RI <span class="sc">+</span> Al, <span class="at">data =</span> glass)</span>
<span id="cb97-14"><a href="#cb97-14" aria-hidden="true" tabindex="-1"></a><span class="do">## Proportion of Trace gives the percentage separation achieved by each discriminant function</span></span>
<span id="cb97-15"><a href="#cb97-15" aria-hidden="true" tabindex="-1"></a>lda</span>
<span id="cb97-16"><a href="#cb97-16" aria-hidden="true" tabindex="-1"></a><span class="do">## predict</span></span>
<span id="cb97-17"><a href="#cb97-17" aria-hidden="true" tabindex="-1"></a>ghat <span class="ot">&lt;-</span> <span class="fu">predict</span>(lda)<span class="sc">$</span>class</span>
<span id="cb97-18"><a href="#cb97-18" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="at">prediced =</span> ghat, <span class="at">observed =</span> glass<span class="sc">$</span>Type)</span>
<span id="cb97-19"><a href="#cb97-19" aria-hidden="true" tabindex="-1"></a><span class="do">## missclassification rate?</span></span>
<span id="cb97-20"><a href="#cb97-20" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(ghat <span class="sc">!=</span> glass<span class="sc">$</span>Type)</span>
<span id="cb97-21"><a href="#cb97-21" aria-hidden="true" tabindex="-1"></a><span class="do">## group means</span></span>
<span id="cb97-22"><a href="#cb97-22" aria-hidden="true" tabindex="-1"></a>centers  <span class="ot">&lt;-</span>  lda<span class="sc">$</span>means</span>
<span id="cb97-23"><a href="#cb97-23" aria-hidden="true" tabindex="-1"></a><span class="do">### plot the discriminant surface</span></span>
<span id="cb97-24"><a href="#cb97-24" aria-hidden="true" tabindex="-1"></a><span class="do">## Set up grid for prediction, 100 x 100 that covers the data range of the two variables</span></span>
<span id="cb97-25"><a href="#cb97-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-26"><a href="#cb97-26" aria-hidden="true" tabindex="-1"></a>grid.df <span class="ot">&lt;-</span>  <span class="fu">with</span>(glass,</span>
<span id="cb97-27"><a href="#cb97-27" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">expand.grid</span>(<span class="at">RI =</span> <span class="fu">make1Dgrid</span>(RI),</span>
<span id="cb97-28"><a href="#cb97-28" aria-hidden="true" tabindex="-1"></a>                             <span class="at">Al =</span> <span class="fu">make1Dgrid</span>(Al)))</span>
<span id="cb97-29"><a href="#cb97-29" aria-hidden="true" tabindex="-1"></a>ghat <span class="ot">&lt;-</span> <span class="fu">predict</span>(lda, <span class="at">newdata =</span> grid.df)</span>
<span id="cb97-30"><a href="#cb97-30" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(ghat)</span>
<span id="cb97-31"><a href="#cb97-31" aria-hidden="true" tabindex="-1"></a><span class="do">## plot</span></span>
<span id="cb97-32"><a href="#cb97-32" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(glass, <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> RI, <span class="at">y =</span> Al))<span class="sc">+</span></span>
<span id="cb97-33"><a href="#cb97-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb97-34"><a href="#cb97-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Type), <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb97-35"><a href="#cb97-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_raster</span>(<span class="fu">aes</span>(<span class="at">fill =</span> ghat<span class="sc">$</span>class),</span>
<span id="cb97-36"><a href="#cb97-36" aria-hidden="true" tabindex="-1"></a>                <span class="at">data =</span> grid.df, <span class="at">alpha =</span> <span class="fl">0.25</span>, <span class="at">interpolate =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb97-37"><a href="#cb97-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">data =</span> <span class="fu">as_tibble</span>(centers), <span class="at">pch =</span> <span class="st">"+"</span>, <span class="at">size =</span> <span class="dv">8</span>)</span>
<span id="cb97-38"><a href="#cb97-38" aria-hidden="true" tabindex="-1"></a><span class="do">### So, not doing great! Can you improve?</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="sec-peershare19" class="level2">
<h2 class="anchored" data-anchor-id="sec-peershare19">Q&amp;A and peer-share</h2>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Peer-Share
</div>
</div>
<div class="callout-body-container callout-body">
<p>Go over an activity from the class again. Feel free to ask Charlotte a question or two about it!</p>
</div>
</div>
</section>
</section>
<section id="may-30th" class="level1">
<h1>May 30th</h1>
<section id="rough-timeline-20" class="level2">
<h2 class="anchored" data-anchor-id="rough-timeline-20">Rough timeline</h2>
<section id="pm---0355-pm-stats-bake-off-report-roulette" class="level3">
<h3 class="anchored" data-anchor-id="pm---0355-pm-stats-bake-off-report-roulette"><strong>02:05 pm - 03:55 pm</strong> Stats Bake-Off &amp; Report Roulette</h3>
<div class="callout callout-style-default callout-warning no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa fa-question-circle" aria-hidden="true"></i> <i class="fa fa-users" aria-hidden="true"></i> <i class="fa fa-pencil-square" aria-hidden="true"></i> Organize yourselves into groups of four.
</div>
</div>
<div class="callout-body-container callout-body">
<section id="determine-who-is-filling-which-role-listed-below-by-order-of-next-birthdays-the-person-with-the-next-birthday-will-be-facilitator-the-next-birthday-after-that-will-be-the-notetaker-the-third-birthday-will-be-the-coder-and-if-your-group-has-four-members-the-fourth-member-will-be-the-reporter." class="level4">
<h4 class="anchored" data-anchor-id="determine-who-is-filling-which-role-listed-below-by-order-of-next-birthdays-the-person-with-the-next-birthday-will-be-facilitator-the-next-birthday-after-that-will-be-the-notetaker-the-third-birthday-will-be-the-coder-and-if-your-group-has-four-members-the-fourth-member-will-be-the-reporter.">Determine who is filling which role listed below by order of next birthdays: the person with the next birthday will be facilitator, the next birthday after that will be the notetaker, the third birthday will be the coder, and if your group has four members, the fourth member will be the reporter.</h4>
</section>
<section id="team-roles-responsibilities" class="level4">
<h4 class="anchored" data-anchor-id="team-roles-responsibilities">👥 Team Roles &amp; Responsibilities</h4>
<p>Each team member should actively contribute to <strong>both the discussion and the analysis</strong>. Use <a href="https://posit.cloud/">Posit Cloud</a> to collaborate in real-time, with the <em>Coder</em> hosting the shared project. Roles are designed to support equitable collaboration — not divide tasks in silos.</p>
<hr>
<ol type="1">
<li>🤝 <strong>Facilitator</strong></li>
</ol>
<ul>
<li>Ensures inclusive and collaborative participation.</li>
<li>Encourages all members to contribute ideas and opinions.</li>
<li>Summarizes progress and decisions during discussions.</li>
<li>Helps keep the group focused and on task.</li>
</ul>
<hr>
<ol start="2" type="1">
<li>📝 <strong>Notetaker</strong></li>
</ol>
<ul>
<li>Records key points, decisions, and reasoning during the task.</li>
<li>Maintains a digital “lab notebook” (e.g., <code>.txt</code> or <code>.Rmd</code> file) in Posit Cloud.</li>
<li>Logs assumptions, issues, and questions for the teaching team.</li>
<li>Supports the Reporter by providing clear summaries of group work.</li>
</ul>
<hr>
<ol start="3" type="1">
<li>💻 <strong>Coder</strong> <em>(Posit Cloud Project Host)</em></li>
</ol>
<ul>
<li>Creates and shares a <a href="https://posit.cloud/">Posit Cloud project</a> for the group (<em>yet to be given by Charlotte</em>).</li>
<li>Organizes code files clearly, using comments and logical structure.</li>
<li>Collaboratively implements the analysis — with input from all members.</li>
<li>Helps debug code and ensure everyone can run scripts smoothly.</li>
</ul>
<hr>
<ol start="4" type="1">
<li>📢 <strong>Reporter</strong></li>
</ol>
<ul>
<li>Drafts the <strong>press release-style report</strong> in <code>.Rmd</code> format in the shared Posit Cloud project.</li>
<li>Leads group consensus on interpretations and key messages.</li>
<li>Presents the final group findings (in the last 15 minutes of this workshop to the other teams).</li>
<li>Reflects on both the scientific analysis and group collaboration.</li>
</ul>
<hr>
<p>🔁 Shared Responsibilities Regardless of role, everyone should: - Contribute to the <strong>statistical analysis and interpretation</strong>. - Review and give feedback on code and written outputs. - Ask questions, support others, and engage with the task.</p>
</section>
<section id="then-1-choose-a-team-name-and-2-each-set-up-a-posit-cloud-free-account.-now-the-facilitator-should-come-up-to-me-giving-me-the-teams-name-and-the-coders-email-address-associated-with-posit-cloud.-i-will-then-randomly-allocate-your-team-a-scenario-which-will-determine-your-groups-next-step." class="level4">
<h4 class="anchored" data-anchor-id="then-1-choose-a-team-name-and-2-each-set-up-a-posit-cloud-free-account.-now-the-facilitator-should-come-up-to-me-giving-me-the-teams-name-and-the-coders-email-address-associated-with-posit-cloud.-i-will-then-randomly-allocate-your-team-a-scenario-which-will-determine-your-groups-next-step.">Then 1) choose a team name and 2) each set up a <a href="https://posit.cloud/">Posit Cloud (Free)</a> account. Now, the <strong>facilitator should come up to me</strong> giving me the team’s name and the coder’s email address associated with <a href="https://posit.cloud/">Posit Cloud</a>. I will then randomly allocate your team a scenario, which will determine your groups’ next step<a href="#fn11" class="footnote-ref" id="fnref11" role="doc-noteref"><sup>11</sup></a>.</h4>
<p><em>[NOTE: Role responsibilities were ammended from those we’ve used earlier in the course with the assistance of ChatGPT (OpenAI, 2025).]</em></p>
</section>
</div>
</div>
<!--{{< include runsheets/runsheet_21.qmd >}}-->
</section>
</section>
</section>
<section id="sec-solutions" class="level1">
<h1>Activity solutions</h1>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Suggested <code>R</code> code solutions
</div>
</div>
<div class="callout-body-container callout-body">
<p>This section will be update with my suggested solutions to the class wide activities after their conclusion. I <strong>strongly</strong> recommend that you review these outside of class if only to familiarise yourself with a different approach.</p>
</div>
</div>
<section id="suggested-solution-to-sec-gorilla" class="level2">
<h2 class="anchored" data-anchor-id="suggested-solution-to-sec-gorilla">Suggested solution to <a href="#sec-gorilla" class="quarto-xref">Section&nbsp;2.2</a></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(tidyverse)</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a><span class="do">## read in data</span></span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/BMI.csv"</span>)</span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>data</span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Mean BMI</span></span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb98-7"><a href="#cb98-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="fu">mean</span>(bmi)) <span class="sc">%&gt;%</span></span>
<span id="cb98-8"><a href="#cb98-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">round</span>()</span>
<span id="cb98-9"><a href="#cb98-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Max BMI males</span></span>
<span id="cb98-10"><a href="#cb98-10" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb98-11"><a href="#cb98-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(gender <span class="sc">==</span> <span class="st">"male"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb98-12"><a href="#cb98-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(bmi <span class="sc">==</span> <span class="fu">max</span>(bmi))</span>
<span id="cb98-13"><a href="#cb98-13" aria-hidden="true" tabindex="-1"></a><span class="do">## t-test BMI males vs females</span></span>
<span id="cb98-14"><a href="#cb98-14" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(bmi <span class="sc">~</span> gender, <span class="at">data =</span> data)</span>
<span id="cb98-15"><a href="#cb98-15" aria-hidden="true" tabindex="-1"></a><span class="do">## t-test steps males vs females</span></span>
<span id="cb98-16"><a href="#cb98-16" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(steps <span class="sc">~</span> gender, <span class="at">data =</span> data) </span>
<span id="cb98-17"><a href="#cb98-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-18"><a href="#cb98-18" aria-hidden="true" tabindex="-1"></a><span class="do">## plotting. ALWAYS PLOT YOUR DATA!</span></span>
<span id="cb98-19"><a href="#cb98-19" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> steps, <span class="at">y =</span> bmi, <span class="at">col =</span> gender)) <span class="sc">+</span></span>
<span id="cb98-20"><a href="#cb98-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="suggested-solution-to-sec-honesty" class="level2">
<h2 class="anchored" data-anchor-id="suggested-solution-to-sec-honesty">Suggested solution to <a href="#sec-honesty" class="quarto-xref">Section&nbsp;3.5</a></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/retracted-honesty-study_week-02.csv"</span>)</span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a><span class="do">## set up descriptive treatments</span></span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span> <span class="fu">mutate</span> (<span class="at">treatment =</span> <span class="fu">ifelse</span>(Cond<span class="sc">==</span><span class="dv">0</span>, <span class="st">"No signature"</span>,</span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a>                                     <span class="fu">ifelse</span>(Cond<span class="sc">==</span><span class="dv">1</span>, <span class="st">"Sign-Top"</span>,</span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a>                                     <span class="fu">ifelse</span>(Cond<span class="sc">==</span><span class="dv">2</span>, <span class="st">"Sign-Bottom"</span> , <span class="cn">NA</span>))))</span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a><span class="do">## remove control condition (here we're only interested in comparing sign-top and sign-bottom)</span></span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a>data_sub <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(., Cond <span class="sc">!=</span> <span class="dv">0</span>)</span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a><span class="do">## MATCH PAPER</span></span>
<span id="cb99-10"><a href="#cb99-10" aria-hidden="true" tabindex="-1"></a><span class="do">## "lowered the share of people over-reporting their math puzzle performance from 79% to 37%"</span></span>
<span id="cb99-11"><a href="#cb99-11" aria-hidden="true" tabindex="-1"></a><span class="do">## with "chi sq test p(0.0013)"</span></span>
<span id="cb99-12"><a href="#cb99-12" aria-hidden="true" tabindex="-1"></a><span class="fu">prop.table</span>(<span class="fu">table</span>(data_sub<span class="sc">$</span>CheatedOnMatrixTax, data_sub<span class="sc">$</span>treatment), <span class="at">margin =</span> <span class="dv">2</span>)</span>
<span id="cb99-13"><a href="#cb99-13" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(data_sub<span class="sc">$</span>CheatedOnMatrixTax, data_sub<span class="sc">$</span>treatment) <span class="sc">|&gt;</span> <span class="fu">chisq.test</span>()</span>
<span id="cb99-14"><a href="#cb99-14" aria-hidden="true" tabindex="-1"></a><span class="do">## "lowered the average amount of over-reporting from 3.94 puzzles to 0.77 puzzles (p &lt; .00001)"</span></span>
<span id="cb99-15"><a href="#cb99-15" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(OverReport <span class="sc">~</span> treatment, <span class="at">data =</span> data_sub)</span>
<span id="cb99-16"><a href="#cb99-16" aria-hidden="true" tabindex="-1"></a><span class="do">## "nearly halved the average amount of claimed commuting expenses, from $9.62 to $5.27 (p = .0014)"</span></span>
<span id="cb99-17"><a href="#cb99-17" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(SumDeduction <span class="sc">~</span> treatment, <span class="at">data =</span> data_sub)</span>
<span id="cb99-18"><a href="#cb99-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-19"><a href="#cb99-19" aria-hidden="true" tabindex="-1"></a><span class="do">## DATA DETECTIVING</span></span>
<span id="cb99-20"><a href="#cb99-20" aria-hidden="true" tabindex="-1"></a><span class="do">## lm</span></span>
<span id="cb99-21"><a href="#cb99-21" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(SumDeduction <span class="sc">~</span> treatment, <span class="at">data =</span> data_sub)</span>
<span id="cb99-22"><a href="#cb99-22" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span> <span class="fu">summary</span>() <span class="do">## matches t-test above (is the same thing)</span></span>
<span id="cb99-23"><a href="#cb99-23" aria-hidden="true" tabindex="-1"></a><span class="do">## plot makes sense</span></span>
<span id="cb99-24"><a href="#cb99-24" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data_sub, <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> SumDeduction)) <span class="sc">+</span></span>
<span id="cb99-25"><a href="#cb99-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_violin</span>() <span class="sc">+</span> <span class="fu">geom_jitter</span>()</span>
<span id="cb99-26"><a href="#cb99-26" aria-hidden="true" tabindex="-1"></a><span class="do">## mapping flagged observations to colour</span></span>
<span id="cb99-27"><a href="#cb99-27" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data_sub, <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> SumDeduction)) <span class="sc">+</span></span>
<span id="cb99-28"><a href="#cb99-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_violin</span>() <span class="sc">+</span> <span class="fu">geom_jitter</span>(<span class="fu">aes</span>(<span class="at">col =</span> <span class="fu">as.factor</span>(flag)))</span>
<span id="cb99-29"><a href="#cb99-29" aria-hidden="true" tabindex="-1"></a><span class="do">## residuals</span></span>
<span id="cb99-30"><a href="#cb99-30" aria-hidden="true" tabindex="-1"></a>data_sub<span class="sc">$</span>residuals <span class="ot">&lt;-</span> <span class="fu">residuals</span>(mod)</span>
<span id="cb99-31"><a href="#cb99-31" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(data_sub, <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> residuals)) <span class="sc">+</span></span>
<span id="cb99-32"><a href="#cb99-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">col =</span> <span class="fu">as.factor</span>(flag))) <span class="sc">+</span></span>
<span id="cb99-33"><a href="#cb99-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">"grey"</span>, <span class="at">linewidth =</span> <span class="dv">2</span>, <span class="at">linetype =</span> <span class="dv">2</span>)</span>
<span id="cb99-34"><a href="#cb99-34" aria-hidden="true" tabindex="-1"></a><span class="do">## now removing flagged observations</span></span>
<span id="cb99-35"><a href="#cb99-35" aria-hidden="true" tabindex="-1"></a><span class="do">## lm</span></span>
<span id="cb99-36"><a href="#cb99-36" aria-hidden="true" tabindex="-1"></a>data_sub <span class="sc">%&gt;%</span></span>
<span id="cb99-37"><a href="#cb99-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(., flag <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb99-38"><a href="#cb99-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">t.test</span>(SumDeduction <span class="sc">~</span> treatment, <span class="at">data =</span> .)</span>
<span id="cb99-39"><a href="#cb99-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-40"><a href="#cb99-40" aria-hidden="true" tabindex="-1"></a><span class="do">## see https://datacolada.org/109</span></span>
<span id="cb99-41"><a href="#cb99-41" aria-hidden="true" tabindex="-1"></a><span class="do">## and/or https://www.theguardian.com/education/2023/jun/25/harvard-professor-data-fraud</span></span>
<span id="cb99-42"><a href="#cb99-42" aria-hidden="true" tabindex="-1"></a><span class="do">## for the story</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="suggested-solution-to-sec-draw" class="level2">
<h2 class="anchored" data-anchor-id="suggested-solution-to-sec-draw">Suggested solution to <a href="#sec-draw" class="quarto-xref">Section&nbsp;6.3</a></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="do">## data source: https://docs.google.com/spreadsheets/d/1YSJ4ypkYLq6j1mIBJCgUHhHjJZQ0Rkfe1qW2WC5HLiw/edit#gid=748627588</span></span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/marvel_movies.csv"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">critics_score =</span> <span class="fu">as.numeric</span>(<span class="fu">str_replace_all</span>(<span class="st">`</span><span class="at">critics % score</span><span class="st">`</span>, <span class="st">"%"</span>, <span class="st">""</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">budget_recovered =</span> <span class="fu">as.numeric</span>(<span class="fu">str_replace_all</span>(<span class="st">`</span><span class="at">% budget recovered</span><span class="st">`</span>, <span class="st">"%"</span>, <span class="st">""</span>)))</span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a><span class="do">##### ##### ##### </span><span class="al">###</span></span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a><span class="do">##### Model 1 #######</span></span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a><span class="do">##### ##### ##### </span><span class="al">###</span></span>
<span id="cb100-9"><a href="#cb100-9" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb100-10"><a href="#cb100-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lm</span>(budget_recovered <span class="sc">~</span> critics_score, <span class="at">data =</span> .) <span class="sc">|&gt;</span></span>
<span id="cb100-11"><a href="#cb100-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summary</span>()</span>
<span id="cb100-12"><a href="#cb100-12" aria-hidden="true" tabindex="-1"></a><span class="do">## Ans</span></span>
<span id="cb100-13"><a href="#cb100-13" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb100-14"><a href="#cb100-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">y =</span> budget_recovered, <span class="at">x =</span> critics_score)) <span class="sc">+</span></span>
<span id="cb100-15"><a href="#cb100-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">" Estimated budget_recovered"</span>)</span>
<span id="cb100-16"><a href="#cb100-16" aria-hidden="true" tabindex="-1"></a><span class="do">## w. data</span></span>
<span id="cb100-17"><a href="#cb100-17" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb100-18"><a href="#cb100-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">y =</span> budget_recovered, <span class="at">x =</span> critics_score)) <span class="sc">+</span></span>
<span id="cb100-19"><a href="#cb100-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span> <span class="fu">geom_point</span>()</span>
<span id="cb100-20"><a href="#cb100-20" aria-hidden="true" tabindex="-1"></a><span class="do">##### ##### ##### </span><span class="al">###</span></span>
<span id="cb100-21"><a href="#cb100-21" aria-hidden="true" tabindex="-1"></a><span class="do">##### Model 2 #######</span></span>
<span id="cb100-22"><a href="#cb100-22" aria-hidden="true" tabindex="-1"></a><span class="do">##### ##### ##### </span><span class="al">###</span></span>
<span id="cb100-23"><a href="#cb100-23" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(<span class="fu">table</span>(data<span class="sc">$</span>category))</span>
<span id="cb100-24"><a href="#cb100-24" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb100-25"><a href="#cb100-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lm</span>(budget_recovered <span class="sc">~</span>  category, <span class="at">data =</span> .) </span>
<span id="cb100-26"><a href="#cb100-26" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span></span>
<span id="cb100-27"><a href="#cb100-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summary</span>()</span>
<span id="cb100-28"><a href="#cb100-28" aria-hidden="true" tabindex="-1"></a><span class="do">## Ans</span></span>
<span id="cb100-29"><a href="#cb100-29" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>pred_vals <span class="ot">&lt;-</span> <span class="fu">predict</span>(mod)</span>
<span id="cb100-30"><a href="#cb100-30" aria-hidden="true" tabindex="-1"></a><span class="do">## plot</span></span>
<span id="cb100-31"><a href="#cb100-31" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb100-32"><a href="#cb100-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> pred_vals, <span class="at">x =</span> category)) <span class="sc">+</span></span>
<span id="cb100-33"><a href="#cb100-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">15</span>, <span class="at">pch =</span> <span class="st">"-"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">" Estimated budget_recovered"</span>)</span>
<span id="cb100-34"><a href="#cb100-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-35"><a href="#cb100-35" aria-hidden="true" tabindex="-1"></a><span class="do">##### ##### ##### </span><span class="al">###</span></span>
<span id="cb100-36"><a href="#cb100-36" aria-hidden="true" tabindex="-1"></a><span class="do">##### Model 3 #######</span></span>
<span id="cb100-37"><a href="#cb100-37" aria-hidden="true" tabindex="-1"></a><span class="do">##### ##### ##### </span><span class="al">###</span></span>
<span id="cb100-38"><a href="#cb100-38" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb100-39"><a href="#cb100-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lm</span>(budget_recovered <span class="sc">~</span> critics_score <span class="sc">+</span> category, <span class="at">data =</span> .) </span>
<span id="cb100-40"><a href="#cb100-40" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span></span>
<span id="cb100-41"><a href="#cb100-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summary</span>()</span>
<span id="cb100-42"><a href="#cb100-42" aria-hidden="true" tabindex="-1"></a><span class="do">## Ans</span></span>
<span id="cb100-43"><a href="#cb100-43" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>pred_vals <span class="ot">&lt;-</span> <span class="fu">predict</span>(mod)</span>
<span id="cb100-44"><a href="#cb100-44" aria-hidden="true" tabindex="-1"></a><span class="do">## plot</span></span>
<span id="cb100-45"><a href="#cb100-45" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb100-46"><a href="#cb100-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> pred_vals, <span class="at">x =</span> critics_score, <span class="at">color =</span> category)) <span class="sc">+</span></span>
<span id="cb100-47"><a href="#cb100-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>() <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">" Estimated budget_recovered"</span>)</span>
<span id="cb100-48"><a href="#cb100-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-49"><a href="#cb100-49" aria-hidden="true" tabindex="-1"></a><span class="do">##### ##### ##### </span><span class="al">###</span></span>
<span id="cb100-50"><a href="#cb100-50" aria-hidden="true" tabindex="-1"></a><span class="do">##### Model 4 #######</span></span>
<span id="cb100-51"><a href="#cb100-51" aria-hidden="true" tabindex="-1"></a><span class="do">##### ##### ##### </span><span class="al">###</span></span>
<span id="cb100-52"><a href="#cb100-52" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb100-53"><a href="#cb100-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lm</span>(budget_recovered <span class="sc">~</span> critics_score<span class="sc">*</span>category, <span class="at">data =</span> .) </span>
<span id="cb100-54"><a href="#cb100-54" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span></span>
<span id="cb100-55"><a href="#cb100-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summary</span>()</span>
<span id="cb100-56"><a href="#cb100-56" aria-hidden="true" tabindex="-1"></a><span class="do">## Ans</span></span>
<span id="cb100-57"><a href="#cb100-57" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>pred_vals <span class="ot">&lt;-</span> <span class="fu">predict</span>(mod)</span>
<span id="cb100-58"><a href="#cb100-58" aria-hidden="true" tabindex="-1"></a><span class="do">## plot</span></span>
<span id="cb100-59"><a href="#cb100-59" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb100-60"><a href="#cb100-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> pred_vals, <span class="at">x =</span> critics_score, <span class="at">color =</span> category)) <span class="sc">+</span></span>
<span id="cb100-61"><a href="#cb100-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>() <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">" Estimated budget_recovered"</span>)</span>
<span id="cb100-62"><a href="#cb100-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-63"><a href="#cb100-63" aria-hidden="true" tabindex="-1"></a><span class="do">##### ##### ##### </span><span class="al">###</span></span>
<span id="cb100-64"><a href="#cb100-64" aria-hidden="true" tabindex="-1"></a><span class="do">##### Model 5 #######</span></span>
<span id="cb100-65"><a href="#cb100-65" aria-hidden="true" tabindex="-1"></a><span class="do">##### ##### ##### </span><span class="al">###</span></span>
<span id="cb100-66"><a href="#cb100-66" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb100-67"><a href="#cb100-67" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rename</span>(., <span class="st">"worldwide"</span> <span class="ot">=</span> <span class="st">`</span><span class="at">worldwide gross ($m)</span><span class="st">`</span>, <span class="st">"domestic"</span> <span class="ot">=</span> <span class="st">`</span><span class="at">domestic gross ($m)</span><span class="st">`</span>,</span>
<span id="cb100-68"><a href="#cb100-68" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"international"</span> <span class="ot">=</span> <span class="st">`</span><span class="at">international gross ($m)</span><span class="st">`</span>)</span>
<span id="cb100-69"><a href="#cb100-69" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(worldwide <span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> domestic <span class="sc">+</span> international, <span class="at">data =</span> data) </span>
<span id="cb100-70"><a href="#cb100-70" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span> <span class="fu">summary</span>()</span>
<span id="cb100-71"><a href="#cb100-71" aria-hidden="true" tabindex="-1"></a><span class="do">## Ans</span></span>
<span id="cb100-72"><a href="#cb100-72" aria-hidden="true" tabindex="-1"></a><span class="do">## plot</span></span>
<span id="cb100-73"><a href="#cb100-73" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(rsm) <span class="do">## install if not available</span></span>
<span id="cb100-74"><a href="#cb100-74" aria-hidden="true" tabindex="-1"></a><span class="fu">persp</span>(mod, <span class="at">form =</span>  <span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> domestic <span class="sc">+</span> international)</span>
<span id="cb100-75"><a href="#cb100-75" aria-hidden="true" tabindex="-1"></a><span class="fu">image</span>(mod, <span class="at">form =</span>  <span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> domestic <span class="sc">+</span> international)</span>
<span id="cb100-76"><a href="#cb100-76" aria-hidden="true" tabindex="-1"></a><span class="do">##### ##### ##### </span><span class="al">###</span></span>
<span id="cb100-77"><a href="#cb100-77" aria-hidden="true" tabindex="-1"></a><span class="do">##### Model 6 #######</span></span>
<span id="cb100-78"><a href="#cb100-78" aria-hidden="true" tabindex="-1"></a><span class="do">##### ##### ##### </span><span class="al">###</span></span>
<span id="cb100-79"><a href="#cb100-79" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(worldwide <span class="sc">~</span> budget<span class="sc">*</span>critics_score, <span class="at">data =</span> data) </span>
<span id="cb100-80"><a href="#cb100-80" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span>  <span class="fu">summary</span>()</span>
<span id="cb100-81"><a href="#cb100-81" aria-hidden="true" tabindex="-1"></a><span class="do">## Ans</span></span>
<span id="cb100-82"><a href="#cb100-82" aria-hidden="true" tabindex="-1"></a><span class="do">## plot</span></span>
<span id="cb100-83"><a href="#cb100-83" aria-hidden="true" tabindex="-1"></a><span class="fu">persp</span>(mod, <span class="at">form =</span>  <span class="sc">~</span> budget<span class="sc">*</span>critics_score)</span>
<span id="cb100-84"><a href="#cb100-84" aria-hidden="true" tabindex="-1"></a><span class="fu">image</span>(mod, <span class="at">form =</span>  <span class="sc">~</span> budget<span class="sc">*</span>critics_score)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="suggested-solution-to-sec-fry" class="level2">
<h2 class="anchored" data-anchor-id="suggested-solution-to-sec-fry">Suggested solution to <a href="#sec-fry" class="quarto-xref">Section&nbsp;6.4.2</a></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="do">## paper https://www.sciencedirect.com/science/article/pii/S1658077X17302655?via%3Dihub#t0005</span></span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 1 data 'extracted' by https://automeris.io/WebPlotDigitizer/</span></span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb101-5"><a href="#cb101-5" aria-hidden="true" tabindex="-1"></a>fries <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/fries.csv"</span>)</span>
<span id="cb101-6"><a href="#cb101-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-7"><a href="#cb101-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(fries, <span class="fu">aes</span>(<span class="at">x =</span> frying_time_s, <span class="at">y =</span> frying_temp_c, <span class="at">col =</span> method)) <span class="sc">+</span></span>
<span id="cb101-8"><a href="#cb101-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>() <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>)</span>
<span id="cb101-9"><a href="#cb101-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-10"><a href="#cb101-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(fries, <span class="fu">aes</span>(<span class="at">x =</span> frying_time_s, <span class="at">y =</span> frying_temp_c, <span class="at">col =</span> method)) <span class="sc">+</span></span>
<span id="cb101-11"><a href="#cb101-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>() <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">stat_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>,<span class="at">formula =</span> y<span class="sc">~</span><span class="fu">poly</span>(x, <span class="dv">2</span>, <span class="at">raw=</span><span class="cn">TRUE</span>), <span class="at">se =</span> <span class="cn">FALSE</span>)</span>
<span id="cb101-12"><a href="#cb101-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-13"><a href="#cb101-13" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(fries, <span class="fu">aes</span>(<span class="at">x =</span> frying_time_s, <span class="at">y =</span> frying_temp_c, <span class="at">col =</span> method)) <span class="sc">+</span></span>
<span id="cb101-14"><a href="#cb101-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>() <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">stat_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>,<span class="at">formula =</span> y<span class="sc">~</span><span class="fu">poly</span>(x, <span class="dv">3</span>, <span class="at">raw=</span><span class="cn">TRUE</span>), <span class="at">se =</span> <span class="cn">FALSE</span>)</span>
<span id="cb101-15"><a href="#cb101-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-16"><a href="#cb101-16" aria-hidden="true" tabindex="-1"></a><span class="do">## first degree</span></span>
<span id="cb101-17"><a href="#cb101-17" aria-hidden="true" tabindex="-1"></a><span class="fu">lm</span>(frying_temp_c <span class="sc">~</span> frying_time_s <span class="sc">+</span> method, <span class="at">data =</span> fries) <span class="sc">|&gt;</span></span>
<span id="cb101-18"><a href="#cb101-18" aria-hidden="true" tabindex="-1"></a>    gglm<span class="sc">::</span><span class="fu">gglm</span>()</span>
<span id="cb101-19"><a href="#cb101-19" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(fries, <span class="fu">aes</span>(<span class="at">x =</span> frying_time_s, <span class="at">y =</span> frying_temp_c, <span class="at">col =</span> method)) <span class="sc">+</span></span>
<span id="cb101-20"><a href="#cb101-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>() <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>)</span>
<span id="cb101-21"><a href="#cb101-21" aria-hidden="true" tabindex="-1"></a><span class="do">## second degree</span></span>
<span id="cb101-22"><a href="#cb101-22" aria-hidden="true" tabindex="-1"></a><span class="fu">lm</span>(frying_temp_c <span class="sc">~</span> <span class="fu">I</span>(frying_time_s<span class="sc">^</span><span class="dv">2</span>) <span class="sc">+</span> method, <span class="at">data =</span> fries) <span class="sc">|&gt;</span></span>
<span id="cb101-23"><a href="#cb101-23" aria-hidden="true" tabindex="-1"></a>gglm<span class="sc">::</span><span class="fu">gglm</span>()</span>
<span id="cb101-24"><a href="#cb101-24" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(fries, <span class="fu">aes</span>(<span class="at">x =</span> frying_time_s, <span class="at">y =</span> frying_temp_c, <span class="at">col =</span> method)) <span class="sc">+</span></span>
<span id="cb101-25"><a href="#cb101-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>() <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">stat_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>,<span class="at">formula =</span> y<span class="sc">~</span><span class="fu">poly</span>(x, <span class="dv">2</span>, <span class="at">raw=</span><span class="cn">TRUE</span>), <span class="at">se =</span> <span class="cn">FALSE</span>)</span>
<span id="cb101-26"><a href="#cb101-26" aria-hidden="true" tabindex="-1"></a><span class="do">## third degree</span></span>
<span id="cb101-27"><a href="#cb101-27" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(frying_temp_c <span class="sc">~</span> <span class="fu">I</span>(frying_time_s<span class="sc">^</span><span class="dv">1</span>) <span class="sc">+</span>  <span class="fu">I</span>(frying_time_s<span class="sc">^</span><span class="dv">2</span>) <span class="sc">+</span>  <span class="fu">I</span>(frying_time_s<span class="sc">^</span><span class="dv">3</span>) <span class="sc">+</span> method,</span>
<span id="cb101-28"><a href="#cb101-28" aria-hidden="true" tabindex="-1"></a>          <span class="at">data =</span> fries, <span class="at">na.action =</span> <span class="st">"na.fail"</span>)</span>
<span id="cb101-29"><a href="#cb101-29" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span> <span class="fu">summary</span>()</span>
<span id="cb101-30"><a href="#cb101-30" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span></span>
<span id="cb101-31"><a href="#cb101-31" aria-hidden="true" tabindex="-1"></a>   gglm<span class="sc">::</span><span class="fu">gglm</span>()</span>
<span id="cb101-32"><a href="#cb101-32" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(frying_temp_c <span class="sc">~</span> <span class="fu">poly</span>(frying_time_s, <span class="dv">3</span>, <span class="at">raw =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> method, <span class="at">data =</span> fries) <span class="do">## but these are correlated</span></span>
<span id="cb101-33"><a href="#cb101-33" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span></span>
<span id="cb101-34"><a href="#cb101-34" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>()</span>
<span id="cb101-35"><a href="#cb101-35" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(fries, <span class="fu">aes</span>(<span class="at">x =</span> frying_time_s, <span class="at">y =</span> frying_temp_c, <span class="at">col =</span> method)) <span class="sc">+</span></span>
<span id="cb101-36"><a href="#cb101-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>() <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">stat_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>,<span class="at">formula =</span> y<span class="sc">~</span><span class="fu">poly</span>(x, <span class="dv">3</span>, <span class="at">raw=</span><span class="cn">TRUE</span>), <span class="at">se =</span> <span class="cn">FALSE</span>)</span>
<span id="cb101-37"><a href="#cb101-37" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span>  gglm<span class="sc">::</span><span class="fu">gglm</span>()</span>
<span id="cb101-38"><a href="#cb101-38" aria-hidden="true" tabindex="-1"></a><span class="do">## orthogonal model</span></span>
<span id="cb101-39"><a href="#cb101-39" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(frying_temp_c <span class="sc">~</span> <span class="fu">poly</span>(frying_time_s, <span class="dv">3</span>) <span class="sc">+</span> method, <span class="at">data =</span> fries, <span class="at">na.action =</span> <span class="st">"na.fail"</span>) <span class="do">## see https://stackoverflow.com/questions/29999900/poly-in-lm-difference-between-raw-vs-orthogonal</span></span>
<span id="cb101-40"><a href="#cb101-40" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span></span>
<span id="cb101-41"><a href="#cb101-41" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>()</span>
<span id="cb101-42"><a href="#cb101-42" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span>  gglm<span class="sc">::</span><span class="fu">gglm</span>()</span>
<span id="cb101-43"><a href="#cb101-43" aria-hidden="true" tabindex="-1"></a><span class="do">## remove the 180 obvs as these are not 'data' they are determined</span></span>
<span id="cb101-44"><a href="#cb101-44" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> fries <span class="sc">%&gt;%</span></span>
<span id="cb101-45"><a href="#cb101-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(frying_temp_c <span class="sc">!=</span> <span class="dv">180</span>)</span>
<span id="cb101-46"><a href="#cb101-46" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb101-47"><a href="#cb101-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">sqrt</span>(frying_time_s), <span class="at">y =</span> frying_temp_c, <span class="at">col =</span> method)) <span class="sc">+</span></span>
<span id="cb101-48"><a href="#cb101-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">geom_point</span>() </span>
<span id="cb101-49"><a href="#cb101-49" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(frying_temp_c <span class="sc">~</span> <span class="fu">poly</span>(<span class="fu">log</span>(frying_time_s), <span class="dv">3</span>) <span class="sc">+</span> method, <span class="at">data =</span> data) </span>
<span id="cb101-50"><a href="#cb101-50" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span></span>
<span id="cb101-51"><a href="#cb101-51" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>()</span>
<span id="cb101-52"><a href="#cb101-52" aria-hidden="true" tabindex="-1"></a>mod <span class="sc">|&gt;</span>  gglm<span class="sc">::</span><span class="fu">gglm</span>()</span>
<span id="cb101-53"><a href="#cb101-53" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span></span>
<span id="cb101-54"><a href="#cb101-54" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">preds =</span> <span class="fu">predict</span>(mod)) <span class="sc">%&gt;%</span></span>
<span id="cb101-55"><a href="#cb101-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">sqrt</span>(frying_time_s), <span class="at">y =</span> frying_temp_c, <span class="at">col =</span> method)) <span class="sc">+</span></span>
<span id="cb101-56"><a href="#cb101-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> preds))</span>
<span id="cb101-57"><a href="#cb101-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-58"><a href="#cb101-58" aria-hidden="true" tabindex="-1"></a><span class="do">## Model diagnostics not looking "great" still. But how can we choose the "best" possible?</span></span>
<span id="cb101-59"><a href="#cb101-59" aria-hidden="true" tabindex="-1"></a><span class="do">## We could manually compare pairs of models using ANOVA/AIC etc. or...</span></span>
<span id="cb101-60"><a href="#cb101-60" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(MuMIn) <span class="do">## remembering which letters are capatalised is a pain</span></span>
<span id="cb101-61"><a href="#cb101-61" aria-hidden="true" tabindex="-1"></a><span class="do">## stands for Multi Model Inference</span></span>
<span id="cb101-62"><a href="#cb101-62" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(frying_temp_c <span class="sc">~</span> <span class="fu">poly</span>(<span class="fu">log</span>(frying_time_s), <span class="dv">3</span>) <span class="sc">+</span> method, <span class="at">data =</span> data, <span class="at">na.action =</span> na.fail)</span>
<span id="cb101-63"><a href="#cb101-63" aria-hidden="true" tabindex="-1"></a><span class="do">## na.action = na.fail is so MuMIn dredge works</span></span>
<span id="cb101-64"><a href="#cb101-64" aria-hidden="true" tabindex="-1"></a>MuMIn<span class="sc">::</span><span class="fu">dredge</span>(mod)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="code-example-to-support-sec-mods-sec-ss" class="level2">
<h2 class="anchored" data-anchor-id="code-example-to-support-sec-mods-sec-ss">Code example to support <a href="#sec-mods" class="quarto-xref">Section&nbsp;8.2</a> / <a href="#sec-ss" class="quarto-xref">Section&nbsp;8.2.2</a></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>base_url <span class="ot">&lt;-</span> <span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/"</span></span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>factorial <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">paste</span>(base_url, <span class="st">"factorial_expt.csv"</span>, <span class="at">sep =</span> <span class="st">""</span>))</span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>factorial<span class="sc">$</span>Disease <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(factorial<span class="sc">$</span>Disease)</span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>factorial<span class="sc">$</span>Organ <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(factorial<span class="sc">$</span>Organ)</span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a>unbalanced <span class="ot">&lt;-</span> factorial[<span class="sc">-</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="dv">10</span>), ]</span>
<span id="cb102-6"><a href="#cb102-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Null model.</span></span>
<span id="cb102-7"><a href="#cb102-7" aria-hidden="true" tabindex="-1"></a>fit.null <span class="ot">&lt;-</span> <span class="fu">lm</span>(logAUC <span class="sc">~</span> <span class="dv">1</span>, <span class="at">data =</span> unbalanced)</span>
<span id="cb102-8"><a href="#cb102-8" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(fit.null)</span>
<span id="cb102-9"><a href="#cb102-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Only Factor Disease.</span></span>
<span id="cb102-10"><a href="#cb102-10" aria-hidden="true" tabindex="-1"></a>fit.a <span class="ot">&lt;-</span> <span class="fu">lm</span>(logAUC <span class="sc">~</span> Disease, <span class="at">data =</span> unbalanced)</span>
<span id="cb102-11"><a href="#cb102-11" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(fit.a)</span>
<span id="cb102-12"><a href="#cb102-12" aria-hidden="true" tabindex="-1"></a><span class="do">## Factors Disease and Organ.</span></span>
<span id="cb102-13"><a href="#cb102-13" aria-hidden="true" tabindex="-1"></a>fit.ab <span class="ot">&lt;-</span> <span class="fu">lm</span>(logAUC <span class="sc">~</span> Disease <span class="sc">+</span> Organ, <span class="at">data =</span> unbalanced)</span>
<span id="cb102-14"><a href="#cb102-14" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(fit.ab)</span>
<span id="cb102-15"><a href="#cb102-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Factors Disease and Organ with interaction.</span></span>
<span id="cb102-16"><a href="#cb102-16" aria-hidden="true" tabindex="-1"></a>fit.abi <span class="ot">&lt;-</span> <span class="fu">lm</span>(logAUC <span class="sc">~</span> Disease<span class="sc">*</span>Organ, <span class="at">data =</span> unbalanced)</span>
<span id="cb102-17"><a href="#cb102-17" aria-hidden="true" tabindex="-1"></a><span class="do">## ANOVA table</span></span>
<span id="cb102-18"><a href="#cb102-18" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(fit.abi)</span>
<span id="cb102-19"><a href="#cb102-19" aria-hidden="true" tabindex="-1"></a><span class="do">## First line.</span></span>
<span id="cb102-20"><a href="#cb102-20" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">residuals</span>(fit.null)<span class="sc">^</span><span class="dv">2</span>) <span class="sc">-</span> <span class="fu">sum</span>(<span class="fu">residuals</span>(fit.a)<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb102-21"><a href="#cb102-21" aria-hidden="true" tabindex="-1"></a><span class="do">## Second line.</span></span>
<span id="cb102-22"><a href="#cb102-22" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">residuals</span>(fit.a)<span class="sc">^</span><span class="dv">2</span>) <span class="sc">-</span> <span class="fu">sum</span>(<span class="fu">residuals</span>(fit.ab)<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb102-23"><a href="#cb102-23" aria-hidden="true" tabindex="-1"></a><span class="do">## Third line.</span></span>
<span id="cb102-24"><a href="#cb102-24" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">residuals</span>(fit.ab)<span class="sc">^</span><span class="dv">2</span>) <span class="sc">-</span> <span class="fu">sum</span>(<span class="fu">residuals</span>(fit.abi)<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb102-25"><a href="#cb102-25" aria-hidden="true" tabindex="-1"></a><span class="do">## Final line.</span></span>
<span id="cb102-26"><a href="#cb102-26" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">residuals</span>(fit.abi)<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb102-27"><a href="#cb102-27" aria-hidden="true" tabindex="-1"></a><span class="do">## Type II sums of squares</span></span>
<span id="cb102-28"><a href="#cb102-28" aria-hidden="true" tabindex="-1"></a>car<span class="sc">::</span><span class="fu">Anova</span>(fit.abi, <span class="at">type =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="suggested-solution-to-sec-ss" class="level2">
<h2 class="anchored" data-anchor-id="suggested-solution-to-sec-ss">Suggested solution to <a href="#sec-ss" class="quarto-xref">Section&nbsp;8.2.2</a></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="do">## using "gorilla" data from lecture 01</span></span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a><span class="do">## example only</span></span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(tidyverse)</span>
<span id="cb103-4"><a href="#cb103-4" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"https://raw.githubusercontent.com/STATS-UOA/databunker/master/data/BMI.csv"</span>)</span>
<span id="cb103-5"><a href="#cb103-5" aria-hidden="true" tabindex="-1"></a>data <span class="sc">%&gt;%</span> </span>
<span id="cb103-6"><a href="#cb103-6" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">grand_mean =</span> <span class="fu">mean</span>(bmi)) <span class="sc">%&gt;%</span></span>
<span id="cb103-7"><a href="#cb103-7" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">sstotal =</span> (bmi <span class="sc">-</span> grand_mean) <span class="sc">^</span><span class="dv">2</span> ) <span class="sc">%&gt;%</span></span>
<span id="cb103-8"><a href="#cb103-8" aria-hidden="true" tabindex="-1"></a>   <span class="fu">group_by</span>(gender) <span class="sc">%&gt;%</span></span>
<span id="cb103-9"><a href="#cb103-9" aria-hidden="true" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="at">treatment_mean =</span> <span class="fu">mean</span>(bmi),</span>
<span id="cb103-10"><a href="#cb103-10" aria-hidden="true" tabindex="-1"></a>          <span class="at">sse =</span> (bmi <span class="sc">-</span> treatment_mean)<span class="sc">^</span><span class="dv">2</span>,</span>
<span id="cb103-11"><a href="#cb103-11" aria-hidden="true" tabindex="-1"></a>          <span class="at">tmp =</span> (treatment_mean <span class="sc">-</span> grand_mean)<span class="sc">^</span><span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb103-12"><a href="#cb103-12" aria-hidden="true" tabindex="-1"></a>   <span class="fu">summarise</span>(<span class="at">ss_tot =</span> <span class="fu">sum</span>(sstotal), <span class="at">ss_e =</span> <span class="fu">sum</span>(sse), <span class="at">ss_treat =</span> <span class="fu">sum</span>(tmp)) <span class="sc">%&gt;%</span></span>
<span id="cb103-13"><a href="#cb103-13" aria-hidden="true" tabindex="-1"></a>   <span class="fu">pull</span>(ss_treat) <span class="sc">%&gt;%</span> <span class="fu">sum</span>()</span>
<span id="cb103-14"><a href="#cb103-14" aria-hidden="true" tabindex="-1"></a><span class="do">## compare to SS value in</span></span>
<span id="cb103-15"><a href="#cb103-15" aria-hidden="true" tabindex="-1"></a><span class="fu">aov</span>(bmi <span class="sc">~</span> gender, <span class="at">data =</span> data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="suggested-reasoning-for-the-sec-distributions-chooks" class="level2">
<h2 class="anchored" data-anchor-id="suggested-reasoning-for-the-sec-distributions-chooks">Suggested reasoning for the <a href="#sec-distributions" class="quarto-xref">Section&nbsp;11.2</a> chooks</h2>
<p><img src="runsheets/img/chooks.png" class="img-fluid"></p>
</section>
<section id="suggested-solutions-to-sec-distribution" class="level2">
<h2 class="anchored" data-anchor-id="suggested-solutions-to-sec-distribution">Suggested solutions to <a href="#sec-distribution" class="quarto-xref">Section&nbsp;12.2</a></h2>
<ol>
<li>
<strong>Poisson</strong>
</li>
<li>
<strong>Geometric</strong> (number of independent trials until the first success)
</li>
<li>
<strong>Hypergeometric</strong> (sampling without replacement from a finite group)
</li>
<li>
<strong>Binomial</strong>
</li>
<li>
<strong>Normal</strong>
</li>
<li>
<strong>Exponential</strong> (waiting time between independent events)
</li>
<li>
<strong>Gamma</strong>
</li>
<li>
<strong>Uniform</strong><a href="#fn12" class="footnote-ref" id="fnref12" role="doc-noteref"><sup>12</sup></a>
</li>
</ol>
</section>
<section id="scoring-for-quiz-in-sec-bayes" class="level2">
<h2 class="anchored" data-anchor-id="scoring-for-quiz-in-sec-bayes">Scoring for quiz in <a href="#sec-bayes" class="quarto-xref">Section&nbsp;14.2</a></h2>
<ol type="1">
<li>a = 1 point, b = 3 points, c = 2 points; 2. a = 1 point, b = 3 points, c = 1 point; 3. a = 3 points, b = 1 point; 4. a = 3 points, b = 1 point. According to <span class="citation" data-cites="bayesrules">Johnson, Ott, and Dogucu (<a href="#ref-bayesrules" role="doc-biblioref">2021</a>)</span> <em>Totals from 4–5 indicate that your current thinking is fairly frequentist, whereas totals from 9–12 indicate alignment with the Bayesian philosophy. In between these extremes, totals from 6–8 indicate that you see strengths in both philosophies</em></li>
</ol>
</section>
</section>
<section id="references" class="level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-james" class="csl-entry" role="listitem">
Abaz, J., S. J. Walsh, J. M. Curran, D. S. Moss, J. Cullen, J. A. Bright, G. A. Crowe, S. L. Cockerton, and T. E. Power. 2002. <span>“Comparison of the Variables Affecting the Recovery of DNA from Common Drinking Containers.”</span> <em>Forensic Science International</em> 126 (3): 233–40. <a href="https://doi.org/10.1016/S0379-0738(02)00047-0">https://doi.org/10.1016/S0379-0738(02)00047-0</a>.
</div>
<div id="ref-fries" class="csl-entry" role="listitem">
Abdul Hamid, N., S. Omar, and M. Sanny. 2019. <span>“Effect of Thawing Conditions and Corresponding Frying Temperature Profiles on the Formation of Acrylamide in French Fries.”</span> <em>Journal of the Saudi Society of Agricultural Sciences</em> 18 (4): 396–400. https://doi.org/<a href="https://doi.org/10.1016/j.jssas.2018.01.002">https://doi.org/10.1016/j.jssas.2018.01.002</a>.
</div>
<div id="ref-lizards" class="csl-entry" role="listitem">
Álvarez-Ruiz, L, J Belliure, and JG Pausas. 2021. <span>“Fire-Driven Behavioral Response to Smoke in a Mediterranean Lizard.”</span> <em>Behavioral Ecology</em> 32 (4): 662–67. <a href="https://doi.org/10.1093/beheco/arab010">https://doi.org/10.1093/beheco/arab010</a>.
</div>
<div id="ref-bats" class="csl-entry" role="listitem">
Barlow, K. E., P. A. Briggs, K. A. Haysom, A. M. Hutson, N. L. Lechiara, P. A. Racey, A. L. Walsh, and S. D. Langton. 2015. <span>“Citizen Science Reveals Trends in Bat Populations: The National Bat Monitoring Programme in Great Britain.”</span> <em>Biological Conservation</em> 182: 14–26. https://doi.org/<a href="https://doi.org/10.1016/j.biocon.2014.11.022">https://doi.org/10.1016/j.biocon.2014.11.022</a>.
</div>
<div id="ref-monkeydata" class="csl-entry" role="listitem">
Bliss-Moreau, Eliza, and Mark Baxter. 2020. <span>“Data from: Interest in Nonsocial Novel Stimuli as a Function of Age in Rhesus Monkeys.”</span> Dryad. <a href="https://doi.org/10.5061/dryad.1bj133v">https://doi.org/10.5061/dryad.1bj133v</a>.
</div>
<div id="ref-monkey" class="csl-entry" role="listitem">
Bliss-Moreau, Eliza, and Mark G. Baxter. 2019. <span>“Interest in Non-Social Novel Stimuli as a Function of Age in Rhesus Monkeys.”</span> <em>Royal Society Open Science</em> 6 (1): 182237. <a href="https://doi.org/10.1098/rsos.182237">https://doi.org/10.1098/rsos.182237</a>.
</div>
<div id="ref-danblog" class="csl-entry" role="listitem">
Dan. 2022. <span>“Everybody’s Got Something to Hide Except Me and My Monkey.”</span> <a href="https://dansblog.netlify.app/posts/2022-09-04-everybodys-got-something-to-hide-except-me-and-my-monkey/everybodys-got-something-to-hide-except-me-and-my-monkey.html">https://dansblog.netlify.app/posts/2022-09-04-everybodys-got-something-to-hide-except-me-and-my-monkey/everybodys-got-something-to-hide-except-me-and-my-monkey.html</a>.
</div>
<div id="ref-fisher" class="csl-entry" role="listitem">
Fisher, Ronald Aylmer. 1926. <span>“The Arrangement of Field Experiments.”</span> <em>Journal of the Ministry of Agriculture</em> 33: 505–15. <a href="https://doi.org/doi.org/10.23637/rothamsted.8v61q">https://doi.org/doi.org/10.23637/rothamsted.8v61q</a>.
</div>
<div id="ref-birds" class="csl-entry" role="listitem">
García-Navas, Vicente, Carlos Martínez-Núñez, Rubén Tarifa, José L. Molina-Pardo, Francisco Valera, Teresa Salido, Francisco M. Camacho, and Pedro J. Rey. 2022. <span>“Partitioning Beta Diversity to Untangle Mechanisms Underlying the Assembly of Bird Communities in Mediterranean Olive Groves.”</span> <em>Diversity and Distributions</em> 28 (1): 112–27. https://doi.org/<a href="https://doi.org/10.1111/ddi.13445">https://doi.org/10.1111/ddi.13445</a>.
</div>
<div id="ref-gelman2005analysis" class="csl-entry" role="listitem">
Gelman, Andrew. 2005. <span>“Analysis of Variance—Why It Is More Important Than Ever.”</span> <em>The Annals of Statistics</em> 33 (1): 1–53. <a href="https://doi.org/10.1214/009053604000001048">https://doi.org/10.1214/009053604000001048</a>.
</div>
<div id="ref-olivier" class="csl-entry" role="listitem">
Gimenez, Olivier. 2023. <span>“Bayesian Statistics with r.”</span> <a href="https://oliviergimenez.github.io/bayesian-stats-with-R/">https://oliviergimenez.github.io/bayesian-stats-with-R/</a>.
</div>
<div id="ref-alec" class="csl-entry" role="listitem">
Helsdingen, A. B. M. van, T. A. Marques, and C. M. Jones-Todd. 2025. <span>“An Inhomogeneous Weibull–Hawkes Process to Model Underdispersed Acoustic Cues.”</span> <em>Journal of Agricultural, Biological and Environmental Statistics</em> 30: 39–62. <a href="https://doi.org/10.1007/s13253-024-00626-w">https://doi.org/10.1007/s13253-024-00626-w</a>.
</div>
<div id="ref-bullshit" class="csl-entry" role="listitem">
Hicks, M. T., J. Humphries, and J. Slater. 2024. <span>“ChatGPT Is Bullshit.”</span> <em>Ethics and Information Technology</em> 26 (38). <a href="https://doi.org/10.1007/s10676-024-09775-5">https://doi.org/10.1007/s10676-024-09775-5</a>.
</div>
<div id="ref-autism" class="csl-entry" role="listitem">
Howsmon, DP, U Kruger, S Melnyk, SJ James, and J Hahn. 2017. <span>“Classification and Adaptive Behavior Prediction of Children with Autism Spectrum Disorder Based Upon Multivariate Data Analysis of Markers of Oxidative Stress and DNA Methylation.”</span> <em>PLoS Computational Biology</em> 13 (3): e1005385. <a href="https://doi.org/10.1371/journal.pcbi.1005385">https://doi.org/10.1371/journal.pcbi.1005385</a>.
</div>
<div id="ref-bayesrules" class="csl-entry" role="listitem">
Johnson, Alicia A., Miles Q. Ott, and Mine Dogucu. 2021. <em>Bayes Rules!: An Introduction to Applied Bayesian Modeling</em>. Boca Raton, FL: CRC Press.
</div>
<div id="ref-cancer" class="csl-entry" role="listitem">
Jones-Todd, Charlotte M., Peter Caie, Janine B. Illian, Ben C. Stevenson, Anne Savage, David J. Harrison, and James L. Bown. 2018. <span>“Identifying Prognostic Structural Features in Tissue Sections of Colon Cancer Patients Using Point Pattern Analysis.”</span> <em>Statistics in Medicine</em> 38 (8): 1421–41. https://doi.org/<a href="https://doi.org/10.1002/sim.8046">https://doi.org/10.1002/sim.8046</a>.
</div>
<div id="ref-stelfi" class="csl-entry" role="listitem">
Jones-Todd, Charlotte M., and Alec B. M. van Helsdingen. 2024. <span>“Stelfi: An r Package for Fitting Hawkes and Log-Gaussian Cox Point Process Models.”</span> <em>Ecology and Evolution</em> 14 (2): e11005. https://doi.org/<a href="https://doi.org/10.1002/ece3.11005">https://doi.org/10.1002/ece3.11005</a>.
</div>
<div id="ref-whales" class="csl-entry" role="listitem">
Jones-Todd, Charlotte M., Enrico Pirotta, John W. Durban, Diane E. Claridge, Robin W. Baird, Erin A. Falcone, Gregory S. Schorr, Stephanie Watwood, and Len Thomas. 2021. <span>“Discrete-Space Continuous-Time Models of Marine Mammal Exposure to Navy Sonar.”</span> <em>Ecological Applications</em> 32 (1): e02475. https://doi.org/<a href="https://doi.org/10.1002/eap.2475">https://doi.org/10.1002/eap.2475</a>.
</div>
<div id="ref-lawson2014design" class="csl-entry" role="listitem">
Lawson, Colin. 2014. <em>Design and Analysis of Experiments with r</em>. CRC Press. <a href="https://www.routledge.com/Design-and-Analysis-of-Experiments-with-R/Lawson/p/book/9781439868133">https://www.routledge.com/Design-and-Analysis-of-Experiments-with-R/Lawson/p/book/9781439868133</a>.
</div>
<div id="ref-tiago" class="csl-entry" role="listitem">
Marques, T. A., L. Thomas, S. W. Martin, D. K. Mellinger, J. A. Ward, D. J. Moretti, D. Harris, and P. L. Tyack. 2013. <span>“Estimating Animal Population Density Using Passive Acoustics.”</span> <em>Biological Reviews</em> 88 (2): 287–309. <a href="https://doi.org/10.1111/brv.12001">https://doi.org/10.1111/brv.12001</a>.
</div>
<div id="ref-dna" class="csl-entry" role="listitem">
Park, Jae Il, Jin Hoon Kim, Eunji Seo, Dong Hoon Bae, Sung Yoon Kim, Hyun Cheol Lee, Kyung Min Woo, and Young Seok Kim. 2016. <span>“Identification and Evaluation of Age-Correlated DNA Methylation Markers for Forensic Use.”</span> <em>Forensic Science International: Genetics</em> 23: 64–70. <a href="https://doi.org/10.1016/j.fsigen.2016.03.005">https://doi.org/10.1016/j.fsigen.2016.03.005</a>.
</div>
<div id="ref-herbicide" class="csl-entry" role="listitem">
Peterson, PG, MF Merrett, SV Fowler, DP Barrett, and Q Paynter. 2020. <span>“Comparing Biocontrol and Herbicide for Managing an Invasive Non-Native Plant Species: Efficacy, Non-Target Effects and Secondary Invasion.”</span> <em>Journal of Applied Ecology</em> 57 (10): 1876–84. <a href="https://doi.org/10.1111/1365-2664.13691">https://doi.org/10.1111/1365-2664.13691</a>.
</div>
<div id="ref-clustglm" class="csl-entry" role="listitem">
Pledger, Shirley, and Richard Arnold. 2014. <span>“Multivariate Methods Using Mixtures: Correspondence Analysis, Scaling and Pattern-Detection.”</span> <em>Computational Statistics &amp; Data Analysis</em> 71: 241–61.
</div>
<div id="ref-robots" class="csl-entry" role="listitem">
Prather, James, Paul Denny, Juho Leinonen, Brett A. Becker, Ibrahim Albluwi, Michelle Craig, Hieke Keuning, et al. 2023. <span>“The Robots Are Here: Navigating the Generative AI Revolution in Computing Education.”</span> In. New York, NY, USA: Association for Computing Machinery. <a href="https://doi.org/10.1145/3623762.3633499">https://doi.org/10.1145/3623762.3633499</a>.
</div>
<div id="ref-cannabis" class="csl-entry" role="listitem">
Schwabe, Anna L., Vanessa Johnson, Joshua Harrelson, and Mitchell E. McGlaughlin. 2023. <span>“Uncomfortably High: Testing Reveals Inflated THC Potency on Retail Cannabis Labels.”</span> <em>PLOS ONE</em> 18 (4): e0282396. <a href="https://doi.org/10.1371/journal.pone.0282396">https://doi.org/10.1371/journal.pone.0282396</a>.
</div>
<div id="ref-instructor" class="csl-entry" role="listitem">
Sheard, Judy, Paul Denny, Arto Hellas, Juho Leinonen, Lauri Malmi, and Simon. 2024. <span>“Instructor Perceptions of AI Code Generation Tools - a Multi-Institutional Interview Study.”</span> In. New York, NY, USA: Association for Computing Machinery. <a href="https://doi.org/10.1145/3626252.3630880">https://doi.org/10.1145/3626252.3630880</a>.
</div>
<div id="ref-retracted" class="csl-entry" role="listitem">
Shu, Lisa L., Nina Mazar, Francesca Gino, Dan Ariely, and Max H. Bazerman. 2012. <span>“RETRACTED: Signing at the Beginning Makes Ethics Salient and Decreases Dishonest Self-Reports in Comparison to Signing at the End.”</span> <em>Proceedings of the National Academy of Sciences</em> 109 (38): 15197–200. <a href="https://doi.org/10.1073/pnas.1209746109">https://doi.org/10.1073/pnas.1209746109</a>.
</div>
<div id="ref-grass" class="csl-entry" role="listitem">
Steinauer, Katja, David Tilman, Peter D. Wragg, Simone Cesarz, Jane M. Cowles, Karin Pritsch, Peter B. Reich, Wolfgang W. Weisser, and Nico Eisenhauer. 2015. <span>“Plant Diversity Effects on Soil Microbial Functions and Enzymes Are Stronger Than Warming in a Grassland Experiment.”</span> <em>Ecology</em> 96 (1): 99–112. https://doi.org/<a href="https://doi.org/10.1890/14-0088.1">https://doi.org/10.1890/14-0088.1</a>.
</div>
<div id="ref-mvabund" class="csl-entry" role="listitem">
Wang, Yi, Ulrike Naumann, Stephen T. Wright, and David I. Warton. 2012. <span>“Mvabund– an r Package for Model-Based Analysis of Multivariate Abundance Data.”</span> <em>Methods in Ecology and Evolution</em> 3 (3): 471–74. https://doi.org/<a href="https://doi.org/10.1111/j.2041-210X.2012.00190.x">https://doi.org/10.1111/j.2041-210X.2012.00190.x</a>.
</div>
<div id="ref-ioa" class="csl-entry" role="listitem">
Ward, Monica, Fiona O’Riordan, Danielle Logan-Fleming, Dervila Cooke, Tara Concannon-Gibney, Marina Efthymiou, and Niamh Watkins and. 2024. <span>“Interactive Oral Assessment Case Studies: An Innovative, Academically Rigorous, Authentic Assessment Approach.”</span> <em>Innovations in Education and Teaching International</em> 61 (5): 930–47. <a href="https://doi.org/10.1080/14703297.2023.2251967">https://doi.org/10.1080/14703297.2023.2251967</a>.
</div>
<div id="ref-lobster" class="csl-entry" role="listitem">
Wilkinson, Erin B., Jonathan H. Grabowski, Graham D. Sherwood, and Philip O. Yund. 2015. <span>“Influence of Predator Identity on the Strength of Predator Avoidance Responses in Lobsters.”</span> <em>Journal of Experimental Marine Biology and Ecology</em> 465: 107–12. https://doi.org/<a href="https://doi.org/10.1016/j.jembe.2015.01.002">https://doi.org/10.1016/j.jembe.2015.01.002</a>.
</div>
<div id="ref-goodprogramming" class="csl-entry" role="listitem">
Wilson, Greg, Jennifer Bryan, Karen Cranston, Justin Kitzes, Lex Nederbragt, and Tracy K. Teal. 2017. <span>“Good Enough Practices in Scientific Computing.”</span> <em>PLoS Computational Biology</em> 13 (6): e1005510. <a href="https://doi.org/10.1371/journal.pcbi.1005510">https://doi.org/10.1371/journal.pcbi.1005510</a>.
</div>
</div>
</section>


<div id="quarto-appendix" class="default"><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>

<ol>
<li id="fn1"><p>Note I have intentionally only used published plots that I made, I consent to the roasting!<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2"><p>The former example is a solution posed on <a href="https://stackoverflow.com/questions/19341554/regular-expression-in-base-r-regex-to-identify-email-address">this StackOverflow post</a> asking about email address validation. The latter snippet is my attempt at a long winded (and far less stable) approach.<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3"><p>See <a href="https://www.theguardian.com/education/2023/jun/25/harvard-professor-data-fraud">this Guardian article</a> about the debacle<a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn4"><p><a href="https://en.wikipedia.org/wiki/Ronald_Fisher">R. A. Fisher’s</a> work in the area of experimental design is, perhaps, the most well known in the field. The principles he devised we still abide by today. Note, however, to give a balanced view of the celebrated mathematician many of his views (on eugenics and race in particular) are considered unethical by many. I would urge you to read up on <a href="https://www.nature.com/articles/s41437-020-00394-6">his work in this area and come to your own conclusions</a>.<a href="#fnref4" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn5"><p>Note that I do like the writing style of the blog but don’t totally agree with the author’s conclusions around repeated measurements; if you want to know why, ask me!<a href="#fnref5" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn6"><p>The title stolen shamelessly from <a href="https://emitanaka.org/edibble-book/">Emi Tanaka’s ebook</a><a href="#fnref6" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn7"><p>Modified from Figure 1.1 of <span class="citation" data-cites="lawson2014design">Lawson (<a href="#ref-lawson2014design" role="doc-biblioref">2014</a>)</span><a href="#fnref7" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn8"><p>often termed nuisance variables<a href="#fnref8" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn9"><p>Random effect models to the rescue!<a href="#fnref9" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn10"><p>questions taken verbatim from <span class="citation" data-cites="bayesrules">Johnson, Ott, and Dogucu (<a href="#ref-bayesrules" role="doc-biblioref">2021</a>)</span><a href="#fnref10" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn11"><p>more specifically, randomly choose which one of <span class="citation" data-cites="lizards">Álvarez-Ruiz, Belliure, and Pausas (<a href="#ref-lizards" role="doc-biblioref">2021</a>)</span>, <span class="citation" data-cites="autism">Howsmon et al. (<a href="#ref-autism" role="doc-biblioref">2017</a>)</span>, or <span class="citation" data-cites="herbicide">Peterson et al. (<a href="#ref-herbicide" role="doc-biblioref">2020</a>)</span> you will be considering in this workshop.<a href="#fnref11" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn12"><p>This was based on an IRL <a href="https://en.wikipedia.org/wiki/German_tank_problem">WWII problem</a><a href="#fnref12" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>